<!DOCTYPE html>
<html lang="en">
<head>
<!--  2025-02-01 Sat 13:23  -->
<meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>‎</title>
<meta name="author" content="basti" />
<meta name="generator" content="Org Mode" />

<link type="text/css" rel="stylesheet" href="org_html_export.css" />
<script type="text/javascript">

document.addEventListener('DOMContentLoaded', function() {
    // Set the initial width for subfigures and their images
    document.querySelectorAll('figure.subfigure[data-width]').forEach(function(subfigure) {
        subfigure.style.width = subfigure.getAttribute('data-width');
    });

    document.querySelectorAll('figure.subfigure img[data-width]').forEach(function(img) {
        img.style.width = img.getAttribute('data-width');
    });

    // Event listener for images within subfigures
    document.querySelectorAll('figure.subfigure img[data-width]').forEach(function(img) {
        img.addEventListener('click', function() {
            // Determine the .figure-wrapper that contains the clicked image
            let wrapper = this.closest('.figure-wrapper');

            // If found, adjust the sizes of its subfigures
            if (wrapper) {
                let subfigures = wrapper.querySelectorAll('figure.subfigure');
                subfigures.forEach(subfigure => {
                    let width = parseFloat(subfigure.getAttribute('data-width'));
                    if (width < 50) { // Double size if less than 50%
                        subfigure.setAttribute('data-width', (width * 2) + '%');
                        subfigure.style.width = (width * 2) + '%';
                    } else { // Halve size otherwise
                        subfigure.setAttribute('data-width', (width / 2) + '%');
                        subfigure.style.width = (width / 2) + '%';
                    }
                });
            }
        });
    });
});

document.addEventListener('DOMContentLoaded', function () {
    var headers = document.querySelectorAll('h1 .extended, h2 .extended, h3 .extended, h4 .extended, h5 .extended, h6 .extended');

    headers.forEach(function (header) {
        var foldableHeader = header.closest('h1, h2, h3, h4, h5, h6');
        if (foldableHeader) {
            foldableHeader.classList.add('foldable-header');
            var nextElement = foldableHeader.nextElementSibling;
            var contentToFold = [];
            while (nextElement && !nextElement.matches('h1, h2, h3, h4, h5, h6')) {
                contentToFold.push(nextElement);
                nextElement = nextElement.nextElementSibling;
            }
            contentToFold.forEach(function (element) {
                element.classList.add('folded-content');
            });

            foldableHeader.addEventListener('click', function () {
                contentToFold.forEach(function (element) {
                    if (element.style.display === 'none' || element.style.display == '') {
                        element.style.display = 'block';
                    } else {
                        element.style.display = 'none';
                    }
                });
            });
        }
    });
});

</script>

<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script type="text/x-mathjax-config">
<!-- /* --><![CDATA[/*><!--*/
    MathJax.Hub.Config({
        jax: ["input/TeX", "output/HTML-CSS"],
        extensions: ["tex2jax.js","TeX/AMSmath.js","TeX/AMSsymbols.js",
                     "TeX/noUndefined.js", "[Contrib]/siunitx/siunitx.js", "[Contrib]/mhchem/mhchem.js"],
        tex2jax: {
            inlineMath: [ ["\\(","\\)"], ['$', '$'], ],
            displayMath: [ ['$$','$$'], ["\\[","\\]"], ["\\begin{displaymath}","\\end{displaymath}"] ],
            skipTags: ["script","noscript","style","textarea","pre","code"],
            ignoreClass: "tex2jax_ignore",
            processEscapes: false,
            processEnvironments: true,
            preview: "TeX"
        },
        showProcessingMessages: true,
        displayAlign: "center",
        displayIndent: "2em",

        "HTML-CSS": {
             scale: 1.0,
             availableFonts: ["STIX","TeX"],
             preferredFont: "TeX",
             webFont: "TeX",
             imageFont: "TeX",
             showMathMenu: true,
        },
        MMLorHTML: {
             prefer: {
                 MSIE:    "MML",
                 Firefox: "MML",
                 Opera:   "HTML",
                 other:   "HTML"
             }
        },
        TeX: {
              equationNumbers: { autoNumber: 'AMS' },
              Macros: {
                  ccsini: '{\\mathrm{Si}₃\\mathrm{N}₄}',
                  cefe: '{\\ce{^{55}Fe}}',
                  vektor: ['{\\begin{pmatrix}#1\\end{pmatrix}}', 1],
                  mtrix: ['{\\begin{pmatrix}#1\\end{pmatrix}}', 1],
                  cp: '{\\mathrm{CP}}',
                  cpt: '{\\mathrm{CPT}}',
                  dd: '{\\mathop{}\\!{\\mathrm{d}}}',
                  sinc: '{\\mathrm{sinc}}'
              }
        }
    });
/*]]>*///--&gt;
</script>
</head>
<body>
<div id="content" class="content">
<nav id="table-of-contents" role="doc-toc">
<h2>Table of Contents</h2>
<div id="text-table-of-contents" role="doc-toc">
<ul>
<li>  <a href="./errata.html#sec:errata">1. Errata   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./introduction.html#sec:introduction">2. Introduction   <span class="tag">    <span class="Intro">Intro</span>  </span></a></li>
<li><a href="./about_thesis.html#sec:about_thesis">3. About this thesis   <span class="tag">  <span class="Intro">Intro</span></span></a>
<ul>
<li>  <a href="./about_thesis.html#orgfe77eb6">3.1. Extended notes for the extended thesis <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./about_thesis.html#orgf40a93d">3.2. Why Org mode   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./about_thesis.html#org607bd89">3.3. Notes for future PhD students and IAXO analyses   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
</ul>
</li>
<li><a href="./theory.html#sec:theory">4. Theory of axions   <span class="tag">  <span class="Theory">Theory</span></span></a>
<ul>
<li>  <a href="./theory.html#sec:theory:useful_reading_material">4.1. Useful reading material   <span class="tag">    <span class="optional">optional</span>  </span></a></li>
<li>  <a href="./theory.html#sec:theory:illustration_cpt">4.2. Illustration of the \cpt symmetry <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./theory.html#sec:theory:invisible_axion_models">4.3. Invisible axion models and axion couplings</a></li>
<li>  <a href="./theory.html#sec:theory:axion_interactions">4.4. Implications for axion interactions - conversion probability</a></li>
<li>  <a href="./theory.html#sec:theory:solar_axion_flux">4.5. Solar axion flux</a></li>
<li>  <a href="./theory.html#sec:theory:chameleon">4.6. Chameleons</a></li>
<li>  <a href="./theory.html#sec:theory:current_bounds">4.7. Current bounds on coupling constants</a></li>
</ul>
</li>
<li><a href="./helioscopes.html#sec:helioscopes">5. Axion helioscopes   <span class="tag">  <span class="Theory">Theory</span></span></a>
<ul>
<li>  <a href="./helioscopes.html#sec:helioscopes:cast">5.1. CERN Axion Solar Telescope (CAST)</a></li>
<li>  <a href="./helioscopes.html#sec:helioscopes:iaxo">5.2. International AXion Observatory (IAXO)</a></li>
</ul>
</li>
<li><a href="./theory_detector.html#sec:theory_detector">6. X-rays, cosmic muons and gaseous detectors   <span class="tag">  <span class="Theory">Theory</span></span></a>
<ul>
<li>  <a href="./theory_detector.html#sec:theory:particle_int">6.1. Particle interactions with matter</a></li>
<li>  <a href="./theory_detector.html#sec:theory:cosmic_radiation">6.2. Cosmic rays</a></li>
<li>  <a href="./theory_detector.html#sec:theory:gas_fundamentals">6.3. Gaseous detector fundamentals</a></li>
</ul>
</li>
<li><a href="./septemboard.html#sec:septemboard">7. Septemboard detector   <span class="tag">  <span class="Detector">Detector</span></span></a>
<ul>
<li>  <a href="./septemboard.html#sec:detector:micromegas">7.1. Micromegas working principle</a></li>
<li>  <a href="./septemboard.html#sec:detector:timepix">7.2. Timepix ASIC</a></li>
<li>  <a href="./septemboard.html#sec:detector:gridpix">7.3. GridPix</a></li>
<li>  <a href="./septemboard.html#sec:detector:detector_2014_15">7.4. 2014 / 2015 GridPix detector for CAST</a></li>
<li>  <a href="./septemboard.html#sec:detector:detector_overview">7.5. Septemboard detector overview</a></li>
<li>  <a href="./septemboard.html#org2a0c0f4">7.6. Detector readout system</a></li>
<li>  <a href="./septemboard.html#sec:detector:scintillators">7.7. Scintillator vetoes</a></li>
<li>  <a href="./septemboard.html#sec:detector:fadc">7.8. FADC</a></li>
<li>  <a href="./septemboard.html#sec:detector:sin_window">7.9. SiN window</a></li>
<li>  <a href="./septemboard.html#sec:detector:septemboard">7.10. Septemboard - 6 GridPixes around a center one</a></li>
<li>  <a href="./septemboard.html#sec:detector:water_cooling">7.11. Water cooling and temperature readout for the septemboard</a></li>
<li>  <a href="./septemboard.html#sec:septem:efficiency">7.12. Detector efficiency</a></li>
<li>  <a href="./septemboard.html#sec:detector:daq">7.13. Data acquisition and detector monitoring</a></li>
</ul>
</li>
<li><a href="./operation_calibration.html#sec:operation_calibration">8. Detector calibration for operation   <span class="tag">  <span class="Detector">Detector</span></span></a>
<ul>
<li>  <a href="./operation_calibration.html#sec:operation_calibration:timepix">8.1. Timepix calibrations</a></li>
<li>  <a href="./operation_calibration.html#sec:operation_calibration:fadc">8.2. FADC calibration</a></li>
<li>  <a href="./operation_calibration.html#sec:operation_calibration:scintillators">8.3. Scintillator calibration</a></li>
</ul>
</li>
<li><a href="./reconstruction.html#sec:reconstruction">9. Data reconstruction   <span class="tag">  <span class="Reconstruction">Reconstruction</span></span></a>
<ul>
<li>  <a href="./reconstruction.html#sec:reco:tpa">9.1. <code>TimepixAnalysis</code> and Nim</a></li>
<li>  <a href="./reconstruction.html#sec:reco:tos_data_parsing">9.2. TOS data parsing</a></li>
<li>  <a href="./reconstruction.html#sec:reco:event_shape">9.3. Expectation of event shapes</a></li>
<li>  <a href="./reconstruction.html#sec:reco:data_reconstruction">9.4. Data reconstruction</a></li>
<li>  <a href="./reconstruction.html#sec:reco:fadc_data">9.5. FADC reconstruction</a></li>
<li>  <a href="./reconstruction.html#sec:reco:scintillator_data">9.6. Scintillator data</a></li>
</ul>
</li>
<li><a href="./cast.html#sec:cast">10. Detector installation &amp; data taking at CAST   <span class="tag">  <span class="CAST">CAST</span></span></a>
<ul>
<li>  <a href="./cast.html#sec:cast:timeline">10.1. Timeline</a></li>
<li>  <a href="./cast.html#sec:cast:alignment">10.2. Alignment</a></li>
<li>  <a href="./cast.html#sec:cast:detector_setup">10.3. Detector setup at CAST</a></li>
<li>  <a href="./cast.html#sec:cast:window_accident">10.4. Window accident</a></li>
<li>  <a href="./cast.html#sec:cast:data_taking_woes">10.5. Data taking woes</a></li>
<li>  <a href="./cast.html#sec:cast:data_taking_campaigns">10.6. Summary of CAST data taking</a></li>
</ul>
</li>
<li><a href="./calibration.html#sec:calibration">11. Data calibration   <span class="tag">  <span class="Calibration">Calibration</span></span></a>
<ul>
<li>  <a href="./calibration.html#sec:calibration:energy">11.1. Energy calibration - in principle</a></li>
<li>  <a href="./calibration.html#sec:calib:detector_behavior_over_time">11.2. Detector behavior over time</a></li>
<li>  <a href="./calibration.html#sec:calib:final_energy_calibration">11.3. Energy calibration dependence on the gas gain</a></li>
<li>  <a href="./calibration.html#sec:calib:fadc">11.4. FADC</a></li>
</ul>
</li>
<li><a href="./background.html#sec:background">12. Finding signal and defining background | Background rate computation   <span class="tag">  <span class="Analysis">Analysis</span></span></a>
<ul>
<li>  <a href="./background.html#sec:background:likelihood_method">12.1. Likelihood method</a></li>
<li>  <a href="./background.html#sec:cdl">12.2. CAST Detector Lab</a></li>
<li>  <a href="./background.html#sec:background:likelihood_cut">12.3. Application of likelihood cut for background rate</a></li>
<li>  <a href="./background.html#sec:background:mlp">12.4. Artificial neural networks as cluster classifiers</a></li>
<li>  <a href="./background.html#sec:background:additional_vetoes">12.5. Additional detector features as vetoes</a></li>
<li>  <a href="./background.html#sec:background:all_vetoes_combined">12.6. Background rates of combined vetoes and efficiencies</a></li>
</ul>
</li>
<li><a href="./limit.html#sec:limit">13. Limit calculation   <span class="tag">  <span class="Limit">Limit</span></span></a>
<ul>
<li>  <a href="./limit.html#sec:limit:method_introduction">13.1. Limit method - introduction</a></li>
<li>  <a href="./limit.html#sec:limit:method_likelihood">13.2. Limit method - likelihood function \(\mathcal{L}\)</a></li>
<li>  <a href="./limit.html#sec:limit:method_computing_L">13.3. Limit method - computing \(\mathcal{L}\)</a></li>
<li>  <a href="./limit.html#sec:limit:method_computing_a_limit">13.4. Limit method - computing a limit</a></li>
<li>  <a href="./limit.html#sec:limit:method_expected_limit">13.5. Limit method - toy candidate sets and expected limits</a></li>
<li>  <a href="./limit.html#sec:limit:method_systematics">13.6. Limit method - extending \(\mathcal{L}\) for systematics</a></li>
<li>  <a href="./limit.html#sec:limit:method_mcmc">13.7. Limit method - evaluating \(\mathcal{L}\) with nuisance parameters</a></li>
<li>  <a href="./limit.html#org5995639">13.8. Note about likelihood integral   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./limit.html#orgc3e5f13">13.9. Derivation of short form of \(\mathcal{L}\) <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./limit.html#sec:limit:ingredients">13.10. Likelihood ingredients in detail</a></li>
<li>  <a href="./limit.html#sec:limit:systematics">13.11. Systematics</a></li>
<li>  <a href="./limit.html#sec:limit:mcmc_calc_limit">13.12. MCMC to sample the distribution and compute a limit</a></li>
<li>  <a href="./limit.html#sec:limit:expected_limits">13.13. Expected limits of different setups</a></li>
<li>  <a href="./limit.html#sec:limit:candidates">13.14. Solar tracking candidates</a></li>
<li>  <a href="./limit.html#sec:limit:observed_limit">13.15. Observed limit - \(g_{ae}\)</a></li>
<li>  <a href="./limit.html#sec:limit:other_couplings">13.16. Other coupling constants</a></li>
<li>  <a href="./limit.html#org75d3b34">13.17. Comparison to 2013 limit (using their method)   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./limit.html#orgfcbf364">13.18. Observed limit for different axion masses   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
</ul>
</li>
<li>  <a href="./outlook.html#sec:outlook">14. Outlook</a></li>
<li>  <a href="./summary.html#sec:summary">15. Summary &amp; conclusion</a></li>
<li>  <a href="./bibliography.html#sec:bibliography">16. Bibliography   <span class="tag">    <span class="html">html</span>  </span></a></li>
<li><a href="./daq.html#sec:daq">17. Data acquisition and detector monitoring   <span class="tag"><span class="Appendix">Appendix</span> <span class="Detector">Detector</span></span></a>
<ul>
<li>  <a href="./daq.html#sec:daq:tof">17.1. Timepix Operating Firmware - TOF</a></li>
<li>  <a href="./daq.html#sec:daq:tos">17.2. Timepix Operating Software - TOS</a></li>
<li>  <a href="./daq.html#sec:daq:septemboard_event_display">17.3. Septemboard event display</a></li>
</ul>
</li>
<li><a href="./appendix_configuration.html#sec:appendix:configuration">18. Configuration and TOS / TOF versions   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_configuration.html#sec:appendix:configuration:tos_config">18.1. TOS configuration file</a></li>
<li>  <a href="./appendix_configuration.html#sec:appendix:configuration:tos_tof_versions">18.2. TOS and TOF versions used at CAST</a></li>
</ul>
</li>
<li><a href="./appendix_calibration.html#sec:appendix:calibration">19. Calibrations   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_calibration.html#sec:appendix:calibration:timepix">19.1. Timepix calibrations</a></li>
<li>  <a href="./appendix_calibration.html#sec:appendix:septemboard_calibrations">19.2. Septemboard calibration</a></li>
<li>  <a href="./appendix_calibration.html#sec:appendix:scintillator_calibration_notes">19.3. Calibration measurements of the veto scintillator paddle   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
</ul>
</li>
<li><a href="./appendix_cast_operations.html#sec:appendix:cast_operations">20. CAST operation procedures   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_cast_operations.html#sec:appendix:cast_operations:terminology">20.1. CAST terminology</a></li>
<li>  <a href="./appendix_cast_operations.html#sec:cast:high_voltage">20.2. High voltage supply</a></li>
<li>  <a href="./appendix_cast_operations.html#sec:cast:vacuum_system">20.3. Vacuum system</a></li>
<li>  <a href="./appendix_cast_operations.html#sec:cast:watercooling_gas">20.4. Watercooling system &amp; gas supply</a></li>
<li>  <a href="./appendix_cast_operations.html#sec:cast:interlock_systems">20.5. Interlock systems</a></li>
<li>  <a href="./appendix_cast_operations.html#sec:appendix:cast_log_files">20.6. CAST log files</a></li>
</ul>
</li>
<li><a href="./appendix_cast_run_list.html#sec:appendix:cast_run_list">21. CAST data taking run list   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_cast_run_list.html#orgb52235a">21.1. Full version   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
</ul>
</li>
<li><a href="./appendix_cast_data_taking_notes.html#sec:appendix:cast_data_taking_notes">22. CAST data taking notes <code>[0/1]</code>   <span class="tag"><span class="Appendix">Appendix</span> <span class="extended">extended</span></span></a>
<ul>
<li>  <a href="./appendix_cast_data_taking_notes.html#orgd947e01">22.1. Run table</a></li>
<li>  <a href="./appendix_cast_data_taking_notes.html#orgf8afac6">22.2. Data runs</a></li>
<li>  <a href="./appendix_cast_data_taking_notes.html#orgfe96ddc">22.3. Calibration runs</a></li>
<li>  <a href="./appendix_cast_data_taking_notes.html#orgac5a852">22.4. Automatically generated run list</a></li>
<li>  <a href="./appendix_cast_data_taking_notes.html#orgd5cdef1">22.5. Automatically calculated total run times</a></li>
<li>  <a href="./appendix_cast_data_taking_notes.html#org2406344">22.6. InGrid temperature from shift forms</a></li>
</ul>
</li>
<li><a href="./appendix_cabling_and_softwar_setup.html#sec:appendix:cabling_and_softwar_setup">23. Cabling &amp; software setup <code>[/]</code>   <span class="tag"><span class="Appendix">Appendix</span> <span class="extended">extended</span></span></a>
<ul>
<li>  <a href="./appendix_cabling_and_softwar_setup.html#orge32a9e1">23.1. Virtex V6 cabling</a></li>
<li>  <a href="./appendix_cabling_and_softwar_setup.html#orge17b088">23.2. Detector cabling</a></li>
<li>  <a href="./appendix_cabling_and_softwar_setup.html#org93c1c60">23.3. Vivado / ISE on void linux &amp; flashing Virtex V6</a></li>
<li>  <a href="./appendix_cabling_and_softwar_setup.html#org9625327">23.4. Setting up the chips in TOS</a></li>
</ul>
</li>
<li><a href="./appendix_vacuum_contamination.html#sec:appendix:vacuum_contamination">24. Window rupture and vacuum contamination   <span class="tag"><span class="Appendix">Appendix</span> <span class="extended">extended</span></span></a>
<ul>
<li>  <a href="./appendix_vacuum_contamination.html#org3cd49bd">24.1. Calculation of vacuum volume</a></li>
<li>  <a href="./appendix_vacuum_contamination.html#orgc39e76f">24.2. Calculation of potential influx of gas</a></li>
<li>  <a href="./appendix_vacuum_contamination.html#orgc67dc29">24.3. Consider pumping of pumps</a></li>
<li>  <a href="./appendix_vacuum_contamination.html#orgbac7f48">24.4. Calculation of possible contamination</a></li>
</ul>
</li>
<li><a href="./appendix_detector_time_behavior.html#sec:appendix:detector_time_behavior">25. Detector behavior over time   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_detector_time_behavior.html#sec:appendix:choice_gas_gain_binning">25.1. Choice of gas gain binning time interval</a></li>
<li>  <a href="./appendix_detector_time_behavior.html#sec:appendix:correlation_gas_gain_ambient_temp">25.2. Correlation of gas gain and ambient CAST temperature</a></li>
</ul>
</li>
<li><a href="./appendix_cast_detector_lab.html#sec:appendix:cast_detector_lab">26. CAST Detector Lab data   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_cast_detector_lab.html#orgc899fb3">26.1. Generate all spectrum plots split by run   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_cast_detector_lab.html#sec:appendix:cdl_spectra_by_run">26.2. All spectra split by run</a></li>
<li>  <a href="./appendix_cast_detector_lab.html#sec:appendix:cdl:all_spectra_fits">26.3. All CDL spectra with line fits   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_cast_detector_lab.html#sec:appendix:cdl:all_spectra_fits_by_run">26.4. All CDL spectra with line fits by run</a></li>
</ul>
</li>
<li><a href="./appendix_fit_by_run_justification.html#sec:appendix:fit_by_run_justification">27. CAST Detector Lab variations and fitting by run   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_fit_by_run_justification.html#sec:appendix:fit_by_run:gas_gain_var_cluster_prop">27.1. Influence of gas gain variations on cluster properties</a></li>
<li>  <a href="./appendix_fit_by_run_justification.html#org13d2559">27.2. Data overview with pixel spectra <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
</ul>
</li>
<li><a href="./appendix_morphing_cdl_spectra.html#sec:appendix:morphing_cdl_spectra">28. Morphing of CDL reference spectra   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_morphing_cdl_spectra.html#org78be87a">28.1. Generate all morphing / tile related plots   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_morphing_cdl_spectra.html#orgdadd742">28.2. Generate plot comparing likelihood behavior   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_morphing_cdl_spectra.html#sec:appendix:morphing_cdl_spectra:tilemaps">28.3. Tilemap of each likelihood dataset</a></li>
<li>  <a href="./appendix_morphing_cdl_spectra.html#sec:appendix:morphing_cdl_spectra:interpolation_raster">28.4. Interpolation of each likelihood dataset</a></li>
<li>  <a href="./appendix_morphing_cdl_spectra.html#sec:appendix:morphing_cdl_spectra:binwise_linear">28.5. Binwise linear interpolations for each likelihood dataset</a></li>
<li>  <a href="./appendix_morphing_cdl_spectra.html#sec:appendix:morphing_cdl_spectra:notes">28.6. Notes on CDL morphing development   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
</ul>
</li>
<li><a href="./appendix_occupancy.html#sec:appendix:occupancy">29. Occupancy maps   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_occupancy.html#org71d0d19">29.1. Run-2</a></li>
<li>  <a href="./appendix_occupancy.html#orgcbfcb5e">29.2. Run-3</a></li>
<li>  <a href="./appendix_occupancy.html#org3a727ee">29.3. Generate occupancy map plots   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
</ul>
</li>
<li><a href="./appendix_fadc.html#sec:appendix:fadc">30. FADC   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_fadc.html#sec:appendix:fadc:rise_fall">30.1. FADC rise and fall time</a></li>
<li>  <a href="./appendix_fadc.html#sec:appendix:background:fadc">30.2. FADC veto</a></li>
<li>  <a href="./appendix_fadc.html#org595d68a">30.3. Generate plot of rise time vs skewness   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_fadc.html#sec:appendix:fadc_veto_empirical_cluster_length">30.4. Expected cluster size</a></li>
</ul>
</li>
<li><a href="./appendix_background_rates.html#sec:appendix:background_rates">31. Raw data and background rates   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_background_rates.html#sec:appendix:background_rates:full_chip">31.1. Background rates over full chip</a></li>
<li>  <a href="./appendix_background_rates.html#org6b8736f">31.2. Generate rate without any vetoes over full chip   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_background_rates.html#org7827cf1">31.3. Generate background rates over full chip   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_background_rates.html#org88d5321">31.4. Generate table of background rates for all setups   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
</ul>
</li>
<li>  <a href="./appendix_background_interpolation_chip_area.html#sec:appendix:background_interpolation_chip_area">32. Background interpolation chip cutout correction   <span class="tag">    <span class="Appendix">Appendix</span>  </span></a></li>
<li><a href="./appendix_limit_additional.html#sec:appendix:limit_additional">33. Additional limit information   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_limit_additional.html#sec:appendix:conversion_probability">33.1. Conversion probability as a function of mass   <span class="tag">    <span class="Appendix">Appendix</span>  </span></a></li>
<li>  <a href="./appendix_limit_additional.html#sec:appendix:exp_limit_percentiles">33.2. Expected limit table with percentiles</a></li>
<li>  <a href="./appendix_limit_additional.html#sec:appendix:limit_additional:axion_photon">33.3. Observed limit - axion photon \(g_{aγ}\)</a></li>
<li>  <a href="./appendix_limit_additional.html#sec:appendix:limit_additional:chameleon">33.4. Observed limit - chameleon \(β_γ\)</a></li>
</ul>
</li>
<li><a href="./appendix_software.html#sec:appendix:software">34. Software   <span class="tag"><span class="Software">Software</span> <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_software.html#orge51f988">34.1. Why did I start writing my own analysis framework?   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_software.html#org378ccdf">34.2. Nim   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_software.html#sec:appendix:timepix_analysis">34.3. TimepixAnalysis</a></li>
<li>  <a href="./appendix_software.html#org92b1f48">34.4. Other libraries relevant for TimepixAnalysis</a></li>
</ul>
</li>
<li><a href="./appendix_full_data_reconstruction.html#sec:appendix:full_data_reconstruction">35. Full data reconstruction   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_full_data_reconstruction.html#orgc87960b">35.1. Raw data parsing and reconstruction</a></li>
<li>  <a href="./appendix_full_data_reconstruction.html#org652a080">35.2. Parse and reconstruct the CDL data</a></li>
<li>  <a href="./appendix_full_data_reconstruction.html#org5831301">35.3. Add tracking information to background files</a></li>
<li>  <a href="./appendix_full_data_reconstruction.html#org904d73c">35.4. Using <code>runAnalysisChain</code></a></li>
<li>  <a href="./appendix_full_data_reconstruction.html#orgb4de31c">35.5. Applying a classifier</a></li>
<li>  <a href="./appendix_full_data_reconstruction.html#org9252218">35.6. Computing limits</a></li>
</ul>
</li>
<li><a href="./appendix_average_depth_xrays_argon.html#sec:appendix:average_depth_xrays_argon">36. Average distance X-rays travel in argon at CAST conditions   <span class="tag">  <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_average_depth_xrays_argon.html#org89ae1e7">36.1. Reference to original document   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_average_depth_xrays_argon.html#orgc035fda">36.2. Calculate conversion point numerically</a></li>
<li>  <a href="./appendix_average_depth_xrays_argon.html#orgc101b38">36.3. Compiling and running the code   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
</ul>
</li>
<li><a href="./appendix_raytracing.html#sec:appendix:raytracing">37. Raytracing   <span class="tag"><span class="Software">Software</span> <span class="Appendix">Appendix</span></span></a>
<ul>
<li>  <a href="./appendix_raytracing.html#sec:appendix:raytracing:traxer">37.1. TrAXer - An interactive axion raytracer</a></li>
<li>  <a href="./appendix_raytracing.html#sec:appendix:raytracing:llnl_telescope">37.2. A few more details about the LLNL telescope</a></li>
<li>  <a href="./appendix_raytracing.html#sec:appendix:raytracing:panter">37.3. Comparison of TrAXer results with PANTER measurements</a></li>
<li>  <a href="./appendix_raytracing.html#sec:appendix:raytracing:axion_image">37.4. Computing an axion image with TrAXer</a></li>
<li>  <a href="./appendix_raytracing.html#orgdb277d0">37.5. Reproducing an X-ray finger run with TrAXer   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_raytracing.html#org33f56e2">37.6. <span class="done DONE">DONE</span> Can we finish our interactive ray tracer?   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_raytracing.html#org66d4b31">37.7. Rerunning Al Kα after replacing target   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
<li>  <a href="./appendix_raytracing.html#orgab3dcf7">37.8. Figure error development notes   <span class="tag">    <span class="extended">extended</span>  </span></a></li>
</ul>
</li>
<li>  <a href="./appendix_acknowledgments.html#sec:appendix:acknowledgments">38. Acknowledgments   <span class="tag">    <span class="Ack">Ack</span>  </span></a></li>
</ul>
</div>
</nav>
<div id="outline-container-sec:background" class="outline-2">
<h2 id="sec:background">  <a href="#sec:background">  <span class="section-number-2">12.</span> Finding signal and defining background | Background rate computation   <span class="tag">    <span class="Analysis">Analysis</span>  </span></a></h2>
<div id="text-sec:background" class="outline-text-2">
<p>
With the CAST data fully reconstructed and energy calibrated, it is
time to define the methods used to extract axion candidates and derive
a background rate from the data. We first introduce the likelihood cut
method in sec. <a href="./background.html#sec:background:likelihood_method">12.1</a> to motivate the need
for reference X-ray data from an X-ray tube. Such data, taken at the
CAST Detector Lab (CDL) will be discussed in depth in
sec. <a href="./background.html#sec:cdl">12.2</a>. We then see how the reference data is used in the
likelihood method to act as an event classifier in
sec. <a href="./background.html#sec:background:likelihood_cut">12.3</a>. As an alternative to the
likelihood cut, we will introduce another classifier in the form of a
simple artificial neural network in sec. <a href="./background.html#sec:background:mlp">12.4</a>. This is
another in depth discussion, as the selection of the training data and
verification is non-trivial. With both classifiers discussed, it is
time to include all other Septemboard detector features as additional
vetoes in sec. <a href="./background.html#sec:background:additional_vetoes">12.5</a>. At the very end we
will look at background rates for different cases,
sec. <a href="./background.html#sec:background:all_vetoes_combined">12.6</a>, motivating the approach of
our limit calculation in the next chapter.
</p>

<blockquote>
<p>
Note: The methods discussed in this chapter are generally classifiers that
predict how &apos;signal-like&apos; a cluster is. Based on this prediction we
will usually define a cut value as to keep a cluster as a potential
signal. This means that if we apply the method to background data
(that is, CAST data taken outside of solar trackings) we recover the
&apos;background rate&apos;; the irreducible amount of background left (at a
certain efficiency), which is signal-like. If instead we apply the
same methods to CAST solar tracking data we get instead a set of
&apos;axion induced X-ray candidates&apos;. <sup>  <a id="fnr.1" href="#fn.1" class="footref" role="doc-backlink">1</a></sup>
In the context of the chapter we commonly talk about &quot;background
rates&quot;, but the equivalent meaning in terms of tracking data and
candidates should be kept in mind.
</p>
</blockquote>
</div>
<div id="outline-container-sec:background:likelihood_method" class="outline-3">
<h3 id="sec:background:likelihood_method">  <a href="#sec:background:likelihood_method">  <span class="section-number-3">12.1.</span> Likelihood method</a></h3>
<div id="text-sec:background:likelihood_method" class="outline-text-3">
<p>
The detection principle of the GridPix detector implies physically
different kinds of events will have different geometrical shapes. An
example can be seen in
fig. <a href="#fig:background:eccentricity_signal_background">1(a)</a>, comparing
the cluster eccentricity of \cefe calibration events with background
data. This motivates usage of a geometry based approach to determine
likely signal-like or background-like clusters. The method to
distinguish the two types of events is a likelihood cut, based on the
one in (<a href="./bibliography.html#citeproc_bib_item_65">Christoph Krieger 2018</a>). It effectively assigns a single value
to each cluster for the likelihood that it is a signal-like event. 
</p>

<p>
Specifically, this likelihood method is based on three different
geometric properties (also see sec. <a href="./reconstruction.html#sec:reco:cluster_geometry">9.4.2</a>):  
</p>
<ol class="org-ol">
<li>the eccentricity \(ε\) of the cluster, determined by
computing the long and short axis of the two dimensional cluster and
then computing the ratio of the RMS of the projected positions of
all active pixels within the cluster along each axis.</li>
<li>the fraction of all pixels within a circle of the radius of one
transverse RMS from the cluster center, \(f\).</li>
<li>the length of the cluster (full extension along the long axis)
divided by the transverse RMS, \(l\).</li>
</ol>

<p>
These variables are obviously highly correlated, but still provide a
very good separation between the typical shapes of X-rays and
background events. They mainly characterize the &quot;spherical-ness&quot; as
well as the density near the center of the cluster, which is precisely
the intuitive sense in which these type of events differ. For each of
these properties we define a probability density function
\(\mathcal{P}_i\), which can then be used to define the likelihood of a
cluster with properties \((ε, f, l)\) to be signal-like:
</p>

\begin{equation}
\label{eq:background:likelihood_def}
\mathcal{L}(ε, f, l) = \mathcal{P}_{ε}(ε) \cdot \mathcal{P}_{f}(f)
\cdot \mathcal{P}_l(l)
\end{equation}

<p>
where the subscript is denoting the individual probability density 
and the argument corresponds to the individual value of each
property.
</p>

<p>
This raises the important question of what defines each individual
probability density \(\mathcal{P}_i\). In principle it can be
defined by computing a normalized density distribution of a known
dataset, which contains representative signal-like data. The \cefe
calibration data from CAST contains such representative data, if not
for one problem: the properties used in the likelihood method are
energy dependent, as seen in
fig. <a href="#fig:background:eccentricity_photo_escape">1(b)</a>, a comparison of
the eccentricity of X-rays from the photopeak of the \cefe calibration
source compared to those from the escape peak. The CAST calibration
data can only characterize two different energies, but the expected
axion signal is a (model dependent) continuous spectrum. 
</p>

<p>
For this reason data was taken using an X-ray tube with 8 different
target / filter combinations to provide the needed data to compute
likelihood distributions for X-rays of a range of different
energies. The details will be discussed in the next section,
<a href="./background.html#sec:cdl">12.2</a>. 
</p>


<figure id="fig:background:properties_energy_type_dependent" class="figure-wrapper">
<figure id="fig:background:eccentricity_signal_background" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/background/eccentricity_calibration_background.svg" data-width="99%" />  <figcaption>Figure 1(a): Calibration \&amp; background</figcaption></figure> <figure id="fig:background:eccentricity_photo_escape" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/background/eccentricity_photo_escape_peak.svg" data-width="99%" />  <figcaption>Figure 1(b): Photopeak \&amp; escape peak</figcaption></figure>
<figcaption>Figure 1: <a href="#fig:background:eccentricity_signal_background">1(a)</a>Comparison of the eccentricity of clusters from a calibration and a background run.
           Background events are clearly much more eccentric on average. <a href="#fig:background:eccentricity_photo_escape">1(b)</a>The X-ray properties are energy dependent as can be seen in the comparison between
           the eccentricity of X-rays from the photopeak compared to those from the escape
           peak. A kernel density estimation is used in both figures.</figcaption>
</figure>
</div>
<div id="outline-container-org0011a4b" class="outline-4">
<h4 id="org0011a4b">  <a href="#org0011a4b">  <span class="section-number-4">12.1.1.</span> Generate plot of eccentricity signal vs. background   <span class="tag">    <span class="extended">extended</span>  </span></a></h4>
<div id="text-12-1-1" class="outline-text-4">
<p>
We will now generate a plot comparing the eccentricity of signal
events from calibration data to background events. In addition another
comparison will be made between photopeak photons and escape peak
photons to show that events at different energies have different
shapes, motivating the need for the X-ray tube data at different energies.
</p>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> ggplotnim, nimhdf5
<span style="color: #F92672;">import</span> ingrid / <span style="color: #AE81FF;">[</span>tos_helpers, ingrid_types<span style="color: #AE81FF;">]</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">read</span><span style="color: #AE81FF;">(</span>f: <span style="color: #66D9EF;">string</span>, run: <span style="color: #66D9EF;">int</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">DataFrame</span> =
  withH5<span style="color: #AE81FF;">(</span>f, <span style="color: #E6DB74;">&quot;r&quot;</span><span style="color: #AE81FF;">)</span>:
    <span style="color: #FD971F;">result</span> = h5f.readRunDsets<span style="color: #AE81FF;">(</span>
      run = run,
      chipDsets = some<span style="color: #66D9EF;">(</span><span style="color: #A6E22E;">(</span>chip: <span style="font-style: italic;">3</span>, dsets: @<span style="color: #E6DB74;">[</span><span style="color: #E6DB74;">&quot;eccentricity&quot;</span>, <span style="color: #E6DB74;">&quot;centerX&quot;</span>, <span style="color: #E6DB74;">&quot;centerY&quot;</span>, <span style="color: #E6DB74;">&quot;energyFromCharge&quot;</span><span style="color: #E6DB74;">]</span><span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">)</span>
    <span style="color: #AE81FF;">)</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">main</span><span style="color: #AE81FF;">(</span>calib, back: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span> =

  <span style="color: #75715E;"># </span><span style="color: #75715E;">read data from each file, one fixed run with good statistics</span>
  <span style="color: #75715E;"># </span><span style="color: #75715E;">cut on silver region</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfC </span>= read<span style="color: #AE81FF;">(</span>calib, <span style="font-style: italic;">128</span><span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfB </span>= read<span style="color: #AE81FF;">(</span>back, <span style="font-style: italic;">124</span><span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">df </span>= bind_rows<span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">[</span><span style="color: #A6E22E;">(</span><span style="color: #E6DB74;">&quot;Calibration&quot;</span>, dfC<span style="color: #A6E22E;">)</span>, <span style="color: #A6E22E;">(</span><span style="color: #E6DB74;">&quot;Background&quot;</span>, dfB<span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">]</span>, <span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #AE81FF;">)</span>
    .filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">float</span> -&gt; <span style="color: #66D9EF;">bool</span>: inRegion<span style="color: #A6E22E;">(</span>`centerX`, `centerY`, crSilver<span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">}</span>,
            f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">float</span>: `eccentricity` &lt; <span style="font-style: italic;">10</span>.<span style="font-style: italic;">0</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
  ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;eccentricity&quot;</span>, fill = <span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
    <span style="color: #75715E;">#</span><span style="color: #75715E;">geom_histogram(bins = 100,</span>
    <span style="color: #75715E;">#               </span><span style="color: #75715E;">hdKind = hdOutline, </span>
    <span style="color: #75715E;">#               </span><span style="color: #75715E;">position = &quot;identity&quot;,</span>
    <span style="color: #75715E;">#               </span><span style="color: #75715E;">alpha = 0.5,</span>
    <span style="color: #75715E;">#               </span><span style="color: #75715E;">density = true) +</span>
    xlab<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Eccentricity&quot;</span><span style="color: #AE81FF;">)</span> + ylab<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Density&quot;</span><span style="color: #AE81FF;">)</span> +   
    geom_density<span style="color: #AE81FF;">(</span>color = <span style="color: #E6DB74;">&quot;black&quot;</span>, size = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span>, alpha = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">7</span>, normalize = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span> + 
    ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Eccentricity of calibration and background data&quot;</span><span style="color: #AE81FF;">)</span> +
    themeLatex<span style="color: #AE81FF;">(</span>fWidth = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">5</span>, width = <span style="font-style: italic;">600</span>, baseTheme = sideBySide<span style="color: #AE81FF;">)</span> +
    margin<span style="color: #AE81FF;">(</span>left = <span style="font-style: italic;">2</span>.<span style="font-style: italic;">2</span>, right = <span style="font-style: italic;">5</span>.<span style="font-style: italic;">2</span><span style="color: #AE81FF;">)</span> +         
    ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Figs/background/eccentricity_calibration_background.pdf&quot;</span>, useTeX = <span style="color: #AE81FF;">true</span>, standalone = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>

  <span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">splitPeaks</span><span style="color: #AE81FF;">(</span>x: <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> =
    <span style="color: #F92672;">if</span> x &gt;= <span style="font-style: italic;">2</span>.<span style="font-style: italic;">75</span> <span style="color: #F92672;">and</span> x &lt;= <span style="font-style: italic;">3</span>.<span style="font-style: italic;">25</span>:
      <span style="color: #E6DB74;">&quot;Escapepeak&quot;</span>
    <span style="color: #F92672;">elif</span> x &gt;= <span style="font-style: italic;">5</span>.<span style="font-style: italic;">55</span> <span style="color: #F92672;">and</span> x &lt;= <span style="font-style: italic;">6</span>.<span style="font-style: italic;">25</span>:
      <span style="color: #E6DB74;">&quot;Photopeak&quot;</span>
    <span style="color: #F92672;">else</span>:
      <span style="color: #E6DB74;">&quot;Unclear&quot;</span>

  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfP </span>= dfC
      .mutate<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">float</span>: <span style="color: #E6DB74;">&quot;Peak&quot;</span> ~ splitPeaks<span style="color: #A6E22E;">(</span>`energyFromCharge`<span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
      .filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">string</span>: `<span style="color: #66D9EF;">Peak</span>` != <span style="color: #E6DB74;">&quot;Unclear&quot;</span><span style="color: #66D9EF;">}</span>,
              f<span style="color: #66D9EF;">{</span>`eccentricity` &lt;= <span style="font-style: italic;">2</span>.<span style="font-style: italic;">0</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
  ggplot<span style="color: #AE81FF;">(</span>dfP, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;eccentricity&quot;</span>, fill = <span style="color: #E6DB74;">&quot;Peak&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
    <span style="color: #75715E;">#</span><span style="color: #75715E;">geom_histogram(bins = 50,</span>
    <span style="color: #75715E;">#               </span><span style="color: #75715E;">hdKind = hdOutline, </span>
    <span style="color: #75715E;">#               </span><span style="color: #75715E;">position = &quot;identity&quot;,</span>
    <span style="color: #75715E;">#               </span><span style="color: #75715E;">alpha = 0.5,</span>
    <span style="color: #75715E;">#               </span><span style="color: #75715E;">density = true) +</span>
    xlab<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Eccentricity&quot;</span><span style="color: #AE81FF;">)</span> + ylab<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Density&quot;</span><span style="color: #AE81FF;">)</span> + 
    geom_density<span style="color: #AE81FF;">(</span>color = <span style="color: #E6DB74;">&quot;black&quot;</span>, size = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span>, alpha = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">7</span>, normalize = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span> +
    ggtitle<span style="color: #AE81FF;">(</span>r<span style="color: #E6DB74;">&quot;$^{55}\text{Fe}$ photopeak (5.9 keV) and escapepeak (3 keV)&quot;</span><span style="color: #AE81FF;">)</span> +
    themeLatex<span style="color: #AE81FF;">(</span>fWidth = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">5</span>, width = <span style="font-style: italic;">600</span>, baseTheme = sideBySide<span style="color: #AE81FF;">)</span> +
    margin<span style="color: #AE81FF;">(</span>left = <span style="font-style: italic;">2</span>.<span style="font-style: italic;">2</span>, right = <span style="font-style: italic;">5</span>.<span style="font-style: italic;">2</span><span style="color: #AE81FF;">)</span> +     
    ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Figs/background/eccentricity_photo_escape_peak.pdf&quot;</span>, useTeX = <span style="color: #AE81FF;">true</span>, standalone = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>
    
<span style="color: #F92672;">when</span> <span style="color: #AE81FF;">isMainModule</span>:
  <span style="color: #F92672;">import</span> cligen
  dispatch main
</pre>
</div>

<p>
yielding
<img alt="eccentricity_calibration_background.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/eccentricity_calibration_background.svg" />
and
<img alt="eccentricity_photo_escape_peak.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/eccentricity_photo_escape_peak.svg" />
</p>

<div class="org-src-container">
<pre class="src src-sh">
</pre>
</div>
</div>
</div>
</div>
<div id="outline-container-sec:cdl" class="outline-3">
<h3 id="sec:cdl">  <a href="#sec:cdl">  <span class="section-number-3">12.2.</span> CAST Detector Lab</a></h3>
<div id="text-sec:cdl" class="outline-text-3">
<p>
In this section we will cover the X-ray tube data taken at the CAST
Detector Lab (CDL) at CERN. First, we will show the setup in
sec. <a href="./background.html#sec:cdl:setup">12.2.1</a>. Next the different target / filter combinations
that were used will be discussed and the measurements presented in
sec. <a href="./background.html#sec:cdl:measurements">12.2.2</a>. These measurements then are used to define
the probability densities for the likelihood method, see
sec. <a href="./background.html#sec:cdl:derive_probability_density">12.2.5</a>. Further, in
sec. <a href="./background.html#sec:cdl:cdl_morphing">12.2.6</a>, we cover a few more details on a linear
interpolation we perform to compute a likelihood distribution at an
arbitrary energy. And finally, they can be used to measure the energy
resolution of the detector at different energies,
sec. <a href="./background.html#sec:cdl:energy_resolution">12.2.9</a>.
</p>

<p>
The data presented here was also part of the master thesis of Hendrik
Schmick (<a href="./bibliography.html#citeproc_bib_item_193">Schmick 2019</a>). Further, the selection of lines and
approach follows the ideas used for the single GridPix detector in
(<a href="./bibliography.html#citeproc_bib_item_65">Christoph Krieger 2018</a>) with some notable differences. For a reasoning
for one particular difference in terms of data treatment, see appendix
<a href="./appendix_fit_by_run_justification.html#sec:appendix:fit_by_run_justification">27</a>.
</p>

<blockquote>
<p>
Note: in the course of the CDL related sections the term
target/filter combination (implicitly including the used high voltage
setting) is used interchangeably with the main fluorescence line (or
just &apos;the fluorescence line&apos;) targeted in a specific measurement. Be
careful while reading about applied high voltages in \(\si{kV}\) and
energies in \(\si{keV}\). The produced fluorescence lines typically have
about &apos;half&apos; the energy in \(\si{keV}\) as the applied voltage in
\(\si{kV}\). See tab. <a href="#tab:cdl:run_overview_tab">19</a> in
sec. <a href="./background.html#sec:cdl:measurements">12.2.2</a> for the precise relation.
</p>
</blockquote>
</div>
<div id="outline-container-sec:cdl:setup" class="outline-4">
<h4 id="sec:cdl:setup">  <a href="#sec:cdl:setup">  <span class="section-number-4">12.2.1.</span> CDL setup</a></h4>
<div id="text-sec:cdl:setup" class="outline-text-4">
<p>
The CAST detector lab provides a vacuum test stand, which contains an
X-ray tube. A Micromegas-like detector can easily be mounted to the
rear end of the test stand. An X-ray tube uses a filament to produce
free electrons, which are then accelerated with a high voltage of the
order of multiple \(\si{kV}\). The main part of the setup is a
rotateable wheel inside the vacuum chamber, which contains a set of 18
different positions with 8 different target materials, as seen in
tab. <a href="#tab:cdl:targets">17</a>. The highly energetic electrons interacting with
the target material generate a continuous Bremsstrahlung spectrum with
characteristic lines depending on the target. A second rotateable
wheel contains a set of 11 different filters, see
tab. <a href="#tab:cdl:filters">18</a>. These can be used to filter out undesired parts
of the generated spectrum by choosing a filter that is opaque in the
those energy ranges.
</p>

<p>
As mentioned previously in sec. <a href="./cast.html#sec:cast:timeline">10.1</a>, the detector was
dismounted from CAST in Feb. 2019 and installed in the CAST detector
lab on <span class="timestamp-wrapper">  <span class="timestamp">&lt;2019-02-14 Thu&gt;</span></span>. The week from <span class="timestamp-wrapper">  <span class="timestamp">&lt;2019-02-15 Fri&gt; </span></span> to
<span class="timestamp-wrapper">  <span class="timestamp">&lt;2019-02-21 Thu&gt; </span></span> X-ray tube data was taken in the CDL.
</p>

<p>
Fig. <a href="#fig:cdl:cdl_setup">2(a)</a> shows the whole vacuum test stand, which
contains the X-ray tube on the front left end and the Septemboard
detector installed on the rear right, visible by the red HV cables and
yellow HDMI cable. In fig. <a href="#fig:cdl:detector_setup">2(b)</a> we see the
whole detector part from above, with the Septemboard detector
installed to the vacuum test stand on the left side. The water cooling
system is seen in the bottom right, with the power supply above
that. The copper shielded box slightly right of the center is the
Ortec pre-amplifier of the FADC, which is connected via a heavily
shielded LEMO cable to the Septemboard detector. This cable was
available in the CAST detector lab and proved invaluable for the
measurements as it essentially removed any noise visible in the FADC
signals (compared to the significant noise issues encountered at CAST,
sec. <a href="./cast.html#sec:cast:data_taking_woes_2017">10.5.1</a>). Given the activation threshold
of the FADC with CAST amplifier settings (see
sec. <a href="./calibration.html#sec:calib:fadc:amplifier_settings">11.4.2</a>) at around \(\SI{2}{keV}\), the
amplification needed to be adjusted in the CDL on a per-target
basis. The shielded cable allowed the FADC to even act as a trigger
for \(\SI{277}{eV}\) \(\ce{C}_{Kα}\) X-rays without any noise problems. <sup>  <a id="fnr.2" href="#fn.2" class="footref" role="doc-backlink">2</a></sup>
</p>


<figure id="fig:cdl:cdl" class="figure-wrapper">
<figure id="fig:cdl:cdl_setup" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/CDL/IMG_20190214_155459.jpg" data-width="94%" />  <figcaption>Figure 2(a): Full setup</figcaption></figure> <figure id="fig:cdl:detector_setup" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/CDL/IMG_20190214_191752.jpg" data-width="94%" />  <figcaption>Figure 2(b): Detector setup</figcaption></figure>
<figcaption>Figure 2: <a href="#fig:cdl:cdl_setup">2(a)</a> shows the full vacuum test stand containing the X-ray tube with 
          the Septemboard detector installed at the rear, visible by the red HV and yellow HDMI cables. <a href="#fig:cdl:detector_setup">2(b)</a> is a view of the detector setup from above. On the left hand side
           is the detector mounted to the vacuum setup. The water cooling is seen in the bottom right, connected via
           the blue tubes. The gas supply is in red tubing and the power supply is visible on the right above the
           water cooling (with a green Phoenix connector). The copper shielded cable is a LEMO cable for the FADC
           signal going to the pre-amplifier (with the University of Bonn sticker).</figcaption>
</figure>


<table id="tab:cdl:targets">
<caption class="t-above"><span class="table-number">Table 17:</span> Table of all available target materials in the CAST detector lab and their respective position on a rotateable wheel.</caption>

<colgroup>
<col class="org-left" />

<col class="org-right" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">Target Material</th>
<th scope="col" class="org-right">Position</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">Ti</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Ag</td>
<td class="org-right">2</td>
</tr>

<tr>
<td class="org-left">Mn</td>
<td class="org-right">3</td>
</tr>

<tr>
<td class="org-left">C</td>
<td class="org-right">4</td>
</tr>

<tr>
<td class="org-left">BN</td>
<td class="org-right">5</td>
</tr>

<tr>
<td class="org-left">Au</td>
<td class="org-right">6</td>
</tr>

<tr>
<td class="org-left">Al</td>
<td class="org-right">7 - 11</td>
</tr>

<tr>
<td class="org-left">Cu</td>
<td class="org-right">12 - 18</td>
</tr>
</tbody>
</table>




<table id="tab:cdl:filters">
<caption class="t-above"><span class="table-number">Table 18:</span> Table of all available filters in the CAST detector lab and their respective position on a rotateable wheel.</caption>

<colgroup>
<col class="org-left" />

<col class="org-right" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">Filter material</th>
<th scope="col" class="org-right">Position</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">Ni 0.1 mm</td>
<td class="org-right">1</td>
</tr>

<tr>
<td class="org-left">Al 5 µm</td>
<td class="org-right">2</td>
</tr>

<tr>
<td class="org-left">PP G12 (EPIC)</td>
<td class="org-right">3</td>
</tr>

<tr>
<td class="org-left">PP G12 (EPIC)</td>
<td class="org-right">4</td>
</tr>

<tr>
<td class="org-left">Cu 10 µm</td>
<td class="org-right">5</td>
</tr>

<tr>
<td class="org-left">Ag 5 µm (99.97%) AG000130/24</td>
<td class="org-right">6</td>
</tr>

<tr>
<td class="org-left">Fe 50 µm</td>
<td class="org-right">7</td>
</tr>

<tr>
<td class="org-left">Mn foil 50 µm (98.7% + permanent polyester)</td>
<td class="org-right">8</td>
</tr>

<tr>
<td class="org-left">Co 50 µm (99.99%) CO0000200</td>
<td class="org-right">9</td>
</tr>

<tr>
<td class="org-left">Cr 40 µm (99.99% + permanent polyester)</td>
<td class="org-right">10</td>
</tr>

<tr>
<td class="org-left">Ti 50 µm (99.97%) TI000315</td>
<td class="org-right">11</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="outline-container-sec:cdl:measurements" class="outline-4">
<h4 id="sec:cdl:measurements">  <a href="#sec:cdl:measurements">  <span class="section-number-4">12.2.2.</span> CDL measurements</a></h4>
<div id="text-sec:cdl:measurements" class="outline-text-4">
<p>
The measurements were performed with 8 different target and filter
combinations, with a higher density towards lower energies due to the
nature of the expected solar axion flux. For each target and filter at
least two runs were measured, one <i>with</i> and one <i>without</i> using the FADC
as a trigger. The latter was taken to collect more statistics in a
shorter amount of time, as the FADC readout slows down the data taking
speed due to increased dead time. Table <a href="#tab:cdl:run_overview_tab">19</a>
provides an overview of all data taking runs, the target, filter and
HV setting, the main X-ray fluorescence line targeted and its
energy. Finally, the mean position of the main fluorescence line in
the charge spectrum and its width as determined by a fit is shown. As
can be seen the position moves in some cases significantly between
different runs, for example in case of the \(\ce{Cu}-\ce{Ni}\)
measurements. Also within a single run some stronger variation can be
seen, evident by the much larger line width for example in run
\(\num{320}\) compared to \(\num{319}\). See appendix
sec. <a href="./appendix_cast_detector_lab.html#sec:appendix:cdl:all_spectra_fits_by_run">26.4</a> for all measured
spectra (pixel and charge spectra), where each plot contains all runs
for a target / filter combination. For example
fig. <a href="#fig:appendix:cdl_charge_Cu-Ni-15kV_by_run">#fig:appendix:cdl_charge_Cu-Ni-15kV_by_run</a> shows runs 319, 320 and
345 of the \(\ce{Cu}-\ce{Ni}\) measurements at \(\SI{15}{kV}\) with the
strong variation in charge position of the peak visible between the runs.
</p>

<p>
The variability both between runs for the same target and filter as
well as within a run shows the detector was undergoing gas gain
changes similar to the variations at CAST
(sec. <a href="./calibration.html#sec:calib:detector_behavior_over_time">11.2</a>). With an understanding
that the variation is correlated to the temperature this can be
explained. The small laboratory underwent significant temperature
changes due to the presence of 3 people, all running machines and
freely opening and closing windows, in particular due to very warm
temperatures relative to a typical February in
Geneva. <sup>  <a id="fnr.3" href="#fn.3" class="footref" role="doc-backlink">3</a></sup> Fig. <a href="#fig:cdl:gas_gain_by_cdl_run">3</a> shows the
calculated gas gains (based on \(\SI{90}{min}\) intervals) for each run
colored by the target/filter combination. It undergoes a strong,
almost exponential change over the course of the data taking
campaign. Because of this variation between runs, each run is treated
fully separately in contrast to (<a href="./bibliography.html#citeproc_bib_item_65">Christoph Krieger 2018</a>), where all
runs for one target / filter combination were combined.
</p>


<figure id="fig:cdl:gas_gain_by_cdl_run">
<img alt="gas_gain_by_run_and_tfkind.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/gas_gain_by_run_and_tfkind.svg" />

<figcaption>Figure 3: <span class="figure-number">Figure 52: </span>Gas gain of each gas gain slice of \(\SI{90}{min}\) in all CDL runs. A significant decrease over the runs (equivalent to the week of data taking) is visible. Multiple points for the same run correspond to multiple gas gain time slices as explained in sec. <a href="./calibration.html#sec:calib:gas_gain_time_binning">11.2.3</a>.</figcaption>
</figure>

<p>
Fortunately, this significant change of gas gain does not have an
impact on the distributions of the cluster properties. See appendix
<a href="./appendix_fit_by_run_justification.html#sec:appendix:fit_by_run:gas_gain_var_cluster_prop">27.1</a> for comparisons of
the cluster properties of the different runs (and thus different gas
gains) for each target and filter combination.
</p>

<p>
As the main purpose is to use the CDL data to generate reference
distributions for certain cluster properties, the relevant clusters
that correspond to known X-ray energies must be extracted from the
data. This is done in two different ways:
</p>
<ol class="org-ol">
<li>A set of fixed cuts (one set for each target/filter combination) is
applied to each run, as presented in
tab. <a href="#tab:cdl:cdl_cleaning_cuts">20</a>. This is the same set as used in
(<a href="./bibliography.html#citeproc_bib_item_65">Christoph Krieger 2018</a>). Its main purpose is to remove events with
multiple clusters and potential background contributions.</li>
<li>By cutting around the main fluorescence line in the charge spectrum
in a \(3σ\) region, for which the spectrum needs to be fitted with
the expected lines, see sec. <a href="./background.html#sec:cdl:fits_to_spectra">12.2.3</a>. This is done
on a run-by-run basis.</li>
</ol>

<p>
The remaining data after both sets of cuts can then be combined for
each target/filter combination to make up the distributions for the
cluster properties as needed for the likelihood method.
</p>

<p>
For a reference of the X-ray fluorescence lines (for more exact values
and \(\alpha_1\), \(\alpha_2\) values etc.) see
tab. <a href="#tab:theory:xray_fluorescence">1</a>. The used EPIC filter refers to a
filter developed for the EPIC camera of the XMM-Newton telescope. It
is a bilayer of \(\SI{1600}{\angstrom}\) polyimide and
\(\SI{800}{\angstrom}\) aluminium. For more information about the EPIC
filters see references
(<a href="./bibliography.html#citeproc_bib_item_216">Strüder et al. 2001</a>; <a href="./bibliography.html#citeproc_bib_item_219">Turner, M. J. L. et al. 2001</a>; <a href="./bibliography.html#citeproc_bib_item_33">Barbera et al. 2003</a>, <a href="./bibliography.html#citeproc_bib_item_34">2016</a>),
in particular (<a href="./bibliography.html#citeproc_bib_item_34">Barbera et al. 2016</a>) for an overview of the materials and production.
</p>

<table id="tab:cdl:run_overview_tab">
<caption class="t-above"><span class="table-number">Table 19:</span> Overview of all runs taken behind the X-ray tube, whether they ran with or without FADC, their targets, filters and high voltage setting and information about the major fluorescence line and its energy. Finally, the mean \(μ\) and width \(σ\) of the main fluorescence line as determined by the fit plus the resulting energy resolution \(μ/σ\) is shown.</caption>

<colgroup>
<col class="org-right" />

<col class="org-left" />

<col class="org-left" />

<col class="org-left" />

<col class="org-right" />

<col class="org-left" />

<col class="org-right" />

<col class="org-left" />

<col class="org-left" />

<col class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-right">Run</th>
<th scope="col" class="org-left">FADC?</th>
<th scope="col" class="org-left">Target</th>
<th scope="col" class="org-left">Filter</th>
<th scope="col" class="org-right">HV [kV]</th>
<th scope="col" class="org-left">Line</th>
<th scope="col" class="org-right">Energy [keV]</th>
<th scope="col" class="org-left">μ [e⁻]</th>
<th scope="col" class="org-left">σ [e⁻]</th>
<th scope="col" class="org-left">σ/μ</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-right">319</td>
<td class="org-left">y</td>
<td class="org-left">Cu</td>
<td class="org-left">Ni</td>
<td class="org-right">15</td>
<td class="org-left">\(\ce{Cu}\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">8.04</td>
<td class="org-left">\(\num{9.509(21)e+05}\)</td>
<td class="org-left">\(\num{7.82(18)e+04}\)</td>
<td class="org-left">\(\num{8.22(19)e-02}\)</td>
</tr>

<tr>
<td class="org-right">320</td>
<td class="org-left">n</td>
<td class="org-left">Cu</td>
<td class="org-left">Ni</td>
<td class="org-right">15</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{9.102(22)e+05}\)</td>
<td class="org-left">\(\num{1.010(19)e+05}\)</td>
<td class="org-left">\(\num{1.110(21)e-01}\)</td>
</tr>

<tr>
<td class="org-right">345</td>
<td class="org-left">y</td>
<td class="org-left">Cu</td>
<td class="org-left">Ni</td>
<td class="org-right">15</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{6.680(12)e+05}\)</td>
<td class="org-left">\(\num{7.15(11)e+04}\)</td>
<td class="org-left">\(\num{1.070(16)e-01}\)</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">315</td>
<td class="org-left">y</td>
<td class="org-left">Mn</td>
<td class="org-left">Cr</td>
<td class="org-right">12</td>
<td class="org-left">\(\ce{Mn}\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">5.89</td>
<td class="org-left">\(\num{6.321(29)e+05}\)</td>
<td class="org-left">\(\num{9.44(26)e+04}\)</td>
<td class="org-left">\(\num{1.494(41)e-01}\)</td>
</tr>

<tr>
<td class="org-right">323</td>
<td class="org-left">n</td>
<td class="org-left">Mn</td>
<td class="org-left">Cr</td>
<td class="org-right">12</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{6.328(11)e+05}\)</td>
<td class="org-left">\(\num{7.225(89)e+04}\)</td>
<td class="org-left">\(\num{1.142(14)e-01}\)</td>
</tr>

<tr>
<td class="org-right">347</td>
<td class="org-left">y</td>
<td class="org-left">Mn</td>
<td class="org-left">Cr</td>
<td class="org-right">12</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{4.956(10)e+05}\)</td>
<td class="org-left">\(\num{6.211(82)e+04}\)</td>
<td class="org-left">\(\num{1.253(17)e-01}\)</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">325</td>
<td class="org-left">y</td>
<td class="org-left">Ti</td>
<td class="org-left">Ti</td>
<td class="org-right">9</td>
<td class="org-left">\(\ce{Ti}\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">4.51</td>
<td class="org-left">\(\num{4.83(31)e+05}\)</td>
<td class="org-left">\(\num{4.87(83)e+04}\)</td>
<td class="org-left">\(\num{1.01(18)e-01}\)</td>
</tr>

<tr>
<td class="org-right">326</td>
<td class="org-left">n</td>
<td class="org-left">Ti</td>
<td class="org-left">Ti</td>
<td class="org-right">9</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{4.615(87)e+05}\)</td>
<td class="org-left">\(\num{4.93(25)e+04}\)</td>
<td class="org-left">\(\num{1.068(57)e-01}\)</td>
</tr>

<tr>
<td class="org-right">349</td>
<td class="org-left">y</td>
<td class="org-left">Ti</td>
<td class="org-left">Ti</td>
<td class="org-right">9</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{3.90(23)e+05}\)</td>
<td class="org-left">\(\num{4.57(57)e+04}\)</td>
<td class="org-left">\(\num{1.17(16)e-01}\)</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">328</td>
<td class="org-left">y</td>
<td class="org-left">Ag</td>
<td class="org-left">Ag</td>
<td class="org-right">6</td>
<td class="org-left">\(\ce{Ag}\) \(\text{L}_{\alpha}\)</td>
<td class="org-right">2.98</td>
<td class="org-left">\(\num{3.0682(97)e+05}\)</td>
<td class="org-left">\(\num{3.935(79)e+04}\)</td>
<td class="org-left">\(\num{1.283(26)e-01}\)</td>
</tr>

<tr>
<td class="org-right">329</td>
<td class="org-left">n</td>
<td class="org-left">Ag</td>
<td class="org-left">Ag</td>
<td class="org-right">6</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{3.0349(51)e+05}\)</td>
<td class="org-left">\(\num{4.004(40)e+04}\)</td>
<td class="org-left">\(\num{1.319(13)e-01}\)</td>
</tr>

<tr>
<td class="org-right">351</td>
<td class="org-left">y</td>
<td class="org-left">Ag</td>
<td class="org-left">Ag</td>
<td class="org-right">6</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{2.5432(63)e+05}\)</td>
<td class="org-left">\(\num{3.545(49)e+04}\)</td>
<td class="org-left">\(\num{1.394(20)e-01}\)</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">332</td>
<td class="org-left">y</td>
<td class="org-left">Al</td>
<td class="org-left">Al</td>
<td class="org-right">4</td>
<td class="org-left">\(\ce{Al}\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">1.49</td>
<td class="org-left">\(\num{1.4868(50)e+05}\)</td>
<td class="org-left">\(\num{2.027(38)e+04}\)</td>
<td class="org-left">\(\num{1.364(26)e-01}\)</td>
</tr>

<tr>
<td class="org-right">333</td>
<td class="org-left">n</td>
<td class="org-left">Al</td>
<td class="org-left">Al</td>
<td class="org-right">4</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{1.3544(30)e+05}\)</td>
<td class="org-left">\(\num{2.539(24)e+04}\)</td>
<td class="org-left">\(\num{1.875(18)e-01}\)</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">335</td>
<td class="org-left">y</td>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-right">2</td>
<td class="org-left">\(\ce{Cu}\) \(\text{L}_{\alpha}\)</td>
<td class="org-right">0.930</td>
<td class="org-left">\(\num{8.885(99)e+04}\)</td>
<td class="org-left">\(\num{1.71(11)e+04}\)</td>
<td class="org-left">\(\num{1.93(13)e-01}\)</td>
</tr>

<tr>
<td class="org-right">336</td>
<td class="org-left">n</td>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-right">2</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{7.777(94)e+04}\)</td>
<td class="org-left">\(\num{2.39(14)e+04}\)</td>
<td class="org-left">\(\num{3.08(19)e-01}\)</td>
</tr>

<tr>
<td class="org-right">337</td>
<td class="org-left">n</td>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-right">2</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{7.86(15)e+04}\)</td>
<td class="org-left">\(\num{2.47(11)e+04}\)</td>
<td class="org-left">\(\num{3.14(15)e-01}\)</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">339</td>
<td class="org-left">y</td>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-right">0.9</td>
<td class="org-left">\(\ce{O }\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">0.525</td>
<td class="org-left">\(\num{5.77(11)e+04}\)</td>
<td class="org-left">\(\num{1.38(22)e+04}\)</td>
<td class="org-left">\(\num{2.39(39)e-01}\)</td>
</tr>

<tr>
<td class="org-right">340</td>
<td class="org-left">n</td>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-right">0.9</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{4.778(31)e+04}\)</td>
<td class="org-left">\(\num{1.230(50)e+04}\)</td>
<td class="org-left">\(\num{2.58(11)e-01}\)</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">342</td>
<td class="org-left">y</td>
<td class="org-left">C</td>
<td class="org-left">EPIC</td>
<td class="org-right">0.6</td>
<td class="org-left">\(\ce{C }\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">0.277</td>
<td class="org-left">\(\num{4.346(36)e+04}\)</td>
<td class="org-left">\(\num{1.223(29)e+04}\)</td>
<td class="org-left">\(\num{2.814(70)e-01}\)</td>
</tr>

<tr>
<td class="org-right">343</td>
<td class="org-left">n</td>
<td class="org-left">C</td>
<td class="org-left">EPIC</td>
<td class="org-right">0.6</td>
<td class="org-left"> </td>
<td class="org-right"> </td>
<td class="org-left">\(\num{3.952(20)e+04}\)</td>
<td class="org-left">\(\num{1.335(14)e+04}\)</td>
<td class="org-left">\(\num{3.379(40)e-01}\)</td>
</tr>
</tbody>
</table>

<table id="tab:cdl:cdl_cleaning_cuts">
<caption class="t-above"><span class="table-number">Table 20:</span> Cuts applied to the CDL datasets in order to roughly clean of potential background events and double hits (recognized as single cluster). RMS refers to the transverse RMS of the clusters.</caption>

<colgroup>
<col class="org-left" />

<col class="org-left" />

<col class="org-left" />

<col class="org-right" />

<col class="org-right" />

<col class="org-right" />

<col class="org-right" />

<col class="org-right" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">Target</th>
<th scope="col" class="org-left">Filter</th>
<th scope="col" class="org-left">line</th>
<th scope="col" class="org-right">HV</th>
<th scope="col" class="org-right">length</th>
<th scope="col" class="org-right">RMS<sub>T</sub><sub>min</sub></th>
<th scope="col" class="org-right">RMS<sub>T</sub><sub>max</sub></th>
<th scope="col" class="org-right">eccentricity</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">Cu</td>
<td class="org-left">Ni</td>
<td class="org-left">\(\ce{Cu}\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">15</td>
<td class="org-right"> </td>
<td class="org-right">0.1</td>
<td class="org-right">1.0</td>
<td class="org-right">1.3</td>
</tr>

<tr>
<td class="org-left">Mn</td>
<td class="org-left">Cr</td>
<td class="org-left">\(\ce{Mn}\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">12</td>
<td class="org-right"> </td>
<td class="org-right">0.1</td>
<td class="org-right">1.0</td>
<td class="org-right">1.3</td>
</tr>

<tr>
<td class="org-left">Ti</td>
<td class="org-left">Ti</td>
<td class="org-left">\(\ce{Ti}\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">9</td>
<td class="org-right"> </td>
<td class="org-right">0.1</td>
<td class="org-right">1.0</td>
<td class="org-right">1.3</td>
</tr>

<tr>
<td class="org-left">Ag</td>
<td class="org-left">Ag</td>
<td class="org-left">\(\ce{Ag}\) \(\text{L}_{\alpha}\)</td>
<td class="org-right">6</td>
<td class="org-right">6.0</td>
<td class="org-right">0.1</td>
<td class="org-right">1.0</td>
<td class="org-right">1.4</td>
</tr>

<tr>
<td class="org-left">Al</td>
<td class="org-left">Al</td>
<td class="org-left">\(\ce{Al}\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">4</td>
<td class="org-right"> </td>
<td class="org-right">0.1</td>
<td class="org-right">1.1</td>
<td class="org-right">2.0</td>
</tr>

<tr>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-left">\(\ce{Cu}\) \(\text{L}_{\alpha}\)</td>
<td class="org-right">2</td>
<td class="org-right"> </td>
<td class="org-right">0.1</td>
<td class="org-right">1.1</td>
<td class="org-right">2.0</td>
</tr>

<tr>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-left">\(\ce{O }\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">0.9</td>
<td class="org-right"> </td>
<td class="org-right">0.1</td>
<td class="org-right">1.1</td>
<td class="org-right">2.0</td>
</tr>

<tr>
<td class="org-left">C</td>
<td class="org-left">EPIC</td>
<td class="org-left">\(\ce{C }\) \(\text{K}_{\alpha}\)</td>
<td class="org-right">0.6</td>
<td class="org-right">6.0</td>
<td class="org-right">0.1</td>
<td class="org-right">1.1</td>
<td class="org-right"> </td>
</tr>
</tbody>
</table>
</div>
<div id="outline-container-org533a54c" class="outline-5">
<h5 id="org533a54c">  <a href="#org533a54c">  <span class="section-number-5">12.2.2.1.</span> Table of fit lines   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-2-1" class="outline-text-5">
<p>
This is the full table.
</p>

<table id="test">
<caption class="t-above"><span class="table-number">Table 21:</span> The fun table!</caption>

<colgroup>
<col class="org-left" />

<col class="org-left" />

<col class="org-left" />

<col class="org-left" />

<col class="org-left" />

<col class="org-left" />

<col class="org-left" />

<col class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">cuNi15</th>
<th scope="col" class="org-left">mnCr12</th>
<th scope="col" class="org-left">tiTi9</th>
<th scope="col" class="org-left">agAg6</th>
<th scope="col" class="org-left">alAl4</th>
<th scope="col" class="org-left">cuEpic2</th>
<th scope="col" class="org-left">cuEpic0<sub>9</sub></th>
<th scope="col" class="org-left">cEpic0<sub>6</sub></th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">\(\text{EG}(\ce{Cu}_{Kα})\)</td>
<td class="org-left">\(\text{EG}(\ce{Mn}_{Kα})\)</td>
<td class="org-left">\(\text{EG}(\ce{Ti}_{Kα})\)</td>
<td class="org-left">\(\text{EG}(\ce{Ag}_{Lα})\)</td>
<td class="org-left">\(\text{EG}(Al_{Kα})\)</td>
<td class="org-left">\(\text{G}(\ce{Cu}_{Lα})\)</td>
<td class="org-left">\(\text{G}(O_{Kα})\)</td>
<td class="org-left">\(\text{G}(\ce{C}_{Kα})\)</td>
</tr>

<tr>
<td class="org-left">\(\text{EG}(\ce{Cu}^{\text{esc}})\)</td>
<td class="org-left">\(\text{G}(\ce{Mn}^{\text{esc}})\)</td>
<td class="org-left">\(\text{G}(\ce{Ti}^{\text{esc}}_{Kα})\)</td>
<td class="org-left">G:</td>
<td class="org-left"> </td>
<td class="org-left">G:</td>
<td class="org-left">#G:</td>
<td class="org-left">G:</td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">#name = \(\ce{Ti}^{\text{esc}}_{Kα}\)</td>
<td class="org-left">name = \(\ce{Ag}_{Lβ}\)</td>
<td class="org-left"> </td>
<td class="org-left">name = \(\ce{Cu}_{Lβ}\)</td>
<td class="org-left">#  name = \(\ce{C}_{Kα}\)</td>
<td class="org-left">name = \(\ce{O}_{Kα}\)</td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left"># \(μ = eμ(\ce{Ti}_{Kα}) · \frac{1.537}{4.511}\)</td>
<td class="org-left">\(N = eN(\ce{Ag}_{Lα}) · 0.1\)</td>
<td class="org-left"> </td>
<td class="org-left">\(N = N(\ce{Cu}_{Lα}) / 5.0\)</td>
<td class="org-left">#  \(μ = μ(O_{Kα}) · (0.277/0.525)\)</td>
<td class="org-left">\(μ = μ(\ce{C}_{Kα}) · \frac{0.525}{0.277}\)</td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">G:</td>
<td class="org-left">\(μ = eμ(\ce{Ag}_{Lα}) · \frac{3.151}{2.984}\)</td>
<td class="org-left"> </td>
<td class="org-left">\(μ = μ(\ce{Cu}_{Lα}) · \frac{0.9498}{0.9297}\)</td>
<td class="org-left">#  \(σ = σ(O_{Kα})\)</td>
<td class="org-left">\(σ = σ(\ce{C}_{Kα})\)</td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">name = \(\ce{Ti}^{\text{esc}}_{Kβ}\)</td>
<td class="org-left">\(σ = eσ(\ce{Ag}_{Lα})\)</td>
<td class="org-left"> </td>
<td class="org-left">\(σ = σ(\ce{Cu}_{Lα})\)</td>
<td class="org-left">#G:</td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(μ = eμ(\ce{Ti}_{Kα}) · \frac{1.959}{4.511}\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">G:</td>
<td class="org-left">#  name = \(\ce{Fe}_{Lα}β\)</td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(σ = σ(\ce{Ti}^{\text{esc}}_{Kα})\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">name = \(\ce{O}_{Kα}\)</td>
<td class="org-left">#  \(μ = μ(O_{Kα}) · \frac{0.71}{0.525}\)</td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">G:</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(N = N(\ce{Cu}_{Lα}) / 3.5\)</td>
<td class="org-left">#  \(σ = σ(O_{Kα})\)</td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">name = \(\ce{Ti}_{Kβ}\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(μ = μ(\ce{Cu}_{Lα}) · \frac{0.5249}{0.9297}\)</td>
<td class="org-left">#G:</td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(μ = eμ(\ce{Ti}_{Kα}) · \frac{4.932}{4.511}\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(σ = σ(\ce{Cu}_{Lα}) / 2.0\)</td>
<td class="org-left">#  name = \(\ce{Ni}_{Lα}β\)</td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(σ = eσ(\ce{Ti}_{Kα})\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">#  \(μ = μ(O_{Kα}) · \frac{0.86}{0.525}\)</td>
<td class="org-left"> </td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-left">cuNi15 Q</td>
<td class="org-left">mnCr12 Q</td>
<td class="org-left">tiTi9 Q</td>
<td class="org-left">agAg6 Q</td>
<td class="org-left">alAl4 Q</td>
<td class="org-left">cuEpic2 Q</td>
<td class="org-left">cuEpic0<sub>9</sub> Q</td>
<td class="org-left">cEpic0<sub>6</sub> Q</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-left">\(\text{G}(\ce{Cu}_{Kα})\)</td>
<td class="org-left">\(\text{G}(\ce{Mn}_{Kα})\)</td>
<td class="org-left">\(\text{G}(\ce{Ti}_{Kα})\)</td>
<td class="org-left">\(\text{G}(\ce{Ag}_{Lα})\)</td>
<td class="org-left">\(\text{G}(Al_{Kα})\)</td>
<td class="org-left">\(\text{G}(\ce{Cu}_{Lα})\)</td>
<td class="org-left">\(\text{G}(O_{Kα})\)</td>
<td class="org-left">\(\text{G}(\ce{C}_{Kα})\)</td>
</tr>

<tr>
<td class="org-left">\(\text{G}(\ce{Cu}^{\text{esc}})\)</td>
<td class="org-left">\(\text{G}(\ce{Mn}^{\text{esc}})\)</td>
<td class="org-left">\(\text{G}(\ce{Ti}^{\text{esc}}_{Kα})\)</td>
<td class="org-left">G:</td>
<td class="org-left"> </td>
<td class="org-left">G:</td>
<td class="org-left">G:</td>
<td class="org-left">G:</td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">#name = \(\ce{Ti}^{\text{esc}}_{Kα}\)</td>
<td class="org-left">name = \(\ce{Ag}_{Lβ}\)</td>
<td class="org-left"> </td>
<td class="org-left">name = \(\ce{Cu}_{Lβ}\)</td>
<td class="org-left">name = \(\ce{C}_{Kα}\)</td>
<td class="org-left">name = \(\ce{O}_{Kα}\)</td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left"># \(μ = eμ(\ce{Ti}_{Kα}) · \frac{1.537}{4.511}\)</td>
<td class="org-left">\(N = N(\ce{Ag}_{Lα}) · 0.1\)</td>
<td class="org-left"> </td>
<td class="org-left">\(N = N(\ce{Cu}_{Lα}) / 5.0\)</td>
<td class="org-left">\(N = N(O_{Kα}) / 10.0\)</td>
<td class="org-left">\(μ = μ(\ce{C}_{Kα}) · \frac{0.525}{0.277}\)</td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">G:</td>
<td class="org-left">\(μ = μ(\ce{Ag}_{Lα}) · \frac{3.151}{2.984}\)</td>
<td class="org-left"> </td>
<td class="org-left">\(μ = μ(\ce{Cu}_{Lα}) · \frac{0.9498}{0.9297}\)</td>
<td class="org-left">\(μ = μ(O_{Kα}) · \frac{277.0}{524.9}\)</td>
<td class="org-left">\(σ = σ(\ce{C}_{Kα})\)</td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">name = \(\ce{Ti}^{\text{esc}}_{Kβ}\)</td>
<td class="org-left">\(σ = σ(\ce{Ag}_{Lα})\)</td>
<td class="org-left"> </td>
<td class="org-left">\(σ = σ(\ce{Cu}_{Lα})\)</td>
<td class="org-left">\(σ = σ(O_{Kα})\)</td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(μ = μ(\ce{Ti}_{Kα}) · \frac{1.959}{4.511}\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left"># $\text{G}:</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(σ = σ(\ce{Ti}^{\text{esc}}_{Kα})\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">#  name = \(\ce{O}_{Kα}\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">G:</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">#  \(N = N(\ce{Cu}_{Lα}) / 4.0\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">name = \(\ce{Ti}_{Kβ}\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">#  \(μ = μ(\ce{Cu}_{Lα}) · \frac{0.5249}{0.9297}\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(μ = μ(\ce{Ti}_{Kα}) · \frac{4.932}{4.511}\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">#  \(σ = σ(\ce{Cu}_{Lα}) / 2.0\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left">\(σ = σ(\ce{Ti}_{Kα})\)</td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left"> </td>
<td class="org-left"> </td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="outline-container-org88c716e" class="outline-5">
<h5 id="org88c716e">  <a href="#org88c716e">  <span class="section-number-5">12.2.2.2.</span> Extra info on target materials   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-2-2" class="outline-text-5">
<ul class="org-ul">
<li class="on"><code>[X]</code> <b>EXTEND THIS TO INCLUDE CITATIONS</b></li>
<li class="off"><code>[ ]</code> <b>ADD OUR OWN POLYIAMID + ALUMINUM TRANSMISSION PLOT</b>
-&gt; 1600 Å polyimide + 800 Å Al
-&gt; Q: What do we use for the polyimide? Guess below text can tell
us the atomic fractions in principle.</li>
</ul>

<p>
EPIC filters:
<a href="https://www.cosmos.esa.int/web/xmm-newton/technical-details-epic">https://www.cosmos.esa.int/web/xmm-newton/technical-details-epic</a>
section 6 about filters contains:
</p>
<blockquote>
<p>
There are four filters in each EPIC camera. Two are thin filters made
of 1600 Å of poly-imide film with 400 Å of aluminium evaporated on to
one side; one is the medium filter made of the same material but with
800 Å of aluminium deposited on it; and one is the thick filter. This
is made of 3300 Å thick Polypropylene with 1100 Å of aluminium and 450
Å of tin evaporated on the film.
</p>
</blockquote>
<p>
i.e. the EPIC filters contain aluminum. That could explain why the
Cu-EPIC 2kV data contains something that might be either aluminum
fluorescence or at least just continuous spectrum that is not filtered
due to the absorption edge of aluminum there!
</p>

<p>
Relevant references:
(<a href="./bibliography.html#citeproc_bib_item_216">Strüder et al. 2001</a>; <a href="./bibliography.html#citeproc_bib_item_219">Turner, M. J. L. et al. 2001</a>; <a href="./bibliography.html#citeproc_bib_item_33">Barbera et al. 2003</a>, <a href="./bibliography.html#citeproc_bib_item_34">2016</a>)
</p>

<p>
In particular (<a href="./bibliography.html#citeproc_bib_item_34">Barbera et al. 2016</a>) contains much more details about
the EPIC filters and its actual composition:
</p>
<blockquote>
<p>
Filter manufacturing process The EPIC Thin and Medium filters
manufactured by MOXTEX consist of a thin film of polyimide, with
nominal thickness of 160 nm, coated with a single layer of aluminum
whose nominal thickness is 40 nm for the Thin and 80 nm for the Medium
filters, respectively. The polyimide thin films are produced by
spin-coating of a polyamic acid (PAA) solution obtained by dissolving
two precursor monomers (an anhydride and an amine) in an organic polar
solvent. For the EPIC Thin and Medium filters the two precursors are
the Biphenyldianhydride (BPDA) and the p-Phenyldiamine (PDA) (Dupont
PI-2610), and the solvent is N-methyl-2-pyrrolidone (NMP) and
Propylene Glycol Monomethyl Ether (Dupont T9040 thinner). To convert
the PAA into polyimide, the solution is heated up to remove the NMP
and to induce the imidization through the evaporation of water
molecules. The film thickness is controlled by spin coating
parameters, PAA viscosity, and curing temperature [19]. The polyimide
thin membrane is attached with epoxy onto a transfer ring and the
aluminum is evaporated in a few runs, distributed over 2–3 days, each
one depositing a metal layer of about 20 nm thickness.
</p>

<p>
The EPIC Thin and Medium flight qualified filters have been
manufactured during a period of 1 year, from January’96 to
January’97. Table 1 lists the full set of flight-qualified filters
(Flight Model and Flight Spare) delivered to the EPIC consortium,
together with their most relevant parameters. Along with the
production of the flight qualified filters, the prototypes and the
qualification filters (not included in this list) have been
manufactured and tested for the construction of the filter
transmission model and to assess the stability in time of the
Optical/UV transparency (opacity). Among these qualification filters
are T4, G12, G18, and G19 that have been previously mentioned.
</p>
</blockquote>

<p>
Further it states that &apos;G12&apos; refers to the <b>medium filter</b>
</p>
<blockquote>
<p>
UV/Vis transmission measurements in the range 190–1000 nm have been
performed between May 1997 and July 2002 on one Thin (T4) and one
medium (G12) EPIC on-ground qualification filters to monitor their
time stability [16].
</p>
</blockquote>

<p>
PP G12 is the name written in the CDL documentation! Mystery solved.
</p>
</div>
</div>
<div id="outline-container-orgb6c61e2" class="outline-5">
<h5 id="orgb6c61e2">  <a href="#orgb6c61e2">  <span class="section-number-5">12.2.2.3.</span> Reconstruct all CDL data   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-2-3" class="outline-text-5">
<p>
Reconstructing all CDL data is done by either using <code>runAnalysisChain</code>
on the directory (currently not tested) or by manually running
<code>raw_data_manipulation</code> and <code>reconstruction</code> as follows:
</p>

<p>
Take note that you may change the paths of course. The paths chosen
here are those in use during the writing process of the thesis.
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #F92672;">cd</span> ~/CastData/data/CDL_2019
raw_data_manipulation -p . -r Xray -o ~/CastData/data/CDL_2019/CDL_2019_Raw.h5
</pre>
</div>

<p>
And now for the reconstruction:
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #F92672;">cd</span> ~/CastData/data/CDL_2019
reconstruction -i ~/CastData/data/CDL_2019/CDL_2019_Raw.h5 -o ~/CastData/data/CDL_2019/CDL_2019_Reco.h5
reconstruction -i ~/CastData/data/CDL_2019/CDL_2019_Reco.h5 --only_charge
reconstruction -i ~/CastData/data/CDL_2019/CDL_2019_Reco.h5 --only_fadc
reconstruction -i ~/CastData/data/CDL_2019/CDL_2019_Reco.h5 --only_gas_gain
reconstruction -i ~/CastData/data/CDL_2019/CDL_2019_Reco.h5 --only_energy_from_e
</pre>
</div>

<p>
At this point the reconstructed CDL H5 file is generally done and
ready to be used in the next section.
</p>
</div>
</div>
<div id="outline-container-sec:background:gen_plots_cdl_data" class="outline-5">
<h5 id="sec:background:gen_plots_cdl_data">  <a href="#sec:background:gen_plots_cdl_data">  <span class="section-number-5">12.2.2.4.</span> Generate plots and tables for this section   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:gen_plots_cdl_data" class="outline-text-5">
<p>
To generate the plots of the above section (and much more) as well as
the table summarizing all the runs and their fits, we continue
with the <code>cdl_spectrum_creation</code> tool as follows:
</p>

<p>
Make sure the config file uses <code>fitByRun</code> to reproduce the same plots!
</p>

<p>
<code>ESCAPE_LATEX</code> performs replacement of characters like <code>&amp;</code> used in
titles. 
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">F_WIDTH</span>=0.9 <span style="color: #FD971F;">ESCAPE_LATEX</span>=true <span style="color: #FD971F;">USE_TEX</span>=true <span style="color: #E6DB74; font-weight: bold;">\</span>
            cdl_spectrum_creation <span style="color: #E6DB74; font-weight: bold;">\</span>
            -i ~/CastData/data/CDL_2019/CDL_2019_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
            --cutcdl --dumpAccurate --hideNloptFit <span style="color: #E6DB74; font-weight: bold;">\</span>
            --plotPath ~/phd/Figs/CDL/fWidth0.9/
</pre>
</div>

<p>
Note: 
</p>

<p>
This generates lots of plots, all placed in the output directory. The
<code>fWidth0.9</code> subdirectory is because here we wish to produce them with
slightly smaller fonts as the single charge spectrum we use in the
next section is inserted standalone.
</p>

<p>
Among them are:
</p>
<ul class="org-ul">
<li>plots of the target / filter kinds with all fits and the raw / cut
data as histogram</li>
<li>plots of histograms of the raw data split <b>by run</b>
-&gt; useful to see the detector variability!</li>
<li>energy resolution plot</li>
<li>peak position of hits / charge vs energy (to see if indeed linear)</li>
<li>calibrated, normalized histogram / kde plot of the target/filter
combinations in energy (calibrated using the main line that was
fitted)</li>
<li>ridgeline plots of KDEs of all the geometric cluster properties
split by target/filter kind and run</li>
<li>gas gain of each gas gain slice in the CDL data, split by run &amp;
target filter kind</li>
<li>temperature data of those CDL runs that contain it</li>
<li>finally it generates the (almost complete) tables as shown in the
above section for the data overview including μ, σ and μ/σ</li>
</ul>

<p>
Note about temperature plots:
</p>
<ul class="org-ul">
<li>  <img alt="septem_temperature_cdl.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/septem_temperature_cdl.svg" /></li>
<li><img alt="septem_imb_temperature_facet_cdl.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/septem_imb_temperature_facet_cdl.svg" />
-&gt; This plot including the IMB temperature shows us precisely what
we expect: the temperature of the IMB and the Septemboard is
directly related and just an offset of one another. This is very
valuable information to have as a reference</li>
<li>  <img alt="septem_temperature_facet_cdl.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/septem_temperature_facet_cdl.svg" /></li>
<li>  <img alt="septem_temperature_facet_cdl_time_since_start.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/septem_temperature_facet_cdl_time_since_start.svg" /></li>
</ul>

<p>
As we ran the command from <code>/tmp/</code> the output plots will be in a
<code>/tmp/out/CDL*</code> directory. We copy over the generated files files
including <code>calibrated_cdl_energy_histos.pdf</code> and
<code>calibrated_cdl_energy_kde.pdf</code> here to <code>/phd/Figs/CDL/</code>.
</p>

<p>
Finally, running the code snippet as mentioned above also produces
table <a href="#tab:cdl:run_overview_tab">19</a> as well as the equivalent for the pixel
spectra and writes them to stdout at the end!
</p>
</div>
</div>
<div id="outline-container-org0655fda" class="outline-5">
<h5 id="org0655fda">  <a href="#org0655fda">  <span class="section-number-5">12.2.2.5.</span> Generate the <code>calibration-cdl-2018.h5</code> file   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-2-5" class="outline-text-5">
<p>
This is also done using the <code>cdl_spectrum_creation</code>. Assuming the
reconstructed CDL H5 file exists, it is as simple as:
</p>
<div class="org-src-container">
<pre class="src src-sh">cdl_spectrum_creation -i ~/CastData/data/CDL_2019/CDL_2019_Reco.h5 --genCdlFile <span style="color: #E6DB74; font-weight: bold;">\</span>
                      --outfile ~/CastData/data/CDL_2019/calibration-cdl
</pre>
</div>
<p>
which generates <code>calibration-cdl-2018.h5</code> for us.
</p>

<p>
Make sure the config file uses <code>fitByRun</code> to reproduce the same plots!
</p>
</div>
</div>
<div id="outline-container-orgc53bd9b" class="outline-5">
<h5 id="orgc53bd9b">  <a href="#orgc53bd9b">  <span class="section-number-5">12.2.2.6.</span> Get the table of fit lines from code   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-2-6" class="outline-text-5">
<p>
Our code <code>cdl_spectrum_creation</code> can be used to output the final
fitting functions in a format like the table inserted in the main
text, thanks to using a CT declarative definition of the fit
functions.
</p>

<p>
We do this by running:
</p>
<div class="org-src-container">
<pre class="src src-sh">cdl_spectrum_creation --printFunctions
</pre>
</div>
</div>
</div>
<div id="outline-container-org98ee5be" class="outline-5">
<h5 id="org98ee5be">  <a href="#org98ee5be">  <span class="section-number-5">12.2.2.7.</span> Historical weather data for Geneva during CDL data taking   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-2-7" class="outline-text-5">
<p>
More or less location of CDL (side of building 17 at Meyrin CERN site):
</p>
<div class="org-src-container">
<pre class="src src-nil">46.22965, 6.04984
</pre>
</div>
<p>
(<a href="https://www.openstreetmap.org/search?whereami=1&amp;query=46.22965%2C6.04984#map=19/46.22965/6.04984">https://www.openstreetmap.org/search?whereami=1&amp;query=46.22965%2C6.04984#map=19/46.22965/6.04984</a>)
</p>

<p>
Here we can see historic weather data plots from Meyrin from the
relevant time range:
<a href="https://meteostat.net/en/place/ch/meyrin?s=06700&amp;t=2019-02-15/2019-02-21">https://meteostat.net/en/place/ch/meyrin?s=06700&amp;t=2019-02-15/2019-02-21</a>
</p>

<p>
This at least proves the weather was indeed very nice outside, sunny
and over 10°C peak temperatures during the day!
</p>

<p>
I exported the data and it&apos;s available here:
<a href="./resources/weather_data_meyrin_cdl_data_taking.csv">./resources/weather_data_meyrin_cdl_data_taking.csv</a>
</p>

<p>
Legend of the columns:
</p>
<table>


<colgroup>
<col class="org-right" />

<col class="org-left" />

<col class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-right">#</th>
<th scope="col" class="org-left">Column</th>
<th scope="col" class="org-left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-right">1</td>
<td class="org-left">time</td>
<td class="org-left">Time</td>
</tr>

<tr>
<td class="org-right">2</td>
<td class="org-left">temp</td>
<td class="org-left">Temperature</td>
</tr>

<tr>
<td class="org-right">3</td>
<td class="org-left">dwpt</td>
<td class="org-left">Dew Point</td>
</tr>

<tr>
<td class="org-right">4</td>
<td class="org-left">rhum</td>
<td class="org-left">Relative Humidity</td>
</tr>

<tr>
<td class="org-right">5</td>
<td class="org-left">prcp</td>
<td class="org-left">Total Precipitation</td>
</tr>

<tr>
<td class="org-right">6</td>
<td class="org-left">snow</td>
<td class="org-left">Snow Depth</td>
</tr>

<tr>
<td class="org-right">7</td>
<td class="org-left">wdir</td>
<td class="org-left">Wind Direction</td>
</tr>

<tr>
<td class="org-right">8</td>
<td class="org-left">wspd</td>
<td class="org-left">Wind Speed</td>
</tr>

<tr>
<td class="org-right">9</td>
<td class="org-left">wpgt</td>
<td class="org-left">Peak Gust</td>
</tr>

<tr>
<td class="org-right">10</td>
<td class="org-left">pres</td>
<td class="org-left">Air Pressure</td>
</tr>

<tr>
<td class="org-right">11</td>
<td class="org-left">tsun</td>
<td class="org-left">Sunshine Duration</td>
</tr>

<tr>
<td class="org-right">12</td>
<td class="org-left">coco</td>
<td class="org-left">Weather Condition Code</td>
</tr>
</tbody>
</table>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> ggplotnim, times
<span style="color: #F92672;">var</span> <span style="color: #FD971F;">df </span>= readCsv<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;/home/basti/phd/resources/weather_data_meyrin_cdl_data_taking.csv&quot;</span><span style="color: #AE81FF;">)</span>
  .mutate<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">string</span> -&gt; <span style="color: #66D9EF;">int</span>: <span style="color: #E6DB74;">&quot;timestamp&quot;</span> ~ parseTime<span style="color: #A6E22E;">(</span>`time`, <span style="color: #E6DB74;">&quot;yyyy-MM-dd HH:mm:ss&quot;</span>, local<span style="color: #E6DB74;">()</span><span style="color: #A6E22E;">)</span>.toUnix<span style="color: #A6E22E;">()</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
  .rename<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #E6DB74;">&quot;Temperature [°C]&quot;</span> &lt;- <span style="color: #E6DB74;">&quot;temp&quot;</span><span style="color: #66D9EF;">}</span>, f<span style="color: #66D9EF;">{</span><span style="color: #E6DB74;">&quot;Pressure [mbar]&quot;</span> &lt;- <span style="color: #E6DB74;">&quot;pres&quot;</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
df = df.gather<span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">[</span><span style="color: #E6DB74;">&quot;Temperature [°C]&quot;</span>, <span style="color: #E6DB74;">&quot;Pressure [mbar]&quot;</span><span style="color: #66D9EF;">]</span>, <span style="color: #E6DB74;">&quot;Data&quot;</span>, <span style="color: #E6DB74;">&quot;Value&quot;</span><span style="color: #AE81FF;">)</span>
<span style="color: #F92672;">echo</span> df
ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;timestamp&quot;</span>, <span style="color: #E6DB74;">&quot;Value&quot;</span>, color = <span style="color: #E6DB74;">&quot;Data&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
  facet_wrap<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Data&quot;</span>, scales = <span style="color: #E6DB74;">&quot;free&quot;</span><span style="color: #AE81FF;">)</span> +
  facetMargin<span style="color: #AE81FF;">(</span><span style="font-style: italic;">0</span>.<span style="font-style: italic;">5</span><span style="color: #AE81FF;">)</span> + 
  geom_line<span style="color: #AE81FF;">()</span> +
  xlab<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Date&quot;</span>, rotate = -<span style="font-style: italic;">45</span>.<span style="font-style: italic;">0</span>, alignTo = <span style="color: #E6DB74;">&quot;right&quot;</span>, margin = <span style="font-style: italic;">2</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
  margin<span style="color: #AE81FF;">(</span>bottom = <span style="font-style: italic;">2</span>.<span style="font-style: italic;">5</span>, right = <span style="font-style: italic;">4</span>.<span style="font-style: italic;">75</span><span style="color: #AE81FF;">)</span> +
  legendPosition<span style="color: #AE81FF;">(</span><span style="font-style: italic;">0</span>.<span style="font-style: italic;">84</span>, <span style="font-style: italic;">0</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> + 
  scale_x_date<span style="color: #AE81FF;">(</span>isTimestamp = <span style="color: #AE81FF;">true</span>,
               dateSpacing = initDuration<span style="color: #66D9EF;">(</span>hours = <span style="font-style: italic;">12</span><span style="color: #66D9EF;">)</span>,
               formatString = <span style="color: #E6DB74;">&quot;yyyy-MM-dd HH:mm&quot;</span>, 
               timeZone = local<span style="color: #66D9EF;">()</span><span style="color: #AE81FF;">)</span> +
  ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Meyrin, Geneva weather during CDL data taking campaign&quot;</span><span style="color: #AE81FF;">)</span> + 
  ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;/home/basti/phd/Figs/CDL/weather_meyrin_during_cdl_data_taking.pdf&quot;</span>, width = <span style="font-style: italic;">1000</span>, height = <span style="font-style: italic;">600</span><span style="color: #AE81FF;">)</span>
</pre>
</div>

<p>
The weather during the data taking campaign is shown in
fig. <a href="#fig:cdl:weather_meyrin_during_cdl_data_taking">4</a>. It&apos;s good to see
my memory served me right.
</p>


<figure id="fig:cdl:weather_meyrin_during_cdl_data_taking">
<img alt="weather_meyrin_during_cdl_data_taking.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/weather_meyrin_during_cdl_data_taking.svg" />

<figcaption>Figure 4: <span class="figure-number">Figure 53: </span>Weather during the data taking campaign at the CDL in Meyrin, Geneva. The weather was sunny and warm for a Februray during the day. As a result the small laboratory heat up significantly and opening / closing windows lead to significant temperature changes.</figcaption>
</figure>
</div>
</div>
</div>
<div id="outline-container-sec:cdl:fits_to_spectra" class="outline-4">
<h4 id="sec:cdl:fits_to_spectra">  <a href="#sec:cdl:fits_to_spectra">  <span class="section-number-4">12.2.3.</span> Charge spectra of the CDL data</a></h4>
<div id="text-sec:cdl:fits_to_spectra" class="outline-text-4">
<p>
To the spectra of each run of the charge data a mixture of Gaussian
functions is fitted. <sup>  <a id="fnr.4" href="#fn.4" class="footref" role="doc-backlink">4</a></sup>
</p>

<p>
Specifically the Gaussian expressed as
</p>
\begin{equation}
G(E; \mu, \sigma, N) = \frac{N}{\sqrt{2 \pi}} \exp\left(-\frac{(E - \mu)^2}{2\sigma^2}\right),
\end{equation}
<p>
is used and will be referenced as \(G\) with possible arguments from
here on. Note that while the physical X-ray transition lines are
Lorentzian shaped (<a href="./bibliography.html#citeproc_bib_item_228">Weisskopf and Wigner 1997</a>), the lines as detected
by a gaseous detector are entirely dominated by detector resolution,
resulting in Gaussian lines. For other types of detectors used in
X-ray fluorescence (XRF) analysis, convolutions of Lorentzian and
Gaussian distributions are used
(<a href="./bibliography.html#citeproc_bib_item_114">Huang and Lim 1986</a>; <a href="./bibliography.html#citeproc_bib_item_106">Heckel and Scholz 1987</a>), called pseudo Voigt
functions (<a href="./bibliography.html#citeproc_bib_item_185">Roberts, Riddle, and Squier 1975</a>; <a href="./bibliography.html#citeproc_bib_item_100">Gunnink 1977</a>).
</p>

<p>
The functions fitted to the different spectra then depend on which
fluorescence lines are visible. The full list of all combinations is
shown in tab. <a href="#tab:cdl:fit_func_charge">22</a>. Typically each line that is
expected from the choice of target, filter and chosen voltage is
fitted, if it can be visually identified in the
data. <sup>  <a id="fnr.5" href="#fn.5" class="footref" role="doc-backlink">5</a></sup> If no &apos;argument&apos; is shown in the table to
\(G\) it means each parameter (\(N, μ, σ\)) is fitted. Any specific
argument given implies that parameter is <span class="underline">fixed</span> relative to another
parameter. For example \(μ^{\ce{Ag}}_{Lα}·\left(\frac{3.151}{2.984}\right)\) fixes
the \(Lβ\) line of silver to the fit parameter \(μ^{\ce{Ag}}_{Lα}\) of the
\(Lα\) line with a multiplicative shift based on the relative eneries of
\(Lα\) to \(Lβ\).  In some cases the amplitude is fixed between different
lines where relative amplitudes cannot be easily predicted or
determined, e.g. in one of the \(\ce{Cu}-\text{EPIC}\) runs, the
\(\ce{C}_{Kα}\) line is fixed to a tenth of the \(\ce{O}_{Kα}\) line. This
is done to get a good fit based on trial and error.
</p>

<p>
Finally, in both \(\ce{Cu}-\text{EPIC}\) lines two &apos;unknown&apos; Gaussians
are added to cover the behavior of the data at higher charges. The
physical origin of this additional contribution is not entirely
clear. The used EPIC filter contains an aluminum coating
(<a href="./bibliography.html#citeproc_bib_item_34">Barbera et al. 2016</a>). As such it has the aluminum absorption edge at
about \(\SI{1.5}{keV}\), possibly matching the additional contribution
for the \(\SI{2}{kV}\) dataset. Whether it is from a continuous part of
the spectrum or a form of aluminum fluorescence is not clear
however. This explanation does not work in the \(\SI{0.9}{kV}\) case,
which is why the line is deemed &apos;unknown&apos;. It may also be a
contribution of the specific polyimide used in the EPIC filter
(<a href="./bibliography.html#citeproc_bib_item_34">Barbera et al. 2016</a>). Another possibility is it is a case of
multi-cluster events, which are too close to be split, but with
properties close enough to a single X-ray as to not be removed by the
cleaning cuts (which gets more likely the lower the energy is).
</p>

<p>
The full set of all fits (including the pixel spectra) is shown in
appendix
<a href="./appendix_cast_detector_lab.html#sec:appendix:cdl:all_spectra_fits_by_run">26.4</a>. Fig. <a href="#fig:cdl:ti_ti_charge_spectrum_run_326">5</a>
shows the charge spectrum of the \(\ce{Ti}\) target and \(\ce{Ti}\) filter
at \(\SI{9}{kV}\) for one of the runs. These plots show the raw data in
the green histogram and the data left after application of the
cleaning cuts (tab. <a href="#tab:cdl:cdl_cleaning_cuts">20</a>) in the purple
histogram. The black line is the result of the fit as described in
tab. <a href="#tab:cdl:fit_func_charge">22</a> with the resulting parameters shown in
the box (parameters that were fixed are not shown). The black straight
lines with grey error bands represent the \(3σ\) region around the main
fluorescence line, which is used to extract those clusters likely from
the fluorescence line and therefore known energy.
</p>


<figure id="fig:cdl:ti_ti_charge_spectrum_run_326">
<img alt="Ti-Ti-9kVCharge-2019_run_326.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/fWidth0.9/Ti-Ti-9kVCharge-2019_run_326.svg" />

<figcaption>Figure 5: <span class="figure-number">Figure 54: </span>Charge spectrum of the \(\ce{Ti}-\ce{Ti}\) spectrum at \(\SI{9}{kV}\) from run 326. The green histogram shows the raw data of this run and the purple histogram indicating the data left after the cleaning cuts are applied. The purple line indicates the result of the fit as described in tab. <a href="#tab:cdl:fit_func_charge">22</a> with the resulting parameters shown in the box. The black lines represent the \(3σ\) region around the main fluorescence line (with grey error bands), which is later used to extract those clusters likely from the fluorescence line and therefore known energy.</figcaption>
</figure>

<p>
\footnotesize
</p>
<table id="tab:cdl:fit_func_charge">
<caption class="t-above"><span class="table-number">Table 22:</span> All fit functions for the charge spectra used for each target / filter combination. Typically each line that is expected and visible in the data is fit. \(G\) is a normal Gaussian. No &apos;argument&apos; to \(G\) means each parameter (\(N, μ, σ\)) is fit. Specific arguments imply this parameter is <span class="underline">fixed</span> relative to another parameter, e.g. \(μ^{\ce{Ag}}_{Lα}·\left(\frac{3.151}{2.984}\right)\) fixes \(Lβ\) line of silver to the fit parameter \(μ^{\ce{Ag}}_{Lα}\) of \(Lα\) with a multiplicative shift based on the relative eneries of \(Lα\) to \(Lβ\). In some cases the amplitude is fixed between different lines, e.g. in one of the \(\ce{Cu}-\text{EPIC}\) runs the \(\ce{C}_{Kα}\) line is fixed to a tenth of the \(\ce{O}_{Kα}\) line. In both \(\ce{Cu}-\text{EPIC}\) lines two &apos;unknown&apos; Gaussians are added to cover the behavior of the data at higher charges. It is unclear what the real cause is, in particular in the lower energy case.</caption>

<colgroup>
<col class="org-left" />

<col class="org-left" />

<col class="org-right" />

<col class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">Target</th>
<th scope="col" class="org-left">Filter</th>
<th scope="col" class="org-right">HV [kV]</th>
<th scope="col" class="org-left">Fit functions</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">Cu</td>
<td class="org-left">Ni</td>
<td class="org-right">15</td>
<td class="org-left">\(G^{\ce{Cu}}_{Kα} + G^{\ce{Cu}, \text{esc}}_{Kα}\)</td>
</tr>

<tr>
<td class="org-left">Mn</td>
<td class="org-left">Cr</td>
<td class="org-right">12</td>
<td class="org-left">\(G^{\ce{Mn}}_{Kα} + G^{\ce{Mn}, \text{esc}}_{Kα}\)</td>
</tr>

<tr>
<td class="org-left">Ti</td>
<td class="org-left">Ti</td>
<td class="org-right">9</td>
<td class="org-left">\(G^{\ce{Ti}}_{Kα} + G^{\ce{Ti}, \text{esc}}_{Kα} + G^{\ce{Ti}}_{Kβ}\left( μ^{\ce{Ti}}_{Kα}·\left(\frac{4.932}{4.511}\right), σ^{\ce{Ti}}_{Kα} \right) + G^{\ce{Ti}, \text{esc}}_{Kβ}\left( μ^{\ce{Ti}}_{Kα}·\left(\frac{1.959}{4.511}\right), σ^{\ce{Ti}, \text{esc}}_{Kα} \right)\)</td>
</tr>

<tr>
<td class="org-left">Ag</td>
<td class="org-left">Ag</td>
<td class="org-right">6</td>
<td class="org-left">\(G^{\ce{Ag}}_{Lα} + G^{\ce{Ag}}_{Lβ}\left( N^{\ce{Ag}}_{Lα}·0.56, μ^{\ce{Ag}}_{Lα}·\left(\frac{3.151}{2.984}\right), σ^{\ce{Ag}}_{Lα} \right)\)</td>
</tr>

<tr>
<td class="org-left">Al</td>
<td class="org-left">Al</td>
<td class="org-right">4</td>
<td class="org-left">\(G^{\ce{Al}}_{Kα}\)</td>
</tr>

<tr>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-right">2</td>
<td class="org-left">\(G^{\ce{Cu}}_{Lα} + G^{\ce{Cu}}_{Lβ}\left( N^{\ce{Cu}}_{Lα}·\left(\frac{0.65}{1.11}\right), μ^{\ce{Cu}}_{Lα}·\left(\frac{0.9498}{0.9297}\right), σ^{\ce{Cu}}_{Lα} \right) + G^{\ce{O}}_{Kα}\left( \frac{N^{\ce{Cu}}_{Lα}}{3.5}, μ^{\ce{Cu}}_{Lα}·\left(\frac{0.5249}{0.9297}\right), \frac{σ^{\ce{Cu}}_{Lα}}{2.0} \right) + G_{\text{unknown}}\)</td>
</tr>

<tr>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-right">0.9</td>
<td class="org-left">\(G^{\ce{O}}_{Kα} + G^{\ce{C}}_{Kα}\left( \frac{N^{\ce{O}}_{Kα}}{10.0}, μ^{\ce{O}}_{Kα}·\left(\frac{277.0}{524.9}\right), σ^{\ce{O}}_{Kα} \right) + G_{\text{unknown}}\)</td>
</tr>

<tr>
<td class="org-left">C</td>
<td class="org-left">EPIC</td>
<td class="org-right">0.6</td>
<td class="org-left">\(G^{\ce{C}}_{Kα} + G^{\ce{O}}_{Kα}\left( μ^{\ce{C}}_{Kα}·\left(\frac{0.525}{0.277}\right), σ^{\ce{C}}_{Kα} \right)\)</td>
</tr>
</tbody>
</table>
<p>
\normalsize
</p>
</div>
<div id="outline-container-org9b64a95" class="outline-5">
<h5 id="org9b64a95">  <a href="#org9b64a95">  <span class="section-number-5">12.2.3.1.</span> Notes on implementation details of fit functions   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-3-1" class="outline-text-5">
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>REWRITE THE BELOW</b> (much of that is irrelevant for the full
thesis)
-&gt; Place into :noexport: section!</li>
</ul>
<p>
The exact implementation in use for both the gaussian:
</p>
<ul class="org-ul">
<li>Gauss:
<a href="https://github.com/Vindaar/seqmath/blob/master/src/seqmath/smath.nim#L997-L1009">https://github.com/Vindaar/seqmath/blob/master/src/seqmath/smath.nim#L997-L1009</a></li>
</ul>

<p>
The fitting was performed both with <a href="https://www.physics.wisc.edu/~craigm/idl/cmpfit.html">MPFit</a> (Levenberg Marquardt C
implementation) as a comparison, but mainly using <a href="https://nlopt.readthedocs.io/en/latest/">NLopt</a> (via
<a href="https://github.com/Vindaar/nimnlopt">https://github.com/Vindaar/nimnlopt</a>). Specifically the gradient based
<a href="http://citeseer.ist.psu.edu/viewdoc/summary?doi=10.1.1.146.5196">&quot;Method of Moving Asymptotes&quot;</a> algorithm was used (NLopt provides a
large number of different minimization / maximization algorithms to
choose from) to perform maximum likelihood estimation written in the
form of a poisson distributed log likelihood \(\chi^2\):
</p>
<p>
where \(n_i\) is the number of events in bin \(i\) and \(y_i\) the model
prediction of events in bin \(i\).
</p>

<p>
The required gradient was calculated simply using the <a href="https://en.wikipedia.org/wiki/Symmetric_derivative">symmetric
derivative</a>. Other algorithms and minimization functions were tried,
but this proved to be the most reliable.  See the implementation:
<a href="https://github.com/Vindaar/TimepixAnalysis/blob/master/Analysis/ingrid/calibration.nim#L131-L162">https://github.com/Vindaar/TimepixAnalysis/blob/master/Analysis/ingrid/calibration.nim#L131-L162</a>
</p>
</div>
</div>
<div id="outline-container-org8f5f033" class="outline-5">
<h5 id="org8f5f033">  <a href="#org8f5f033">  <span class="section-number-5">12.2.3.2.</span> Pixel spectra   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-3-2" class="outline-text-5">
<p>
In case of the pixel spectra the fit functions are generally very
similar, but in some cases the regular Gaussian is replaced by an
&apos;exponential Gaussian&apos; defined as follows:
</p>
<p>
where the constant \(c\) is chosen such that the resulting function is
continuous. The idea being that in the pixel spectra can have a longer
exponential tail on the left side due to threshold effects and
multiple electrons entering a single grid hole.
</p>

<p>
The implementation of the exponential Gaussian is found here:
<a href="https://github.com/Vindaar/TimepixAnalysis/blob/master/Analysis/ingrid/calibration.nim#L182-L194">https://github.com/Vindaar/TimepixAnalysis/blob/master/Analysis/ingrid/calibration.nim#L182-L194</a>
</p>
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>REPLACE LINKS SUCH AS THESE BY TAGGED VERSION AND NOT DIRECT
INLINE LINKS</b></li>
</ul>

<p>
The full list of all fit combinations for the pixel spectra is shown
in tab. <a href="#tab:cdl:fit_funcs_pixel">23</a>. The fitting otherwise works the same
way, using a non-linear least square fit both implemented by hand
using MMA as well as a standard Levenberg-Marquardt fit.
</p>

<table id="tab:cdl:fit_funcs_pixel">
<caption class="t-above"><span class="table-number">Table 23:</span> All fit functions for the pixel spectra for the different combinations. If a function misses a parameter below (out of \((N, μ, σ)\) for the Gaussian and \((a, b, N, μ, σ)\) for the exponential Gaussian) means that parameter has been fixed relative to another.</caption>

<colgroup>
<col class="org-left" />

<col class="org-left" />

<col class="org-right" />

<col class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">Target</th>
<th scope="col" class="org-left">Filter</th>
<th scope="col" class="org-right">HV [kV]</th>
<th scope="col" class="org-left">Fit functions</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">Cu</td>
<td class="org-left">Ni</td>
<td class="org-right">15</td>
<td class="org-left">\(EG^{\ce{Cu}}_{Kα} + EG^{\ce{Cu}, \text{esc}}_{Kα}\)</td>
</tr>

<tr>
<td class="org-left">Mn</td>
<td class="org-left">Cr</td>
<td class="org-right">12</td>
<td class="org-left">\(EG^{\ce{Mn}}_{Kα} + G^{\ce{Mn}, \text{esc}}_{Kα}\)</td>
</tr>

<tr>
<td class="org-left">Ti</td>
<td class="org-left">Ti</td>
<td class="org-right">9</td>
<td class="org-left">\(EG^{\ce{Ti}}_{Kα} + G^{\ce{Ti}, \text{esc}}_{Kα} + G^{\ce{Ti}, \text{esc}}_{Kβ}\left( eμ^{\ce{Ti}}_{Kα}·(\frac{1.959}{4.511}), σ^{\ce{Ti}, \text{esc}}_{Kα} \right) + G^{\ce{Ti}}_{Kβ}\left( eμ^{\ce{Ti}}_{Kα}·(\frac{4.932}{4.511}), eσ^{\ce{Ti}}_{Kα} \right)\)</td>
</tr>

<tr>
<td class="org-left">Ag</td>
<td class="org-left">Ag</td>
<td class="org-right">6</td>
<td class="org-left">\(EG^{\ce{Ag}}_{Lα} + G^{\ce{Ag}}_{Lβ}\left( eN^{\ce{Ag}}_{Lα}·0.56, eμ^{\ce{Ag}}_{Lα}·(\frac{3.151}{2.984}), eσ^{\ce{Ag}}_{Lα} \right)\)</td>
</tr>

<tr>
<td class="org-left">Al</td>
<td class="org-left">Al</td>
<td class="org-right">4</td>
<td class="org-left">\(EG^{\ce{Al}}_{Kα}\)</td>
</tr>

<tr>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-right">2</td>
<td class="org-left">\(G^{\ce{Cu}}_{Lα} + G^{\ce{Cu}}_{Lβ}\left( N^{\ce{Cu}}_{Lα}·(\frac{0.65}{1.11}), μ^{\ce{Cu}}_{Lα}·(\frac{0.9498}{0.9297}), σ^{\ce{Cu}}_{Lα} \right) + G^{\ce{O}}_{Kα}\left( \frac{N^{\ce{Cu}}_{Lα}}{3.5}, μ^{\ce{Cu}}_{Lα}·(\frac{0.5249}{0.9297}), \frac{σ^{\ce{Cu}}_{Lα}}{2.0} \right) + G_{\text{unknown}}\)</td>
</tr>

<tr>
<td class="org-left">Cu</td>
<td class="org-left">EPIC</td>
<td class="org-right">0.9</td>
<td class="org-left">\(G^{\ce{O}}_{Kα} + G_{\text{unknown}}\)</td>
</tr>

<tr>
<td class="org-left">C</td>
<td class="org-left">EPIC</td>
<td class="org-right">0.6</td>
<td class="org-left">\(G^{\ce{C}}_{Kα} + G^{\ce{O}}_{Kα}\left( μ^{\ce{C}}_{Kα}·(\frac{0.525}{0.277}), σ^{\ce{C}}_{Kα} \right)\)</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="outline-container-orgb75f467" class="outline-5">
<h5 id="orgb75f467">  <a href="#orgb75f467">  <span class="section-number-5">12.2.3.3.</span> Generate plot of charge spectrum   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-3-3" class="outline-text-5">
<p>
See sec. <a href="./background.html#sec:background:gen_plots_cdl_data">12.2.2.4</a> for the commands used to
generate all the plots for the pixel and charge spectra, including the
one used in the above section.
</p>
</div>
</div>
</div>
<div id="outline-container-org5cf52be" class="outline-4">
<h4 id="org5cf52be">  <a href="#org5cf52be">  <span class="section-number-4">12.2.4.</span> Overview of CDL data in energy</a></h4>
<div id="text-12-2-4" class="outline-text-4">
<p>
With the fits to the charge spectra performed on a run-by-run basis,
they can be utilized to calibrate the energy of each cluster in the
data. <sup>  <a id="fnr.6" href="#fn.6" class="footref" role="doc-backlink">6</a></sup> This is done by using the linear
relationship between charge and energy and therefore using the charge
of the main fluorescence line as computed from the fit. Each run is
therefore self-calibrated (in contrast to our normal energy
calibration approach
<a href="./calibration.html#sec:calibration:energy">11.1</a>). Fig. <a href="#fig:cdl:calibrated_energy_histos">6</a> shows
normalized histograms of all CDL data after applying basic cuts and
performing said energy calibrations.
</p>


<figure id="fig:cdl:calibrated_energy_histos">
<img alt="calibrated_cdl_energy_histos.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/calibrated_cdl_energy_histos.svg" />

<figcaption>Figure 6: <span class="figure-number">Figure 55: </span>Normalized histograms of all CDL data after applying basic cuts and calibrating the data in energy using the charge of the main fitted line and its known energy as a baseline. Some targets show a wider distribution, due to detector variability which results in different gas gains and thus different charges in different runs.</figcaption>
</figure>
</div>
</div>
<div id="outline-container-sec:cdl:derive_probability_density" class="outline-4">
<h4 id="sec:cdl:derive_probability_density">  <a href="#sec:cdl:derive_probability_density">  <span class="section-number-4">12.2.5.</span> Definition of the reference distributions</a></h4>
<div id="text-sec:cdl:derive_probability_density" class="outline-text-4">
<p>
Having performed fits to all charge spectra of each run and the
position of the main fluorescence line and its width determined, the
reference distributions for the cluster properties entering the
likelihood can be computed. By taking all clusters within the
\(3σ\) bounds around the main fluorescence line of the data used for the
fit, the dataset is selected. This guarantees to leave mainly X-rays
of the targeted energy for each line in the dataset. As the fit is
performed for each run separately, the \(3σ\) charge cut is performed
run by run and then all data combined for each fluorescence line
(target, filter &amp; HV setting). 
</p>

<p>
The desired reference distributions then are simply the normalized
histograms of the clusters in each of the properties. With 8 targeted
fluorescence lines and 3 properties this yields a total of 24
reference distributions. Each histogram is then interpreted as a
probability density function (PDF) for clusters &apos;matching&apos; (more on
this in sec. <a href="./background.html#sec:cdl:cdl_morphing">12.2.6</a>) the energy of its fluorescence
line. An overview of all the reference distributions is shown in
fig. <a href="#fig:cdl:reference_distributions_overview_ridgeline">7</a>. We can see
that all distributions tend to get wider towards lower energies
(towards the top of the plot). This is expected and due to smaller
clusters having less primary electrons and therefore statistical
variations in geometric properties playing a more important
role. <sup>  <a id="fnr.7" href="#fn.7" class="footref" role="doc-backlink">7</a></sup> The binning shown in the figure is the exact
binning used to define the PDF. In case of the fraction of pixels
within a transverse RMS radius, bins with significantly higher counts
are observed at low energies. This is <span class="underline">not</span> a binning artifact, but a
result of the definition of the variable. The property computes the
fraction of pixels that lie within a circle of the radius
corresponding to the transverse RMS of the cluster around the cluster
center (see fig. <a href="#fig:reco:property_explanations">#fig:reco:property_explanations</a>). At energies with few
pixels in total, the integer nature of \(N\) or \(N+1\) primary electrons
(active pixels) inside the radius becomes apparent.
</p>

<p>
The binning in the histograms is chosen by hand such that the binning
is as fine as possible without leading to significant statistical
fluctuations, as those would have a direct effect on the PDFs leading
to unphysical effects on the probabilities. Ideally an approach of
either an automatic bin selection algorithm or something like a kernel
density estimation should be used. However, the latter is slightly
problematic due to the integer effects in the low energies of the
fraction in transverse RMS variable.
</p>

<p>
The summarized &apos;recipe&apos; of the approach is therefore:
</p>
<ol class="org-ol">
<li>apply cleaning cuts according to tab. <a href="#tab:cdl:cdl_cleaning_cuts">20</a>,</li>
<li>perform fits according to tab. <a href="#tab:cdl:fit_func_charge">22</a>,</li>
<li>cut to the \(3σ\) region around the main fluorescence line of the
performed fit (i.e. the first term in tab. <a href="#tab:cdl:fit_func_charge">22</a>),</li>
<li>combine all remaining clusters for the same fluorescence line from
each run,</li>
<li>compute a histogram for each desired cluster property and each
fluorescence line,</li>
<li>normalize the histogram to define the reference distribution, \(\mathcal{P}_i\)</li>
</ol>


<figure id="fig:cdl:reference_distributions_overview_ridgeline">
<img alt="ridgeline_all_properties_side_by_side.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/ridgeline_all_properties_side_by_side.svg" />

<figcaption>Figure 7: <span class="figure-number">Figure 56: </span>Overview of all reference distributions for each target/filter combination and property. The binning is the same as used to compute the probabilities. Towards lower energies (towards the top) the distributions all become wider, as the clusters have fewer electrons and statistical fluctuations play a larger role. The &apos;fraction in transverse RMS&apos; property becomes partially discrete, which is <span class="underline">not</span> a binning effect, but due to integer counting effects of the number of electrons within the transverse RMS radius.</figcaption>
</figure>
</div>
<div id="outline-container-orgaac0e6d" class="outline-5">
<h5 id="orgaac0e6d">  <a href="#orgaac0e6d">  <span class="section-number-5">12.2.5.1.</span> Generate ridgeline plot of all reference distributions   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-5-1" class="outline-text-5">
<p>
The ridgeline plot used in the above section is produced using
<code>TimepixAnalysis/Plotting/plotCdl</code>. See
sec. <a href="./background.html#sec:background:gen_plots_likelihood">12.2.6.1</a> for the exact commands as
more plots are produced used in other sections.
</p>
</div>
</div>
<div id="outline-container-sec:background:correlation_lnL_variables" class="outline-5">
<h5 id="sec:background:correlation_lnL_variables">  <a href="#sec:background:correlation_lnL_variables">  <span class="section-number-5">12.2.5.2.</span> Generate plots for interpolated likelihood distribution and logL variable correlations <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:correlation_lnL_variables" class="outline-text-5">
<p>
In the main text I mentioned that the \(\ln\mathcal{L}\) variables are
all very correlated. Let&apos;s create a few plots to look at how
correlated they actually are. We&apos;ll generate scatter plots of the
three variables against another, using a color scale for the third variable.
</p>

<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>WRITE A FEW WORDS ABOUT THE PLOTS!</b></li>

<li class="on"><code>[X]</code> <p>
Create a plot of the likelihood distributions interpolated over
all energies!
Well, it doesn&apos;t work in the way I thought, because obviously we cannot just compute
the likelihood distributions directly! We can compute <b>a</b> likelihood value for a cluster
at an arbitrary energy, but to get a likelihood distribution we&apos;d need actual X-ray data
at all energies! The closest we have to that is the general CDL data (not just the main
peak &amp; using correct energies for each cluster)
-&gt; Attempt to use the CDL data with only the cleaning cuts
-&gt; Question: Which tool do we add this to? Or separate here using
CDL reconstructed data?
</p>

<p>
How much statistics do we even have if we end up splitting
everything over 1000 energies? Like nothing…
Done in:
<img alt="logL_of_CDL_vs_energy.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/logL_of_CDL_vs_energy.svg" />
now where we compare the no morphing vs. the linear morphing
case. Much more interesting than I would have thought, because one
can clearly see the effect of the morphing on the logL values that
are computed!
This is a pretty nice result to showcase the morphing is actually
useful. Will be put into appendix and linked.
</p></li>

<li class="on"><code>[X]</code> <b>ADD LINE SHOWING WHERE THE CUT VALUES ARE TO THE PLOT</b>
-&gt; The hard corners in the interpolated data is because the
reference distributions are already pre-binned of course! So if by
just changing the bins slightly the ε cut value still lies in the
same bin, of course we see a &apos;straight&apos; line in energy. Hence a non
smooth curve.
We could replace this all:
<ul class="org-ul">
<li>either by a smooth KDE and interpolate based on
that as an alternative. I should really try this, the only
questionable issue is the fracRms distribution and its discrete
features. <b>However</b> we don&apos;t actually guarantee in any way that
in current approach the bins <span class="underline">actually</span> correspond to any fixed
integer values. It is quite likely that the bins that show
smaller/larger values are too wide/small!</li>
<li>or by keeping everything as is, but then performing a spline
interpolation on the <b>distinct</b> (logL, energy) pairs such that the
result is a smooth logL value. &quot;Better bang for buck&quot; than doing
full KDE and avoids the issue of discreteness in the fracRms
distribution. Even though the interpolated values probably
correspond to something as if the fracRms distribution <span class="underline">had</span> been smooth.</li>
</ul></li>
</ul>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> std / <span style="color: #AE81FF;">[</span>os, strutils<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> ingrid / ingrid_types
<span style="color: #F92672;">import</span> ingrid / private / <span style="color: #AE81FF;">[</span>cdl_utils, cdl_cuts, hdf5_utils, likelihood_utils<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> pkg / <span style="color: #AE81FF;">[</span>ggplotnim, nimhdf5<span style="color: #AE81FF;">]</span>


<span style="color: #F92672;">const</span> <span style="color: #66D9EF;">TpxDir</span> = <span style="color: #E6DB74;">&quot;/home/basti/CastData/ExternCode/TimepixAnalysis&quot;</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">cdl_runs_file </span>= <span style="color: #66D9EF;">TpxDir</span> / <span style="color: #E6DB74;">&quot;resources/cdl_runs_2019.org&quot;</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">fname </span>= <span style="color: #E6DB74;">&quot;/home/basti/CastData/data/CDL_2019/CDL_2019_Reco.h5&quot;</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">cdlFile </span>= <span style="color: #E6DB74;">&quot;/home/basti/CastData/data/CDL_2019/calibration-cdl-2018.h5&quot;</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">dsets </span>= @<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;totalCharge&quot;</span>, <span style="color: #E6DB74;">&quot;eccentricity&quot;</span>, <span style="color: #E6DB74;">&quot;lengthDivRmsTrans&quot;</span>, <span style="color: #E6DB74;">&quot;fractionInTransverseRms&quot;</span><span style="color: #AE81FF;">]</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">calcEnergyFromFits</span><span style="color: #AE81FF;">(</span>df: <span style="color: #66D9EF;">DataFrame</span>, fit_μ: <span style="color: #66D9EF;">float</span>, tfKind: <span style="color: #66D9EF;">TargetFilterKind</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">DataFrame</span> =
  <span style="color: #E6DB74;">## Given the fit result of this data type &amp; target/filter combination compute the energy</span>
  <span style="color: #E6DB74;">## of each cluster by using the mean position of the main peak and its known energy</span>
  <span style="color: #FD971F;">result</span> = df
  <span style="color: #FD971F;">result</span><span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;Target&quot;</span><span style="color: #AE81FF;">]</span> = $tfKind
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">invTab </span>= getInverseXrayRefTable<span style="color: #AE81FF;">()</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">energies </span>= getXrayFluorescenceLines<span style="color: #AE81FF;">()</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">lineEnergy </span>= energies<span style="color: #AE81FF;">[</span>invTab<span style="color: #66D9EF;">[</span>$tfKind<span style="color: #66D9EF;">]</span><span style="color: #AE81FF;">]</span>
  <span style="color: #FD971F;">result</span> = <span style="color: #FD971F;">result</span>.mutate<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">float</span>: <span style="color: #E6DB74;">&quot;energy&quot;</span> ~ `totalCharge` / fit_μ * lineEnergy<span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>

<span style="color: #F92672;">let</span> <span style="color: #FD971F;">h5f </span>= <span style="color: #66D9EF;">H5open</span><span style="color: #AE81FF;">(</span>fname, <span style="color: #E6DB74;">&quot;r&quot;</span><span style="color: #AE81FF;">)</span>
<span style="color: #F92672;">var</span> <span style="color: #FD971F;">df </span>= newDataFrame<span style="color: #AE81FF;">()</span>
<span style="color: #F92672;">for</span> tfKind <span style="color: #F92672;">in</span> <span style="color: #66D9EF;">TargetFilterKind</span>:
  <span style="color: #F92672;">for</span> <span style="color: #AE81FF;">(</span>run, grp<span style="color: #AE81FF;">)</span> <span style="color: #F92672;">in</span> tfRuns<span style="color: #AE81FF;">(</span>h5f, tfKind, cdl_runs_file<span style="color: #AE81FF;">)</span>:
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">dfLoc </span>= newDataFrame<span style="color: #AE81FF;">()</span>
    <span style="color: #F92672;">for</span> dset <span style="color: #F92672;">in</span> dsets:
      <span style="color: #F92672;">if</span> dfLoc.<span style="color: #F92672;">len</span> == <span style="font-style: italic;">0</span>:
        dfLoc = toDf<span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">{</span> dset : h5f.readCutCDL<span style="color: #A6E22E;">(</span>run, <span style="font-style: italic;">3</span>, dset, tfKind, <span style="color: #66D9EF;">float64</span><span style="color: #A6E22E;">)</span> <span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">else</span>:
        dfLoc<span style="color: #AE81FF;">[</span>dset<span style="color: #AE81FF;">]</span> = h5f.readCutCDL<span style="color: #AE81FF;">(</span>run, <span style="font-style: italic;">3</span>, dset, tfKind, <span style="color: #66D9EF;">float64</span><span style="color: #AE81FF;">)</span>
    dfLoc<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;runNumber&quot;</span><span style="color: #AE81FF;">]</span> = run
    dfLoc<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;tfKind&quot;</span><span style="color: #AE81FF;">]</span> = $tfKind
    <span style="color: #75715E;"># </span><span style="color: #75715E;">calculate energy from fit</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">fit_μ </span>= grp.attrs<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;fit_μ&quot;</span>, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>
    dfLoc = dfLoc.calcEnergyFromFits<span style="color: #AE81FF;">(</span>fit_μ, tfKind<span style="color: #AE81FF;">)</span>
    df.<span style="color: #F92672;">add</span> dfLoc

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">calcInterp</span><span style="color: #AE81FF;">(</span>ctx: <span style="color: #66D9EF;">LikelihoodContext</span>, df: <span style="color: #66D9EF;">DataFrame</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">DataFrame</span> =
  <span style="color: #75715E;"># </span><span style="color: #75715E;">walk all rows</span>
  <span style="color: #75715E;"># </span><span style="color: #75715E;">feed ecc, ldiv, frac into logL and return a DF with</span>
  <span style="color: #FD971F;">result</span> = df.mutate<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">float</span>: <span style="color: #E6DB74;">&quot;logL&quot;</span> ~ ctx.calcLikelihoodForEvent<span style="color: #A6E22E;">(</span>`energy`,
                                                            `eccentricity`,
                                                            `lengthDivRmsTrans`,
                                                            `fractionInTransverseRms`<span style="color: #A6E22E;">)</span>
  <span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>

<span style="color: #75715E;"># </span><span style="color: #75715E;">first make plots of 3 logL variables to see their correlations  </span>
ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;eccentricity&quot;</span>, <span style="color: #E6DB74;">&quot;lengthDivRmsTrans&quot;</span>, color = <span style="color: #E6DB74;">&quot;fractionInTransverseRms&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
  geom_point<span style="color: #AE81FF;">(</span>size = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
  ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;lnL variables of all (cleaned) CDL data for correlations&quot;</span><span style="color: #AE81FF;">)</span> + 
  ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;~/phd/Figs/background/correlation_ecc_ldiv_frac.pdf&quot;</span>, dataAsBitmap = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>

ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;eccentricity&quot;</span>, <span style="color: #E6DB74;">&quot;fractionInTransverseRms&quot;</span>, color = <span style="color: #E6DB74;">&quot;lengthDivRmsTrans&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
  geom_point<span style="color: #AE81FF;">(</span>size = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
  ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;lnL variables of all (cleaned) CDL data for correlations&quot;</span><span style="color: #AE81FF;">)</span> +   
  ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;~/phd/Figs/background/correlation_ecc_frac_ldiv.pdf&quot;</span>, dataAsBitmap = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>

ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;lengthDivRmsTrans&quot;</span>, <span style="color: #E6DB74;">&quot;fractionInTransverseRms&quot;</span>, color = <span style="color: #E6DB74;">&quot;eccentricity&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
  geom_point<span style="color: #AE81FF;">(</span>size = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
  ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;lnL variables of all (cleaned) CDL data for correlations&quot;</span><span style="color: #AE81FF;">)</span> +   
  ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;~/phd/Figs/background/correlation_ldiv_frac_ecc.pdf&quot;</span>, dataAsBitmap = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>


df = df.filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span>`eccentricity` &lt; <span style="font-style: italic;">2</span>.<span style="font-style: italic;">5</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;eccentricity&quot;</span>, <span style="color: #E6DB74;">&quot;lengthDivRmsTrans&quot;</span>, color = <span style="color: #E6DB74;">&quot;fractionInTransverseRms&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
  geom_point<span style="color: #AE81FF;">(</span>size = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
  ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;lnL variables of all (cleaned) CDL data for correlations (ε &lt; 2.5)&quot;</span><span style="color: #AE81FF;">)</span> +   
  ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;~/phd/Figs/background/correlation_ecc_ldiv_frac_ecc_smaller_2_5.pdf&quot;</span>, dataAsBitmap = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>

ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;eccentricity&quot;</span>, <span style="color: #E6DB74;">&quot;fractionInTransverseRms&quot;</span>, color = <span style="color: #E6DB74;">&quot;lengthDivRmsTrans&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
  geom_point<span style="color: #AE81FF;">(</span>size = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
  ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;lnL variables of all (cleaned) CDL data for correlations (ε &lt; 2.5)&quot;</span><span style="color: #AE81FF;">)</span> +   
  ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;~/phd/Figs/background/correlation_ecc_frac_ldiv_ecc_smaller_2_5.pdf&quot;</span>, dataAsBitmap = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>

ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;lengthDivRmsTrans&quot;</span>, <span style="color: #E6DB74;">&quot;fractionInTransverseRms&quot;</span>, color = <span style="color: #E6DB74;">&quot;eccentricity&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
  geom_point<span style="color: #AE81FF;">(</span>size = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
  ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;lnL variables of all (cleaned) CDL data for correlations (ε &lt; 2.5)&quot;</span><span style="color: #AE81FF;">)</span> +   
  ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;~/phd/Figs/background/correlation_ldiv_frac_ecc_ecc_smaller_2_5.pdf&quot;</span>, dataAsBitmap = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>


<span style="color: #F92672;">from</span> std/sequtils <span style="color: #F92672;">import</span> concat
<span style="color: #75715E;"># </span><span style="color: #75715E;">now generate the plot of the logL values for all cleaned CDL data. We will compare the</span>
<span style="color: #75715E;"># </span><span style="color: #75715E;">case of no morphing with the linear morphing case</span>
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">getLogL</span><span style="color: #AE81FF;">(</span>df: <span style="color: #66D9EF;">DataFrame</span>, mk: <span style="color: #66D9EF;">MorphingKind</span><span style="color: #AE81FF;">)</span>: <span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">DataFrame</span>, <span style="color: #66D9EF;">DataFrame</span><span style="color: #AE81FF;">)</span> = 
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">ctx </span>= initLikelihoodContext<span style="color: #AE81FF;">(</span>cdlFile, yr2018, crGold, igEnergyFromCharge, <span style="color: #66D9EF;">Timepix1</span>, mk, useLnLCut = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">dfMorph </span>= ctx.calcInterp<span style="color: #AE81FF;">(</span>df<span style="color: #AE81FF;">)</span>
  dfMorph<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;Morphing?&quot;</span><span style="color: #AE81FF;">]</span> = $mk
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">cutVals </span>= ctx.calcCutValueTab<span style="color: #AE81FF;">()</span>
  <span style="color: #F92672;">case</span> cutVals.morphKind
  <span style="color: #F92672; font-style: italic;">of</span> mkNone:
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">lineEnergies </span>= getEnergyBinning<span style="color: #AE81FF;">()</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">tab </span>= getInverseXrayRefTable<span style="color: #AE81FF;">()</span>
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">cuts </span>= <span style="color: #F92672;">newSeq</span><span style="color: #AE81FF;">[</span><span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]()</span>
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">energies </span>= @<span style="color: #AE81FF;">[</span><span style="font-style: italic;">0</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">]</span>
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">lastCut </span>= <span style="color: #66D9EF;">Inf</span>
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">lastE </span>= <span style="color: #66D9EF;">Inf</span>
    <span style="color: #F92672;">for</span> k, v <span style="color: #F92672;">in</span> tab:
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">cut </span>= cutVals<span style="color: #AE81FF;">[</span>k<span style="color: #AE81FF;">]</span>
      <span style="color: #F92672;">if</span> classify<span style="color: #AE81FF;">(</span>lastCut<span style="color: #AE81FF;">)</span> != fcInf:
        cuts.<span style="color: #F92672;">add</span> lastCut
        energies.<span style="color: #F92672;">add</span> lastE
      cuts.<span style="color: #F92672;">add</span> cut
      lastCut = cut
      <span style="color: #F92672;">let</span> <span style="color: #66D9EF;">E</span> = lineEnergies<span style="color: #AE81FF;">[</span>v<span style="color: #AE81FF;">]</span>
      energies.<span style="color: #F92672;">add</span> <span style="color: #66D9EF;">E</span>
      lastE = <span style="color: #66D9EF;">E</span>
    cuts.<span style="color: #F92672;">add</span> cuts<span style="color: #AE81FF;">[</span>^<span style="font-style: italic;">1</span><span style="color: #AE81FF;">]</span> <span style="color: #75715E;"># </span><span style="color: #75715E;">add last value again to draw line up </span>
    <span style="color: #F92672;">echo</span> energies.<span style="color: #F92672;">len</span>, <span style="color: #E6DB74;">&quot; vs &quot;</span>, cuts.<span style="color: #F92672;">len</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfCuts </span>= toDf<span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">{</span>energies, cuts, <span style="color: #E6DB74;">&quot;Morphing?&quot;</span> : $cutVals.morphKind<span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
    <span style="color: #FD971F;">result</span> = <span style="color: #AE81FF;">(</span>dfCuts, dfMorph<span style="color: #AE81FF;">)</span>
  <span style="color: #F92672; font-style: italic;">of</span> mkLinear:
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">energies </span>= concat<span style="color: #AE81FF;">(</span>@<span style="color: #66D9EF;">[</span><span style="font-style: italic;">0</span>.<span style="font-style: italic;">0</span><span style="color: #66D9EF;">]</span>, cutVals.lnLCutEnergies, @<span style="color: #66D9EF;">[</span><span style="font-style: italic;">20</span>.<span style="font-style: italic;">0</span><span style="color: #66D9EF;">]</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">cutsSeq </span>= cutVals.lnLCutValues.toSeq1D
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">cuts </span>= concat<span style="color: #AE81FF;">(</span>@<span style="color: #66D9EF;">[</span>cutVals.lnLCutValues<span style="color: #A6E22E;">[</span><span style="font-style: italic;">0</span><span style="color: #A6E22E;">]</span><span style="color: #66D9EF;">]</span>, cutsSeq, @<span style="color: #66D9EF;">[</span>cutsSeq<span style="color: #A6E22E;">[</span>^<span style="font-style: italic;">1</span><span style="color: #A6E22E;">]</span><span style="color: #66D9EF;">]</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfCuts </span>= toDf<span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">{</span><span style="color: #E6DB74;">&quot;energies&quot;</span> : energies, <span style="color: #E6DB74;">&quot;cuts&quot;</span> : cuts, <span style="color: #E6DB74;">&quot;Morphing?&quot;</span> : $cutVals.morphKind<span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
    <span style="color: #FD971F;">result</span> = <span style="color: #AE81FF;">(</span>dfCuts, dfMorph<span style="color: #AE81FF;">)</span>

<span style="color: #F92672;">var</span> <span style="color: #FD971F;">dfMorph </span>= newDataFrame<span style="color: #AE81FF;">()</span>
<span style="color: #F92672;">let</span> <span style="color: #AE81FF;">(</span>dfCutsNone, dfNone<span style="color: #AE81FF;">)</span> = getLogL<span style="color: #AE81FF;">(</span>df, mkNone<span style="color: #AE81FF;">)</span>
<span style="color: #F92672;">let</span> <span style="color: #AE81FF;">(</span>dfCutsLinear, dfLinear<span style="color: #AE81FF;">)</span> = getLogL<span style="color: #AE81FF;">(</span>df, mkLinear<span style="color: #AE81FF;">)</span>
dfMorph.<span style="color: #F92672;">add</span> dfNone
dfMorph.<span style="color: #F92672;">add</span> dfLinear

<span style="color: #F92672;">var</span> <span style="color: #FD971F;">dfCuts </span>= newDataFrame<span style="color: #AE81FF;">()</span>
dfCuts.<span style="color: #F92672;">add</span> dfCutsNone
dfCuts.<span style="color: #F92672;">add</span> dfCutsLinear

<span style="color: #75715E;">#</span><span style="color: #75715E;">dfCuts.showBrowser()</span>

<span style="color: #F92672;">echo</span> dfMorph
ggplot<span style="color: #AE81FF;">(</span>dfMorph, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;logL&quot;</span>, <span style="color: #E6DB74;">&quot;energy&quot;</span>, color = factor<span style="color: #A6E22E;">(</span><span style="color: #E6DB74;">&quot;Target&quot;</span><span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
  facet_wrap<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Morphing?&quot;</span><span style="color: #AE81FF;">)</span> +
  geom_point<span style="color: #AE81FF;">(</span>size = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
  geom_line<span style="color: #AE81FF;">(</span>data = dfCuts, aes = aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;cuts&quot;</span>, <span style="color: #E6DB74;">&quot;energies&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> + <span style="color: #75715E;"># </span><span style="color: #75715E;">, color = &quot;Morphing?&quot;)) + </span>
  ggtitle<span style="color: #AE81FF;">(</span>r<span style="color: #E6DB74;">&quot;$\ln\mathcal{L}$ values of all (cleaned) CDL data against energy&quot;</span><span style="color: #AE81FF;">)</span> +
  themeLatex<span style="color: #AE81FF;">(</span>fWidth = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">9</span>, width = <span style="font-style: italic;">600</span>, baseTheme = singlePlot<span style="color: #AE81FF;">)</span> +
  ylab<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Energy [keV]&quot;</span><span style="color: #AE81FF;">)</span> + xlab<span style="color: #AE81FF;">(</span>r<span style="color: #E6DB74;">&quot;$\ln\mathcal{L}$&quot;</span><span style="color: #AE81FF;">)</span> + 
  ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;~/phd/Figs/background/logL_of_CDL_vs_energy.pdf&quot;</span>, width = <span style="font-style: italic;">1000</span>, height = <span style="font-style: italic;">600</span>, dataAsBitmap = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>
</pre>
</div>
</div>
</div>
</div>
<div id="outline-container-sec:cdl:cdl_morphing" class="outline-4">
<h4 id="sec:cdl:cdl_morphing">  <a href="#sec:cdl:cdl_morphing">  <span class="section-number-4">12.2.6.</span> Definition of the likelihood distribution</a></h4>
<div id="text-sec:cdl:cdl_morphing" class="outline-text-4">
<p>
With our reference distributions defined it is time to look back at
the equation for the definition of the likelihood, eq. \eqref{eq:background:likelihood_def}.
</p>

<blockquote>
<p>
Note: To avoid numerical issues dealing with very small probabilities, the
actual relation evaluated numerically is the negative log of the
likelihood:
</p>

<p>
\[
-\ln \mathcal{L}(ε, f, l) = - \ln \mathcal{P}_ε(ε) -
  \ln \mathcal{P}_f(f) - \ln \mathcal{P}_l(l).
\]
</p>
</blockquote>

<p>
By considering a single fluorescence line, the three reference
distributions \(\mathcal{P}_i(i)\) make up the likelihood
\(\mathcal{L}(ε, l, f)\) <i>function</i> for that energy; a function of the
variables \(ε, f, l\). In order to use the likelihood function as a
classifier for X-ray like clusters we need a one dimensional
expression. This is where the likelihood <i>distribution</i> \(\mathfrak{L}\)
comes in. We reuse all the clusters used to define the reference
distributions \(\mathcal{P}_{ε,f,l}\) and compute their likelihood
values \(\{ \mathcal{L}_j \}\), where \(j\) is the index of the
\(j\text{-th}\) cluster. By then computing the histogram of the set of
all these likelihood values, we obtain the likelihood distribution
</p>

<p>
\[
\mathfrak{L} = \text{histogram}\left( \{ \mathcal{L}_j \} \right).
\]
</p>

<p>
All these distributions are shown in
fig. <a href="#fig:cdl:likelihood_distributions">8</a> as negative log likelihood
distributions. <sup>  <a id="fnr.8" href="#fn.8" class="footref" role="doc-backlink">8</a></sup> We see that the distributions
change slightly in shape and move towards larger \(-\ln \mathcal{L}\)
values towards the lower energy target/filter combinations. The shape
change is most notably due to the significant integer nature of the
&apos;fraction in transverse RMS&apos; \(f\) variable, as seen in
fig. <a href="#fig:cdl:reference_distributions_overview_ridgeline">7</a>. The shift to
larger values expresses that the reference distributions
\(\mathcal{P}_i\) become wider and thus each bin has lower probability.
</p>


<figure id="fig:cdl:likelihood_distributions">
<img alt="logL_ridgeline.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/logL_ridgeline.svg" />

<figcaption>Figure 8: <span class="figure-number">Figure 57: </span>\(-\ln\mathcal{L}\) distributions for each of the targeted fluorescence lines and thus target/filter combinations.</figcaption>
</figure>

<p>
To finally classify events as signal or background using the
likelihood distribution, one sets a desired &quot;software efficiency&quot;
\(ε_{\text{eff}}\), which is defined as:
</p>

\begin{equation}
\label{eq:background:lnL:cut_condition}
ε_{\text{eff}} = \frac{∫_0^{\mathcal{L&apos;}} \mathfrak{L}(\mathcal{L}) \, \mathrm{d}\mathcal{L}}{∫_0^{∞}\mathfrak{L}(\mathcal{L}) \, \mathrm{d} \mathcal{L}}.
\end{equation}

<p>
The likelihood <i>value</i> \(\mathcal{L}&apos;\) is the value corresponding to
the \(ε_{\text{eff}}^{\text{th}}\) percentile of the likelihood
<i>distribution</i> \(\mathfrak{L}\). In practical terms one computes the
normalized cumulative sum of the log likelihood and searches for the
point at which the desired \(ε_{\text{eff}}\) is reached. The typical
software efficiency we aim for is \(\SI{80}{\percent}\). Classification
as X-ray-like then is simply any cluster with a \(-\ln\mathcal{L}\)
value smaller than the cut value \(-\ln\mathcal{L}&apos;\). Note that this
value \(\mathcal{L}&apos;\) has to be determined for <span class="underline">each</span> likelihood
distribution.
</p>

<p>
To summarize the derivation of the likelihood distribution
\(\mathfrak{L}\) and its usage as a classifier as a &apos;recipe&apos;:
</p>
<ol class="org-ol">
<li>compute the reference distributions \(\mathcal{P}_i\) as described in
sec. <a href="./background.html#sec:cdl:derive_probability_density">12.2.5</a>,</li>
<li>take the raw cluster data (unbinned data!) of those clusters that
define the \(\mathcal{P}_i\) and feed each of these into
eq. \eqref{eq:background:likelihood_def} for a single likelihood <i>value</i>
\(\mathcal{L}_i\) each,</li>
<li>compute the <i>histogram</i> of the set of all these likelihood values
\(\{ \mathcal{L}_i \}\) to define the likelihood <i>distribution</i> \(\mathfrak{L}\).</li>
<li>define a desired &apos;software efficiency&apos; \(ε_{\text{eff}}\) and compute the
corresponding likelihood <i>value</i> \(\mathcal{L}_c\) using
eq. \eqref{eq:background:lnL:cut_condition},</li>
<li>any cluster with \(\mathcal{L}_i \leq \mathcal{L}_c\) is considered
X-ray-like with efficiency \(ε_{\text{eff}}\).</li>
</ol>

<p>
Note that due to the usage of negative log likelihoods the
raw data often contains infinities, which are just a side effect of
picking up a zero probability from one of the reference distributions
for a particular cluster. In reality the reference distributions
should be a continuous distribution that is nowhere exactly
zero. However, due to limited statistics there is a small range of
non-zero probabilities (most bins are empty outside the main
range). For all practical purposes this does not matter, but it does
explain the rather hard cutoff from &apos;sensible&apos; likelihood values to
infinities in the raw data.
</p>
</div>
<div id="outline-container-sec:background:gen_plots_likelihood" class="outline-5">
<h5 id="sec:background:gen_plots_likelihood">  <a href="#sec:background:gen_plots_likelihood">  <span class="section-number-5">12.2.6.1.</span> Generate plots of the likelihood and reference distributions   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:gen_plots_likelihood" class="outline-text-5">
<p>
In order to generate plots of the reference distributions as well as
the likelihood distributions, we use the <code>TimepixAnalysis/Plotting/plotCdl</code> tool (adjust the
path to the <code>calibration-cdl-2018.h5</code> according to your system):
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">ESCAPE_LATEX</span>=true <span style="color: #FD971F;">USE_TEX</span>=true <span style="color: #E6DB74; font-weight: bold;">\</span>
./plotCdl <span style="color: #E6DB74; font-weight: bold;">\</span>
    -c ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/CDL
</pre>
</div>
<p>
which generates a range of figures in the local <code>out</code> directory.
</p>

<p>
Among them:
</p>
<ul class="org-ul">
<li>ridgeline plots for each property, as well as facet plots</li>
<li>a combined ridgeline plot of all reference distributions side by
side as a ridgeline each (used in the above section)</li>
<li>plots of the CDL energies using <code>energyFromCharge</code></li>
<li>plots of the likelihood distributions. Both as a outline and
ridgeline plot.</li>
</ul>

<p>
Note that this tool can also create comparisons between a background
dataset and their properties and the reference data!
</p>

<p>
For now we use:
</p>
<ul class="org-ul">
<li>  <code>out/ridgeline_all_properties_side_by_side.pdf</code></li>
<li>  <code>out/eccentricity_facet_calibration-cdl-2018.h5.pdf</code></li>
<li>  <code>out/fractionInTransverseRms_facet_calibration-cdl-2018.h5.pdf</code></li>
<li>  <code>out/lengthDivRmsTrans_facet_calibration-cdl-2018.h5.pdf</code></li>
<li>  <code>out/logL_ridgeline.pdf</code></li>
</ul>

<p>
which also all live in <code>phd/Figs/CDL</code>.
</p>
</div>
</div>
</div>
<div id="outline-container-sec:cdl:cdl_morphing_energy_logL" class="outline-4">
<h4 id="sec:cdl:cdl_morphing_energy_logL">  <a href="#sec:cdl:cdl_morphing_energy_logL">  <span class="section-number-4">12.2.7.</span> Energy interpolation of likelihood distributions</a></h4>
<div id="text-sec:cdl:cdl_morphing_energy_logL" class="outline-text-4">
<p>
For the GridPix detector used in 2014/15, similar X-ray tube data was
taken and each of the 8 X-ray tube energies were assigned an energy
interval. The likelihood distribution to use for each cluster was
chosen based on which energy interval the cluster&apos;s energy falls
into. This leads to discontinuities of the properties at the interval
boundaries due to the energy dependence of the cluster properties as
seen in fig. <a href="#fig:background:eccentricity_photo_escape">1(b)</a>. This change is
of course continuous instead of discrete. It can then lead to jumps in
the efficiency of the background suppression method and thus in the
achieved background rate. It seems a safe assumption that the
reference distributions undergo a continuous change for changing
energies of the X-rays. Therefore, to avoid discontinuities we perform
a linear interpolation for each cluster with energy \(E_β\) between the
closest two neighboring X-ray tube energies \(E_α\) and \(E_γ\) in each
probability density \(\mathcal{P}_i\) at the cluster&apos;s properties. With
\(ΔE = E_α - E_γ\) the difference in energy between the closest two
X-ray tube energies, each probability density is then interpolated to:
</p>

<p>
\[
\mathcal{P}_i(E_β, x_i) = \left(1 - \frac{|E_β - E_α|}{ΔE}\right) · \mathcal{P}_i(E_α, x_i) +
  \left( 1 - \frac{|E_γ - E_β|}{ΔE} \right) · \mathcal{P}_i(E_γ, x_i) .
\]
</p>

<p>
Each probability density of the closest neighbors is evaluated at the
cluster&apos;s property \(x_i\) and the linear interpolation weighted by the
distance to each energy is computed.
</p>

<p>
The choice for a linear interpolation was made after different ideas
were tried. Most importantly, a linear interpolation does not yield
unphysical results (for example negative bin counts in interpolated
data, which can happen in a spline interpolation) and yields very good
results in the cases that can be tested, namely reconstructing a known
likelihood distribution \(B\) by doing a linear interpolation between
the two outer neighbors \(A\) and
\(C\). <a href="#fig:cdl:cdl_morphing_frac_known_lines">9(a)</a> shows this idea using
the fraction in transverse RMS variable. This variable is shown here
as it has the strongest obvious shape differences going from line to
line. The green histogram corresponds to interpolated bins based on
the difference in energy between the line above and below the target
(hence it is not done for the outer lines, as there is no &apos;partner&apos;
above / below). Despite the interpolation covering an energy range
almost twice as large as used in practice, over- or underestimation of
the interpolation (green) is minimal. To give an idea of what this
results in for the interpolation,
fig. <a href="#fig:cdl:cdl_morphing_frac_all_energies">9(b)</a> shows a heatmap of
the same variable showing how the interpolation describes the energy
and fraction in transverse RMS space continuously.
</p>


<figure id="fig:cdl:cdl_morphing_examples" class="figure-wrapper">
<figure id="fig:cdl:cdl_morphing_frac_known_lines" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/CDL/cdlMorphing/fractionInTransverseRms_ridgeline_morph_mtLinear_calibration-cdl-2018.h5_2018.svg" data-width="99%" />  <figcaption>Figure 9(a): Known lines</figcaption></figure> <figure id="fig:cdl:cdl_morphing_frac_all_energies" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/CDL/cdlMorphing/cdl_as_raster_interpolated_morph_mtNone_fractionInTransverseRms.svg" data-width="99%" />  <figcaption>Figure 9(b): All energies</figcaption></figure>
<figcaption>Figure 9: <a href="#fig:cdl:cdl_morphing_frac_known_lines">9(a)</a> shows recovering the known lines (aside from the outer ones)
           using a binwise linear interpolation from the neighbors. While the interpolation (green)
           sometimes over- or undershoots it needs to be kept in mind it covers almost twice the
           energy that is needed. <a href="#fig:cdl:cdl_morphing_frac_all_energies">9(b)</a> is a heatmap of the full energy vs. fraction in transverse RMS
           space interpolated using this binwise linear interpolation. The energy of the fluorescence
           lines is indicated on which the interpolation is based.</figcaption>
</figure>


<p>
Given that such an interpolation works as well as it does on
recovering a known line, implies that a linear interpolation on
&apos;half&apos; <sup>  <a id="fnr.9" href="#fn.9" class="footref" role="doc-backlink">9</a></sup> the energy interval in practice should
yield reasonably realistic distributions, certainly much better than
allowing a discrete jump at specific boundaries.
</p>

<p>
See appendix <a href="./appendix_morphing_cdl_spectra.html#sec:appendix:morphing_cdl_spectra">28</a> for a lot more
information about the ideas considered and comparisons to the approach
without interpolation. In particular
fig. <a href="#fig:appendix:cdl_morphing_logL_vs_energy">140</a>, which computes the
\(\ln\mathcal{L}\) values for all of the (cleaned, cuts
tab. <a href="#tab:cdl:cdl_cleaning_cuts">20</a> applied) CDL data comparing the case of
no interpolation with interpolation showing a clear improvement in the
smoothness of the point cloud.
</p>
</div>
<div id="outline-container-orgb6ac3b5" class="outline-5">
<h5 id="orgb6ac3b5">  <a href="#orgb6ac3b5">  <span class="section-number-5">12.2.7.1.</span> Practical note about interpolation   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-7-1" class="outline-text-5">
<p>
The need to define cut values \(\mathcal{L}&apos;\) for each likelihood
distribution to have a variable to cut on is one reason why the
practical implementation does not use full interpolation to compute
the reference distributions on the fly each time for every individual
cluster energy, but instead uses a pre-calculated high resolution mesh
of \(\num{1000}\) interpolated distributions. This allows to compute the
cut value as well as the distributions before starting to classify all
clusters, saving significant performance. With a mesh of \(\num{1000}\)
energies the largest error on the energy is \(&lt;\SI{5}{eV}\) anyhow. 
</p>
</div>
</div>
<div id="outline-container-sec:background:generated_morphing_plots" class="outline-5">
<h5 id="sec:background:generated_morphing_plots">  <a href="#sec:background:generated_morphing_plots">  <span class="section-number-5">12.2.7.2.</span> Generate plots for morphing / interpolation   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:generated_morphing_plots" class="outline-text-5">
<p>
The plots are created with <code>TimepixAnalysis/Tools/cdlMorphing</code>:
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">WRITE_PLOT_CSV</span>=true <span style="color: #FD971F;">USE_TEX</span>=true ./cdlMorphing <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/CDL/cdlMorphing/
</pre>
</div>

<p>
<b>Note</b>: The script produces the <code>cdl_as_raster_interpolated_*</code>
plots. These use the terminology <code>mtNone</code> for &quot;no&quot; morphing, but that
is only because we perform the morphing &apos;manually&apos; in the code using
the same function as in <code>likelihood_utils.nim</code>. Because the other
morphing happening in that script is one using an offset of 2 (to skip
the center line!).
</p>
</div>
</div>
</div>
<div id="outline-container-org4dd5a17" class="outline-4">
<h4 id="org4dd5a17">  <a href="#org4dd5a17">  <span class="section-number-4">12.2.8.</span> Study of interpolation of CDL distributions <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h4>
<div id="text-12-2-8" class="outline-text-4">
<p>
The full study and notes about how we ended up at using a linear
interpolation, can be found in appendix <a href="./appendix_morphing_cdl_spectra.html#sec:appendix:morphing_cdl_spectra">28</a>.
</p>

<ul class="org-ul">
<li class="on"><code>[X]</code> This is now in the appendix. Good enough? Link</li>
</ul>

<p>
Put here all our studies of how and why we ended up at linear
interpolation!
</p>

<p>
The actual linear interpolation results (i.e. reproducing the &quot;middle&quot;
one will be shown in the actual thesis).
</p>
</div>
</div>
<div id="outline-container-sec:cdl:energy_resolution" class="outline-4">
<h4 id="sec:cdl:energy_resolution">  <a href="#sec:cdl:energy_resolution">  <span class="section-number-4">12.2.9.</span> Energy resolution</a></h4>
<div id="text-sec:cdl:energy_resolution" class="outline-text-4">
<p>
On a slight tangent, with the main fluorescence lines fitted in all
the charge spectra, the position and line width can be used to compute
the energy resolution of the detector:
</p>

<p>
\[
ΔE = \frac{σ}{μ}
\]
</p>

<p>
where \(σ\) is a measure of the line width and \(μ\) the position (in this
case in total cluster charge). Note that in some cases the full width at half
maximum (FWHM) is used and in others the standard deviation (for a normal
distribution the FWHM is about \(2.35 σ\)). The energy resolutions
obtained by our detector in the CDL dataset is seen in
fig. <a href="#fig:cdl:energy_resolution">10</a>. At the lowest energies below
\(\SI{1}{keV}\) the resolution is about \(ΔE \approx \SI{30}{\%}\) and
generally between \(\SIrange{10}{15}{\%}\) from \(\SI{2}{keV}\) on.
</p>


<figure id="fig:cdl:energy_resolution">
<img alt="energyresoplot-2019.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/energyresoplot-2019.svg" />

<figcaption>Figure 10: <span class="figure-number">Figure 58: </span>Energy resolutions depending on the energy of the fluorescence lines based on the charge and pixel spectra. As expected the behavior of the energy resolution is more or less \(1/E\). The uncertainty for each point is the error propagated uncertainty based on the fit parameter uncertainties for the mean position and the line width. There are multiple data points for each energy owing to the fact that each run is fit separately.</figcaption>
</figure>
</div>
<div id="outline-container-org7c210b7" class="outline-5">
<h5 id="org7c210b7">  <a href="#org7c210b7">  <span class="section-number-5">12.2.9.1.</span> Note on energy resolutions   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-9-1" class="outline-text-5">
<p>
Our energy resolutions, if converted to FWHM seem to be rather bad
compared to best in class gaseous detectors, for which sometimes
almost as low as 10% was achieved. 
</p>
</div>
</div>
<div id="outline-container-orga92fe1c" class="outline-5">
<h5 id="orga92fe1c">  <a href="#orga92fe1c">  <span class="section-number-5">12.2.9.2.</span> Generate plot of energy resolution   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-2-9-2" class="outline-text-5">
<p>
This plot is generated as part of the <code>cdl_spectrum_creation.nim</code>
program. See sec. <a href="./background.html#sec:background:gen_plots_cdl_data">12.2.2.4</a> for the commands.
</p>
</div>
</div>
</div>
</div>
<div id="outline-container-sec:background:likelihood_cut" class="outline-3">
<h3 id="sec:background:likelihood_cut">  <a href="#sec:background:likelihood_cut">  <span class="section-number-3">12.3.</span> Application of likelihood cut for background rate</a></h3>
<div id="text-sec:background:likelihood_cut" class="outline-text-3">
<p>
By applying the likelihood cut method introduced in the first part of
this chapter to the background data of CAST, we can extract all
clusters that are X-ray-like and therefore describe the irreducible
background rate. Unless otherwise stated the following plots use a
software efficiency of \(\SI{80}{\%}\).
</p>

<p>
Fig. <a href="#fig:background:cluster_centers_no_vetoes_2017_18">11(a)</a> shows the
cluster centers and their distribution over the whole center GridPix,
which highlights the extremely uneven distribution of background. The
increase towards the edges and in particular the corners is due to
events being cut off. Statistically by cutting off a piece of a
track-like event, the resulting event likely becomes more spherical
than before. In particular in a corner where two sides are cut off
potentially (see also sec. <a href="./septemboard.html#sec:detector:septemboard">7.10</a>). This is an
aspect the detector vetoes help with, see
sec. <a href="./background.html#sec:background:septem_veto">12.5.3</a>. In addition the plot shows some
smaller regions of few pixel diameter that have more activity, due to
minor noise. With about \(\num{74000}\) clusters left on the center
chip, the likelihood cut at \(\SI{80}{\percent}\) software efficiency
represents a background suppression of about a factor \(\num{20}\)
(compare tab. <a href="#tab:cast:data_stats_overview">14</a>, \(\sim\num{1.5e6}\) events
on the center chip over the entire CAST data taking). In the regions
towards the center of the chip, the suppression is of course much
higher. Fig. <a href="#fig:background:background_suppression_tiles_no_vetoes_2017_18">11(b)</a>
shows what the background suppression looks like locally when
comparing the number of clusters in a small region of the chip to the
total number of raw clusters that were detected. Note that this is
based on the assumption that the raw data is homogeneously
distributed. See appendix <a href="./appendix_occupancy.html#sec:appendix:occupancy">29</a> for occupancy maps
of the raw data.
</p>


<figure id="fig:background:background_no_vetoes_clusters" class="figure-wrapper">
<figure id="fig:background:cluster_centers_no_vetoes_2017_18" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/backgroundClusters/background_cluster_centers_lnL80_no_vetoes.svg" data-width="99%" />  <figcaption>Figure 11(a): Cluster centers $\ln\mathcal{L}$</figcaption></figure> <figure id="fig:background:background_suppression_tiles_no_vetoes_2017_18" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/backgroundClusters/background_suppression_tile_map_lnL80_no_vetoes.svg" data-width="99%" />  <figcaption>Figure 11(b): Background suppression</figcaption></figure>
<figcaption>Figure 11: <a href="#fig:background:cluster_centers_no_vetoes_2017_18">11(a)</a> Cluster centers of all X-ray like clusters in the 2017/18 CAST background
           data. The number of these clusters increases drastically towards the edges
           and in particular corners, due to geometric effects. Some regions with
           minor sparking at the edges are visible as small yellow points of few pixel sizes.
           The red outline is the center \goldArea region in which we quote the background rate.
           <a href="#fig:background:background_suppression_tiles_no_vetoes_2017_18">11(b)</a> shows the local background suppression over the total number of raw clusters detected. It assumes a
           homogeneous background distribution in the raw data.</figcaption>
</figure>


<p>
The distribution of the X-ray like clusters in the background data
motivate on the one hand to consider local background rates for a
physics analysis and at the same time the selection of a specific
region in which a benchmark background rate can be defined. For this
purpose (<a href="./bibliography.html#citeproc_bib_item_65">Christoph Krieger 2018</a>) defines different detector regions in
which the background rate is computed and treated as constant. One of
these, termed the &apos;gold region&apos; is a square around the center of
\(\SI{5}{mm}\) side length (visible as red square in fig.
<a href="#fig:background:cluster_centers_no_vetoes_2017_18">11(a)</a> and a bit less
than 3x3 tiles around center in
fig. <a href="#fig:background:background_suppression_tiles_no_vetoes_2017_18">11(b)</a>). All
background rate plots unless otherwise specified in the remainder of
the thesis always refer to this region of low background.
</p>

<p>
Using the \(\ln\mathcal{L}\) approach for the GridPix data taken at CAST
in 2017/18 at a software efficiency of \(ε_{\text{eff}} = \SI{80}{\%}\)
then, a background rate shown in
fig. <a href="#fig:background_rate_eff80_only_center">12</a> is achieved. The average
background rate between \(\SIrange{0}{8}{keV}\) in this case is
\(\SI{2.12423(9666)e-05}{keV⁻¹.cm⁻².s⁻¹}\).
</p>


<figure id="fig:background_rate_eff80_only_center">
<img alt="background_rate_crGold_no_vetoes.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/background_rate_crGold_no_vetoes.svg" />

<figcaption>Figure 12: <span class="figure-number">Figure 59: </span>Background rate achieved based on the \(\ln\mathcal{L}\) method at \(ε_{\text{eff}} = \SI{80}{\percent}\) using the CAST 2017/18 data without the application of any vetoes.</figcaption>
</figure>

<p>
This is comparable to the background rate presented in
(<a href="./bibliography.html#citeproc_bib_item_65">Christoph Krieger 2018</a>), and reproduced here in in
fig. <a href="#fig:detector:background_rate_2014">15</a> for the 2014/15 CAST GridPix
detector. For this result no additional detector features over those
available for the 2014/15 detector are used and the classification
technique is essentially the same. <sup>  <a id="fnr.10" href="#fn.10" class="footref" role="doc-backlink">10</a></sup> In order to
improve on this background rate we will first look at a different
classification technique based on artificial neural networks. Then
afterwards we will go through the different detector features and
discuss how they can further improve the background rate.
</p>
</div>
<div id="outline-container-orga526f26" class="outline-4">
<h4 id="orga526f26">  <a href="#orga526f26">  <span class="section-number-4">12.3.1.</span> Generate background rate and cluster plot <code>[0/2]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h4>
<div id="text-12-3-1" class="outline-text-4">
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>NOTE: THE PLOT WE CURRENTLY GENERATE: DOES IT USE TRACKING INFO
OR NOT?</b></li>

<li class="off"><code>[ ]</code> <b>EXPLAIN AND SHOW HOW TO INSERT TRACKING INFO</b></li>
</ul>
<div class="org-src-container">
<pre class="src src-sh">./cast_log_reader tracking <span style="color: #E6DB74; font-weight: bold;">\</span>
                  -p ../resources/LogFiles/tracking-logs <span style="color: #E6DB74; font-weight: bold;">\</span>
                  --startTime 2018/05/01 <span style="color: #E6DB74; font-weight: bold;">\</span>
                  --endTime 2018/12/31 <span style="color: #E6DB74; font-weight: bold;">\</span>
                  --h5out ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                  --dryRun
</pre>
</div>
<p>
With the <code>dryRun</code> option you are only presented with what would be
written. Run without to actually add the data.  
</p>

<p>
To generate the background rate plot we need:
</p>
<ol class="org-ol">
<li>the reconstructed data files for the background data at CAST
<ul class="org-ul">
<li>(optional for this part): the slow control and tracking logs of
CAST</li>
<li>the tracking information added to the reconstructed background
data files to compute the background rate only for tracking or
only for signal candidate data, done by running the
<code>cast_log_reader</code> with the reconstructed data files as input</li>
</ul></li>
<li>the reconstructed CDL data and the <code>calibration-cdl-2018.h5</code> file
generated from it for the reference and likelihood distributions</li>
</ol>

<p>
To compute the background rate using these inputs we use the
<code>likelihood</code> tool first in order to apply the likelihood cut method
according to a desired software efficiency defined in the
<code>config.toml</code> file. Afterwards we can use another tool to generate the
plot of the background rate (which takes care of scaling the data to a
rate etc.).
</p>

<p>
The relevant section of the <code>config.toml</code> file should look like this:
</p>
<div class="org-src-container">
<pre class="src src-toml">[<span style="color: #66D9EF;">Likelihood</span>]
<span style="color: #75715E;"># </span><span style="color: #75715E;">the signal efficiency to be used for the logL cut (percentage of X-rays of the</span>
<span style="color: #75715E;"># </span><span style="color: #75715E;">reference distributions that will be recovered with the corresponding cut value)</span>
<span style="color: #FD971F;">signalEfficiency</span> = 0.8
<span style="color: #75715E;"># </span><span style="color: #75715E;">the CDL morphing technique to be used (see `MorphingKind` enum), none or linear</span>
<span style="color: #FD971F;">morphingKind</span> = <span style="color: #E6DB74;">&quot;Linear&quot;</span>
<span style="color: #75715E;"># </span><span style="color: #75715E;">clustering algorithm for septem veto</span>
<span style="color: #FD971F;">clusterAlgo</span> = <span style="color: #E6DB74;">&quot;dbscan&quot;</span> <span style="color: #75715E;"># </span><span style="color: #75715E;">choose from {&quot;default&quot;, &quot;dbscan&quot;}</span>
<span style="color: #75715E;"># </span><span style="color: #75715E;">the search radius for the cluster finding algorithm in pixel</span>
<span style="color: #FD971F;">searchRadius</span> = 50 <span style="color: #75715E;"># </span><span style="color: #75715E;">for default clustering algorithm</span>
<span style="color: #FD971F;">epsilon</span> = 65 <span style="color: #75715E;"># </span><span style="color: #75715E;">for DBSCAN algorithm</span>

[<span style="color: #66D9EF;">CDL</span>]
<span style="color: #75715E;"># </span><span style="color: #75715E;">whether to fit the CDL spectra by run or by target/filter combination.</span>
<span style="color: #75715E;"># </span><span style="color: #75715E;">If `true` the resulting `calibration-cdl*.h5` file will contain sub groups</span>
<span style="color: #75715E;"># </span><span style="color: #75715E;">for each run in each target/filter combination group!</span>
<span style="color: #FD971F;">fitByRun</span> = <span style="color: #F92672;">true</span>
</pre>
</div>
<p>
(linear morphing, 80% efficiency, and CDL based on fits per run)
</p>

<p>
In total we&apos;ll want the following files:
</p>
<ul class="org-ul">
<li>for years 2017 &amp; 2018:
<ul class="org-ul">
<li>chip region crAll &amp; crGold:
<ul class="org-ul">
<li>no vetoes</li>
<li>scinti veto</li>
<li>FADC veto</li>
<li>septem veto</li>
<li>line veto</li>
</ul></li>
</ul></li>
<li class="on"><code>[X]</code> <b>USE <code>createAllLikelihoodCombinations</code> for it after short
rewrite</b></li>
</ul>

<p>
In order to generate all likelihood output files (after having
computed the <code>likelihood</code> datasets in the data files and added the
tracking information!) we will use the
<code>createAllLikelihoodCombinations</code> tool (rename please)
</p>


<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>INSERT TRACKING INFORMATION AND ADD THE <code>--tracking</code> FLAG TO <code>LIKELIHOOD</code></b></li>

<li class="off"><code>[ ]</code> <b>RERUN THE BELOW AND CHANGE PATHS TO DIRECTLY IN PHD DIRECTORY!</b></li>
</ul>


<ul class="org-ul">
<li><b>UPDATE</b>: The current files we will likely use are generated by:</li>
</ul>
<p>
All standard variants including the different FADC percentiles:
</p>
<div class="org-src-container">
<pre class="src src-sh">./createAllLikelihoodCombinations <span style="color: #E6DB74; font-weight: bold;">\</span>
    --f2017 ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --f2018 ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --c2017 ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --c2018 ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --regions crGold --regions crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --vetoSets <span style="color: #E6DB74;">&quot;{fkScinti, fkFadc, fkSeptem, fkLineVeto, fkExclusiveLineVeto}&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --fadcVetoPercentiles 0.9 --fadcVetoPercentiles 0.95 --fadcVetoPercentiles 0.99 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --out /t/lhood_outputs_adaptive_fadc <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --multiprocessing
</pre>
</div>

<p>
The septem + line variants <span class="underline">without</span> the FADC:
</p>
<div class="org-src-container">
<pre class="src src-sh">./createAllLikelihoodCombinations <span style="color: #E6DB74; font-weight: bold;">\</span>
    --f2017 ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --f2018 ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --c2017 ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --c2018 ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --regions crGold --regions crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --vetoSets <span style="color: #E6DB74;">&quot;{+fkScinti, fkSeptem, fkLineVeto, fkExclusiveLineVeto}&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --out /t/lhood_outputs_adaptive_fadc <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --multiprocessing
</pre>
</div>
<p>
and the lnL cut efficiency variants for 70 and 90% for the septem +
line + FADC@90% variants:
</p>
<div class="org-src-container">
<pre class="src src-sh">./createAllLikelihoodCombinations <span style="color: #E6DB74; font-weight: bold;">\</span>
    --f2017 ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --f2018 ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --c2017 ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --c2018 ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --regions crGold --regions crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --signalEfficiency 0.7 --signalEfficiency 0.9 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --vetoSets <span style="color: #E6DB74;">&quot;{+fkScinti, +fkFadc, +fkSeptem, fkLineVeto}&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --fadcVetoPercentile 0.9 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --out /t/lhood_outputs_adaptive_fadc <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --multiprocessing <span style="color: #E6DB74; font-weight: bold;">\</span>
    --dryRun
</pre>
</div>
<p>
which for the time being are here:
<a href="./../org/resources/lhood_limits_automation_correct_duration/">./../org/resources/lhood_limits_automation_correct_duration/</a>
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #F92672;">cd</span> $<span style="color: #FD971F;">TPA</span>/Analysis
./createAllLikelihoodCombinations --f2017 ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --f2018 ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --c2017 ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --c2018 ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --regions crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --regions crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --vetoes fkNoVeto <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --vetoes fkScinti <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --vetoes fkFadc <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --vetoes fkSeptem <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --vetoes fkLineVeto <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --vetoes fkExclusiveLineVeto <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --out ~/phd/resources/background/autoGen <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --multiprocessing <span style="color: #E6DB74; font-weight: bold;">\</span>
                                  --dryRun 
</pre>
</div>

<p>
using the <code>--dryRun</code> option first to see the generated commands that
will be run.
</p>

<p>
We will run this over night <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-02-13 Mon 00:37&gt; </span></span> now using the new
default eccentricity cutoff for the line veto of 1.5 (motivated by the
ratio of fraction passing real over fake!) as well as the
<code>lvRegularNoHLC</code> line veto kind (although this should not matter, as
we won&apos;t use the line veto without the septem veto!). The files are
now in <a href="./resources/background/autoGen/">./resources/background/autoGen/</a>.
</p>

<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>WARNING: THE GENERATED FILES HERE STILL USE OUR HACKED IN
CHANGED NUMBER OF BINS FOR THE REFERENCE DISTRIBUTIONS!</b>
-&gt; TWICE AS MANY BINS!</li>
</ul>



<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>THESE ARE OUTDATED. WE USE OUR TOOL</b></li>
</ul>
<p>
First let&apos;s apply the likelihood tool to the 2017 data for the gold
region:
</p>
<div class="org-src-container">
<pre class="src src-sh">likelihood -f ~/CastData/data/DataRuns2017_Reco.h5 --computeLogL <span style="color: #E6DB74; font-weight: bold;">\</span>
           --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --h5out /home/basti/phd/resources/background/lhood_2017_crGold_80eff_no_vetoes.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5
</pre>
</div>


<p>
and the same for the end of 2018 data:
</p>
<div class="org-src-container">
<pre class="src src-sh">likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 --computeLogL <span style="color: #E6DB74; font-weight: bold;">\</span>
           --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --h5out /home/basti/phd/resources/background/lhood_2018_crGold_80eff_no_vetoes.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5
</pre>
</div>

<p>
and now the same for the whole chip:
</p>
<div class="org-src-container">
<pre class="src src-sh">likelihood -f ~/CastData/data/DataRuns2017_Reco.h5 --computeLogL <span style="color: #E6DB74; font-weight: bold;">\</span>
           --region crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --h5out /home/basti/phd/resources/background/lhood_2017_crAll_80eff_no_vetoes.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5
</pre>
</div>
<p>
and the same for the end of 2018 data:
</p>
<div class="org-src-container">
<pre class="src src-sh">likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 --computeLogL <span style="color: #E6DB74; font-weight: bold;">\</span>
           --region crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --h5out /home/basti/phd/resources/background/lhood_2018_crAll_80eff_no_vetoes.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5
</pre>
</div>


<p>
With this we can first create the plot of the cluster centers from the
all chip files using <code>plotClusterCenters</code>:
</p>



<div class="org-src-container">
<pre class="src src-sh">plotBackgroundClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMax 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;X-ray like clusters of CAST data&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/backgroundClusters/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showGoldRegion <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundSuppression <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 --energyMax 12.0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --suffix <span style="color: #E6DB74;">&quot;_lnL80_no_vetoes&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --filterNoisyPixels <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTikZ
</pre>
</div>

<p>
From the logL output files created by
<code>createAllLikelihoodCombinations</code>, let&apos;s now create the &apos;classical&apos;
background rate in the gold region without any vetoes:
</p>

<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --combName 2017/18 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --combYear 2017 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate from CAST data&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_crGold_no_vetoes.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/background/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
</pre>
</div>

<p>
which generates
<a href="./ Figs/background/background_rate_crGold_no_vetoes.pdf">./ Figs/background/background_rate_crGold_no_vetoes.pdf</a> from the TikZ <code>.tex</code>
file of the same name with the following output about the background
rates.
</p>


<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>FINISH THIS</b></li>
</ul>
</div>
</div>
<div id="outline-container-orgccc158b" class="outline-4">
<h4 id="orgccc158b">  <a href="#orgccc158b">  <span class="section-number-4">12.3.2.</span> Background suppression <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h4>
<div id="text-12-3-2" class="outline-text-4">
<p>
In tab. <a href="#tab:cast:data_stats_overview">14</a> we see that the total CAST data
contains \(\num{984319} + \num{470188} = \num{1454507}\)
events. Compared to our \(\num{94625}\) clusters left on the whole chip,
this represents a:
</p>
<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">let</span> <span style="color: #FD971F;">c17 </span>= <span style="font-style: italic;">984319&apos;f64</span>
<span style="color: #F92672;">let</span> <span style="color: #FD971F;">c18 </span>= <span style="font-style: italic;">470188&apos;f64</span>
<span style="color: #F92672;">let</span> <span style="color: #FD971F;">cCut </span>= <span style="font-style: italic;">94625&apos;f64</span>
<span style="color: #F92672;">echo</span> <span style="color: #AE81FF;">(</span>c17 + c18<span style="color: #AE81FF;">)</span> / cCut
</pre>
</div>
<p>
background suppression of about 15.3. 
</p>

<ul class="org-ul">
<li class="on"><code>[X]</code> We could make a plot showing the background suppression? A tile
map with text on taking X times Y squares and printing the
suppression? Would be a nice way to give better understanding how
the vetoes help to improve things over the chip.</li>
</ul>

<p>
Running <code>plotBackgroundClusters</code> now also produces a tile map of the
background suppression over the total raw number of clusters on the
chip!
<code>plots/background_suppression_tile_map.pdf</code> which is in
<img alt="background_suppression_tiles_no_vetoes.svg" class="org-svg" src="./figs/home/basti/phd/Figs/backgroundClusters/background_suppression_tiles_no_vetoes.svg" />.
</p>
</div>
</div>
<div id="outline-container-orgab81f58" class="outline-4">
<h4 id="orgab81f58">  <a href="#orgab81f58">  <span class="section-number-4">12.3.3.</span> Verification of software efficiency using calibration data <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h4>
<div id="text-12-3-3" class="outline-text-4">
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>WRITE ME</b>
-&gt; Important because systematics.</li>
<li class="off"><code>[ ]</code> <b>MAYBE INSTEAD OF EXTENDED, APPENDIX</b></li>
</ul>


<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>SHOW OUR VERIFICATION USING 55FE DATA</b>
-&gt; Check and if not possibly do:
Using fake energies (by leaving out pixel info) to generate
arbitrary energies check the efficiency. I think this is what we did
in that context. Right, that was it:
<a href="./../CastData/ExternCode/TimepixAnalysis/Tools/determineEffectiveEfficiency.nim">./../CastData/ExternCode/TimepixAnalysis/Tools/determineEffectiveEfficiency.nim</a>
-&gt; This is &quot;outdated&quot; in the sense that we have good fake generation
now. Probably good enough to just discuss efficiency that we use for systematic!</li>
<li class="off"><code>[ ]</code> <b>CREATE PLOT OF THE LOGL CUT VALUES TO ACHIEVE THE USED
EFFICIENCY!</b>
-&gt; essentially take our morphed likelihood distributions and for
each of those calculate the logL value for the cut value. Then
create a plot of energy vs. cut value and show that. Should be a
continuous function through the logL values that fixes the desired efficiency.</li>
<li class="off"><code>[ ]</code> <b>THINK ABOUT OPTIMIZING EFFICIENCY</b>
-&gt; We can think about writing a short script to optimize the
efficiency</li>

<li class="off"><code>[ ]</code> <b>AFAIR WE STILL COMPUTE LOGL CUT VALUE BASED ON BINNED
HISTOGRAM. CHANGE THAT TO UNBINNED DATA</b></li>
</ul>
</div>
</div>
</div>
<div id="outline-container-sec:background:mlp" class="outline-3">
<h3 id="sec:background:mlp">  <a href="#sec:background:mlp">  <span class="section-number-3">12.4.</span> Artificial neural networks as cluster classifiers</a></h3>
<div id="text-sec:background:mlp" class="outline-text-3">
<p>
The likelihood cut based method presented in section
<a href="./limit.html#sec:limit:method_likelihood">13.2</a> works well, but does not use the full
potential of the data. It mainly uses length and eccentricity related
properties, which are hand picked and ignores the possible separation
power of other properties.
</p>

<p>
Multiple different ways to use all separation power exist. One
promising approach is the usage of artificial neural networks (ANNs). A
multi-layer perceptron (MLP) (<a href="./bibliography.html#citeproc_bib_item_7">Amari 1967</a>) <sup>  <a id="fnr.11" href="#fn.11" class="footref" role="doc-backlink">11</a></sup> is a
simple supervised ANN model, which consists of an input and output
layer plus one or more fully connected hidden layers. By training such
a network on the already computed geometric properties of the
clusters, computational requirements remain relatively moderate
compared to approaches using – for example – the raw data as inputs.
</p>

<p>
As each neuron on a given layer in an MLP is fully connected to all
neurons on the previous layer, the output of neuron \(k\) on layer \(i\)
is described by
</p>

\begin{equation}
\label{eq:mlp:neuron_output}
y_{k,i} = φ \left( \sum_{j = 0}^m w_{kj} y_{j,i-1} \right)
\end{equation}

<p>
where \(w_{kj}\) is the weight between neuron \(k\) on layer \(i\) and
neuron \(j\) of \(m\) neurons on layer \(i-1\). \(φ\) is a (typically
non-linear) activation function which is applied to saturate a
neuron&apos;s output.
</p>

<p>
If \(y_{j,i-1}\) is considered a vector for all \(j\), \(w_{kj}\) can be
considered a weight matrix and eq. \eqref{eq:mlp:neuron_output} is simply a
matrix product. Each layer is computed iteratively starting from the
input layer towards the output layer.
</p>

<p>
Given that an MLP is a supervised learning algorithm, the desired
target output for a given input during training is known. A loss
function is defined to evaluate the accuracy of the network. Many
different loss functions are used in practice, but in many cases the
mean squared error (MSE; also known as L2 norm) is used as a loss
</p>

<p>
\[
l(\mathbf{y}, \mathbf{\hat{y}}) = \frac{1}{N} \sum_{i = 1}^N \left( y_i - \hat{y}_i \right)²
\]
</p>

<p>
where \(\mathbf{y}\) is a vector \(∈ \mathbb{R}^N\) of the network
outputs and \(\mathbf{\hat{y}}\) the target outputs. The sum runs over
all \(N\) output neurons. <sup>  <a id="fnr.12" href="#fn.12" class="footref" role="doc-backlink">12</a></sup>
</p>

<p>
In order to train a neural network, the initially random weights must
be modified. This is done by computing the gradients of the loss
(given an input) with respect to all the weights of the network,
\(\frac{∂ l(\mathbf{y})}{∂ w_{ij}}\). Effectively the chain rule is used to express
these partial derivatives using the intermediate steps of the
calculation. This leads to an iterative equation for the weights
further up the network (towards the input
layer). Each weight is updated from iteration \(n\) to \(n+1\) according to
</p>

<p>
\[
w^{n+1}_{ij} = w^n_{ij} - η \frac{∂ l(\mathbf{y})}{∂ w^n_{ij}}.
\]
</p>

<p>
where \(η\) is the learning rate. This approach to updating the weights
during training is referred to as
backpropagation. (<a href="./bibliography.html#citeproc_bib_item_188">Rumelhart, Hinton, and Williams 1986</a>) <sup>  <a id="fnr.13" href="#fn.13" class="footref" role="doc-backlink">13</a></sup>
</p>
</div>
<div id="outline-container-sec:background:mlp_for_cast" class="outline-4">
<h4 id="sec:background:mlp_for_cast">  <a href="#sec:background:mlp_for_cast">  <span class="section-number-4">12.4.1.</span> MLP for CAST data</a></h4>
<div id="text-sec:background:mlp_for_cast" class="outline-text-4">
<p>
The simplest approach to using a neural network for data
classification of CAST-like data, is an MLP that uses the pre-computed
geometric properties for each cluster as an input.
</p>

<p>
The choice remains between a single or two output neurons. The more
classical approach is treating signal (X-rays) and background events
as two different classes that the classifier learns to predict, which
we also use. By our convention the target outputs for signal
and background events are
</p>

\begin{align*}
\mathbf{\hat{y}}_{\text{signal}} &amp;= \vektor{ 1 \\ 0 } \\
\mathbf{\hat{y}}_{\text{background}} &amp;= \vektor{ 0 \\ 1 }
\end{align*}

<p>
where the first entry of each \(\mathbf{\hat{y}}\) corresponds to output
neuron 1 and the second to output neuron 2.
</p>

<p>
For the network to generalize to all real CAST data, the training
dataset must be representative of the wide variety of the real
data. For background-like clusters it can be sourced from the
extensive non-tracking dataset recorded at CAST. For the signal-like
X-ray data it is more problematic. The only source of X-rays with
enough statistics are the \(\cefe\) calibration datasets from CAST, but
these only describe X-rays of two energies. The CAST detector lab data
from the X-ray tube are both limited in statistics as well as
suffering from systematic differences to the CAST data due to
different gas gains. <sup>  <a id="fnr.14" href="#fn.14" class="footref" role="doc-backlink">14</a></sup>
</p>

<p>
We will now describe how we generate X-ray clusters for MLP training
by simulating them from a <i>target energy</i>, a specific <i>transverse
diffusion</i> and a desired <i>gas gain</i>.
</p>
</div>
</div>
<div id="outline-container-sec:background:mlp:event_generation" class="outline-4">
<h4 id="sec:background:mlp:event_generation">  <a href="#sec:background:mlp:event_generation">  <span class="section-number-4">12.4.2.</span> Generation of simulated X-rays as MLP training input</a></h4>
<div id="text-sec:background:mlp:event_generation" class="outline-text-4">
<p>
To generate simulated events we wish to use the least amount of
inputs, which still yield events that represent the recorded data as well as
possible. In particular, the systematic variations between different
times and datasets should be reproducible based on these parameters. The
idea is to generate events using the underlying gaseous detector
physics (see sec. <a href="./theory_detector.html#sec:theory_detector">6</a>) and make as few heuristic
modifications to better match the observed (imperfect) data as
possible. This lead to an algorithm, which only uses three
parameters: a target energy for the event (within the typical energy
resolution). A gas gain to encode the gain variation seen. A gas
diffusion coefficient to encode variations in the diffusion
properties (also possibly a result of changing temperatures).
</p>

<p>
In contrast to approaches that simulate the interactions at the
particle level, our simulation is based entirely on the emergent
properties seen as a result of those. The <a href="https://github.com/Vindaar/TimepixAnalysis/blob/master/Analysis/ingrid/fake_generator.nim">basic Monte
Carlo algorithm</a> will now be described in a series of steps.
</p>
<ol class="org-ol">
<li>(optional) sample from different fluorescence lines for an element
given their relative intensities to define a target energy \(E\).</li>
<li>sample from the exponential distribution of the absorption length
(see sec. <a href="./theory_detector.html#sec:theory:xray_matter_gas">6.1.1</a>) for the used gas mixture and
target photon energy to get the conversion point of the X-ray in
the gas (Note: we only sample X-rays that convert; those that would
traverse the whole chamber without conversion are ignored).</li>
<li>sample a target charge to achieve for the cluster, based on target energy by sampling from a
normal distribution with a mean roughly matching detector energy resolution (target energy
inverted to a charge).</li>
<li>sample center position of the cluster within a uniform radius of
\(\SI{4.5}{mm}\) around the chip center.</li>
<li>begin the sampling of each electron in the cluster.</li>
<li>sample a charge (in number of electrons after amplification) for
the electron based on a Pòlya distribution of input gas gain (and
matching normalization &amp; \(θ\) constant). Reject the electron if not
crossing activation threshold (based on real data).</li>
<li>sample a radial distance from the cluster center based on gas
diffusion constant \(D_T\) and diffusion after the remaining drift distance
to the readout plane \(z_{\text{drift}}\) using eq. \eqref{eq:gas_physics:diffusion_after_drift}. Sample twice
from the 1D version using a normal distribution \(\mathcal{N}(μ = 0,
   σ = D_T · \sqrt{z_{\text{drift}}})\) for each dimension. Combine to
a radius from center, \(r = \sqrt{x² + y²}\).</li>
<li>sample a random angle, uniform in \((0, 2π]\).</li>
<li>convert radius and angle to an \((x, y)\) position of the electron,
add to the cluster.</li>
<li>based on a linear approximation activate between 0 to 4
neighboring pixels with a slightly reduced gas gain. We never
activate any neighbors at a charge of \(\SI{1000}{e⁻}\) and always
activate at least one at \(\SI{10000}{e⁻}\). Number of activated
pixels depends on a uniform random number being below one of 4
different thresholds:
\[
    N_{\text{neighbor}} = \text{rand}(0, 1) · N &lt; \text{threshold}
    \]
where the threshold is determined by the linear function described
by the mentioned condition and \(\text{rand}(0, 1)\) is a uniform random
number in \((0, 1)\).</li>
<li>continue sampling electrons until the total charge adds up to the
target charge. We stop at the value closest to the target
(before or after adding a final pixel that crosses the target
charge) to avoid biasing us towards values always larger than the
target.</li>
<li>The final cluster is reconstructed just like any other real
cluster, using the same calibration functions as the real chip
(depending on which dataset it should correspond to).</li>
</ol>

<p>
Note: Neighboring pixels are added to achieve matching eccentricity
distributions between real data and simulated. Activating neighboring
pixels increases the local pixel density randomly, which effectively
increases the weight of some pixels, leading to a slight increase of
eccentricity of a cluster. The approach of activating up to 4 neighbor
pixels and giving them slightly lower charges stems from empirically
matching the simulated data to real data. From a physical perspective
the most likely cause of neighboring pixels is due to UV photons that
are emitted in the amplification region and travel towards the grid
and produce a new electron, which starts an avalanche from there. From
that point of view neighboring pixels should see the full gas
amplification and neighbors other than the direct { up, down, left,
right } neighbors can be activated (depending on an exponential
distribution due to possible absorption of the UV photons in the
gas). See Markus Gruber&apos;s master thesis (<a href="./bibliography.html#citeproc_bib_item_97">Gruber 2018</a>) for a related
study on this for GridPix detectors. 
</p>

<p>
Based on the above algorithm, fake events can be generated that either
match the gas gain and gas diffusion of an existing data taking run
(background or calibration) or any arbitrary combination of
parameters. The former is important for verification of the validity
of the generated events as well as to check the MLP cut efficiency
(more on that in sec. <a href="./background.html#sec:background:mlp:mlp_cut_value">12.4.6</a>). The latter
is used to generate a wide variety of MLP training data. Event
production is a very fast process <sup>  <a id="fnr.15" href="#fn.15" class="footref" role="doc-backlink">15</a></sup>, allowing to produce
large amounts of statistics for MLP training in a reasonable time
frame.
</p>

<p>
For other applications that require higher fidelity simulations,
Degrad (<a href="./bibliography.html#citeproc_bib_item_41">Biagi 1995a</a>), a sister program to Magboltz
(<a href="./bibliography.html#citeproc_bib_item_42">Biagi 1995b</a>), can be used, which simulates the particle
interactions directly. This is utilized in a code by M. Gruber
(<a href="./bibliography.html#citeproc_bib_item_98">Gruber 2023</a>), which uses Degrad and Garfield++ (<a href="./bibliography.html#citeproc_bib_item_242">“Garfield++,” n.d.</a>)
to perform realistic event simulation for GridPix detectors (its data
can also be analyzed by the software presented in sec. <a href="./reconstruction.html#sec:reco:tpa">9.1</a>
and appendix <a href="./appendix_software.html#sec:appendix:software">34</a>).
</p>
</div>
<div id="outline-container-org01e3638" class="outline-5">
<h5 id="org01e3638">  <a href="#org01e3638">  <span class="section-number-5">12.4.2.1.</span> Note on this section and theory   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-2-1" class="outline-text-5">
<p>
The explanation above and the implementation of the fake event
generation is the main motivator for a good chunk of the included
gaseous detector theory fundamentals:
</p>

<ul class="org-ul">
<li>the X-ray matter interaction section is needed to calculate the
absorption length (implemented in
(<a href="./bibliography.html#citeproc_bib_item_199">Schmidt and SciNim contributors 2022</a>)). Understanding the theory is
needed both to understand how to sample in the first place and how
to calculate the absorption length for the gas mixture (and
pressure, temperature etc) of the used detector gas</li>
<li>Dalton&apos;s law is crucial to correctly take into account the behavior
of the gas mixtures and how the percentages given for gas mixtures
are applied to the calculation</li>
<li>Diffusion, how it relates to a random walk, its dimensionality
dependence and interpretation as sampling from a normal distribution
for the resulting diffusion after a certain drift is required for
the sampling of each electron.</li>
<li>the discussion of the \cefe spectrum is required to understand that
an escape photon is not equivalent to a real \(\SI{3}{keV}\) X-ray
(more on that in the next sections)</li>
<li>gas gain understanding is (here also) needed to sample the gain for
each electron</li>
</ul>
</div>
</div>
<div id="outline-container-sec:background:gen_plots_fake_event_comparisons" class="outline-5">
<h5 id="sec:background:gen_plots_fake_event_comparisons">  <a href="#sec:background:gen_plots_fake_event_comparisons">  <span class="section-number-5">12.4.2.2.</span> Generate some example and comparison plots <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:gen_plots_fake_event_comparisons" class="outline-text-5">
<ul class="org-ul">
<li class="on"><code>[X]</code> Comparison of the different properties as a ridgeline plot?
-&gt; similar to plots in appendix
<a href="./appendix_fit_by_run_justification.html#sec:appendix:fit_by_run:gas_gain_var_cluster_prop">27.1</a>
like
<img alt="C-EPIC-0.6kV_ridgeline_kde_by_run.svg" class="org-svg" src="./figs/home/basti/phd/Figs/CDL/C-EPIC-0.6kV_ridgeline_kde_by_run.svg" /></li>
</ul>

<p>
Let&apos;s generate plots comparing the properties used by the MLP of
generated events with those of different runs. The idea is to generate
fake data based on the gas properties of each run, i.e. gas gain, gas
diffusion and the target energy of the X-ray fluorescence line we want
(i.e. the Kα line for \(\ce{Mn}\) in case of the \cefe source or each
target in the X-ray tube data).
</p>

<p>
We&apos;ll generate a ridgeline plot of all these properties normalized to
\(x/\text{max}(\{x\})\) where \(\{x\}\) is the set of all values in the
run to get values between 0 and 1 for all properties such that they
can be compared in a single plot. Each ridge will use a KDE based
density to best highlight any differences in the underlying
distribution.
</p>

<p>
We will first use a tool to generate HDF5 files of simulated events
following a real run. For one run, e.g. 241 CAST calibration, we do
this by:
</p>
<div class="org-src-container">
<pre class="src src-sh">fake_event_generator <span style="color: #E6DB74; font-weight: bold;">\</span>
    like <span style="color: #E6DB74; font-weight: bold;">\</span>
    -p ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --run 241 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp/test_fakegen_run241.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outRun 241 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --tfKind Mn-Cr-12kV <span style="color: #E6DB74; font-weight: bold;">\</span>
    --nmc 50000
</pre>
</div>
<p>
where we specify we want X-rays like the 55Fe source (via <code>tfKind</code>)
and want to simulate 50k X-rays.
</p>

<p>
Then we can compare the properties using <code>plotDatasetGgplot</code>, which
can be found here <a href="./../CastData/ExternCode/TimepixAnalysis/Plotting/plotDsetGgplot/plotDatasetGgplot.nim">./../CastData/ExternCode/TimepixAnalysis/Plotting/plotDsetGgplot/plotDatasetGgplot.nim</a>:
</p>
<div class="org-src-container">
<pre class="src src-sh">plotDatasetGgplot <span style="color: #E6DB74; font-weight: bold;">\</span>
    -f ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    -f /t/test_fakegen_run241.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --run 241 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/phd/Figs/fakeEventSimulation/runComparisons
</pre>
</div>

<p>
Let&apos;s automate this quickly for all \cefe calibration runs.
</p>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> shell, strutils, sequtils
<span style="color: #F92672;">import</span> nimhdf5
<span style="color: #F92672;">import</span> ingrid / <span style="color: #AE81FF;">[</span>ingrid_types, tos_helpers<span style="color: #AE81FF;">]</span>

<span style="color: #F92672;">const</span> <span style="color: #FD971F;">filePath </span>= <span style="color: #E6DB74;">&quot;~/CastData/data/CalibrationRuns</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">_Reco.h5&quot;</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">genData </span>= <span style="color: #E6DB74;">&quot;&quot;&quot;</span>
<span style="color: #E6DB74;">fake_event_generator \</span>
<span style="color: #E6DB74;">    like \</span>
<span style="color: #E6DB74;">    -p $file \</span>
<span style="color: #E6DB74;">    --run $run \</span>
<span style="color: #E6DB74;">    --outpath $fakeFile \</span>
<span style="color: #E6DB74;">    --outRun $run \</span>
<span style="color: #E6DB74;">    --tfKind Mn-Cr-12kV \</span>
<span style="color: #E6DB74;">    --nmc 50000</span>
<span style="color: #E6DB74;">&quot;&quot;&quot;</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">plotData </span>= <span style="color: #E6DB74;">&quot;&quot;&quot;</span>
<span style="color: #E6DB74;">plotDatasetGgplot \</span>
<span style="color: #E6DB74;">    -f $file \</span>
<span style="color: #E6DB74;">    -f $fakeFile \</span>
<span style="color: #E6DB74;">    --names 55Fe --names Simulation \</span>
<span style="color: #E6DB74;">    --run $run \</span>
<span style="color: #E6DB74;">    --plotPath /home/basti/phd/Figs/fakeEventSimulation/runComparisons/ \</span>
<span style="color: #E6DB74;">    --prefix ingrid_properties_run_$run \</span>
<span style="color: #E6DB74;">    --suffix &quot;, run $run&quot;</span>
<span style="color: #E6DB74;">&quot;&quot;&quot;</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">main</span><span style="color: #AE81FF;">(</span>generate = <span style="color: #AE81FF;">false</span>,
          plot = <span style="color: #AE81FF;">false</span>,
          fakeFile = <span style="color: #E6DB74;">&quot;~/org/resources/fake_events_for_runs.h5&quot;</span><span style="color: #AE81FF;">)</span> =
  <span style="color: #F92672;">const</span> <span style="color: #FD971F;">years </span>= <span style="color: #AE81FF;">[</span><span style="font-style: italic;">2017</span>, <span style="font-style: italic;">2018</span><span style="color: #AE81FF;">]</span>
  <span style="color: #F92672;">for</span> year <span style="color: #F92672;">in</span> years:
    <span style="color: #75715E;">#</span><span style="color: #75715E;">if year == &quot;2017&quot;: continue ## skip for now, already done</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">file </span>= filePath % <span style="color: #AE81FF;">[</span>$year<span style="color: #AE81FF;">]</span>
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">runs </span>= <span style="color: #F92672;">newSeq</span><span style="color: #AE81FF;">[</span><span style="color: #66D9EF;">int</span><span style="color: #AE81FF;">]()</span>
    withH5<span style="color: #AE81FF;">(</span>file, <span style="color: #E6DB74;">&quot;r&quot;</span><span style="color: #AE81FF;">)</span>:
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">fileInfo </span>= getFileInfo<span style="color: #AE81FF;">(</span>h5f<span style="color: #AE81FF;">)</span>
      runs = fileInfo.runs
    <span style="color: #F92672;">for</span> run <span style="color: #F92672;">in</span> runs:
      <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;Working on run: &quot;</span>, run
      <span style="color: #F92672;">if</span> generate:
        <span style="color: #F92672;">let</span> <span style="color: #FD971F;">genCmd </span>= genData % <span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;file&quot;</span>, file, <span style="color: #E6DB74;">&quot;run&quot;</span>, $run, <span style="color: #E6DB74;">&quot;fakeFile&quot;</span>, fakeFile<span style="color: #AE81FF;">]</span>
        shell:
          <span style="color: #AE81FF;">(</span>$genCmd<span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">if</span> plot:
        <span style="color: #F92672;">let</span> <span style="color: #FD971F;">plotCmd </span>= plotData % <span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;file&quot;</span>, file, <span style="color: #E6DB74;">&quot;fakeFile&quot;</span>, fakeFile, <span style="color: #E6DB74;">&quot;run&quot;</span>, $run, <span style="color: #E6DB74;">&quot;run&quot;</span>, $run, <span style="color: #E6DB74;">&quot;run&quot;</span>, $run<span style="color: #AE81FF;">]</span>
        shell:
          <span style="color: #AE81FF;">(</span>$plotCmd<span style="color: #AE81FF;">)</span>
<span style="color: #F92672;">when</span> <span style="color: #AE81FF;">isMainModule</span>:
  <span style="color: #F92672;">import</span> cligen
  dispatch main
</pre>
</div>

<p>
which yields all the plots in:
<a href="./Figs/fakeEventSimulation/runComparisons/">./Figs/fakeEventSimulation/runComparisons/</a>
</p>

<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>RERUN WITH CSV OUTPUT!</b></li>
</ul>

<p>
First we generate the HDF5 file containing all the fake data:
</p>
<div class="org-src-container">
<pre class="src src-sh">ntangle thesis.org &amp;&amp; nim c code/generate_all_run_fake_data_plots
code/generate_all_run_fake_data_plots --generate
</pre>
</div>
<p>
and then we can create all the plots. In the section below we will
include the plot for run 241.
</p>

<div class="org-src-container">
<pre class="src src-sh">code/generate_all_run_fake_data_plots --plot
</pre>
</div>

<p>
<b>Note</b>: The <code>plotDatasetGgplot</code> program already uses <code>themeLatex</code> to
get a pretty version of the plot. For that reason no further
modification is needed to get pretty plots.
</p>
</div>
</div>
</div>
<div id="outline-container-sec:background:mlp:determine_gas_diffusion" class="outline-4">
<h4 id="sec:background:mlp:determine_gas_diffusion">  <a href="#sec:background:mlp:determine_gas_diffusion">  <span class="section-number-4">12.4.3.</span> Determination of gas diffusion</a></h4>
<div id="text-sec:background:mlp:determine_gas_diffusion" class="outline-text-4">
<p>
To generate X-ray events with similar physical properties as those
seen in a particular data taking run, we need the three inputs
required: gas gain, gas diffusion and target energy. The determination
of the gas gain is a well defined procedure, explained in sections
<a href="./operation_calibration.html#sec:daq:polya_distribution">8.1.2</a> and
<a href="./calibration.html#sec:calib:gas_gain_time_binning">11.2.3</a>. The target energy is a matter of the
purpose the generated data is for. The gas diffusion however is more
complicated and requires explanation.
</p>

<p>
In theory the gas diffusion is a fixed parameter for a specific gas
mixture at fixed temperature, pressure and electromagnetic fields and
can be computed using Monte Carlo tools as mentioned already in
sec. <a href="./theory_detector.html#sec:theory:gas_diffusion">6.3.4</a>. However, in practice MC tools suffer
from significant uncertainty (in particular in the form of run-by-run
RNG variation) especially at common numbers of MC samples. More
importantly though, real detectors do not have perfectly stable and
known temperatures, pressures and electromagnetic fields (for example
the field inhomogeneity of the drift field in the 7-GridPix detector
is not perfectly known), leading to important differences and
variations from theoretical numbers.
</p>

<p>
Fortunately, similar to the gas gain, the effective numbers for the
gas diffusion can be extracted from real data. The gas diffusion
constant \(D_T\) (introduced in sec. <a href="./theory_detector.html#sec:theory:gas_diffusion">6.3.4</a>)
describes the standard deviation of the transverse distance from the
initial center after drifting a distance \(z\) along a homogeneous
electric field. This parameter is one of our geometric properties, in
the form of the transverse RMS &apos;\rmst&apos; (which is actually the transverse
standard deviation of the electron positions <sup>  <a id="fnr.16" href="#fn.16" class="footref" role="doc-backlink">16</a></sup>) and thus it can be used to
determine the gas diffusion coefficient. <sup>  <a id="fnr.17" href="#fn.17" class="footref" role="doc-backlink">17</a></sup> As it
is a statistical process and \(D_T\) is the standard deviation of the
population a large ensemble of events is needed.
</p>

<p>
This is possible both for calibration data as well as background
data. In both cases it is important to apply minor cuts to remove the
majority of events in which \rmst contains contributions that are not
due to gas diffusion (e.g. a double X-ray event which was not
separated by the cluster finder). For calibration data the cuts should
filter to single X-rays, whereas for background events clean muon
tracks are the target.
</p>

<p>
In theory one could determine it by way of a fit to the \rmst
distribution of all clusters in a data taking run as was done in
(<a href="./bibliography.html#citeproc_bib_item_137">Krieger et al. 2018</a>). This approach is problematic in practice due
to the large variation in possible X-ray conversion points – and
therefore drift distances – as a function of energy. Different
energies lead to different convolutions of exponential distributions
with the gas diffusion. This means a single fit does not describe the
data well in general and the upper limit is <span class="underline">not</span> a good estimator for
the real gas diffusion coefficient (because it is due to those X-rays
converting directly behind the detector window by chance and/or
undergoing statistically large amount of diffusion).
</p>

<p>
Instead of performing an analytical convolution of the exponential
distribution and gas diffusion distributions to determine the correct
fit, a Monte Carlo approach is used here as well. The idea here is to
use a reasonable <sup>  <a id="fnr.18" href="#fn.18" class="footref" role="doc-backlink">18</a></sup> gas diffusion constant to 
generate (simplified <sup>  <a id="fnr.19" href="#fn.19" class="footref" role="doc-backlink">19</a></sup>) fake events. Compute the
\rmst distribution of this fake data and compare to the real \rmst
distribution of the target run based on a test statistic. Next one computes the
derivative of the test statistic as a form of loss function and
adjusts the diffusion coefficient accordingly. Over a number of
iterations the distribution of the simulated \rmst distribution will
converge to the real \rmst distribution, if the choice of test
statistic is suitable. In other words we use gradient descent to find
that diffusion constant \(D_T\), which best reproduces the \rmst
distribution of our target run.
</p>

<p>
One important point with regards to computing it for background data
is the following: when generating fake events for the background
dataset, the equivalent &apos;conversion point&apos; for muons is a uniform
distribution over all distances from the detector window to the
readout. For X-rays it is given by the energy dependent attenuation
length in the detector gas.
</p>

<p>
The test statistic chosen in practice for this purpose is the
Cramér-von-Mises (CvM) criterion, defined by (<a href="./bibliography.html#citeproc_bib_item_71">Cramér 1928</a>; <a href="./bibliography.html#citeproc_bib_item_159">von Mises 1936</a>):
</p>

<p>
\[
ω² = ∫_{-∞}^∞ \left[ F_n(x) - F^*(x) \right]² \, \mathrm{d}F^*(x)
\]
</p>

<p>
where \(F_n(x)\) is an empirical distribution function to test for and
\(F^*(x)\) a cumulative distribution function to compare with. In case
of the two-sample case it can be computed as follows (<a href="./bibliography.html#citeproc_bib_item_10">Anderson 1962</a>):
</p>

<p>
\[
T = \frac{N M}{N + M} ω² = \frac{U}{N M (N + M)} - \frac{4 M N - 1}{6(M + N)}
\]
</p>

<p>
with \(U\):
</p>

<p>
\[
U = N \sum_{i=1}^N \left( r_i - i \right)² + M \sum_{j = 1}^M \left( s_j - j \right)²
\]
</p>

<p>
In contrast to – for example – the Kolmogorov-Smirnov test, it
includes the entire (E)CDF into the statistic instead of just the
largest deviation, which is a useful property to protect against
outliers.
</p>

<p>
The iterative optimization process is therefore
</p>

<p>
\[
D_{T,i+1} = D_{T,i} - η \frac{∂ f(D_T)}{∂ D_T}
\]
</p>

<p>
where \(f(D_T)\) is the Monte Carlo algorithm, which computes the test
statistic for a given \(D_T\) and \(η\) is the step size along the
gradient (a &apos;learning rate&apos;). The derivative is computed using finite
differences. <sup>  <a id="fnr.20" href="#fn.20" class="footref" role="doc-backlink">20</a></sup>
</p>

<p>
Fig. <a href="#fig:background:mlp:gas_diffusion_values">13</a> shows the values of the
transverse gas diffusion constant \(D_T\) as they have been determined for all
runs. The color scale is the value of the Cramér-von-Mises test
criterion and indicates a measure of the uncertainty of the obtained
parameter. We can see that for the CDL datasets (runs above 305) the
uncertainty is larger. This is because reproducing the \rmst
distribution for these datasets is more problematic than for the CAST
\cefe datasets (due to more data impurity caused by X-ray backgrounds
of energies other than the target fluorescence line). Generally we can
see variation in the range from about
\(\SIrange{620}{660}{μm.cm^{-1/2}}\) for the CAST data and larger
variation for the CDL data. The theoretical value we expect is about
\(\SI{670}{μm.cm^{-1/2}}\), implying the approach yields reasonable values.
</p>


<figure id="fig:background:mlp:gas_diffusion_values">
<img alt="σT_per_run.svg" class="org-svg" src="./figs/home/basti/phd/Figs/determineDiffusion/σT_per_run.svg" />

<figcaption>Figure 13: <span class="figure-number">Figure 60: </span>Transverse gas diffusion constant \(D_T\) as determined based on an iterative method attempting to reproduce the transverse RMS distribution using simulated X-ray events, the Cramér-von-Mises (CvM) test criterion and gradient descent. All runs above run 305 are CDL calibration runs. The color scale is the CvM test score. Background runs are marked by crosses.</figcaption>
</figure>

<p>
Because this iterative calculation is not computationally negligible, the
resulting \(D_T\) parameters for each run are cached in an HDF5
file. <sup>  <a id="fnr.21" href="#fn.21" class="footref" role="doc-backlink">21</a></sup>
</p>
</div>
<div id="outline-container-orge5d36cf" class="outline-5">
<h5 id="orge5d36cf">  <a href="#orge5d36cf">  <span class="section-number-5">12.4.3.1.</span> Note on handling background runs   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-3-1" class="outline-text-5">
<p>
When determining the diffusion constant for background runs, the value
we actually determine is about \(\SI{40}{μm.cm^{-1/2}}\) too large when
compared to \cefe calibration runs of the same time period (and thus
likely similar parameters). For this reason we simply subtract this
empirical difference as a constant offset. This is already done in the
plot of fig. <a href="#fig:background:mlp:gas_diffusion_values">13</a>.
</p>

<p>
The reason is almost certainly that the background data still contains
more events that are not very suitable to use for their transverse
RMS. But because it is a stable offset, we simply subtract it globally
for all background deduced values.
</p>

<p>
This guarantees that the cut values we actually produce for the MLP
matches those that would be valid for equivalent real X-ray data. If
we didn&apos;t do that we would likely get completely wrong cut values. The
idea is to only have a reference for the diffusion in the background
runs, which seems to be working correctly for its purpose.
</p>

<p>
A figure of the diffusion without the offset is (this plot is from
development), fig. <a href="#fig:background:diffusion_per_run_raw_background">14</a>. As
one can see the offset is very stable and thus removing it manually
should be fine.
</p>


<figure id="fig:background:diffusion_per_run_raw_background">
<img alt="σT_per_run_more_precise.svg" class="org-svg" src="./figs/home/basti/phd/Figs/gasGainAndNeighbors/σT_per_run_more_precise.svg" />

<figcaption>Figure 14: <span class="figure-number">Figure 61: </span>Development figure of the diffusion constant \(D_T\) (disregard the usage of \(σ_T\) in the figure) which shows the raw background values.</figcaption>
</figure>
</div>
</div>
<div id="outline-container-org829b973" class="outline-5">
<h5 id="org829b973">  <a href="#org829b973">  <span class="section-number-5">12.4.3.2.</span> Generate plot of diffusion constants for all runs <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-3-2" class="outline-text-5">
<p>
The <code>TimepixAnalysis/Tools/determineDiffusion</code> module can be compiled
as a standalone program and then run on an input HDF5 file. It will
perform the gradient descent optimization for every run in the file
and then create the figure of the section above.
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">WRITE_PLOT_CSV</span>=true ./determineDiffusion <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CDL_2019/CDL_2019_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/phd/Figs/determineDiffusion/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --histoPlotPath ~/phd/Figs/determineDiffusion/histograms <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX
</pre>
</div>

<ul class="org-ul">
<li class="on"><code>[X]</code> <b>RERUN THIS WITH TEX AND CSV FILES!</b>
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>RERUN WITH TEX AND CSV FOR SIMULATION ITSELF</b>
i.e. delete the cache tables before rerunning with <code>WRITE_PLOT_CSV=true</code></li>
</ul></li>

<li class="off"><code>[ ]</code> <b>CHECK Ag-Ag TARGET</b> Run 351 and 329</li>
</ul>


<p>
All the histograms created during the optimization process are stored
in the <code>histogram</code> subdirectory. It is one histogram with the \rmst
distribution and a fit similar to (<a href="./bibliography.html#citeproc_bib_item_137">Krieger et al. 2018</a>) and
(possibly multiple) comparison plots of the real and simulated data.
</p>

<p>
See also sec. <a href="./background.html#sec:background:mlp:gen_effective_efficiency_plots">12.4.7.1</a>.
</p>
</div>
</div>
</div>
<div id="outline-container-org6860293" class="outline-4">
<h4 id="org6860293">  <a href="#org6860293">  <span class="section-number-4">12.4.4.</span> Comparison of simulated events and real data</a></h4>
<div id="text-12-4-4" class="outline-text-4">
<p>
Fig. <a href="#fig:background:mlp:ridgeline_simulated_vs_real_run241">15</a> shows the
comparison of all cluster properties of simulated and real data for
one \cefe calibration run, number 241. The data is cut to the
photopeak and normalized to fit into a single plot. Each ridge shows a
kernel density estimation (KDE) of the data. Gas gain and gas diffusion are
extracted from the data as explained in previous sections.
</p>

<p>
All distributions outside the number of hits agree extremely
well. This is expected to not perfectly match. Only the total charge
is used as an input to the MLP and it matches pretty well on its
own. The reason is the neighboring activation logic in the MC
algorithm, which is empirically modified such that it produces correct
<span class="underline">geometric</span> behavior at the cost of slightly over- or underestimating
the number of hits. The Pólya sampling and general neighboring logic
is too simplistic to properly reproduce both aspects at the same
time. Other runs show similar agreement. Plots for all runs like this
can be found in the extended version of the thesis.
</p>


<figure id="fig:background:mlp:ridgeline_simulated_vs_real_run241">
<img alt="ingrid_properties_run_241_ridgeline_kde_by_run.svg" class="org-svg" src="./figs/home/basti/phd/Figs/fakeEventSimulation/runComparisons/ingrid_properties_run_241_ridgeline_kde_by_run.svg" />

<figcaption>Figure 15: <span class="figure-number">Figure 62: </span>Comparison of all geometric properties of simulated and real data of the \cefe photopeak clusters of run 241 in a ridgeline plot. Each ridge shows a KDE of the data. The gas gain and gas diffusion were first extracted from the real run data to simulate events of the photopeak. The two datasets agree very well with the exception of the number of hits (expected).</figcaption>
</figure>
</div>
<div id="outline-container-org2a3756e" class="outline-5">
<h5 id="org2a3756e">  <a href="#org2a3756e">  <span class="section-number-5">12.4.4.1.</span> Plots comparing simulated and real data   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-4-1" class="outline-text-5">
<p>
The figure shown in the main body and all other plots for all runs are
produced in sec. <a href="./background.html#sec:background:gen_plots_fake_event_comparisons">12.4.2.2</a>.
</p>

<p>
As it would also blow up the length of the extended thesis
unnecessarily, all the figures are simply found in:
</p>

<p>
<a href="./Figs/fakeEventSimulation/runComparisons/">./Figs/fakeEventSimulation/runComparisons/</a>
</p>
</div>
</div>
</div>
<div id="outline-container-org91f2d47" class="outline-4">
<h4 id="org91f2d47">  <a href="#org91f2d47">  <span class="section-number-4">12.4.5.</span> Overview of the best performing MLP</a></h4>
<div id="text-12-4-5" class="outline-text-4">
<p>
As a reference <sup>  <a id="fnr.22" href="#fn.22" class="footref" role="doc-backlink">22</a></sup> an MLP was trained on a mix of simulated X-rays and a subset
of the CAST background data. The implementation was done using
<a href="https://github.com/scinim/flambeau">Flambeau</a> (<a href="./bibliography.html#citeproc_bib_item_203">SciNim contributors 2023</a>), a wrapper for <a href="https://pytorch.org/">libtorch</a>
(<a href="./bibliography.html#citeproc_bib_item_166">Paszke et al. 2019</a>) <sup>  <a id="fnr.23" href="#fn.23" class="footref" role="doc-backlink">23</a></sup>, using the
parameters and inputs as shown in
tab. <a href="#tab:background:best_mlp_overview">24</a>. The network layout consists of
\(\num{14}\) input neurons, two hidden layers of only \(\num{30}\) neurons <sup>  <a id="fnr.24" href="#fn.24" class="footref" role="doc-backlink">24</a></sup>
each and \(\num{2}\) neurons on the output layer. We use the Adam
(<a href="./bibliography.html#citeproc_bib_item_132">Kingma and Ba 2014</a>) optimizer and \(\tanh\) as an activation function on
both hidden layers, with \(\text{sigmoid}\) used on the output
layer. The mean squared error between output and target vectors is
used as the loss function.
</p>

<table id="tab:background:best_mlp_overview">
<caption class="t-above"><span class="table-number">Table 24:</span> Overview of all MLP parameters and training information.</caption>

<colgroup>
<col class="org-left" />

<col class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">Property</th>
<th scope="col" class="org-left">Value</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">Input neurons</td>
<td class="org-left">14 (12 geometric, 2 non-geometric)</td>
</tr>

<tr>
<td class="org-left">Hidden layers</td>
<td class="org-left">2</td>
</tr>

<tr>
<td class="org-left">Neurons on hidden layer</td>
<td class="org-left">30</td>
</tr>

<tr>
<td class="org-left">Output neurons</td>
<td class="org-left">2</td>
</tr>

<tr>
<td class="org-left">Activation function</td>
<td class="org-left">\(\tanh\)</td>
</tr>

<tr>
<td class="org-left">Output layer activation function</td>
<td class="org-left">\(\text{sigmoid}\)</td>
</tr>

<tr>
<td class="org-left">Loss function</td>
<td class="org-left">Mean Squared Error (MSE)</td>
</tr>

<tr>
<td class="org-left">Optimizer</td>
<td class="org-left">Adam (<a href="./bibliography.html#citeproc_bib_item_132">Kingma and Ba 2014</a>)</td>
</tr>

<tr>
<td class="org-left">Learning rate</td>
<td class="org-left">\(\num{7e-4}\)</td>
</tr>

<tr>
<td class="org-left">Batch size</td>
<td class="org-left">8192</td>
</tr>

<tr>
<td class="org-left">Training data</td>
<td class="org-left">\(\num{250 000}\) simulated X-rays and \(\num{288 000}\)</td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left">real background events selected from the</td>
</tr>

<tr>
<td class="org-left"> </td>
<td class="org-left"><b>outer chips only</b>, same number for validation</td>
</tr>

<tr>
<td class="org-left">Training epochs</td>
<td class="org-left">\(\num{82 000}\)</td>
</tr>
</tbody>
</table>

<p>
The \(\num{14}\) input neurons were fed with all geometric properties
that do not scale directly with the energy (no number of active pixels
or direct energy) or position on the chip (as the training data is
skewed towards the center) with the exception of the total charge. For
the background data we avoid &apos;contaminating&apos; the training dataset with
CAST clusters that will end up as part of our background rate for the
limit calculation by only selecting clusters from chips other than the
center chip. This way the MLP is trained entirely on data independent
of the CAST data of interest. \(\num{250 000}\) synthetic X-rays and
\(\num{288 000}\) background clusters are used for training and the same
number of validation <sup>  <a id="fnr.25" href="#fn.25" class="footref" role="doc-backlink">25</a></sup>.
</p>

<p>
The first \(\num{25000}\) epochs are trained on a synthetic X-ray
dataset with a strong bias to low energy X-rays (clusters in the range
from \(\SIrange{0}{3}{keV}\) with the frequency dropping linearly with
increasing energy). This is because these are more difficult to
separate from background. Afterwards we switch to a separate set of
synthetic X-rays uniformly distributed in energies. Without this
approach the network can drift towards a local minimum, in which low
energy X-rays are classified as background, while still achieving good
accuracy. At the cost of low energy X-rays, low energy background
clusters are almost perfectly rejected. This is of course undesirable.
</p>

<p>
During the training process of \(\num{82 000}\) epochs, every
\(\num{1000}\) epochs the accuracy and loss are evaluated based on the
test sample and a model checkpoint is stored. The evolution of the
loss function for this network is shown in
fig. <a href="#fig:background:mlp:training_loss">16(a)</a>. We can see that
performance for the test dataset is mostly stagnant from around epoch
\(\num{10 000}\). The network develops minor overtraining, but this is
not important, because no training data is ever used in practice. 
</p>

<p>
Fig. <a href="#fig:background:mlp:validation_output">16(b)</a> shows the output of
neuron 0 (neuron 1 is a mirror image) for the test sample, with signal
like data towards 1 and background like data towards 0. The separation
is almost perfect, but a small amount of each type can be seen over
the entire range. This is expected due to the presence of real X-rays
in the background dataset, low energy events generally being similar
and statistical outliers in the X-ray dataset.
</p>


<figure id="fig:background:mlp:training_plots" class="figure-wrapper">
<figure id="fig:background:mlp:training_loss" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/neuralNetworks/17_11_23_adam_tanh30_sigmoid_mse_3keV/loss.svg" data-width="99%" />  <figcaption>Figure 16(a): Training loss</figcaption></figure> <figure id="fig:background:mlp:validation_output" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/neuralNetworks/17_11_23_adam_tanh30_sigmoid_mse_3keV/test_validation_log10.svg" data-width="99%" />  <figcaption>Figure 16(b): MLP prediction</figcaption></figure>
<figcaption>Figure 16: <a href="#fig:background:mlp:training_loss">16(a)</a> Loss over the training progress, evaluated every $\num{1000}$ epochs. Test data loss mostly stagnant after $\num{10000}$ epochs. <a href="#fig:background:mlp:validation_output">16(b)</a> Output of the validation data sample for neuron 0 as a $\log$ plot.</figcaption>
</figure>


<p>
Beyond the MLP prediction to indicate a measure of classification
efficiency, we can look at the receiver operating characteristic (ROC)
curves for different X-ray reference datasets of the CDL data. For
this we read all real center GridPix background data and all CDL
data. Both of these are split into the corresponding target/filter
combinations based on the energy of each cluster.
</p>

<p>
We can then define the background rejection \(b_{\text{rej}}\) for
different cut values \(c_i\) as
</p>

<p>
\[
b_{\text{rej}} = \frac{\text{card}(\{ b &lt; c_i\})}{\text{card}(\{ b \})},
\]
</p>

<p>
where \(b\) are all the predictions of the MLP for background data and
\(\text{card}\) is the cardinality of the set of numbers, i.e. the
number of entries. \(i\) refers to the \(i^{\text{th}}\) cut value (as
this is computed with discrete bins). In addition we define the signal
efficiency by
</p>


<p>
\[
s_{\text{eff}} = \frac{\text{card}(\{ s \geq c_i\})}{\text{card}(\{ s \})},
\]
</p>

<p>
where similarly \(s\) are the MLP predictions for X-ray clusters.
</p>

<p>
If we plot the pairs of signal efficiency and background rejection
values (one for each cut value), we produce a ROC curve. Done for each
CDL target/filter combination and similarly for the likelihood cut by
replacing the MLP prediction with the likelihood value of each
cluster, we obtain fig. <a href="#fig:roc_curves_logl_mlp">17</a>. The line style
indicates the classifier method (\(\ln\mathcal{L}\) or MLP) and the
color corresponds to the different targets and thus different
fluorescence lines. The improvement in background rejection at a fixed
signal efficiency exceeds \(\SI{10}{\percent}\) at multiple
energies. The only region where the MLP is worse than the likelihood
cut is for the Cu-EPIC \(\SI{0.9}{kV}\) target/filter combination at
signal efficiencies above about \(\SI{92}{\%}\). Especially the shape of
the ROC curve for the MLP at high signal efficiencies implies that it
should produce comparable background rates to the likelihood cut at
higher signal efficiencies.
</p>


<figure id="fig:roc_curves_logl_mlp">
<img alt="roc_curve_combined_split_by_target.svg" class="org-svg" src="./figs/home/basti/phd/Figs/neuralNetworks/17_11_23_adam_tanh30_sigmoid_mse_3keV/roc_curve_combined_split_by_target.svg" />

<figcaption>Figure 17: <span class="figure-number">Figure 63: </span>ROC curves for the comparison of the likelihood cut method (solid lines) to the MLP predictions (dashed lines), colored by the different targets used to generate the reference datasets. The background data used for each target corresponds to background clusters in an energy range around the fluorescence line. The MLP visibly outperforms the likelihood cut method in all energies. At the same signal efficiency (x axis) a significantly higher background rejection is achieved.</figcaption>
</figure>
</div>
<div id="outline-container-org2fa4080" class="outline-5">
<h5 id="org2fa4080">  <a href="#org2fa4080">  <span class="section-number-5">12.4.5.1.</span> Training code   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-5-1" class="outline-text-5">
<p>
The code performing the training is found <a href="https://github.com/Vindaar/TimepixAnalysis/blob/master/Tools/NN_playground/train_ingrid.nim">here</a>.
</p>
</div>
</div>
<div id="outline-container-orge84c3d1" class="outline-5">
<h5 id="orge84c3d1">  <a href="#orge84c3d1">  <span class="section-number-5">12.4.5.2.</span> More thoughts on different types of MLPs   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-5-2" class="outline-text-5">
<p>
As mentioned in the footnote of the main section above, we trained
different networks. Any network trained with stochastic gradient
descent (SGD) tends to require a lot longer to train and generally
struggles more to find the best parameters. Initially I focused on SGD
networks, due to the extreme overtraining found with Adam networks. It
was only when I tried ridiculously small networks that Adam ended up
being very useful after all.
</p>

<p>
The main reason I prefer the 30 neuron network over the mentioned 10
neuron network, is that it tends to generalize a little bit better
between different runs (i.e. gas gain and diffusion parameters) of
X-rays of similar energies. What this means is that the effective
efficiency is a bit more stable, in particular for the CDL data (where
we know the data differs more between runs and compared to our
synthetic data).
</p>

<p>
Next, we use a sigmoid function on the output layer and then simply the mean
squared error as the loss function. The advantage of the sigmoid on
the output is that the neuron predictions will be effectively clamped
between \((0, 1)\). This is useful, because otherwise there is always a
chance the network will produce different &apos;distributions&apos; in the
output prediction. So that inputs with differing parameters produce
strictly different outputs in different ranges. This is not
necessarily a problem (it can even be useful), but it can become one
if the synthetic dataset from which a cut value is computed ends
(partially) in a different output distribution than the target data
(e.g. a \cefe calibration data). It tends to exacerbate potential
differences between synthetic and real data leading to larger
differences in the target and real efficiencies.
</p>

<p>
Also, the nice aspect about the models being so tiny, is that it makes
it much easier to look into the model. In particular the mentioned 10
neuron model (2 layers of 10 neurons each) can easily be represented
as a set of 2 matrices and 2 bias vectors, small enough to look at by
eye. With the 30 neuron model we mostly use finally, this is starting
to be on the annoying and opaque end of course. Still much better than
a 300x300 matrix!
</p>
</div>
</div>
<div id="outline-container-org6fdea35" class="outline-5">
<h5 id="org6fdea35">  <a href="#org6fdea35">  <span class="section-number-5">12.4.5.3.</span> Network validation   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-5-3" class="outline-text-5">
<p>
In order to validate the usability and performance of any MLP we
train, we look at the following things (outside loss, accuracy and train / test
predictions):
</p>
<ol class="org-ol">
<li>the direct MLP prediction values for all the different CDL
runs. This use useful to check that all data, independent of energy
and gas properties is classified as X-rays (as mentioned in the
text, low energy X-rays can end up in the background class).</li>
<li>the effective efficiencies for each calibration and CDL run, as
explained in sec. <a href="./background.html#sec:background:mlp:mlp_cut_value">12.4.6</a> and
<a href="./background.html#sec:background:mlp:effective_efficiency">12.4.7</a>. The real efficiency
should be as close to the target efficiency as possible. Otherwise
the network does not generalize from synthetic data to the real
data as it should.</li>
<li>the ROC curve split by each CDL fluorescence line. Ideally it
should be higher than the likelihood cut ROC curve for all lines
and in all relevant signal efficiency regions. If this is not the
case the network is likely not going to outperform the likelihood
cut method.</li>
</ol>

<p>
All the plots for the MLP we use can be found in XXXXXXXXX   
</p>
</div>
</div>
<div id="outline-container-org5f91c3a" class="outline-5">
<h5 id="org5f91c3a">  <a href="#org5f91c3a">  <span class="section-number-5">12.4.5.4.</span> Generate plots of the loss, accuracy and output <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-5-4" class="outline-text-5">
<p>
The plots we care about for the context of the thesis are:
</p>
<ul class="org-ul">
<li>validation output as log10 plot &lt;- <code>train_ingrid</code></li>
<li>loss &lt;- <code>train_ingrid</code></li>
<li>ROC curve &lt;- <code>train_ingrid</code></li>
</ul>

<p>
The ROC curve is generated by <code>train_ingrid</code> when passing the <code>--predict</code>
argument. The first two are either generated during training or when
using the <code>--skipTraining</code> argument without <code>--predict</code> (due to
different data requirements).
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">WRITE_PLOT_CSV</span>=true <span style="color: #FD971F;">USE_TEX</span>=true ./train_ingrid <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simFiles ~/CastData/data/FakeData/fakeData_500k_uniform_energy_0_10_keV.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --model ~/org/resources/nn_devel_mixing/17_11_23_adam_tanh30_sigmoid_mse_3keV/mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.pt <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/phd/Figs/neuralNetworks/17_11_23_adam_tanh30_sigmoid_mse_3keV/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets eccentricity <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets skewnessLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets skewnessTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets kurtosisLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets kurtosisTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets length <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets width <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rmsLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rmsTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets lengthDivRmsTrans <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rotationAngle <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets fractionInTransverseRms <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets totalCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets σT <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --activation tanh <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outputActivation sigmoid <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lossFunction MSE <span style="color: #E6DB74; font-weight: bold;">\</span>
    --optimizer Adam <span style="color: #E6DB74; font-weight: bold;">\</span>
    --learningRate 7e-4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simulatedData <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundRegion crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --nFake 250_000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 1 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 2 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 6 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --clamp 50 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --predict
</pre>
</div>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">WRITE_PLOT_CSV</span>=true <span style="color: #FD971F;">USE_TEX</span>=true ./train_ingrid <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simFiles ~/CastData/data/FakeData/fakeData_500k_uniform_energy_0_10_keV.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --model ~/org/resources/nn_devel_mixing/17_11_23_adam_tanh30_sigmoid_mse_3keV/mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.pt <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/phd/Figs/neuralNetworks/17_11_23_adam_tanh30_sigmoid_mse_3keV/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets eccentricity <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets skewnessLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets skewnessTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets kurtosisLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets kurtosisTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets length <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets width <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rmsLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rmsTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets lengthDivRmsTrans <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rotationAngle <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets fractionInTransverseRms <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets totalCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets σT <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --activation tanh <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outputActivation sigmoid <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lossFunction MSE <span style="color: #E6DB74; font-weight: bold;">\</span>
    --optimizer Adam <span style="color: #E6DB74; font-weight: bold;">\</span>
    --learningRate 7e-4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simulatedData <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundRegion crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --nFake 250_000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 1 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 2 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 6 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --clamp 50 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --skipTraining
</pre>
</div>


<p>
<a href="./Figs/neuralNetworks/17_11_23_adam_tanh30_mse/">./Figs/neuralNetworks/17_11_23_adam_tanh30_mse/</a>
</p>

<p>
which are a direct copy of the initial plots generated from the call
above.
</p>

<ul class="org-ul">
<li class="off"><code>[ ]</code> We will likely want slightly nicer versions of these plots of
course!
The H5 file
<a href="./../org/resources/nn_devel_mixing/10_05_23_sgd_gauss_diffusion_tanh300_mse_loss/mlp_desc_v2.h5">./../org/resources/nn_devel_mixing/10_05_23_sgd_gauss_diffusion_tanh300_mse_loss/mlp_desc_v2.h5</a>
contains all the required losses etc for the plots!
The model file we use is:
<a href="./../org/resources/nn_devel_mixing/10_05_23_sgd_gauss_diffusion_tanh300_mse_loss/mlp_tanh300_msecheckpoint_epoch_485000_loss_0.0055_acc_0.9933.pt">./../org/resources/nn_devel_mixing/10_05_23_sgd_gauss_diffusion_tanh300_mse_loss/mlp_tanh300_msecheckpoint_epoch_485000_loss_0.0055_acc_0.9933.pt</a></li>
</ul>


<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>CREATE COMMAND TO GENERATE THE PLOTS FROM THE MODEL!</b></li>
</ul>
</div>
</div>
<div id="outline-container-orge51212a" class="outline-5">
<h5 id="orge51212a">  <a href="#orge51212a">  <span class="section-number-5">12.4.5.5.</span> Other networks trained   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-5-5" class="outline-text-5">
<p>
In the course of developing the MLP presented in the thesis, we
trained a large number of different networks.
</p>

<p>
We varied:
</p>
<ul class="org-ul">
<li>learning rates</li>
<li>optimizers (SGD, Adam, AdaGrad)
<ul class="org-ul">
<li>optimizer settings (momentum etc)</li>
</ul></li>
<li>activation functions (\(\tanh\), ReLU, GeLU, softmax, eLU)</li>
<li>loss functions (MSE, L1, …)</li>
<li>linear output layer or non linear output layer</li>
<li>number of hidden layers</li>
<li>number of neurons on hidden layer</li>
<li>input neurons (including / excluding individual variables)</li>
</ul>

<p>
As usual when training machine learning models, it&apos;s all a bit hit and
miss. :)
</p>

<p>
I may upload all the models, plots (in addition to the notes) to a
repository / Zenodo, but there is not <span class="underline">that</span> much of interest there I imagine.
</p>

<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>UPLOAD ALL MODELS</b></li>
</ul>
</div>
</div>
<div id="outline-container-orgf5c3c69" class="outline-5">
<h5 id="orgf5c3c69">  <a href="#orgf5c3c69">  <span class="section-number-5">12.4.5.6.</span> Generate plot of the ROC curve   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-5-6" class="outline-text-5">
<p>
The ROC curve plot is generated by <code>train_ingrid</code> if the <code>--predict</code>
argument is given, specifically via the <code>targetSpecificRocCurve</code> proc.
</p>

<p>
Generate the ROC curve for the MLP trained on the outer chips:
</p>
<div class="org-src-container">
<pre class="src src-sh">./train_ingrid <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --model ~/org/resources/nn_devel_mixing/30_10_23_sgd_tanh300_mse_loss_outer_chips/mlp_tanh_sigmoid_MSE_SGD_300_2.pt <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/phd/Figs/neuralNetworks/30_10_23_sgd_tanh300_mse_outer_chips/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 300 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 300 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --activation tanh <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outputActivation sigmoid <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lossFunction MSE <span style="color: #E6DB74; font-weight: bold;">\</span>
    --optimizer SGD <span style="color: #E6DB74; font-weight: bold;">\</span>
    --learningRate 7e-4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simulatedData <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundRegion crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --nFake 250_000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 1 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 2 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 6 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --predict
</pre>
</div>
</div>
</div>
<div id="outline-container-orgb0c746d" class="outline-5">
<h5 id="orgb0c746d">  <a href="#orgb0c746d">  <span class="section-number-5">12.4.5.7.</span> Training call for the best performing MLP   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-5-7" class="outline-text-5">
<p>
All MLPs we trained, including the one we mainly use in the end, use
the following datasets as inputs:
</p>
<ul class="org-ul">
<li>eccentricity</li>
<li>skewnessLongitudinal</li>
<li>skewnessTransverse</li>
<li>kurtosisLongitudinal</li>
<li>kurtosisTransverse</li>
<li>length</li>
<li>width</li>
<li>rmsLongitudinal</li>
<li>rmsTransverse</li>
<li>lengthDivRmsTrans</li>
<li>rotationAngle</li>
<li>fractionInTransverseRms</li>
<li>totalCharge</li>
<li>σT (&lt;- this is actually \(D_T\))</li>
</ul>

<p>
This section (anything below) is extracted from
<a href="./../org/journal.html">./../org/journal.html</a>.
</p>

<p>
Start training at <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-11-17 Fri 19:42&gt;</span></span>:
</p>
<div class="org-src-container">
<pre class="src src-sh">./train_ingrid <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simFiles ~/CastData/data/FakeData/fakeData_500k_0_to_3keV_decrease.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --modelOutpath ~/org/resources/nn_devel_mixing/17_11_23_adam_tanh30_sigmoid_mse_3keV/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/Sync/17_11_23_adam_tanh30_sigmoid_mse_3keV/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets eccentricity <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets skewnessLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets skewnessTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets kurtosisLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets kurtosisTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets length <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets width <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rmsLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rmsTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets lengthDivRmsTrans <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rotationAngle <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets fractionInTransverseRms <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets totalCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets σT <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --activation tanh <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outputActivation sigmoid <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lossFunction MSE <span style="color: #E6DB74; font-weight: bold;">\</span>
    --optimizer Adam <span style="color: #E6DB74; font-weight: bold;">\</span>
    --learningRate 7e-4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simulatedData <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundRegion crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --nFake 250_000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 1 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 2 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 6 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --clamp 5000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotEvery 1000
</pre>
</div>
<p>
Stopped after 25k <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-11-17 Fri 20:09&gt;</span></span>
</p>

<p>
Now continue with uniform data:
</p>

<div class="org-src-container">
<pre class="src src-sh">./train_ingrid <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simFiles ~/CastData/data/FakeData/fakeData_500k_uniform_energy_0_10_keV.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --model ~/org/resources/nn_devel_mixing/17_11_23_adam_tanh30_sigmoid_mse_3keV/mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_25000_loss_0.0253_acc_0.9657.pt <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/Sync/17_11_23_adam_tanh30_sigmoid_mse_3keV/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets eccentricity <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets skewnessLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets skewnessTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets kurtosisLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets kurtosisTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets length <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets width <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rmsLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rmsTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets lengthDivRmsTrans <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rotationAngle <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets fractionInTransverseRms <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets totalCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets σT <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --activation tanh <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outputActivation sigmoid <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lossFunction MSE <span style="color: #E6DB74; font-weight: bold;">\</span>
    --optimizer Adam <span style="color: #E6DB74; font-weight: bold;">\</span>
    --learningRate 7e-4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simulatedData <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundRegion crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --nFake 250_000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 1 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 2 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 6 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --clamp 5000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotEvery 1000
</pre>
</div>

<p>
Stopped after 82k epochs <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-11-17 Fri 21:14&gt;</span></span>.
</p>
</div>
<ol class="org-ol">
<li><a id="org4673dda"> </a><code>nn_predict</code><br />
<div id="text-12-4-5-7-1" class="outline-text-6">
<p>
Prediction:
</p>
<div class="org-src-container">
<pre class="src src-sh">./nn_predict <span style="color: #E6DB74; font-weight: bold;">\</span>
    --model ~/org/resources/nn_devel_mixing/17_11_23_adam_tanh30_sigmoid_mse_3keV/mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.pt <span style="color: #E6DB74; font-weight: bold;">\</span>
    --signalEff 0.9 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/Sync/17_11_23_adam_tanh30_sigmoid_mse_3keV
</pre>
</div>

<p>
All positive fortunately.
</p>
</div>
</li>
<li><a id="orgbf804da"> </a><code>effective_eff_55fe</code><br />
<div id="text-12-4-5-7-2" class="outline-text-6">
<div class="org-src-container">
<pre class="src src-sh">./effective_eff_55fe <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --model ~/org/resources/nn_devel_mixing/17_11_23_adam_tanh30_sigmoid_mse_3keV/mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.pt <span style="color: #E6DB74; font-weight: bold;">\</span>
    --ε 0.95 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/CDL_2019_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --evaluateFit --plotDatasets <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/Sync/run2_run3_17_11_23_adam_tanh30_sigmoid_mse_82k/ 
</pre>
</div>
<p>
Finished <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-11-17 Fri 21:29&gt;</span></span>.
</p>

<p>
<code>~/Sync/run2_run3_17_11_23_adam_tanh30_sigmoid_mse_82k/efficiency_based_on_fake_data_per_run_cut_val.pdf</code>
<img alt="efficiency_based_on_fake_data_per_run_cut_val.svg" class="org-svg" src="./figs/home/basti/phd/Figs/neuralNetworks/run2_run3_17_11_23_adam_tanh30_sigmoid_mse_82k/efficiency_based_on_fake_data_per_run_cut_val.svg" />
</p>

<p>
-&gt; Somewhat similar to the 10 neuron network <b>BUT</b> the spread is much
smaller and the CDL data is better predicted!
This might our winner.
</p>

<p>
90% <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-11-18 Sat 08:59&gt;</span></span>
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">WRITE_PLOT_CSV</span>=true <span style="color: #FD971F;">USE_TEX</span>=true ./effective_eff_55fe <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --model ~/org/resources/nn_devel_mixing/17_11_23_adam_tanh30_sigmoid_mse_3keV/mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.pt <span style="color: #E6DB74; font-weight: bold;">\</span>
    --ε 0.90 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/CDL_2019_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --evaluateFit --plotDatasets <span style="color: #E6DB74; font-weight: bold;">\</span>
    --generatePlots --generateRunPlots <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/Sync/run2_run3_17_11_23_adam_tanh30_sigmoid_mse_mlp90_82k/ 
</pre>
</div>
</div>
</li>
<li><a id="orgf0f223c"> </a><code>train_ingrid --predict</code><br />
<div id="text-12-4-5-7-3" class="outline-text-6">
<p>
<span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-11-17 Fri 21:30&gt;</span></span>
</p>
<div class="org-src-container">
<pre class="src src-sh">./train_ingrid <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --back ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simFiles ~/CastData/data/FakeData/fakeData_500k_uniform_energy_0_10_keV.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --model ~/org/resources/nn_devel_mixing/17_11_23_adam_tanh30_sigmoid_mse_3keV/mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.pt <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/Sync/17_11_23_adam_tanh30_sigmoid_mse_3keV/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets eccentricity <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets skewnessLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets skewnessTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets kurtosisLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets kurtosisTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets length <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets width <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rmsLongitudinal <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rmsTransverse <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets lengthDivRmsTrans <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets rotationAngle <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets fractionInTransverseRms <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets totalCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --datasets σT <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --numHidden 30 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --activation tanh <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outputActivation sigmoid <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lossFunction MSE <span style="color: #E6DB74; font-weight: bold;">\</span>
    --optimizer Adam <span style="color: #E6DB74; font-weight: bold;">\</span>
    --learningRate 7e-4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --simulatedData <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundRegion crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --nFake 250_000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 1 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 2 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundChips 6 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --clamp 5000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --predict
</pre>
</div>
<p>
Finished <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-11-17 Fri 21:33&gt;</span></span>
</p>

<p>
Compared to the ROC curve of tanh10, this ROC curve looks a little bit
worse. But the effective efficiency predictions are quite a bit
better.
</p>

<p>
So we will choose based on what happens with the background and other
efficiencies.
</p>
</div>
</li>
<li><a id="orgebc9ee7"> </a><code>createAllLikelihoodCombinations</code><br />
<div id="text-12-4-5-7-4" class="outline-text-6">
<p>
Starting <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-11-18 Sat 00:03&gt;</span></span>:
</p>
<div class="org-src-container">
<pre class="src src-sh">./createAllLikelihoodCombinations <span style="color: #E6DB74; font-weight: bold;">\</span>
    --f2017 ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --f2018 ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --c2017 ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --c2018 ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --regions crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --vetoSets <span style="color: #E6DB74;">&quot;{fkMLP, +fkFadc, +fkScinti, +fkSeptem, fkLineVeto, fkExclusiveLineVeto}&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --mlpPath ~/org/resources/nn_devel_mixing/17_11_23_adam_tanh30_sigmoid_mse_3keV/mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.pt <span style="color: #E6DB74; font-weight: bold;">\</span>
    --fadcVetoPercentile 0.99 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --signalEfficiency 0.85 --signalEfficiency 0.9 --signalEfficiency 0.95 --signalEfficiency 0.98 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --out ~/org/resources/lhood_mlp_17_11_23_adam_tanh10_sigmoid_mse_82k/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --multiprocessing <span style="color: #E6DB74; font-weight: bold;">\</span>
    --jobs 4 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --dryRun
</pre>
</div>
<div class="org-src-container">
<pre class="src src-nil">Running all likelihood combinations took 11730.39050412178 s
</pre>
</div>
<p>
Despite only 2 jobs in the multithreaded case below (but excluding
fkMLP alone!!) it was still more than twice slower. So single threaded
is the better way.
</p>

<p>
Multithreaded took:
</p>
<div class="org-src-container">
<pre class="src src-nil">Running all likelihood combinations took 31379.83979129791 s
</pre>
</div>
</div>
</li>
<li><a id="org7a2beb2"> </a><code>plotBackgroundClusters</code>, <code>plotBackgroundRate</code><br />
<ol class="org-ol">
<li><a id="orgde15d6a"> </a>No vetoes<br />
<div id="text-12-4-5-7-5-1" class="outline-text-7">
<p>
Background clusters, 95%:
</p>
<div class="org-src-container">
<pre class="src src-sh">plotBackgroundClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMax 30 --title <span style="color: #E6DB74;">&quot;X-ray like clusters of CAST data MLP@95%&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp/ --filterNoisyPixels <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 --energyMax 12.0 --suffix <span style="color: #E6DB74;">&quot;_mlp_95_adam_tanh30_sigmoid_mse_82k&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundSuppression
</pre>
</div>

<p>
90%
</p>
<div class="org-src-container">
<pre class="src src-sh">plotBackgroundClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.9_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.9_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMax 30 --title <span style="color: #E6DB74;">&quot;X-ray like clusters of CAST data MLP@90%&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp/ --filterNoisyPixels <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 --energyMax 12.0 --suffix <span style="color: #E6DB74;">&quot;_mlp_90_adam_tanh30_sigmoid_mse_82k&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundSuppression
</pre>
</div>


<p>
Background rate, comparison:
</p>
<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.98_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.98_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.9_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.9_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.85_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.85_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;98&quot;</span> --names <span style="color: #E6DB74;">&quot;98&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;95&quot;</span> --names <span style="color: #E6DB74;">&quot;95&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;90&quot;</span> --names <span style="color: #E6DB74;">&quot;90&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;85&quot;</span> --names <span style="color: #E6DB74;">&quot;85&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 --title <span style="color: #E6DB74;">&quot;Background rate from CAST&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters --showTotalTime --topMargin 1.5 --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_adam_tanh30_mse_sigmoid_82k.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2
</pre>
</div>
</div>
</li>
<li><a id="org1ecea8f"> </a>Scinti+FADC+Line<br />
<div id="text-12-4-5-7-5-2" class="outline-text-7">
<p>
Background clusters, 95%:
</p>
<div class="org-src-container">
<pre class="src src-sh">plotBackgroundClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMax 30 --title <span style="color: #E6DB74;">&quot;X-ray like clusters of CAST data MLP@95%+Scinti+FADC+Line&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp/ --filterNoisyPixels <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 --energyMax 12.0 --suffix <span style="color: #E6DB74;">&quot;_mlp_95_scinti_fadc_line_adam_tanh30_sigmoid_mse_82k&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundSuppression
</pre>
</div>

<p>
85%
</p>
<div class="org-src-container">
<pre class="src src-sh">plotBackgroundClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.85_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.85_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMax 30 --title <span style="color: #E6DB74;">&quot;X-ray like clusters of CAST data MLP@85%+Scinti+FADC+Line&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp/ --filterNoisyPixels <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 --energyMax 12.0 --suffix <span style="color: #E6DB74;">&quot;_mlp_85_scinti_fadc_line_adam_tanh30_sigmoid_mse_82k&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundSuppression
</pre>
</div>

<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.98_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.98_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.9_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.9_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.85_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.85_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;98&quot;</span> --names <span style="color: #E6DB74;">&quot;98&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;95&quot;</span> --names <span style="color: #E6DB74;">&quot;95&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;90&quot;</span> --names <span style="color: #E6DB74;">&quot;90&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;85&quot;</span> --names <span style="color: #E6DB74;">&quot;85&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 --title <span style="color: #E6DB74;">&quot;Background rate from CAST&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters --showTotalTime --topMargin 1.5 --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_adam_tanh30_scinti_fadc_line_mse_sigmoid_82k.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2
</pre>
</div>
</div>
</li>
<li><a id="org26259e3"> </a>Scinti+FADC+Septem+Line<br />
<div id="text-12-4-5-7-5-3" class="outline-text-7">
<p>
Background clusters, 95%:
</p>
<div class="org-src-container">
<pre class="src src-sh">plotBackgroundClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMax 30 --title <span style="color: #E6DB74;">&quot;X-ray like clusters of CAST data MLP@95%+Scinti+FADC+Septem+Line&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp/ --filterNoisyPixels <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 --energyMax 12.0 --suffix <span style="color: #E6DB74;">&quot;_mlp_95_scinti_fadc_septem_line_adam_tanh30_sigmoid_mse_82k&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundSuppression
</pre>
</div>

<p>
85%
</p>
<div class="org-src-container">
<pre class="src src-sh">plotBackgroundClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.85_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.85_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMax 30 --title <span style="color: #E6DB74;">&quot;X-ray like clusters of CAST data MLP@85%+Scinti+FADC+Septem+Line&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp/ --filterNoisyPixels <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 --energyMax 12.0 --suffix <span style="color: #E6DB74;">&quot;_mlp_85_scinti_fadc_septem_line_adam_tanh30_sigmoid_mse_82k&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundSuppression
</pre>
</div>

<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.98_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.98_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.9_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.9_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.85_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.85_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;98&quot;</span> --names <span style="color: #E6DB74;">&quot;98&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;95&quot;</span> --names <span style="color: #E6DB74;">&quot;95&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;90&quot;</span> --names <span style="color: #E6DB74;">&quot;90&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;85&quot;</span> --names <span style="color: #E6DB74;">&quot;85&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 --title <span style="color: #E6DB74;">&quot;Background rate from CAST&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters --showTotalTime --topMargin 1.5 --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_adam_tanh30_scinti_fadc_septem_line_mse_sigmoid_82k.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2
</pre>
</div>
</div>
</li>
</ol>
</li>
</ol>
</div>
</div>
<div id="outline-container-sec:background:mlp:mlp_cut_value" class="outline-4">
<h4 id="sec:background:mlp:mlp_cut_value">  <a href="#sec:background:mlp:mlp_cut_value">  <span class="section-number-4">12.4.6.</span> Determination of MLP cut value</a></h4>
<div id="text-sec:background:mlp:mlp_cut_value" class="outline-text-4">
<p>
Based on the output distributions of the MLP,
fig. <a href="#fig:background:mlp:validation_output">16(b)</a>, it can be used to act
as a cluster discriminator in the same way as for the likelihood cut
method, following eq. \eqref{eq:background:lnL:cut_condition}. The likelihood
distribution is replaced by the MLP prediction via either of the two
output neurons. For example, as seen for one output neuron in
fig. <a href="#fig:background:mlp:validation_output">16(b)</a>, the software efficiency
would be determined based on the &apos;prediction&apos; value along the x-axis
such that the desired software efficiency is above the cut value. Any
cluster below would be removed. In practice again the empirical
distribution function of the signal-like output data is computed and
the quantile of the target software efficiency \(ε_{\text{eff}}\) is
determined. Similarly to the likelihood cut method this is done for
\(\num{8}\) different energy ranges corresponding to the CDL
fluorescence lines. However, no interpolation is performed because
only the single output distribution is known. <sup>  <a id="fnr.26" href="#fn.26" class="footref" role="doc-backlink">26</a></sup>
</p>

<p>
Due to the significant differences in gas gain between the CDL dataset
and the CAST \cefe calibration data, we do not use a single cut value
for each energy for all CAST data taking runs. Instead we use the
X-ray cluster simulation machinery to provide run specific X-ray
clusters from which to deduce cut values.
</p>

<p>
For each run we wish to apply the MLP to, we start by computing the
mean gas gain based on all gas gain intervals. Then we determine the
gas diffusion coefficient as explained in
sec. <a href="./background.html#sec:background:mlp:determine_gas_diffusion">12.4.3</a>. With two of the
three required parameters for X-ray generation in hand, we then
simulate X-rays following each X-ray fluorescence line measured in the
CDL dataset, yielding 8 different simulated datasets for each
run. Based on these we compute one cut value on the MLP output
each. The so deduced cut value is applied as the MLP cut to that run
in the valid energy range. The same approach is used for calibration
runs as well as for background runs.
</p>

<p>
This means <i>both</i> the MLP training as well as determination of cut
values is <i>entirely</i> based on synthetic data, only using aggregate
parameters of the real data. 
</p>
</div>
</div>
<div id="outline-container-sec:background:mlp:effective_efficiency" class="outline-4">
<h4 id="sec:background:mlp:effective_efficiency">  <a href="#sec:background:mlp:effective_efficiency">  <span class="section-number-4">12.4.7.</span> Verification of software efficiency using calibration data</a></h4>
<div id="text-sec:background:mlp:effective_efficiency" class="outline-text-4">
<p>
As the simulated X-ray clusters certainly differ from real X-rays,
verification of the software efficiency using calibration data is
required. For all \cefe calibration runs as well as all CDL data we
produce cut values as explained in the previous section. Then we apply
these to the real clusters of those runs and compute the effective
efficiency as
</p>

<p>
\[
ε_{\text{effective}} = \frac{N_{\text{cut}}}{N_{\text{total}}}
\]
</p>

<p>
where \(N_{\text{cut}}\) is the clusters remaining after applying the
MLP cut. \(N_{\text{total}}\) is the total number of clusters after
application of the standard cleaning cuts applied for the \cefe
spectrum fit and the CDL fluorescence line fits. 
</p>

<p>
The resulting efficiency is the effective efficiency the MLP
produces. A close match with the target software efficiency implies
the simulated X-ray data matches the real data well and the network
learned to identify X-rays based on physical properties (and not due
to overtraining). In the limit calculation later, the mean of all these
effective efficiencies of the \cefe calibration runs is used in place
of the target software efficiency as a realistic estimator for the
signal efficiency. The standard deviation of all these effective
efficiencies is one of the systematics used there.
</p>

<p>
Fig. <a href="#fig:background:mlp:effective_efficiencies">18</a> shows the effective
efficiencies obtained for all \cefe calibration and CDL runs using the
MLP introduced earlier. The marker symbol represents the energy of the
target fluorescence line. Note that the \cefe calibration escape peak
near \(\SI{3}{keV}\) is a separate data point <code>&quot;3.0&quot;</code> compared to the
silver fluorescence line given as <code>&quot;2.98&quot;</code>. That is because the two
are fundamentally different things. The escape event is a
\(\SI{5.9}{keV}\) photopeak X-ray entering the detector, which ends up
&apos;losing&apos; about \(\SI{3}{keV}\) due to excitation of an argon
fluorescence Kα X-ray. This means the absorption length is that of a
\(\SI{5.9}{keV}\) photon, whereas the silver fluorescence line
corresponds to a real \(\SI{2.98}{keV}\) photon and thus corresponding
absorption length. As such, the geometric properties on average are
different.
</p>

<p>
The target efficiency in the plot was \(\SI{95}{\%}\) with an achieved
mean effective efficiency close to the target. Values are slightly
lower in Run-2 (runs &lt; 200) and slightly above in Run-3. Variation is
significantly larger in the CDL runs (runs above number 305), but
often towards larger efficiencies. Only one CDL run is at
significantly lower efficiency (\(\SI{83}{\%}\)), with a few other lower
energy runs around \(\SI{91}{\%}\). The data quality in the CDL data is
lower and it has larger variation of the common properties (gas
gain, diffusion) as compared to the CAST dataset. 
</p>


<figure id="fig:background:mlp:effective_efficiencies">
<img alt="efficiency_based_on_fake_data_per_run_cut_val.svg" class="org-svg" src="./figs/home/basti/phd/Figs/neuralNetworks/17_11_23_adam_tanh30_sigmoid_mse_3keV/effectiveEff0.95/efficiency_based_on_fake_data_per_run_cut_val.svg" />

<figcaption>Figure 18: <span class="figure-number">Figure 64: </span>Effective efficiencies obtained using the MLP for a target \(ε = \SI{95}{\%}\) software efficiency. Runs above run 305 are CDL calibration runs. Different symbols represent different target fluorescence lines. The \cefe escape peak is indicated at \(\SI{3}{keV}\) in contrast to the \(\SI{2.98}{keV}\) silver line, due to different physical properties. The photopeak corresponds to <code>5.9</code> and the equivalent CDL Mn target to <code>5.89</code>.</figcaption>
</figure>
</div>
<div id="outline-container-sec:background:mlp:gen_effective_efficiency_plots" class="outline-5">
<h5 id="sec:background:mlp:gen_effective_efficiency_plots">  <a href="#sec:background:mlp:gen_effective_efficiency_plots">  <span class="section-number-5">12.4.7.1.</span> Generate effective efficiency plot   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:mlp:gen_effective_efficiency_plots" class="outline-text-5">
<p>
Before we can compute the effective efficiencies we need the cache
table. In principle we have it already for the data and calibration
runs, but we want a plot of all runs as well:
</p>
<div class="org-src-container">
<pre class="src src-sh">./determineDiffusion <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CDL_2019/CDL_2019_Reco.h5
</pre>
</div>

<p>
This currently produces the plot <code>~/Sync/σT_per_run.pdf</code>, which we
copied here:
<img alt="σT_per_run.svg" class="org-svg" src="./figs/home/basti/phd/Figs/σT_per_run.svg" />
</p>

<p>
Now we compute the effective efficiencies:
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">WRITE_PLOT_CSV</span>=true <span style="color: #FD971F;">USE_TEX</span>=true ./effective_eff_55fe <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/CastData/data/CalibrationRuns2017_Reco.h5 ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --model ~/org/resources/nn_devel_mixing/17_11_23_adam_tanh30_sigmoid_mse_3keV/mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.pt <span style="color: #E6DB74; font-weight: bold;">\</span>
    --ε 0.95 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/CDL_2019_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --evaluateFit --plotDatasets <span style="color: #E6DB74; font-weight: bold;">\</span>
    --generatePlots --generateRunPlots <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/phd/Figs/neuralNetworks/17_11_23_adam_tanh30_sigmoid_mse_3keV/effectiveEff0.95/
</pre>
</div>
</div>
</div>
<div id="outline-container-org0bd3020" class="outline-5">
<h5 id="org0bd3020">  <a href="#org0bd3020">  <span class="section-number-5">12.4.7.2.</span> Note on effective efficiency for CDL data   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-4-7-2" class="outline-text-5">
<p>
Of course it is not ideal that the CDL data sees larger variation than
the CAST data. However, on the one hand I&apos;m reasonably certain that
the variation would be quite a bit lower if the CDL data quality was
higher and more similar to the CAST data.
</p>

<p>
On the other hand, unfortunately we simply do not have the data to
cross check with the efficiency for each run. The parameters going
into the MLP (total charge and diffusion) depending on the detector
behavior are vitally important to get good cut values. If we had more
data for different energies, we could use better calculate effective
efficiencies, maybe individually for each target. But as of this
dataset, I would not fully trust those numbers.
</p>

<p>
And we can see that there are enough runs of low energy X-rays in the
CDL data with higher signal efficiency. This gives us confidence that
the synthetic data doesn&apos;t just give us cut values that are generally
too strict (underestimating the effective efficiency for low
energy X-rays always). 
</p>
</div>
</div>
</div>
<div id="outline-container-sec:background:mlp:background_rate" class="outline-4">
<h4 id="sec:background:mlp:background_rate">  <a href="#sec:background:mlp:background_rate">  <span class="section-number-4">12.4.8.</span> Background rate using MLP</a></h4>
<div id="text-sec:background:mlp:background_rate" class="outline-text-4">
<p>
Applying the MLP cut as explained leads to background rates as
presented in fig. <a href="#fig:background:mlp:background_rate">19</a>, where the MLP is
compared to the \(\ln\mathcal{L}\) cut (\(ε = \SI{80}{\%}\)) at a similar software
efficiency \(ε = \SI{84.7}{\%}\) as well as at a significantly higher
efficiency of \(ε = \SI{94.4}{\%}\). Note the efficiencies are effective
efficiencies as explained in
sec. <a href="./background.html#sec:background:mlp:effective_efficiency">12.4.7</a>.
</p>

<p>
The background suppression is significantly improved in particular at
lower energies, implying other cluster properties provide better
separation at those than the three inputs used for the likelihood
cut. The mean background rates between \(\SIrange{0.2}{8}{keV}\) for
each of these are <sup>  <a id="fnr.27" href="#fn.27" class="footref" role="doc-backlink">27</a></sup>:
</p>

\begin{align*}
b_{\ln\mathcal{L} @ \SI{80}{\%}} &amp;= \SI{2.06142(9643)e-05}{keV^{-1}.cm^{-2}.s^{-1}} \\
b_{\text{MLP} @ \SI{85}{\%}} &amp;= \SI{1.56523(8403)e-05}{keV^{-1}.cm^{-2}.s^{-1}} \\
b_{\text{MLP} @ \SI{95}{\%}} &amp;= \SI{2.01631(9537)e-05}{keV^{-1}.cm^{-2}.s^{-1}}
\end{align*}

<p>
The most important aspect is that the MLP allows to achieve very
comparable background rates at significantly higher
efficiencies. Accepting lower signal efficiencies does not come with a
noticeable improvement, implying that for the remaining clusters the
distinction between background and X-ray properties is very small (and
some remaining clusters surely are of X-ray origin).
</p>


<figure id="fig:background:mlp:background_rate">
<img alt="background_rate_gold_mlp_0.95_0.8_lnL.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/background_rate_gold_mlp_0.95_0.8_lnL.svg" />

<figcaption>Figure 19: <span class="figure-number">Figure 65: </span>Comparison of the background rate in the center region of the MLP at different efficiencies and the standard \(\ln\mathcal{L}\). The MLP improves most at low energies and achieves comparable or better rates at higher efficiencies.</figcaption>
</figure>
</div>
<div id="outline-container-sec:background:gen_bck_rate_mlp" class="outline-5">
<h5 id="sec:background:gen_bck_rate_mlp">  <a href="#sec:background:gen_bck_rate_mlp">  <span class="section-number-5">12.4.8.1.</span> Generate background rate plot for MLP <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:gen_bck_rate_mlp" class="outline-text-5">
<p>
Let&apos;s generate the background rate plot for the MLP.
</p>

<p>
We&apos;ll want to show 85% (84.7% effective) as direct comparison to LnL,
95% effective (94.4% effective) as reference because we use it later and
compare both to the 80% LnL.
</p>

<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.85_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.85_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@0.95&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@0.95&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@0.85&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@0.85&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL@0.8&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL@0.8&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate in center 5·5 mm², MLP at different ε&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_gold_mlp_0.95_0.8_lnL.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/background/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
</pre>
</div>
</div>
</div>
</div>
</div>
<div id="outline-container-sec:background:additional_vetoes" class="outline-3">
<h3 id="sec:background:additional_vetoes">  <a href="#sec:background:additional_vetoes">  <span class="section-number-3">12.5.</span> Additional detector features as vetoes</a></h3>
<div id="text-sec:background:additional_vetoes" class="outline-text-3">
<p>
Next we will cover how the additional detector features can be used as
vetoes to suppress even more background. We start by looking at the
scintillators in sec. <a href="./background.html#sec:background:scinti_veto">12.5.1</a>. Then we consider
the FADC as a veto based on time evolution of the events in
sec. <a href="./background.html#sec:background:fadc_veto">12.5.2</a>. Finally, we consider the outer GridPix
ring as further geometric vetoes in sec. <a href="./background.html#sec:background:septem_veto">12.5.3</a>
and sec. <a href="./background.html#sec:background:line_veto">12.5.4</a> in the form of the &apos;septem veto&apos;
and the &apos;line veto&apos;.
</p>

<p>
Note, in the context of these detector vetoes we will generally
apply these vetoes on top of the the \(ε = \SI{80}{\%}\)
\(\ln\mathcal{L}\) method. We will later come back to the MLP in sec. <a href="./background.html#sec:background:all_vetoes_combined">12.6</a>.
</p>
</div>
<div id="outline-container-sec:background:scinti_veto" class="outline-4">
<h4 id="sec:background:scinti_veto">  <a href="#sec:background:scinti_veto">  <span class="section-number-4">12.5.1.</span> Scintillators as vetoes</a></h4>
<div id="text-sec:background:scinti_veto" class="outline-text-4">
<p>
As introduced in theory section <a href="./theory_detector.html#sec:theory:xray_fluorescence">6.1.4</a> one
problematic source of background is X-ray fluorescence created due to
abundant muons interacting with material of – or close to – the
detector. The resulting X-rays, if detected, are indistinguishable
from those X-rays originating from an axion (or other particle of
study). And given the required background levels, cosmic induced X-ray
fluorescence plays a significant role in the datasets. Scintillators
(ideally 4π around the whole detector setup) can be very helpful to
reduce the influence of such background events by &apos;tagging&apos; certain
events. For a fully encapsulated detector, any (at least) muon induced
X-ray would be preceded by a signal in one (or multiple)
scintillators. As such, if the time \(t_s\) between the scintillator
trigger and the time of activity recorded with the GridPix is small
enough, the two are likely to be in real coincidence. 
</p>

<p>
In the setup used at CAST with a \(\SI{42}{\cm}\) times \(\SI{82}{\cm}\)
scintillator paddle (see sec. <a href="./septemboard.html#sec:detector:scintillators">7.7</a>), about
\(\sim\SI{30}{cm}\) above the detector center – and a \(\cos²(θ)\)
distribution for muons – a significant fraction of muons should be
tagged. Similarly, the second scintillator on the Septemboard
detector, the small SiPM behind the readout area should trigger
precisely in those cases where a muon traverses the detector from the
front or back such that the muon track is orthogonal to the readout
plane.
</p>

<p>
The term &apos;non trivial trigger&apos; in the following indicates events in
which a scintillator triggered and the number of clock cycles to the
GridPix readout was larger than \(\num{0}\) (scintillator had a trigger in
the first place) and smaller than \(\num{300}\). The latter cut is
somewhat arbitrary, as the idea is two things: 1. no clock cycles
of \(\num{4095}\) (indicates clock ran over) and 2. the physics involved
leading to coincidences typically takes place on time scales shorter
than \(\SI{100}{clock\;cycles} = \SI{2.5}{μs}\) (see
sec. <a href="./septemboard.html#sec:detector:scintillators">7.7</a>). Anything above should just be a
random coincidence. \(\num{300}\) is therefore just chosen to have a
buffer to where physical clock cycles start.
</p>

<p>
During the Run-3 data taking period a total of \(\num{69243}\) non
trivial veto paddle triggers were
recorded. <sup>  <a id="fnr.28" href="#fn.28" class="footref" role="doc-backlink">28</a></sup> The distribution of the
clock cycles after which the GridPix readout happened is shown in
fig. <a href="#fig:background:scintillator_clock_cycles">20</a> on the right. The narrow
peak at \(\SI{255}{clock\;cycles}\) seems to be some kind of artifact,
potentially a firmware bug. The corresponding GridPix data was
investigated and there is nothing unusual about it (neither cluster
properties nor individual events). The source therefore remains
unclear, but a physical origin is extremely unlikely as the peak is
exactly one clock cycle wide (unrealistic for a physical process in
this context) and coincidentally exactly \(\num{255}\) (<code>0xFF</code> in
hexadecimal), hinting towards a counting issue in the firmware. The
actual distribution looks as expected, being a more or less flat
distribution corresponding to muons traversing at different distances
to the readout. The real distribution does not start at exactly
\(\SI{0}{clock\;cycles}\) due to inherent processing delays and even
close tracks requiring some time to drift to the readout and
activating the FADC. Further, geometric effects play a role. Very
close to the grid, only perfectly parallel tracks can achieve low clock
cycles, but the further away different angles contribute to the same
times.
</p>

<p>
The SiPM recorded \(\num{4298}\) non trivial triggers in the same time,
which are shown in fig. <a href="#fig:background:scintillator_clock_cycles">20</a> on
the left. <sup>  <a id="fnr.29" href="#fn.29" class="footref" role="doc-backlink">29</a></sup> Also this distribution
looks more or less as expected, showing a peak towards low clock
cycles (typical ionization and therefore similar times to accumulate
enough charge to trigger) with a tail for less and less ionizing
tracks. The same physical cutoff as in the veto paddle distribution
is visible corresponding to the full \(\SI{3}{cm}\) of drift time.
</p>

<p>
Both of these distributions and the physically motivated cutoffs in
clock cycles motivate a scintillator veto at clock cycles somewhere
above \(\SIrange{60}{70}{clock\;cycles}\). To be on the conservative side
and because random coincidences are very unlikely (on the time scales
of several clock cycles; picking a value slightly larger implies a
negligible dead time), a scintillator veto cut of
\(\SI{150}{clock\;cycles}\) was chosen.
</p>

<p>
The resulting improvement of the background rate is shown in
fig. <a href="#fig:background:background_rate_scinti_veto">21</a>, albeit only for the
end of 2018 (Run-3) data as the scintillator trigger was not working
correctly in Run-2 (as mentioned in
sec. <a href="./cast.html#sec:cast:data_taking_woes">10.5</a>). The biggest improvements can be seen
in the \(\SI{3}{keV}\) and \(\SI{8}{keV}\) peaks, both of which are likely
X-ray fluorescence (\(\ce{Ar}_{Kα}\) \(\SI{3}{keV}\) &amp; \(\ce{Cu}_{Kα}\)
\(\SI{8}{keV}\)) and orthogonal muons (\(&gt;\SI{8}{keV}\)). Arguably the
improvements could be bigger, but the efficiency of the scintillator
was not ideal, resulting in some likely muon induced X-rays to remain
untagged. Lastly, the coverage of the scintillators is leaving large
angular areas without a possibility to be tagged.
</p>

<p>
For a future detector an almost \(4π\) scintillator setup and correctly
calibrated and tested scintillators are an extremely valuable upgrade.
</p>


<figure id="fig:background:scintillator_clock_cycles">
<img alt="scintillators_facet_main_run-1.svg" class="org-svg" src="./figs/home/basti/phd/Figs/scintillators/scintillators_facet_main_run-1.svg" />

<figcaption>Figure 20: <span class="figure-number">Figure 66: </span>Clock cycle distributions of both scintillators of the end of 2018 data. The data is filtered to all non-trivial triggers (non zero and less than \num{300}; there are individual random coincidences in values up to \num{4095} where all triggers whose counter overran are). The origin of the peak at \SI{255}{clocks} in the data of the paddle is unclear.</figcaption>
</figure>


<figure id="fig:background:background_rate_scinti_veto">
<img alt="background_rate_crGold_scinti_run3.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/background_rate_crGold_scinti_run3.svg" />

<figcaption>Figure 21: <span class="figure-number">Figure 67: </span>Background rate based on the Run-3 CAST data achieved by the addition of a scintillator cut veto of \(\SI{3.75}{μs}\) ($\num{150} clock cycles) for any cluster that initially passes the log likelihood cut. The biggest improvements can be seen in the \(\SI{3}{keV}\) and \(\SI{8}{keV}\) peaks, both of which are likely X-ray fluorescence (Cu &amp; Ar; both energies) and orthogonal muons (\(&gt;\SI{8}{keV}\)).</figcaption>
</figure>
</div>
<div id="outline-container-org49ff111" class="outline-5">
<h5 id="org49ff111">  <a href="#org49ff111">  <span class="section-number-5">12.5.1.1.</span> Generate background rate plot with scintillator vetoes   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-1-1" class="outline-text-5">
<p>
Let&apos;s generate the plot, first for all data (effect of scinti only
visible in 2018 though!):
</p>

<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate from CAST data, incl. scinti veto&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_crGold_scinti.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/background/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
</pre>
</div>
<p>
(old files:
    ~/phd/resources/background/autoGen/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold.h5</sub> \
    ~/phd/resources/background/autoGen/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold.h5</sub> \
    ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold</sub><sub>scinti.h5</sub> \
    ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti.h5</sub> \
)
</p>

<p>
And now only for Run-3:
</p>
<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate from CAST data in Run-3, incl. scinti veto&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_crGold_scinti_run3.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/background/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
</pre>
</div>

<p>
(old files:
    ~/phd/resources/background/autoGen/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold.h5</sub> \
    ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti.h5</sub> \
)
</p>
</div>
</div>
<div id="outline-container-org2dd9794" class="outline-5">
<h5 id="org2dd9794">  <a href="#org2dd9794">  <span class="section-number-5">12.5.1.2.</span> Angle estimate covered by veto paddle   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-1-2" class="outline-text-5">
<p>
Given a veto paddle size of \(\SI{42}{\cm}\) times \(\SI{82}{\cm}\) we can
calculate the rough arc angle covered by the paddle for muons hitting
detector material. The exact calculation of course would need to take
into account the extent of the detector, the fact that there is copper
in the vacuum tube in front of the detector etc. For now let&apos;s just
look at the center of the detector volume and see what is covered by
the paddle.
</p>

<p>
The lead shielding is \(\SI{10}{cm}\) above the detector. There are
maybe a bit less than another \(\SI{10}{cm}\) above the lead shielding
to the scintillator and the detector center is also a bit less than
\(\SI{10}{cm}\) away from the lead shielding above (\(\SI{78}{mm}\)
diameter gas volume diameter, with the rest of the housing and a
couple of centimeter spacing to the lead shielding).
</p>

<p>
Assuming \(\SI{30}{cm}\) from the center to the veto scintillator, the
angle to the sides is:
</p>

<p>
\[
α = \arctan\left(\frac{\SI{21}{cm}}{\SI{30}{cm}}\right) = \SI{35}{°}
\]
</p>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> math
<span style="color: #F92672;">echo</span> arctan<span style="color: #AE81FF;">(</span><span style="font-style: italic;">21</span>.<span style="font-style: italic;">0</span>/<span style="font-style: italic;">30</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span>.radToDeg
</pre>
</div>

<p>
so about \(\SI{35}{°}\) covering to either side. This implies
\(\SI{70}{°}\) coverage above the detector. In the long direction it&apos;s
even more due to the length of the paddle.
</p>

<p>
Given a roughly \(\cos²(θ)\) distribution of the muon background, this
should cover
</p>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> numericalnim, math

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">cos2</span><span style="color: #AE81FF;">(</span>θ: <span style="color: #66D9EF;">float</span>, ctx: <span style="color: #66D9EF;">NumContext</span><span style="color: #66D9EF;">[</span><span style="color: #66D9EF;">float</span>, <span style="color: #66D9EF;">float</span><span style="color: #66D9EF;">]</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">float</span> = cos<span style="color: #AE81FF;">(</span>θ<span style="color: #AE81FF;">)</span>*cos<span style="color: #AE81FF;">(</span>θ<span style="color: #AE81FF;">)</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">integrate</span><span style="color: #AE81FF;">(</span>start, stop: <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">float</span> =
  <span style="color: #FD971F;">result</span> = simpson<span style="color: #AE81FF;">(</span>cos2, start, stop<span style="color: #AE81FF;">)</span>
<span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;Integrate from -π/2 to π/2 = &quot;</span>, integrate<span style="color: #AE81FF;">(</span>-<span style="color: #66D9EF;">PI</span>/<span style="font-style: italic;">2</span>.<span style="font-style: italic;">0</span>, <span style="color: #66D9EF;">PI</span>/<span style="font-style: italic;">2</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span>, <span style="color: #E6DB74;">&quot; = π/2&quot;</span>
<span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;Integrate from -π/2 to π/2 = &quot;</span>, integrate<span style="color: #AE81FF;">(</span>-<span style="font-style: italic;">35</span>.<span style="font-style: italic;">0</span>.degToRad, <span style="font-style: italic;">35</span>.<span style="font-style: italic;">0</span>.degToRad<span style="color: #AE81FF;">)</span>
<span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;Fraction of total flux *in this dimension* = &quot;</span>, integrate<span style="color: #AE81FF;">(</span>-<span style="font-style: italic;">35</span>.<span style="font-style: italic;">0</span>.degToRad, <span style="font-style: italic;">35</span>.<span style="font-style: italic;">0</span>.degToRad<span style="color: #AE81FF;">)</span> / <span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">PI</span>/<span style="font-style: italic;">2</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span>
</pre>
</div>

<p>
So assuming a 2D case it already covers about 68% of the expected muon
flux, assuming a perfect trigger efficiency of the scintillator.
</p>
</div>
</div>
<div id="outline-container-org93d4f8a" class="outline-5">
<h5 id="org93d4f8a">  <a href="#org93d4f8a">  <span class="section-number-5">12.5.1.3.</span> Estimate expected number of muons in data taking   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-1-3" class="outline-text-5">
<p>
The detection volume of relevance for the center chip can be roughly
said to be \(\SI{4.2}{cm²}\) (1.4 cm wide chip, 3 cm height orthogonal
to sky). Assuming a muon rate of 1 cm⁻²·min⁻¹ yields for 1125 h of
background time:
</p>
<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> unchained
<span style="color: #F92672;">let</span> <span style="color: #FD971F;">rate </span>= <span style="font-style: italic;">1</span>.cm⁻²•min⁻¹
<span style="color: #F92672;">let</span> <span style="color: #FD971F;">time </span>= <span style="font-style: italic;">1125</span>.h
<span style="color: #F92672;">let</span> <span style="color: #FD971F;">area </span>= <span style="font-style: italic;">4</span>.<span style="font-style: italic;">2</span>.cm²
<span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;Expected number of muon counts = &quot;</span>, rate * time * area
</pre>
</div>

<p>
So about \num{284500} expected muons in front of the
detector. Assuming I can trust the <code>scintiInfo</code> tool of an output of
(see next section):
</p>
<blockquote>
<p>
Main triggers: 69243
FADC triggers 211683
Relevant triggers 142440
Open shutter time in s: 13.510434
Scinti1 rate: 0.3050257995269533 s<sup>-1</sup>
Scinti2 rate: 57.21503839180888 s<sup>-1</sup>
</p>
</blockquote>
<p>
so about 69k veto paddle triggers. That puts the efficiency at knowing
the paddle will only see about 68% of all muons
\(\sim\SI{35.8}{\%}\). Not great, but also not horrible. Given that we
know our threshold was likely a bit high, this seems to be a possible
ball park estimate.
</p>
</div>
</div>
<div id="outline-container-org9b158a2" class="outline-5">
<h5 id="org9b158a2">  <a href="#org9b158a2">  <span class="section-number-5">12.5.1.4.</span> Generate plots of scintillator data, study of 255 peak   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-1-4" class="outline-text-5">
<p>
The tool we use to plot the scintillator data is <code>Tools/scintiInfo</code>.
</p>

<p>
It generates output into the <code>out/&lt;plot path&gt;/</code> directory:
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">WRITE_PLOT_CSV</span>=true <span style="color: #FD971F;">USE_TEX</span>=true ./scintiInfo -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
             --plotPath ~/phd/Figs/scintillators/
</pre>
</div>
<p>
which generates 3 plots for each scintillator containing all data,
all non trivial data and all hits &gt; 0 and &lt; 300.
</p>

<p>
Now let&apos;s investigate what the events of the veto paddle look like in
which we have a peak &gt;250 clocks.
</p>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> std / <span style="color: #AE81FF;">[</span>sequtils, strformat<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> ingrid / <span style="color: #AE81FF;">[</span>tos_helpers, ingrid_types<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> pkg / <span style="color: #AE81FF;">[</span>ggplotnim, nimhdf5, datamancer, ginger<span style="color: #AE81FF;">]</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">allData</span><span style="color: #AE81FF;">(</span>h5f: <span style="color: #66D9EF;">H5File</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">DataFrame</span> =
  <span style="color: #FD971F;">result</span> = h5f.readDsets<span style="color: #AE81FF;">(</span>chipDsets = some<span style="color: #66D9EF;">(</span><span style="color: #A6E22E;">(</span>chip: <span style="font-style: italic;">3</span>, dsets: <span style="color: #66D9EF;">TpaIngridDsetKinds</span>.mapIt<span style="color: #E6DB74;">(</span>it.toDset<span style="color: #FD971F;">()</span><span style="color: #E6DB74;">)</span><span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">)</span>,
                         commonDsets = @<span style="color: #66D9EF;">[</span><span style="color: #E6DB74;">&quot;fadcReadout&quot;</span>, <span style="color: #E6DB74;">&quot;szint1ClockInt&quot;</span>, <span style="color: #E6DB74;">&quot;szint2ClockInt&quot;</span><span style="color: #66D9EF;">]</span><span style="color: #AE81FF;">)</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">plotEvents</span><span style="color: #AE81FF;">(</span>df: <span style="color: #66D9EF;">DataFrame</span>, run: <span style="color: #66D9EF;">int</span>, numEvents: <span style="color: #66D9EF;">int</span>, plotCount: <span style="color: #F92672;">var</span> <span style="color: #66D9EF;">int</span>,
                outpath: <span style="color: #66D9EF;">string</span>,
                fadcRun: <span style="color: #66D9EF;">ReconstructedFadcRun</span><span style="color: #AE81FF;">)</span> =
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">showFadc </span>= fadcRun.eventNumber.<span style="color: #F92672;">len</span> &gt; <span style="font-style: italic;">0</span>
  <span style="color: #F92672;">for</span> <span style="color: #AE81FF;">(</span>tup, subDf<span style="color: #AE81FF;">)</span> <span style="color: #F92672;">in</span> groups<span style="color: #AE81FF;">(</span>df.group_by<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;eventNumber&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span>:
    <span style="color: #F92672;">if</span> numEvents &gt; <span style="font-style: italic;">0</span> <span style="color: #F92672;">and</span> plotCount &gt; numEvents: <span style="color: #F92672;">break</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfEv </span>= subDf.dfToSeptemEvent<span style="color: #AE81FF;">()</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">eventNumber </span>= tup<span style="color: #AE81FF;">[</span><span style="font-style: italic;">0</span><span style="color: #AE81FF;">][</span><span style="font-style: italic;">1</span><span style="color: #AE81FF;">]</span>.<span style="color: #F92672;">toInt</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">pltSeptem </span>= ggplot<span style="color: #AE81FF;">(</span>dfEv, aes<span style="color: #66D9EF;">(</span>x, y, color = <span style="color: #E6DB74;">&quot;charge&quot;</span><span style="color: #66D9EF;">)</span>, backend = bkCairo<span style="color: #AE81FF;">)</span> +
      geom_point<span style="color: #AE81FF;">(</span>size = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
      xlim<span style="color: #AE81FF;">(</span><span style="font-style: italic;">0</span>, <span style="font-style: italic;">768</span><span style="color: #AE81FF;">)</span> + ylim<span style="color: #AE81FF;">(</span><span style="font-style: italic;">0</span>, <span style="font-style: italic;">768</span><span style="color: #AE81FF;">)</span> + scale_x_continuous<span style="color: #AE81FF;">()</span> + scale_y_continuous<span style="color: #AE81FF;">()</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>y = <span style="font-style: italic;">0</span>, xMin = <span style="font-style: italic;">128</span>, xMax = <span style="font-style: italic;">640</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>y = <span style="font-style: italic;">256</span>, xMin = <span style="font-style: italic;">0</span>, xMax = <span style="font-style: italic;">768</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>y = <span style="font-style: italic;">512</span>, xMin = <span style="font-style: italic;">0</span>, xMax = <span style="font-style: italic;">768</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>y = <span style="font-style: italic;">768</span>, xMin = <span style="font-style: italic;">128</span>, xMax = <span style="font-style: italic;">640</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>x = <span style="font-style: italic;">0</span>, yMin = <span style="font-style: italic;">256</span>, yMax = <span style="font-style: italic;">512</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>x = <span style="font-style: italic;">256</span>, yMin = <span style="font-style: italic;">256</span>, yMax = <span style="font-style: italic;">512</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>x = <span style="font-style: italic;">512</span>, yMin = <span style="font-style: italic;">256</span>, yMax = <span style="font-style: italic;">512</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>x = <span style="font-style: italic;">768</span>, yMin = <span style="font-style: italic;">256</span>, yMax = <span style="font-style: italic;">512</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>x = <span style="font-style: italic;">128</span>, yMin = <span style="font-style: italic;">0</span>, yMax = <span style="font-style: italic;">256</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>x = <span style="font-style: italic;">384</span>, yMin = <span style="font-style: italic;">0</span>, yMax = <span style="font-style: italic;">256</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>x = <span style="font-style: italic;">640</span>, yMin = <span style="font-style: italic;">0</span>, yMax = <span style="font-style: italic;">256</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>x = <span style="font-style: italic;">128</span>, yMin = <span style="font-style: italic;">512</span>, yMax = <span style="font-style: italic;">768</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>x = <span style="font-style: italic;">384</span>, yMin = <span style="font-style: italic;">512</span>, yMax = <span style="font-style: italic;">768</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_linerange<span style="color: #AE81FF;">(</span>aes = aes<span style="color: #66D9EF;">(</span>x = <span style="font-style: italic;">640</span>, yMin = <span style="font-style: italic;">512</span>, yMax = <span style="font-style: italic;">768</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      margin<span style="color: #AE81FF;">(</span>top = <span style="font-style: italic;">1</span>.<span style="font-style: italic;">5</span><span style="color: #AE81FF;">)</span> +
      ggtitle<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;Septem event of event {eventNumber} and run {run}. &quot;</span><span style="color: #AE81FF;">)</span>

    <span style="color: #F92672;">if</span> <span style="color: #F92672;">not</span> showFadc:
      pltSeptem + ggsave<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;{outpath}/septemEvents/septemEvent_run_{run}_event_{eventNumber}.pdf&quot;</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">else</span>:
      <span style="color: #75715E;"># </span><span style="color: #75715E;">prepare FADC plot, create canvas and place both next to one another</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">eventIdx </span>= fadcRun.eventNumber.<span style="color: #F92672;">find</span><span style="color: #AE81FF;">(</span>eventNumber<span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfFadc </span>= toDf<span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">{</span> <span style="color: #E6DB74;">&quot;x&quot;</span>         : toSeq<span style="color: #A6E22E;">(</span><span style="font-style: italic;">0</span> ..&lt; <span style="font-style: italic;">2560</span><span style="color: #A6E22E;">)</span>,
                          <span style="color: #E6DB74;">&quot;data&quot;</span>      : fadcRun.fadcData<span style="color: #A6E22E;">[</span>eventIdx, _<span style="color: #A6E22E;">]</span>.squeeze <span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">pltFadc </span>= ggplot<span style="color: #AE81FF;">(</span>dfFadc, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;x&quot;</span>, <span style="color: #E6DB74;">&quot;data&quot;</span><span style="color: #66D9EF;">)</span>, backend = bkCairo<span style="color: #AE81FF;">)</span> +
        geom_line<span style="color: #AE81FF;">()</span> +
        geom_point<span style="color: #AE81FF;">(</span>color = <span style="color: #E6DB74;">&quot;black&quot;</span>, alpha = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">1</span><span style="color: #AE81FF;">)</span> +
        ggtitle<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;Fadc signal of event {eventNumber} and run {run}&quot;</span><span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">var</span> <span style="color: #FD971F;">img </span>= initViewport<span style="color: #AE81FF;">(</span>wImg = <span style="font-style: italic;">1200</span>, hImg = <span style="font-style: italic;">600</span>, backend = bkCairo<span style="color: #AE81FF;">)</span>
      img.layout<span style="color: #AE81FF;">(</span><span style="font-style: italic;">2</span>, rows = <span style="font-style: italic;">1</span><span style="color: #AE81FF;">)</span>
      img.embedAsRelative<span style="color: #AE81FF;">(</span><span style="font-style: italic;">0</span>, ggcreate<span style="color: #66D9EF;">(</span>pltSeptem<span style="color: #66D9EF;">)</span>.view<span style="color: #AE81FF;">)</span>
      img.embedAsRelative<span style="color: #AE81FF;">(</span><span style="font-style: italic;">1</span>, ggcreate<span style="color: #66D9EF;">(</span>pltFadc<span style="color: #66D9EF;">)</span>.view<span style="color: #AE81FF;">)</span>      

      <span style="color: #F92672;">var</span> <span style="color: #FD971F;">area </span>= img.addViewport<span style="color: #AE81FF;">()</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">title </span>= &amp;<span style="color: #E6DB74;">&quot;Septemboard event and FADC signal for event {eventNumber}&quot;</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">text </span>= area.initText<span style="color: #AE81FF;">(</span>c<span style="color: #66D9EF;">(</span><span style="font-style: italic;">0</span>.<span style="font-style: italic;">5</span>, <span style="font-style: italic;">0</span>.<span style="font-style: italic;">05</span>, ukRelative<span style="color: #66D9EF;">)</span>, title, goText, taCenter, font = some<span style="color: #66D9EF;">(</span>font<span style="color: #A6E22E;">(</span><span style="font-style: italic;">16</span>.<span style="font-style: italic;">0</span><span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span>
      area.addObj text
      img.children.<span style="color: #F92672;">add</span> area
      img.draw<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;{outpath}/septemEvents/septem_fadc_run_{run}_event_{eventNumber}.pdf&quot;</span><span style="color: #AE81FF;">)</span>
    
    <span style="color: #F92672;">inc</span> plotCount

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">plotSzinti</span><span style="color: #AE81FF;">(</span>h5f: <span style="color: #66D9EF;">H5file</span>, df: <span style="color: #66D9EF;">DataFrame</span>, cutFn: <span style="color: #66D9EF;">FormulaNode</span>,
                title: <span style="color: #66D9EF;">string</span>, outpath: <span style="color: #66D9EF;">string</span>,
                fname: <span style="color: #66D9EF;">string</span>,
                numEventPlots: <span style="color: #66D9EF;">int</span>,
                plotEvents: <span style="color: #66D9EF;">bool</span>,
                showFadc: <span style="color: #66D9EF;">bool</span> = <span style="color: #AE81FF;">false</span>
               <span style="color: #AE81FF;">)</span> =
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">toGather </span>= df.getKeys<span style="color: #AE81FF;">()</span>.filterIt<span style="color: #AE81FF;">(</span>it <span style="color: #F92672;">notin</span> <span style="color: #66D9EF;">[</span><span style="color: #E6DB74;">&quot;runNumber&quot;</span>, <span style="color: #E6DB74;">&quot;eventNumber&quot;</span><span style="color: #66D9EF;">]</span><span style="color: #AE81FF;">)</span>
    
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfF </span>= df.filter<span style="color: #AE81FF;">(</span>cutFn<span style="color: #AE81FF;">)</span>
    .filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span>`eccentricity` &lt; <span style="font-style: italic;">10</span>.<span style="font-style: italic;">0</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
    .gather<span style="color: #AE81FF;">(</span>toGather, <span style="color: #E6DB74;">&quot;key&quot;</span>, <span style="color: #E6DB74;">&quot;value&quot;</span><span style="color: #AE81FF;">)</span>
    
  <span style="color: #F92672;">echo</span> dfF
  ggplot<span style="color: #AE81FF;">(</span>dfF, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;value&quot;</span>, fill = <span style="color: #E6DB74;">&quot;key&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
    facet_wrap<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;key&quot;</span>, scales = <span style="color: #E6DB74;">&quot;free&quot;</span><span style="color: #AE81FF;">)</span> + 
    geom_histogram<span style="color: #AE81FF;">(</span>position = <span style="color: #E6DB74;">&quot;identity&quot;</span>, binBy = <span style="color: #E6DB74;">&quot;subset&quot;</span><span style="color: #AE81FF;">)</span> +
    legendPosition<span style="color: #AE81FF;">(</span><span style="font-style: italic;">0</span>.<span style="font-style: italic;">90</span>, <span style="font-style: italic;">0</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
    ggtitle<span style="color: #AE81FF;">(</span>title<span style="color: #AE81FF;">)</span> + 
    ggsave<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;{outpath}/{fname}&quot;</span>, width = <span style="font-style: italic;">2000</span>, height = <span style="font-style: italic;">2000</span><span style="color: #AE81FF;">)</span>

  <span style="color: #F92672;">if</span> plotEvents:
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">plotCount </span>= <span style="font-style: italic;">0</span>
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">fadcRun</span>: <span style="color: #66D9EF;">ReconstructedFadcRun</span>
    <span style="color: #F92672;">for</span> <span style="color: #AE81FF;">(</span>tup, subDf<span style="color: #AE81FF;">)</span> <span style="color: #F92672;">in</span> dfF.group_by<span style="color: #AE81FF;">(</span>@<span style="color: #66D9EF;">[</span><span style="color: #E6DB74;">&quot;runNumber&quot;</span><span style="color: #66D9EF;">]</span><span style="color: #AE81FF;">)</span>.groups:
      <span style="color: #F92672;">if</span> numEventPlots &gt; <span style="font-style: italic;">0</span> <span style="color: #F92672;">and</span> plotCount &gt; numEventPlots: <span style="color: #F92672;">break</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">run </span>= tup<span style="color: #AE81FF;">[</span><span style="font-style: italic;">0</span><span style="color: #AE81FF;">][</span><span style="font-style: italic;">1</span><span style="color: #AE81FF;">]</span>.<span style="color: #F92672;">toInt</span>
      <span style="color: #F92672;">if</span> showFadc:
        fadcRun = h5f.readRecoFadcRun<span style="color: #AE81FF;">(</span>run<span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;Run &quot;</span>, run
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">events </span>= subDf<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;eventNumber&quot;</span>, <span style="color: #66D9EF;">int</span><span style="color: #AE81FF;">]</span>.toSeq1D
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfS </span>= getSeptemDataFrame<span style="color: #AE81FF;">(</span>h5f, run<span style="color: #AE81FF;">)</span>
        .filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">int</span>: `eventNumber` <span style="color: #F92672;">in</span> events<span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">echo</span> dfS
      plotEvents<span style="color: #AE81FF;">(</span>dfS, run, numEventPlots, plotCount, outpath, fadcRun<span style="color: #AE81FF;">)</span>  

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">main</span><span style="color: #AE81FF;">(</span>fname: <span style="color: #66D9EF;">string</span>,
          peakAt255 = <span style="color: #AE81FF;">false</span>,
          vetoPaddle = <span style="color: #AE81FF;">false</span>,
          sipm = <span style="color: #AE81FF;">false</span>,
          sipmXrayLike = <span style="color: #AE81FF;">false</span>,
          plotEvents = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span> =
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">h5f </span>= <span style="color: #66D9EF;">H5open</span><span style="color: #AE81FF;">(</span>fname, <span style="color: #E6DB74;">&quot;r&quot;</span><span style="color: #AE81FF;">)</span>
  <span style="color: #75715E;">#</span><span style="color: #75715E;">let fileInfo = h5f.getFileInfo()</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">df </span>= h5f.allData<span style="color: #AE81FF;">()</span>

  <span style="color: #75715E;"># </span><span style="color: #75715E;">first the veto paddle around the 255 peak (plot all events)</span>
  <span style="color: #F92672;">if</span> peakAt255:
    h5f.plotSzinti<span style="color: #AE81FF;">(</span>df,
                   f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">int</span>: `szint2ClockInt` &gt; <span style="font-style: italic;">250</span> <span style="color: #F92672;">and</span> `szint2ClockInt` &lt; <span style="font-style: italic;">265</span><span style="color: #66D9EF;">}</span>,
                   <span style="color: #E6DB74;">&quot;Cluster properties of all events with veto paddle trigger clock cycles = 255&quot;</span>,
                   <span style="color: #E6DB74;">&quot;Figs/scintillators/peakAt255&quot;</span>,
                   <span style="color: #E6DB74;">&quot;cluster_properties_peak_at_255.pdf&quot;</span>,
                   -<span style="font-style: italic;">1</span>,
                   plotEvents<span style="color: #AE81FF;">)</span>
  <span style="color: #75715E;"># </span><span style="color: #75715E;">now the veto paddle generally</span>
  <span style="color: #F92672;">if</span> vetoPaddle:
    h5f.plotSzinti<span style="color: #AE81FF;">(</span>df,
                   f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">int</span>: `szint2ClockInt` &gt; <span style="font-style: italic;">0</span> <span style="color: #F92672;">and</span> `szint2ClockInt` &lt; <span style="font-style: italic;">200</span><span style="color: #66D9EF;">}</span>,
                   <span style="color: #E6DB74;">&quot;Cluster properties of all events with veto paddle &gt; 0 &amp;&amp; &lt; 200&quot;</span>,
                   <span style="color: #E6DB74;">&quot;Figs/scintillators/veto_paddle/&quot;</span>,
                   <span style="color: #E6DB74;">&quot;cluster_properties_veto_paddle_less200.pdf&quot;</span>,
                   <span style="font-style: italic;">200</span>,
                   plotEvents<span style="color: #AE81FF;">)</span>
  <span style="color: #75715E;"># </span><span style="color: #75715E;">finally the SiPM</span>
  <span style="color: #F92672;">if</span> sipm:
    h5f.plotSzinti<span style="color: #AE81FF;">(</span>df,
                   f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">int</span>: `szint1ClockInt` &gt; <span style="font-style: italic;">0</span> <span style="color: #F92672;">and</span> `szint1ClockInt` &lt; <span style="font-style: italic;">200</span><span style="color: #66D9EF;">}</span>,
                   <span style="color: #E6DB74;">&quot;Cluster properties of all events with SiPM &gt; 0 &amp;&amp; &lt; 200&quot;</span>,
                   <span style="color: #E6DB74;">&quot;Figs/scintillators/sipm/&quot;</span>,
                   <span style="color: #E6DB74;">&quot;cluster_properties_sipm_less200.pdf&quot;</span>,                 
                   <span style="font-style: italic;">200</span>,
                   plotEvents<span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">if</span> sipmXrayLike:
    h5f.plotSzinti<span style="color: #AE81FF;">(</span>df,
                   f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">float</span>: `szint1ClockInt` &gt; <span style="font-style: italic;">0</span> <span style="color: #F92672;">and</span> `szint1ClockInt` &lt; <span style="font-style: italic;">200</span> <span style="color: #F92672;">and</span>
                     `energyFromCharge` &gt; <span style="font-style: italic;">7</span>.<span style="font-style: italic;">0</span> <span style="color: #F92672;">and</span> `energyFromCharge` &lt; <span style="font-style: italic;">9</span>.<span style="font-style: italic;">0</span> <span style="color: #F92672;">and</span>
                     `length` &lt; <span style="font-style: italic;">7</span>.<span style="font-style: italic;">0</span><span style="color: #66D9EF;">}</span>,
                   <span style="color: #E6DB74;">&quot;Cluster properties of all events with SiPM &gt; 0 &amp;&amp; &lt; 200, 7 keV &lt; energy &lt; 9 keV, length &lt; 7mm&quot;</span>,
                   <span style="color: #E6DB74;">&quot;Figs/scintillators/sipmXrayLike/&quot;</span>,
                   <span style="color: #E6DB74;">&quot;cluster_properties_sipm_less200_7_energy_9_length_7.pdf&quot;</span>,                 
                   -<span style="font-style: italic;">1</span>,
                   plotEvents,
                   showFadc = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>
    

  <span style="color: #F92672;">discard</span> h5f.<span style="color: #F92672;">close</span><span style="color: #AE81FF;">()</span>
    
<span style="color: #F92672;">when</span> <span style="color: #AE81FF;">isMainModule</span>:
  <span style="color: #F92672;">import</span> cligen
  dispatch main
</pre>
</div>

<p>
-&gt; Conclusion from all this:
I don&apos;t see any real cause for these kinds of events. They are all
relatively busy background events of mostly standard tracks.
Maybe these events are the events that follow the events where the
FPGA has a hiccup and doesn&apos;t take the FADC trigger? Who knows or it
may be something completely different. At 255 * 25ns = 6.375μs I don&apos;t
see any physical source. More importantly are two things:
</p>
<ol class="org-ol">
<li>a physical source is <b>extremely unlikely</b> to be so narrow that it
<span class="underline">always</span> give 255 clock cycles.</li>
<li>255 is 2<sup>8</sup> when counting from 0 (= it&apos;s <code>1111 1111</code>), which implies
there is a high chance it is a bug in the counting logic where
maybe the clock stopped at 2<sup>8</sup> instead of 2<sup>12</sup> (4095) or something
along those lines.</li>
</ol>
<p>
A physical event that always appears at 255 clock cycles is rather
unlikely! See fig. <a href="#fig:scintillators:peak_at_255_cluster_properties">22</a>
for the cluster properties and the
<code>Figs/scintillators/peakAt255/septemEvents</code> directory for all the
event displays of these events.
</p>


<figure id="fig:scintillators:peak_at_255_cluster_properties">
<img alt="cluster_properties.svg" class="org-svg" src="./figs/home/basti/phd/Figs/scintillators/peakAt255/cluster_properties.svg" />

<figcaption>Figure 22: <span class="figure-number">Figure 68: </span>Overview of the cluster properties of all the events with a trigger at \(\SI{255}{clock\;cycles}\). Nothing out of the ordinary for background data here.</figcaption>
</figure>

<p>
-&gt; Conclusion regarding regular veto paddle plots and events:
The plots look pretty much like what we expect here, namely mainly 14
mm long tracks in the histogram
and looking at events the story continues like that. It&apos;s essentially
all just tracks that go through horizontally. It&apos;s still useful
though, as sometimes the statistical process of ionization leaves
blobs that could maybe be interpreted as X-rays, in particular in
corner cutting tracks.
-&gt; As such this is also a good dataset for candidates for corner
cutting cases as we know the data is very pure &quot;good&quot; tracks.
</p>


<figure id="fig:scintillators:veto_paddle_cluster_properties">
<img alt="cluster_properties_veto_paddle_less200.svg" class="org-svg" src="./figs/home/basti/phd/Figs/scintillators/veto_paddle/cluster_properties_veto_paddle_less200.svg" />

<figcaption>Figure 23: <span class="figure-number">Figure 69: </span>Overview of the cluster properties of all the events with a trigger at \(&gt;\SI{0}{clock\;cycles}\) and \(&lt;\SI{200}{clock;cycles}\) for the veto paddle. As one might expect it&apos;s dominated by full chip long tracks (peak at 14mm).</figcaption>
</figure>


<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>SIMILARLY DO THE SAME FOR SIPM</b>
-&gt; Extend the code above such that we also do plots for the SiPM. so
properties &amp; events. In addition to that add the FADC events! Read
FADC data and place it next to the SiPM event displays. What do we
see?
-&gt; In particular this is at the same time an extremely valuable
&quot;dataset&quot; of events where we expect the FADC to have different
shapes. Then when comparing the distribution of those events with
55Fe events, do we see a difference?</li>
</ul>

<p>
-&gt; Conclusions regarding the SiPM. The data is a bit surprising in
some ways, but explained by looking at it. There is more 14 mm like
tracks in the data than I would have assumed. But looking at the
individual events there is a common trend of tracks that are pretty
steep as to go through the SiPM (it&apos;s much bigger than one chip of
14mm after all!), but shallow enough to still cover the full chip more
or less!
Some events <b>do</b> have clusters that are <b>very</b> dense and likely
orthogonal muons though. In addition there is a large number of
extremely busy events in these plots.
As there are few events that are really X-ray like in nature, looking
at the FADC data for <b>most</b> of them is likely not very
interesting. But we should cut on the energy (7 - 9 keV) and having
triggered and a length maybe 7 mm or so. What remains is a target for
event displays including the FADC.
</p>

<p>
Also looking at the event displays with the FADC signal of those event
that are not very track like and in energies between 7 and 9 keV shows
that there are indeed quite a few whose fall and rise time is
<b>significantly longer</b> than the typical O(&lt;100 clock cycles). This
implies there is really a &apos;long time&apos; accumulation going on. Our
expectation of 1.5μs until all the track is accumulated, so this might
actually be <span class="underline">an</span> explanation. 
</p>


<figure id="fig:scintillators:sipm_cluster_properties">
<img alt="cluster_properties_sipm_less200.svg" class="org-svg" src="./figs/home/basti/phd/Figs/scintillators/sipm/cluster_properties_sipm_less200.svg" />

<figcaption>Figure 24: <span class="figure-number">Figure 70: </span>Overview of the cluster properties of all the events with a trigger at \(&gt;\SI{0}{clock\;cycles}\) and \(&lt;\SI{200}{clock;cycles}\) of the SiPM. Surprisingly the data also contains significant amounts of 14 mm events.</figcaption>
</figure>
</div>
</div>
<div id="outline-container-org8b4a118" class="outline-5">
<h5 id="org8b4a118">  <a href="#org8b4a118">  <span class="section-number-5">12.5.1.5.</span> Muon calculations for expected SiPM rates   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-1-5" class="outline-text-5">
<ul class="org-ul">
<li class="off"><code>[ ]</code> Place our calcs here as no export?
-&gt; Refers to calculations of muon rates etc that partially are
mentioned in the theory section!
-&gt; Yes, put here!</li>
</ul>
</div>
</div>
</div>
<div id="outline-container-sec:background:fadc_veto" class="outline-4">
<h4 id="sec:background:fadc_veto">  <a href="#sec:background:fadc_veto">  <span class="section-number-4">12.5.2.</span> FADC veto</a></h4>
<div id="text-sec:background:fadc_veto" class="outline-text-4">
<p>
As previously mentioned in <a href="./septemboard.html#sec:detector:fadc">7.8</a> the FADC not only serves
as a trigger for the readout and reference time for the scintillator
triggers. Because of its high temporal resolution it can in principle
act as a veto of its own by providing insight into the longitudinal
cluster shape.
</p>

<p>
A cluster drifting towards the readout and finally through the grid
induces a voltage measured by the FADC. As such the length of the FADC
signal is a function of the time it takes the cluster to drift
&apos;through&apos; the grid. The kind of orthogonal muon events that should be
triggered by the SiPM as explained in the previous section
<a href="./background.html#sec:background:scinti_veto">12.5.1</a> for example should also be detectable by
the FADC in the form of longer signal rise times than typical in an X-ray.
</p>

<p>
From gaseous detector physics theory we can estimate the typical sizes
and therefore expected signal rise times for an X-ray if we know the
gas of our detector. For the \(\SI{1050}{mbar}\), \(97.7/2.3\,\%\)
\(\ce{Ar}/\ce{iC4H10}\) mixture used with a \(\SI{500}{V.cm⁻¹}\) drift
field in the Septemboard detector at CAST, the relevant parameters
are <sup>  <a id="fnr.30" href="#fn.30" class="footref" role="doc-backlink">30</a></sup>
</p>
<ul class="org-ul">
<li>drift velocity \(v = \SI{2.28}{cm.μs⁻¹}\)</li>
<li>transverse diffusion \(D_T = \SI{670}{μm.cm^{-1/2}}\)</li>
<li>longitudinal diffusion \(D_L = \SI{270}{μm.cm^{-1/2}}\)</li>
</ul>
<p>
As the detector has a height of \(\SI{3}{cm}\) we expect a typical X-ray
interacting close to the cathode to form a cluster of
\(\sqrt{\SI{3}{cm}}·\SI{670}{μm.cm^{-1/2}} \approx \SI{1160.5}{μm}\)
transverse extent and \(\sqrt{\SI{3}{cm}}·\SI{270}{μm.cm^{-1/2}} \approx
\SI{467.5}{μm}\) in longitudinal size, where this corresponds to a \(1
σ\) environment. To get a measure for the cluster size, a rough estimate
for an upper limit is a \(3 σ\) distance away from the center in each
direction. For the transverse size it leads to about \(\SI{7}{mm}\) and
in longitudinal about \(\SI{2.8}{mm}\). From the CAST \cefe data we see
a peak at around \(\SI{6}{mm}\) of transverse cluster size along the
longer axis, which matches well with our expectation (see appendix
<a href="./appendix_fadc.html#sec:appendix:fadc_veto_empirical_cluster_length">30.4</a> for the length
data). From the drift velocity and the upper bound on the longitudinal
cluster size we can therefore also compute an equivalent effective
drift time <span class="underline">seen by the FADC</span>. This comes out to be
</p>

<p>
\[
t = \SI{2.8}{mm} / \SI{22.8}{mm.μs⁻¹} = \SI{0.123}{μs}
\]
</p>

<p>
or about \(\SI{123}{ns}\), equivalent to \(\SI{123}{clock\;cycle}\) of the
FADC clock. Alternatively, we can compute the most likely rise time
based on the known peak of the cluster length, \(\SI{6}{mm}\) and the
ratio of the transverse and longitudinal diffusion, \(D_T / D_L \approx
2.5\) to end up at a time of \(\frac{\SI{6}{mm}}{2.5} ·
\SI{22.8}{mm.μs⁻¹} = \SI{105}{ns}\).
</p>

<p>
Fig. <a href="#fig:background:fadc_rise_time">25</a> shows the measured rise time for
the \cefe calibration data compared to the background data. The peak
of the calibration rise time is at about \(\SI{55}{ns}\). While this is
almost a factor of 2 smaller than the theoretical values mentioned
before, this is expected as those first of all are an upper bound and
secondly the &quot;rise time&quot; here is not the full time due to our
definition starting from \(\SI{10}{\%}\) below the baseline and stopping
\(\SI{2.5}{\%}\) before the peak, shortening our time
(ref. sec. <a href="./reconstruction.html#sec:fadc:definition_baseline_rise_fall_time">9.5.3</a>). At the same
time we see that the background data has a much longer tail towards
higher clock cycle counts, as one expects. This implies that we can
perform a cut on the rise time and utilize it as an additional
veto. The \cefe data allows us to define a cut based on a known and
desired signal efficiency. This is done by identifying a desired
percentile on both ends of the distribution of a cleaned X-ray
dataset. It is important to note that the cut values need to be
determined for each of the used FADC amplifier settings
separately. This is done automatically based on the known runs
corresponding to each setting.
</p>


<figure id="fig:background:fadc_rise_time">
<img alt="fadc_riseTime_kde_signal_vs_background_run3.svg" class="org-svg" src="./figs/home/basti/phd/Figs/FADC/fadc_riseTime_kde_signal_vs_background_run3.svg" />

<figcaption>Figure 25: <span class="figure-number">Figure 71: </span>KDE of the rise time of the FADC signals in the \cefe and background data of the CAST Run-3 dataset. The X-ray data is a single peak with a mean of about \(\SI{55}{ns}\) while the background distribution is extremely wide, motivating a veto based on this data.</figcaption>
</figure>

<p>
For the signal decay time we do a simpler cut, with less of a physical
motivation. As the decay time is highly dependent on the resistance
and capacitance properties of the grid, high voltage and FADC
amplifier settings, we simply introduce a conservative cut in such a
range as to avoid cutting away any X-rays. Any background removed is
just a bonus.
</p>

<p>
Finally, the FADC is only used as a veto if the individual spectrum is
not considered noisy (see sec. <a href="./calibration.html#sec:calibration:fadc_noise">11.4.1</a>). This is
determined by 4 local dominant peaks based on a peak finding algorithm
and in addition a general skewnees of the total signal larger than
\(\num{-0.4}\). The skewness is an empirical cutoff as good FADC signals
typically lie at larger negative skewness values (due to them having a
signal towards negative values). See appendix
<a href="./appendix_fadc.html#sec:appendix:background:fadc">30.2</a> for a scatter plot of FADC rise times
and skewness values. A very low percentage of false positives (good
signals appearing &apos;noisy&apos;) is accepted for the certainty of not
falsely rejecting noisy FADC events (as they would represent a random
coincidence to the event on the center chip).
</p>

<p>
Applying the FADC veto with a target percentile of 1 from both the lower
and upper end (\(1^{\text{st}}\) and \(99^{\text{th}}\) percentiles; thus
a \(\SI{98}{\%}\) signal efficiency) in addition to the scintillator
veto presented in the previous section, results in a background rate as
seen in fig. <a href="#fig:background:background_rate_fadc_veto">26</a>. Different
percentiles (and associated efficiencies) were be tested for their
impact on the expected axion limit. Lower percentiles than 99 did not
yield a significant improvement in the background suppression over the
resulting signal efficiency loss. This is visible due to the sharp
edge of the rise time for X-rays (green) in
fig. <a href="#fig:background:fadc_rise_time">25</a> when comparing to its overlap with
background (purple).
</p>

<p>
The achieved background rate of the methods leads to a background rate
of \(\SI{9.0305(7277)e-06}{keV⁻¹.cm⁻².s⁻¹}\) in the energy range
\(\SIrange{2}{8}{keV}\). And between \(\SIrange{4}{8}{keV}\) even
\(\SI{4.5739(6343)e-06}{keV⁻¹.cm⁻².s⁻¹}\). The veto brings improvements
across the whole energy range, in which the FADC
triggers. Interestingly, in some cases even below that range. The
events removed in the first two bins are events with a big spark on
the upper right GridPix, which induced an X-ray like low energy
cluster on the central chip and triggered the FADC. In the range at
around \(\SI{8}{keV}\), likely candidates for removed events are cases
of orthogonal muons that did not trigger the SiPM (but have a rise
time incompatible with X-rays).
</p>


<figure id="fig:background:background_rate_fadc_veto">
<img alt="background_rate_crGold_scinti_fadc.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/background_rate_crGold_scinti_fadc.svg" />

<figcaption>Figure 26: <span class="figure-number">Figure 72: </span>Comparison of pure \(\ln\mathcal{L}\) cuts, additional scintillator veto and finally also FADC veto at an efficiency of \(\SI{98}{\%}\). In the range \(\SIrange{2}{8}{keV}\) a background rate of \(\SI{9.0305(7277)e-06}{keV⁻¹.cm⁻².s⁻¹}\) is achieved and between \(\SIrange{4}{8}{keV}\) even \(\SI{4.5739(6343)e-06}{keV⁻¹.cm⁻².s⁻¹}\).</figcaption>
</figure>

<p>
Generally, appendix <a href="./appendix_fadc.html#sec:appendix:background:fadc">30.2</a> contains more figures of
the FADC data. Rise times, fall times, how the different FADC
settings affect these and an efficiency curve of a one-sided cut on
the rise time for the signal efficiency and background suppression.
</p>
</div>
<div id="outline-container-org5f98dff" class="outline-5">
<h5 id="org5f98dff">  <a href="#org5f98dff">  <span class="section-number-5">12.5.2.1.</span> Thoughts on the FADC as a ToA tool for orthognal muons   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-2-1" class="outline-text-5">
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>THIS STUFF HAS ALREADY BEEN MENTIONED IN CONTEXT OF
SCINTILLATORS!</b>
-&gt; Rewrite it! However, the scinti part does not explain it as well
and leaves out some pieces (transverse distribution along Z)
-&gt; But explanation better in <a href="./septemboard.html#sec:detector:scintillators">7.7</a>.
-&gt; Most of this below does not belong here. It belongs into the
motivation, which was mentioned before. <span class="underline">Some</span> of the stuff may be
useful in an interpretation of the application of the FADC veto.</li>
</ul>

<p>
While the FADC serves as a very useful trigger to read out events,
decrease the likelihood of multiple independent clusters on the center
chip and serves as a reference time for triggers of the scintillators,
its time information can theoretically provide an additional veto.
</p>

<p>
Even for an almost fully 4π coverage of scintillators around the
detector setup, there is exactly one part that is never covered: the
window of the detector towards the magnet bore. If a muon traverses
through this path it is unlikely to be
shielded. <sup>  <a id="fnr.31" href="#fn.31" class="footref" role="doc-backlink">31</a></sup> In particular during solar
trackings of a helioscope experiment this should contribute a more
significant fraction of background due to the distribution of muons at
surface level following a roughly \(\cos²(θ)\) distribution (\(θ\) the
zenith angle).
</p>

<p>
Muons entering in such a way traverse the gas volume orthogonally to
the readout plane. The projection on the readout is thus roughly
spherical (same as an X-ray). The speed of the muon compared to the
time scale of gas diffusion &amp; clock speed of the readout electronic
means that a muon ionizes the gas along its path effectively
instantly.
</p>

<p>
Two distinctions can be made between X-rays and such muons:
</p>
<ol class="org-ol">
<li>The effective instant ionization implies that such an cluster has a
&apos;duration&apos; that is equivalent to the drift velocity times the gas
volume height. For a typical GridPix setup of height \(\SI{3}{cm}\)
with an argon/isobutane gas mixture, this implies times of
\(\mathcal{O}(\SI{2}{μs})\). Compared to that the duration of an
X-ray is roughly the equivalent of the geometrical size in the
readout plane, which for the same conditions is
\(\mathcal{O}(\SI{3}{mm})\), expressed as a time under the drift
velocity.
Thus, if the duration of a cluster is reasonably well known, this
can easily separate the two types of clusters.</li>
<li>Each ionized electron undergoes diffusion along its path to the
readout plane. For these tracks those electrons that are produced
far away from the readout yield large diffusion, whereas those
close to the readout very little. Assuming a constant ionization
per distance, this implies the muon track is formed like a
cone. The electrons arriving at the readout first are almost
exactly on the muon path and the later ones have more and more
diffusion. This latter effect is invisible in a setup using the
kind of FADC as used in this detector combined with the Timepix1
based GridPix, because the FADC records no spatial
information. However, a detector based on GridPixes using Timepix3,
which allows for ToA readout at the same time as <code>ToT</code> this effect
may be visible and for this reason is important to keep in mind for
the future.</li>
</ol>

<p>
For a argon/isobutane mixture at \(\SI{1050}{mbar}\) the energy
deposition of most muons will be in the range of
\(\SIrange{8}{10}{keV}\), which in any case is outside of the main
region of interest for axion searches.
</p>

<p>
Note: for a detector with an almost perfect 4π coverage, the
scintillators <b>behind</b> the detector would of course trigger for such
muons. Indeed, it is likely that using that information would already
be enough to remove such events. A discrimination based on time
information yields a more certain result than a large scintillator
might, which (even if to a small degree) does introduce random
coincidences.
</p>
</div>
</div>
<div id="outline-container-org680442c" class="outline-5">
<h5 id="org680442c">  <a href="#org680442c">  <span class="section-number-5">12.5.2.2.</span> About FADC rise time from theory and reality <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-2-2" class="outline-text-5">
<p>
The theoretical numbers of about 100 ns do not really match very well
with our experimental numbers anymore after the recent
changes. However, this is mainly due to three aspects:
</p>
<ol class="org-ol">
<li>The offsets from the baseline and to the peak shorten the rise time
by at least (10 + 2.5) %, so our peak at ~55ns -&gt; 61.875
ns.</li>
<li>Realistically it is also though not taking into account that
the bottom of the signal is likely still part of the actual
signal. Once the trailing electrons of the cloud traverse through
the grid, the bulk has already been deposited on the pixels and the
discharge of the grid is underway. It&apos;s a convolution of the still
incoming electrons and their induction and the discharge of the
previous.</li>
<li>Our calculations use a rather random \(3 σ\) size of the clusters
(compared to the RMS). This is of course an extremely rough
estimate for the cluster size, which mainly will be correct in the
upper outliers.</li>
</ol>

<p>
All three aspects combined likely very well explain the difference. It
would be interesting to add to our simulation of the <code>rmsTransverse</code>
distributions an additional step of trying to simulate the ionization
&amp; discharge convolution to see what the actual physical signals might
look like! For that it might be enough to assume that each electron
once it passes through a grid hole &quot;instantaneously&quot; is amplified by
the gas gain leading to a fixed amount of induction. Would be
interesting.
</p>
<ul class="org-ul">
<li class="off"><code>[ ]</code> Do this!</li>
</ul>
</div>
</div>
<div id="outline-container-org4a6c88b" class="outline-5">
<h5 id="org4a6c88b">  <a href="#org4a6c88b">  <span class="section-number-5">12.5.2.3.</span> Note on events being removed below FADC trigger   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-2-3" class="outline-text-5">
<p>
See sec. <a href="./../org/Doc/StatusAndProgress.html">./../org/Doc/StatusAndProgress.html</a>
in <code>statusAndProgress</code>. There we look at precisely those events that
are removed despite in theory having no reason to be removed.
</p>
</div>
</div>
<div id="outline-container-org039095f" class="outline-5">
<h5 id="org039095f">  <a href="#org039095f">  <span class="section-number-5">12.5.2.4.</span> Further notes about FADC   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-2-4" class="outline-text-5">
<p>
For more detailed notes about the FADC discussed above, see my
development notes about the FADC here <a href="./../org/Doc/StatusAndProgress.html">./../org/Doc/StatusAndProgress.html</a>.
</p>
</div>
</div>
<div id="outline-container-org077467e" class="outline-5">
<h5 id="org077467e">  <a href="#org077467e">  <span class="section-number-5">12.5.2.5.</span> Generate plot for background rate including FADC   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-2-5" class="outline-text-5">
<p>
Let&apos;s generate the plot:
</p>

<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;FADC&quot;</span> --names <span style="color: #E6DB74;">&quot;FADC&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate from CAST data, incl. scinti and FADC veto&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_crGold_scinti_fadc.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/background/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
</pre>
</div>

<p>
(old files:
 ~/phd/resources/background/autoGen/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold.h5</sub> \
 ~/phd/resources/background/autoGen/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold</sub><sub>scinti.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
)
</p>
</div>
</div>
<div id="outline-container-sec:background:fadc_veto:gen_signal_back_fadc_plots" class="outline-5">
<h5 id="sec:background:fadc_veto:gen_signal_back_fadc_plots">  <a href="#sec:background:fadc_veto:gen_signal_back_fadc_plots">  <span class="section-number-5">12.5.2.6.</span> Generate plots of rise/fall time for CDL data   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:fadc_veto:gen_signal_back_fadc_plots" class="outline-text-5">
<p>
This code snippet is the original code that we used for the first
plots, when it was still part of
<a href="./../org/Doc/StatusAndProgress.html">./../org/Doc/StatusAndProgress.html</a>. It has since been moved to <a href="./../CastData/ExternCode/TimepixAnalysis/Plotting/plotFadc/plotFadc.nim">./../CastData/ExternCode/TimepixAnalysis/Plotting/plotFadc/plotFadc.nim</a>.
</p>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> nimhdf5, ggplotnim
<span style="color: #F92672;">import</span> std / <span style="color: #AE81FF;">[</span>strutils, os, sequtils, sets, strformat<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> ingrid / <span style="color: #AE81FF;">[</span>tos_helpers, ingrid_types<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> ingrid / calibration / <span style="color: #AE81FF;">[</span>calib_fitting, calib_plotting<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> ingrid / calibration 

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">plotFallTimeRiseTime</span><span style="color: #AE81FF;">(</span>df: <span style="color: #66D9EF;">DataFrame</span>, suffix: <span style="color: #66D9EF;">string</span>, riseTimeHigh: <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">)</span> =
  <span style="color: #E6DB74;">## Given a full run of FADC data, create the</span>
  <span style="color: #E6DB74;">## Note: it may be sensible to compute a truncated mean instead</span>
  <span style="color: #75715E;"># </span><span style="color: #75715E;">local copy filtered to maximum allowed rise time</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">df </span>= df.filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span>`riseTime` &lt;= riseTimeHigh<span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
  
  <span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">plotDset</span><span style="color: #AE81FF;">(</span>dset: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span> =

    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfCalib </span>= df.filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span>`<span style="color: #66D9EF;">Type</span>` == <span style="color: #E6DB74;">&quot;⁵⁵Fe&quot;</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;============================== &quot;</span>, dset, <span style="color: #E6DB74;">&quot; ==============================&quot;</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;Percentiles:&quot;</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t 1-th: &quot;</span>, dfCalib<span style="color: #AE81FF;">[</span>dset, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>.percentile<span style="color: #AE81FF;">(</span><span style="font-style: italic;">1</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t 5-th: &quot;</span>, dfCalib<span style="color: #AE81FF;">[</span>dset, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>.percentile<span style="color: #AE81FF;">(</span><span style="font-style: italic;">5</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t50-th: &quot;</span>, dfCalib<span style="color: #AE81FF;">[</span>dset, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>.percentile<span style="color: #AE81FF;">(</span><span style="font-style: italic;">50</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t mean: &quot;</span>, dfCalib<span style="color: #AE81FF;">[</span>dset, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>.mean
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t95-th: &quot;</span>, dfCalib<span style="color: #AE81FF;">[</span>dset, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>.percentile<span style="color: #AE81FF;">(</span><span style="font-style: italic;">95</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t99-th: &quot;</span>, dfCalib<span style="color: #AE81FF;">[</span>dset, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>.percentile<span style="color: #AE81FF;">(</span><span style="font-style: italic;">99</span><span style="color: #AE81FF;">)</span>
    
    ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span>dset, fill = <span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> + 
      geom_histogram<span style="color: #AE81FF;">(</span>position = <span style="color: #E6DB74;">&quot;identity&quot;</span>, bins = <span style="font-style: italic;">100</span>, hdKind = hdOutline, alpha = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">7</span><span style="color: #AE81FF;">)</span> +
      ggtitle<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;FADC signal {dset} in ⁵⁵Fe vs background data in </span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">&quot;</span> % suffix<span style="color: #AE81FF;">)</span> +
      xlab<span style="color: #AE81FF;">(</span>dset &amp; <span style="color: #E6DB74;">&quot; [ns]&quot;</span><span style="color: #AE81FF;">)</span> + 
      ggsave<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;Figs/statusAndProgress/FADC/fadc_{dset}_signal_vs_background_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.pdf&quot;</span> % suffix<span style="color: #AE81FF;">)</span>
    ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span>dset, fill = <span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> + 
      geom_density<span style="color: #AE81FF;">(</span>normalize = <span style="color: #AE81FF;">true</span>, alpha = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">7</span>, adjust = <span style="font-style: italic;">2</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
      ggtitle<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;FADC signal {dset} in ⁵⁵Fe vs background data in </span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">&quot;</span> % suffix<span style="color: #AE81FF;">)</span> +
      xlab<span style="color: #AE81FF;">(</span>dset &amp; <span style="color: #E6DB74;">&quot; [ns]&quot;</span><span style="color: #AE81FF;">)</span> +      
      ggsave<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;Figs/statusAndProgress/FADC/fadc_{dset}_kde_signal_vs_background_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.pdf&quot;</span> % suffix<span style="color: #AE81FF;">)</span>
  plotDset<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;fallTime&quot;</span><span style="color: #AE81FF;">)</span>
  plotDset<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;riseTime&quot;</span><span style="color: #AE81FF;">)</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">read</span><span style="color: #AE81FF;">(</span>fname, typ: <span style="color: #66D9EF;">string</span>, eLow, eHigh: <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">DataFrame</span> =
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">h5f </span>= <span style="color: #66D9EF;">H5open</span><span style="color: #AE81FF;">(</span>fname, <span style="color: #E6DB74;">&quot;r&quot;</span><span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">fileInfo </span>= h5f.getFileInfo<span style="color: #AE81FF;">()</span>
  
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">peakPos </span>= <span style="color: #F92672;">newSeq</span><span style="color: #AE81FF;">[</span><span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]()</span>
  <span style="color: #FD971F;">result</span> = newDataFrame<span style="color: #AE81FF;">()</span>
  <span style="color: #F92672;">for</span> run <span style="color: #F92672;">in</span> fileInfo.runs:
    <span style="color: #F92672;">if</span> recoBase<span style="color: #AE81FF;">()</span> &amp; $run / <span style="color: #E6DB74;">&quot;fadc&quot;</span> <span style="color: #F92672;">notin</span> h5f: <span style="color: #F92672;">continue</span> <span style="color: #75715E;"># </span><span style="color: #75715E;">skip runs that were without FADC</span>
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">df </span>= h5f.readRunDsets<span style="color: #AE81FF;">(</span>
      run,
      <span style="color: #75715E;">#</span><span style="color: #75715E;">chipDsets = some((chip: 3, dsets: @[&quot;eventNumber&quot;])), # XXX: causes problems?? Removes some FADC data</span>
                                                             <span style="color: #75715E;"># </span><span style="color: #75715E;">but not due to events!</span>
      fadcDsets = @<span style="color: #66D9EF;">[</span><span style="color: #E6DB74;">&quot;eventNumber&quot;</span>,
                    <span style="color: #E6DB74;">&quot;baseline&quot;</span>,
                    <span style="color: #E6DB74;">&quot;riseStart&quot;</span>,
                    <span style="color: #E6DB74;">&quot;riseTime&quot;</span>,                  
                    <span style="color: #E6DB74;">&quot;fallStop&quot;</span>,
                    <span style="color: #E6DB74;">&quot;fallTime&quot;</span>,
                    <span style="color: #E6DB74;">&quot;minvals&quot;</span>,
                    <span style="color: #E6DB74;">&quot;argMinval&quot;</span><span style="color: #66D9EF;">]</span>                 
    <span style="color: #AE81FF;">)</span>
    <span style="color: #75715E;"># </span><span style="color: #75715E;">in calibration case filter to </span>
    <span style="color: #F92672;">if</span> typ == <span style="color: #E6DB74;">&quot;⁵⁵Fe&quot;</span>:
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">xrayRefCuts </span>= getXrayCleaningCuts<span style="color: #AE81FF;">()</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">cut </span>= xrayRefCuts<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;Mn-Cr-12kV&quot;</span><span style="color: #AE81FF;">]</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">grp </span>= h5f<span style="color: #AE81FF;">[</span><span style="color: #66D9EF;">(</span>recoBase<span style="color: #A6E22E;">()</span> &amp; $run / <span style="color: #E6DB74;">&quot;chip_3&quot;</span><span style="color: #66D9EF;">)</span>.grp_str<span style="color: #AE81FF;">]</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">passIdx </span>= cutOnProperties<span style="color: #AE81FF;">(</span>
        h5f,
        grp,
        crSilver, <span style="color: #75715E;"># </span><span style="color: #75715E;">try cutting to silver</span>
        <span style="color: #66D9EF;">(</span>toDset<span style="color: #A6E22E;">(</span>igRmsTransverse<span style="color: #A6E22E;">)</span>, cut.minRms, cut.maxRms<span style="color: #66D9EF;">)</span>,
        <span style="color: #66D9EF;">(</span>toDset<span style="color: #A6E22E;">(</span>igRmsTransverse<span style="color: #A6E22E;">)</span>, <span style="font-style: italic;">0</span>.<span style="font-style: italic;">0</span>, cut.maxEccentricity<span style="color: #66D9EF;">)</span>,        
        <span style="color: #66D9EF;">(</span>toDset<span style="color: #A6E22E;">(</span>igLength<span style="color: #A6E22E;">)</span>, <span style="font-style: italic;">0</span>.<span style="font-style: italic;">0</span>, cut.maxLength<span style="color: #66D9EF;">)</span>,
        <span style="color: #66D9EF;">(</span>toDset<span style="color: #A6E22E;">(</span>igHits<span style="color: #A6E22E;">)</span>, cut.minPix, <span style="color: #66D9EF;">Inf</span><span style="color: #66D9EF;">)</span>,
        <span style="color: #66D9EF;">(</span>toDset<span style="color: #A6E22E;">(</span>igEnergyFromCharge<span style="color: #A6E22E;">)</span>, eLow, eHigh<span style="color: #66D9EF;">)</span>
      <span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfChip </span>= h5f.readRunDsets<span style="color: #AE81FF;">(</span>run, chipDsets = some<span style="color: #66D9EF;">(</span><span style="color: #A6E22E;">(</span>chip: <span style="font-style: italic;">3</span>, dsets: @<span style="color: #E6DB74;">[</span><span style="color: #E6DB74;">&quot;eventNumber&quot;</span><span style="color: #E6DB74;">]</span><span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">allEvNums </span>= dfChip<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;eventNumber&quot;</span>, <span style="color: #66D9EF;">int</span><span style="color: #AE81FF;">]</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">evNums </span>= passIdx.mapIt<span style="color: #AE81FF;">(</span>allEvNums<span style="color: #66D9EF;">[</span>it<span style="color: #66D9EF;">]</span><span style="color: #AE81FF;">)</span>.toSet
      df = df.filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">int</span>: `eventNumber` <span style="color: #F92672;">in</span> evNums<span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
    df<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;runNumber&quot;</span><span style="color: #AE81FF;">]</span> = run
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> df
  <span style="color: #FD971F;">result</span><span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #AE81FF;">]</span> = typ
  <span style="color: #F92672;">echo</span> <span style="color: #FD971F;">result</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">main</span><span style="color: #AE81FF;">(</span>back, calib: <span style="color: #66D9EF;">string</span>, year: <span style="color: #66D9EF;">int</span>,
          energyLow = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">0</span>, energyHigh = <span style="color: #66D9EF;">Inf</span>,
          riseTimeHigh = <span style="color: #66D9EF;">Inf</span>
         <span style="color: #AE81FF;">)</span> =
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">df </span>= newDataFrame<span style="color: #AE81FF;">()</span>
  df.<span style="color: #F92672;">add</span> read<span style="color: #AE81FF;">(</span>back, <span style="color: #E6DB74;">&quot;Background&quot;</span>, energyLow, energyHigh<span style="color: #AE81FF;">)</span>
  df.<span style="color: #F92672;">add</span> read<span style="color: #AE81FF;">(</span>calib, <span style="color: #E6DB74;">&quot;⁵⁵Fe&quot;</span>, energyLow, energyHigh<span style="color: #AE81FF;">)</span>
  
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">is2017 </span>= year == <span style="font-style: italic;">2017</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">is2018 </span>= year == <span style="font-style: italic;">2018</span>
  <span style="color: #F92672;">if</span> <span style="color: #F92672;">not</span> is2017 <span style="color: #F92672;">and</span> <span style="color: #F92672;">not</span> is2018:
    <span style="color: #F92672;">raise</span> <span style="color: #F92672;">newException</span><span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">IOError</span>, <span style="color: #E6DB74;">&quot;The input file is neither clearly a 2017 nor 2018 calibration file!&quot;</span><span style="color: #AE81FF;">)</span>
  
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">yearToRun </span>= <span style="color: #F92672;">if</span> is2017: <span style="font-style: italic;">2</span> <span style="color: #F92672;">else</span>: <span style="font-style: italic;">3</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">suffix </span>= <span style="color: #E6DB74;">&quot;Run-</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">&quot;</span> % $yearToRun
  plotFallTimeRiseTime<span style="color: #AE81FF;">(</span>df, suffix, riseTimeHigh<span style="color: #AE81FF;">)</span> 
      
<span style="color: #F92672;">when</span> <span style="color: #AE81FF;">isMainModule</span>:
  <span style="color: #F92672;">import</span> cligen
  dispatch main
</pre>
</div>


<p>
Run-2:
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">WRITE_PLOT_CSV</span>=true <span style="color: #FD971F;">ESCAPE_LATEX</span>=true <span style="color: #FD971F;">USE_TEX</span>=true plotFadc <span style="color: #E6DB74; font-weight: bold;">\</span>
    -c ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    -b ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --year 2017 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/FADC/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --riseTimeHigh 500
</pre>
</div>
<p>
Run-3:
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">WRITE_PLOT_CSV</span>=true <span style="color: #FD971F;">ESCAPE_LATEX</span>=true <span style="color: #FD971F;">USE_TEX</span>=true plotFadc <span style="color: #E6DB74; font-weight: bold;">\</span>
    -c ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    -b ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --year 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/FADC/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --riseTimeHigh 500
</pre>
</div>
</div>
</div>
<div id="outline-container-orgd590e66" class="outline-5">
<h5 id="orgd590e66">  <a href="#orgd590e66">  <span class="section-number-5">12.5.2.7.</span> Generate plots comparing rise/fall time of X-rays and background   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-2-7" class="outline-text-5">
<p>
<b>UPDATE</b>: <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-12-14 Thu 18:50&gt;</span></span>: The below is also outdated! See
<code>plotFadc</code> above.
</p>

<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>MOVE THE CODE CURRENTLY IN STATUS TO TPA!</b></li>
</ul>

<p>
Use our FADC plotting tool to generate the plots for the rise and fall
time with a custom upper end
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #F92672;">cd</span> /tmp
ntangle ~/org/Doc/StatusAndProgress.org &amp;&amp; nim c -d:danger /t/fadc_rise_fall_signal_vs_background
./fadc_rise_fall_signal_vs_background <span style="color: #E6DB74; font-weight: bold;">\</span>
    -c ~/CastData/data/CalibrationRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    -b ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --year 2017 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --riseTimeHigh 600
./fadc_rise_fall_signal_vs_background <span style="color: #E6DB74; font-weight: bold;">\</span>
    -c ~/CastData/data/CalibrationRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    -b ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --year 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --riseTimeHigh 600
</pre>
</div>

<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>RE GENERATE PLOT FOR 2018 WHEN SEPTEM+LINE VETO APPLICATION DONE!</b></li>
</ul>


<ul class="org-ul">
<li class="on"><code>[X]</code> fixed the issue causing the flat offset / background</li>
</ul>

<p>
This is somewhat of a continuation of
sec. <a href="./calibration.html#sec:reco:fadc_rise_fall_plots">11.4.2.2</a>.
</p>

<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>REVISIT THIS WITH 2018 DATA</b></li>
</ul>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> nimhdf5, ggplotnim
<span style="color: #F92672;">import</span> std / <span style="color: #AE81FF;">[</span>strutils, os, sequtils, sets, strformat<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> ingrid / <span style="color: #AE81FF;">[</span>tos_helpers, ingrid_types<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> ingrid / calibration / <span style="color: #AE81FF;">[</span>calib_fitting, calib_plotting<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> ingrid / calibration 

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">stripPrefix</span><span style="color: #AE81FF;">(</span>s, p: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> =
  <span style="color: #FD971F;">result</span> = s
  <span style="color: #FD971F;">result</span>.removePrefix<span style="color: #AE81FF;">(</span>p<span style="color: #AE81FF;">)</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">plotFallTimeRiseTime</span><span style="color: #AE81FF;">(</span>df: <span style="color: #66D9EF;">DataFrame</span>, suffix: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span> =
  <span style="color: #E6DB74;">## Given a full run of FADC data, create the</span>
  <span style="color: #E6DB74;">## Note: it may be sensible to compute a truncated mean instead</span>
  <span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">plotDset</span><span style="color: #AE81FF;">(</span>dset: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span> =

    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfCalib </span>= df.filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span>`<span style="color: #66D9EF;">Type</span>` == <span style="color: #E6DB74;">&quot;calib&quot;</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;============================== &quot;</span>, dset, <span style="color: #E6DB74;">&quot; ==============================&quot;</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;Percentiles:&quot;</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t 1-th: &quot;</span>, dfCalib<span style="color: #AE81FF;">[</span>dset, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>.percentile<span style="color: #AE81FF;">(</span><span style="font-style: italic;">1</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t 5-th: &quot;</span>, dfCalib<span style="color: #AE81FF;">[</span>dset, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>.percentile<span style="color: #AE81FF;">(</span><span style="font-style: italic;">5</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t95-th: &quot;</span>, dfCalib<span style="color: #AE81FF;">[</span>dset, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>.percentile<span style="color: #AE81FF;">(</span><span style="font-style: italic;">95</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t99-th: &quot;</span>, dfCalib<span style="color: #AE81FF;">[</span>dset, <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]</span>.percentile<span style="color: #AE81FF;">(</span><span style="font-style: italic;">99</span><span style="color: #AE81FF;">)</span>    
    
    ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span>dset, fill = <span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> + 
      geom_histogram<span style="color: #AE81FF;">(</span>position = <span style="color: #E6DB74;">&quot;identity&quot;</span>, bins = <span style="font-style: italic;">100</span>, hdKind = hdOutline, alpha = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">7</span><span style="color: #AE81FF;">)</span> +
      ggtitle<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;Comparison of FADC signal {dset} in ⁵⁵Fe vs background data in </span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">&quot;</span> % suffix<span style="color: #AE81FF;">)</span> +
      ggsave<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;Figs/statusAndProgress/FADC/fadc_{dset}_signal_vs_background_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.pdf&quot;</span> % suffix<span style="color: #AE81FF;">)</span>
    ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span>dset, fill = <span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> + 
      geom_density<span style="color: #AE81FF;">(</span>normalize = <span style="color: #AE81FF;">true</span>, alpha = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">7</span>, adjust = <span style="font-style: italic;">2</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">)</span> +
      ggtitle<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;Comparison of FADC signal {dset} in ⁵⁵Fe vs background data in </span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">&quot;</span> % suffix<span style="color: #AE81FF;">)</span> +
      ggsave<span style="color: #AE81FF;">(</span>&amp;<span style="color: #E6DB74;">&quot;Figs/statusAndProgress/FADC/fadc_{dset}_kde_signal_vs_background_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.pdf&quot;</span> % suffix<span style="color: #AE81FF;">)</span>
  plotDset<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;fallTime&quot;</span><span style="color: #AE81FF;">)</span>
  plotDset<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;riseTime&quot;</span><span style="color: #AE81FF;">)</span>

  <span style="color: #F92672;">when</span> <span style="color: #AE81FF;">false</span>:
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfG </span>= df.group_by<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;runNumber&quot;</span><span style="color: #AE81FF;">)</span>.summarize<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">float</span>: <span style="color: #E6DB74;">&quot;riseTime&quot;</span> &lt;&lt; truncMean<span style="color: #A6E22E;">(</span>col<span style="color: #E6DB74;">(</span><span style="color: #E6DB74;">&quot;riseTime&quot;</span><span style="color: #E6DB74;">)</span>.toSeq1D, <span style="font-style: italic;">0</span>.<span style="font-style: italic;">05</span><span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">}</span>,
                                                 f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">float</span>: <span style="color: #E6DB74;">&quot;fallTime&quot;</span> &lt;&lt; truncMean<span style="color: #A6E22E;">(</span>col<span style="color: #E6DB74;">(</span><span style="color: #E6DB74;">&quot;fallTime&quot;</span><span style="color: #E6DB74;">)</span>.toSeq1D, <span style="font-style: italic;">0</span>.<span style="font-style: italic;">05</span><span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
    ggplot<span style="color: #AE81FF;">(</span>dfG, aes<span style="color: #66D9EF;">(</span>runNumber, riseTime, color = fallTime<span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> + 
      geom_point<span style="color: #AE81FF;">()</span> +
      ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Comparison of FADC signal rise times in ⁵⁵Fe data for all runs in </span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">&quot;</span> % suffix<span style="color: #AE81FF;">)</span> +
      ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Figs/FADC/fadc_mean_riseTime_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.pdf&quot;</span> % suffix<span style="color: #AE81FF;">)</span>
    ggplot<span style="color: #AE81FF;">(</span>dfG, aes<span style="color: #66D9EF;">(</span>runNumber, fallTime, color = riseTime<span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> + 
      geom_point<span style="color: #AE81FF;">()</span> +
      ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Comparison of FADC signal fall times in ⁵⁵Fe data for all runsin </span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">&quot;</span> % suffix<span style="color: #AE81FF;">)</span> +
      ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Figs/FADC/fadc_mean_fallTime_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.pdf&quot;</span> % suffix<span style="color: #AE81FF;">)</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">read</span><span style="color: #AE81FF;">(</span>fname, typ: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">DataFrame</span> =
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">h5f </span>= <span style="color: #66D9EF;">H5open</span><span style="color: #AE81FF;">(</span>fname, <span style="color: #E6DB74;">&quot;r&quot;</span><span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">fileInfo </span>= h5f.getFileInfo<span style="color: #AE81FF;">()</span>
  
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">peakPos </span>= <span style="color: #F92672;">newSeq</span><span style="color: #AE81FF;">[</span><span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">]()</span>
  <span style="color: #FD971F;">result</span> = newDataFrame<span style="color: #AE81FF;">()</span>
  <span style="color: #F92672;">for</span> run <span style="color: #F92672;">in</span> fileInfo.runs:
    <span style="color: #F92672;">if</span> recoBase<span style="color: #AE81FF;">()</span> &amp; $run / <span style="color: #E6DB74;">&quot;fadc&quot;</span> <span style="color: #F92672;">notin</span> h5f: <span style="color: #F92672;">continue</span> <span style="color: #75715E;"># </span><span style="color: #75715E;">skip runs that were without FADC</span>
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">df </span>= h5f.readRunDsets<span style="color: #AE81FF;">(</span>
      run,
      <span style="color: #75715E;">#</span><span style="color: #75715E;">chipDsets = some((chip: 3, dsets: @[&quot;eventNumber&quot;])), # XXX: causes problems?? Removes some FADC data</span>
                                                             <span style="color: #75715E;"># </span><span style="color: #75715E;">but not due to events!</span>
      fadcDsets = @<span style="color: #66D9EF;">[</span><span style="color: #E6DB74;">&quot;eventNumber&quot;</span>,
                    <span style="color: #E6DB74;">&quot;baseline&quot;</span>,
                    <span style="color: #E6DB74;">&quot;riseStart&quot;</span>,
                    <span style="color: #E6DB74;">&quot;riseTime&quot;</span>,                  
                    <span style="color: #E6DB74;">&quot;fallStop&quot;</span>,
                    <span style="color: #E6DB74;">&quot;fallTime&quot;</span>,
                    <span style="color: #E6DB74;">&quot;minvals&quot;</span>,
                    <span style="color: #E6DB74;">&quot;argMinval&quot;</span><span style="color: #66D9EF;">]</span>                 
    <span style="color: #AE81FF;">)</span>
    <span style="color: #75715E;"># </span><span style="color: #75715E;">in calibration case filter to </span>
    <span style="color: #F92672;">if</span> typ == <span style="color: #E6DB74;">&quot;calib&quot;</span>:
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">xrayRefCuts </span>= getXrayCleaningCuts<span style="color: #AE81FF;">()</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">cut </span>= xrayRefCuts<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;Mn-Cr-12kV&quot;</span><span style="color: #AE81FF;">]</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">grp </span>= h5f<span style="color: #AE81FF;">[</span><span style="color: #66D9EF;">(</span>recoBase<span style="color: #A6E22E;">()</span> &amp; $run / <span style="color: #E6DB74;">&quot;chip_3&quot;</span><span style="color: #66D9EF;">)</span>.grp_str<span style="color: #AE81FF;">]</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">passIdx </span>= cutOnProperties<span style="color: #AE81FF;">(</span>
        h5f,
        grp,
        crSilver, <span style="color: #75715E;"># </span><span style="color: #75715E;">try cutting to silver</span>
        <span style="color: #66D9EF;">(</span>toDset<span style="color: #A6E22E;">(</span>igRmsTransverse<span style="color: #A6E22E;">)</span>, cut.minRms, cut.maxRms<span style="color: #66D9EF;">)</span>,
        <span style="color: #66D9EF;">(</span>toDset<span style="color: #A6E22E;">(</span>igRmsTransverse<span style="color: #A6E22E;">)</span>, <span style="font-style: italic;">0</span>.<span style="font-style: italic;">0</span>, cut.maxEccentricity<span style="color: #66D9EF;">)</span>,        
        <span style="color: #66D9EF;">(</span>toDset<span style="color: #A6E22E;">(</span>igLength<span style="color: #A6E22E;">)</span>, <span style="font-style: italic;">0</span>.<span style="font-style: italic;">0</span>, cut.maxLength<span style="color: #66D9EF;">)</span>,
        <span style="color: #66D9EF;">(</span>toDset<span style="color: #A6E22E;">(</span>igHits<span style="color: #A6E22E;">)</span>, cut.minPix, <span style="color: #66D9EF;">Inf</span><span style="color: #66D9EF;">)</span>,
        <span style="color: #75715E;">#</span><span style="color: #75715E;">(toDset(igEnergyFromCharge), 2.5, 3.5)</span>
      <span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">dfChip </span>= h5f.readRunDsets<span style="color: #AE81FF;">(</span>run, chipDsets = some<span style="color: #66D9EF;">(</span><span style="color: #A6E22E;">(</span>chip: <span style="font-style: italic;">3</span>, dsets: @<span style="color: #E6DB74;">[</span><span style="color: #E6DB74;">&quot;eventNumber&quot;</span><span style="color: #E6DB74;">]</span><span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">allEvNums </span>= dfChip<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;eventNumber&quot;</span>, <span style="color: #66D9EF;">int</span><span style="color: #AE81FF;">]</span>
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">evNums </span>= passIdx.mapIt<span style="color: #AE81FF;">(</span>allEvNums<span style="color: #66D9EF;">[</span>it<span style="color: #66D9EF;">]</span><span style="color: #AE81FF;">)</span>.toSet
      df = df.filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">int</span>: `eventNumber` <span style="color: #F92672;">in</span> evNums<span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
    df<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;runNumber&quot;</span><span style="color: #AE81FF;">]</span> = run
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> df
  <span style="color: #FD971F;">result</span><span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #AE81FF;">]</span> = typ
  <span style="color: #F92672;">echo</span> <span style="color: #FD971F;">result</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">main</span><span style="color: #AE81FF;">(</span>back, calib: <span style="color: #66D9EF;">string</span>, year: <span style="color: #66D9EF;">int</span><span style="color: #AE81FF;">)</span> =
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">df </span>= newDataFrame<span style="color: #AE81FF;">()</span>
  df.<span style="color: #F92672;">add</span> read<span style="color: #AE81FF;">(</span>back, <span style="color: #E6DB74;">&quot;back&quot;</span><span style="color: #AE81FF;">)</span>
  df.<span style="color: #F92672;">add</span> read<span style="color: #AE81FF;">(</span>calib, <span style="color: #E6DB74;">&quot;calib&quot;</span><span style="color: #AE81FF;">)</span>
  
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">is2017 </span>= year == <span style="font-style: italic;">2017</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">is2018 </span>= year == <span style="font-style: italic;">2018</span>
  <span style="color: #F92672;">if</span> <span style="color: #F92672;">not</span> is2017 <span style="color: #F92672;">and</span> <span style="color: #F92672;">not</span> is2018:
    <span style="color: #F92672;">raise</span> <span style="color: #F92672;">newException</span><span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">IOError</span>, <span style="color: #E6DB74;">&quot;The input file is neither clearly a 2017 nor 2018 calibration file!&quot;</span><span style="color: #AE81FF;">)</span>
  
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">yearToRun </span>= <span style="color: #F92672;">if</span> is2017: <span style="font-style: italic;">2</span> <span style="color: #F92672;">else</span>: <span style="font-style: italic;">3</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">suffix </span>= <span style="color: #E6DB74;">&quot;run</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">&quot;</span> % $yearToRun
  plotFallTimeRiseTime<span style="color: #AE81FF;">(</span>df, suffix<span style="color: #AE81FF;">)</span> 
      
<span style="color: #F92672;">when</span> <span style="color: #AE81FF;">isMainModule</span>:
  <span style="color: #F92672;">import</span> cligen
  dispatch main
</pre>
</div>

<ul class="org-ul">
<li><b>EXPLANATION FOR FLAT BACKGROUND IN RISE / FALL TIME:</b>
The &quot;dead&quot; register causes our fall / rise time calculation to
break! This leads to a &apos;background&apos; of homogeneous rise / fall times
-&gt; THIS NEEDS TO BE FIXED FIRST!!
-&gt; Has been fixed since.</li>
</ul>
</div>
</div>
<div id="outline-container-orgb18ecab" class="outline-5">
<h5 id="orgb18ecab">  <a href="#orgb18ecab">  <span class="section-number-5">12.5.2.8.</span> Calculate expected cluster sizes and rise times <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-2-8" class="outline-text-5">
<ul class="org-ul">
<li class="off"><code>[ ]</code> Once happy with the text from <code>statusAndProgress</code>, move all the
related parts, concluding in the explicit code using the gas
property CSV to compute the expected values explicitly.</li>
</ul>

<p>
We estimate the rise times in sec. <a href="./../org/Doc/StatusAndProgress.html">./../org/Doc/StatusAndProgress.html</a>.
</p>
</div>
</div>
</div>
<div id="outline-container-sec:background:septem_veto" class="outline-4">
<h4 id="sec:background:septem_veto">  <a href="#sec:background:septem_veto">  <span class="section-number-4">12.5.3.</span> Outer GridPix as veto - &apos;septem veto&apos;</a></h4>
<div id="text-sec:background:septem_veto" class="outline-text-4">
<p>
The final hardware feature that is used to improve the background rate
is the outer ring of GridPixes. The size of large clusters is a
significant fraction of a single GridPix. This means that depending on
the cluster center position, parts of the cluster may very well be
outside of the chip. While the most important area of the chip is the
center area (due to the X-ray optics focusing the axion induced
X-rays), misalignment and the extended nature of the &apos;axion image&apos;
mean that a significant portion of the chip should be as low in
background as
possible. Fig. <a href="#fig:background:cluster_centers_no_vetoes_2017_18">11(a)</a>
as we saw in sec. <a href="./background.html#sec:background:likelihood_cut">12.3</a> shows a significant
increase in cluster counts towards the edges and corners of the center
GridPix. The GridPix ring can help us reduce this.
</p>

<p>
Normally the individual chips are treated separately in the analysis
chain. The &apos;septem veto&apos; is the name for an additional veto step,
which can be optionally applied to the center chip. With it, each
cluster that is considered signal-like based on the likelihood cut (or
MLP), will be analyzed in a second step. The full event is
reconstructed again from the beginning, but this time as an event
covering <i>all 7</i> chips. This allows the cluster finding algorithm to
detect clusters beyond the center chip boundaries. After finding all
clusters, the normal cluster reconstruction to compute all properties
is done again. Finally, for each cluster in the event the likelihood
method or MLP is applied. If now all clusters whose center is on the
central chip are considered background-like, the event is vetoed,
because the initial signal-like cluster turned out to be part of a
larger cluster covering more than one chip.
</p>

<p>
There is a slight complication here. The layout of the septemboard
includes spacing (see for example
fig. <a href="#fig:detector:occupancy_sparking_run_241">25</a> or
<a href="#fig:reco:fadc_reco_example">34</a> for the layout), which is required to mount
the chips. This spacing, in addition to general lower efficiency
towards the edges of a GridPix, means significant information is
lost. When reconstructing the full &apos;septemboard events&apos; this spacing
would break the cluster finding algorithms, as the spacing might
extend the distance over the cutoff
criterion <sup>  <a id="fnr.32" href="#fn.32" class="footref" role="doc-backlink">32</a></sup>. For this reason the cluster finding
algorithm is actually performed on a &apos;tight layout&apos; where the spacing
has been reduced to zero. For the calculation of the geometric
properties however, the clusters are transformed into the real
septemboard coordinates including spacing. <sup>  <a id="fnr.33" href="#fn.33" class="footref" role="doc-backlink">33</a></sup>
</p>

<p>
In contrast to the regular single chip analysis performed before,
which uses a simple, bespoke clustering algorithm (see
sec. <a href="./reconstruction.html#sec:reco:cluster_finding">9.4.1</a>, &apos;Default&apos;), the full &apos;septemboard
reconstruction&apos; uses the DBSCAN clustering algorithm. The equivalent
search radius is extended a bit over the normal default (65 pixels
compared to 50, unless changed in configuration). DBSCAN produces
better results in terms of likely related clusters (over chip
boundaries). <sup>  <a id="fnr.34" href="#fn.34" class="footref" role="doc-backlink">34</a></sup>
</p>

<p>
An example that shows two clusters on the center chip, one of which
was initially interpreted as a signal-like event before being vetoed
by the &apos;septem veto&apos;, is shown in
fig. <a href="#fig:example_clusters_septem_veto">27</a>. The colors indicate the
clusters each pixel belongs to according to the cluster finding
algorithm. As the chips are treated separately initially, there are
two clusters found on the center chip. The green and purple cluster
&quot;portions&quot; of the center chip. The cyan part passes the likelihood cut
initially, which triggers the &apos;septem veto&apos; (X-rays at such low
energies are much less spherical on average; same diffusion, but less
electrons). A full 7 GridPix event reconstruction shows the additional
parts of the two clusters. The cyan cluster is finally rejected. It is
a good example, as it shows a cluster that is still relatively close
to the center, and yet still &apos;connects&apos; to another chip.
</p>


<figure id="fig:example_clusters_septem_veto">
<img alt="septemEvent_run_162_event_80301.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/exampleEvents/septemEvent_run_162_event_80301.svg" />

<figcaption>Figure 27: <span class="figure-number">Figure 73: </span>An example event showing all 7 GridPix of the CAST GridPix1 detector. The outlines are the boundaries of each chip and the color of each point indicates the cluster which it is part of according to the cluster finder. Initially the cyan cluster (center chip portion) passes the log likelihood cut (i.e. is signal like), but is vetoed by the &apos;septem veto&apos; as there are more pixels outside the center chip that are part of this cluster. The green cluster in the bottom left of the center chip is in addition a good example of how in particular cutting a track in the corners leads to a much more spherical cluster.</figcaption>
</figure>

<p>
The background rate with the septem veto is shown in
fig. <a href="#fig:background:background_rate_septem_veto">28</a>, where we see that most of the
improvement is in the lower energy range \(&lt; \SI{2}{keV}\). This is the
<i>most important region for the solar axion flux for the axion-electron
coupling</i>. Looking at the mean background rate in intervals of
interest, between \(\SIrange{2}{8}{keV}\) it is
\(\SI{8.0337(6864)e-06}{keV⁻¹.cm⁻².s⁻¹}\) and \(\SIrange{4}{8}{keV} =
\SI{4.0462(5966)e-06}{keV⁻¹.cm⁻².s⁻¹}\). And even in the full range down to
\(\SI{0}{keV}\) the mean is in the \(10⁻⁶\) range, \(\SI{9.5436(6479)e-06}{keV⁻¹.cm⁻².s⁻¹}\).
</p>


<figure id="fig:background:background_rate_septem_veto">
<img alt="background_rate_crGold_scinti_fadc_septem.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/background_rate_crGold_scinti_fadc_septem.svg" />

<figcaption>Figure 28: <span class="figure-number">Figure 74: </span>Background rate achieved based on the &apos;septem veto&apos; (in addition to the scintillator cut and FADC veto) for the full 2017/18 dataset within the center \(\SI[parse-numbers=false]{5 \times 5}{mm²}\). Significant improvement in the \(&lt; \SI{2}{keV}\) range, which is most important for axion-electron coupling solar flux. The background rate between \(\SIrange{2}{8}{keV}\) is \(\SI{8.0337(6864)e-06}{keV⁻¹.cm⁻².s⁻¹}\) and \(\SIrange{4}{8}{keV} = \SI{4.0462(5966)e-06}{keV⁻¹.cm⁻².s⁻¹}\).</figcaption>
</figure>
</div>
<div id="outline-container-org8de1171" class="outline-5">
<h5 id="org8de1171">  <a href="#org8de1171">  <span class="section-number-5">12.5.3.1.</span> Generate plot for background rate including septem veto   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-3-1" class="outline-text-5">
<p>
Let&apos;s generate the plot:
</p>

<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_septem_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_septem_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;FADC&quot;</span> --names <span style="color: #E6DB74;">&quot;FADC&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;Septem&quot;</span> --names <span style="color: #E6DB74;">&quot;Septem&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate from CAST data, incl. scinti, FADC, septem veto&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_crGold_scinti_fadc_septem.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/background/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
</pre>
</div>

<p>
(old files:
 ~/phd/resources/background/autoGen/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold.h5</sub> \
 ~/phd/resources/background/autoGen/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold</sub><sub>scinti.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
)
</p>
</div>
</div>
<div id="outline-container-orgefd7928" class="outline-5">
<h5 id="orgefd7928">  <a href="#orgefd7928">  <span class="section-number-5">12.5.3.2.</span> Generate example plot of vetoed septemboard event   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-3-2" class="outline-text-5">
<p>
The septemboard event is generated as a byproduct of the <code>likelihood</code>
program, if it is being run with <code>--plotSeptem</code> as a command line
argument.
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">PLOT_SEPTEM_E_CUTOFF</span>=1.0 <span style="color: #FD971F;">PLOT_SEPTEM_EVENT_NUMBER</span>=80301 <span style="color: #E6DB74; font-weight: bold;">\</span>
    likelihood <span style="color: #E6DB74; font-weight: bold;">\</span>
    -f ~/CastData/data/DataRuns2017_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --run 162 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --h5out /tmp/dummy.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lnL <span style="color: #E6DB74; font-weight: bold;">\</span>
    --signalEfficiency 0.8 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --septemveto <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotSeptem <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/phd/Figs/background/exampleEvents/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX
</pre>
</div>
<p>
-&gt; Works (just produces too many plots!)
-&gt; Add an &quot;event index&quot; filter for the plotSeptem. Done.
</p>
</div>
</div>
</div>
<div id="outline-container-sec:background:line_veto" class="outline-4">
<h4 id="sec:background:line_veto">  <a href="#sec:background:line_veto">  <span class="section-number-4">12.5.4.</span> &apos;Line veto&apos;</a></h4>
<div id="text-sec:background:line_veto" class="outline-text-4">
<p>
There is one more further optional veto, dubbed the &apos;line veto&apos;. It
checks whether there are clusters on the outer chips whose long axis
&quot;points at&quot; clusters on the center chip (passing the likelihood or MLP
cut). The idea being that there is a high chance that such clusters
are correlated, especially because ionization is an inherently
statistical process. It can be used in addition to the &apos;septem veto&apos;
or standalone. The approach uses the same general approach as the
septem veto by reconstructing the full events as &apos;septemboard events&apos;
again. The difference to the septem veto is that it is not reliant on
the cluster finding search radius.
</p>

<p>
The center cluster that initially passed the likelihood or MLP cut
will be rejected if the long axis of the outer chip cluster cuts
within \(3·\left(\frac{\text{RMS}_T + \text{RMS}_L}{2}\right)\). In other words
within \(3 σ\) of the mean (along long and short axis) standard
distribution of the pixels <sup>  <a id="fnr.35" href="#fn.35" class="footref" role="doc-backlink">35</a></sup>.
</p>

<p>
If desired not every outer chip cluster is considered for the &apos;line
veto&apos;. An \(ε_{\text{cut}}\) parameter can be adjusted to only allow
those clusters to contribute with an eccentricity larger than
\(ε_{\text{cut}}\). The value of this cutoff impacts the efficiency, but
also the expected random coincidence rate of the veto. More on that in
sec. <a href="./background.html#sec:background:estimate_veto_efficiency">12.5.5</a>.
</p>

<p>
An example of an event being vetoed by the &apos;line veto&apos; is shown in
fig. <a href="#fig:background:example_clusters_line_veto">29</a>. The black circle
around the center chip cluster indicates the radius in which the orange
line (extension of the long axis of the top, green cluster) needs to cut the
center cluster. 
</p>


<figure id="fig:background:example_clusters_line_veto">
<img alt="septemEvent_run_261_event_44109.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/exampleEvents/septemEvent_run_261_event_44109.svg" />

<figcaption>Figure 29: <span class="figure-number">Figure 75: </span>Example event, which highlights the use case of the &apos;line veto&apos;. The green cluster in the upper chip is eccentric and its long axis &apos;points&apos; towards the purple center cluster (which initially passes the \(\ln\mathcal{L}\) cut). The black circle is a measure for the radius of the center cluster. If the line of the eccentric cluster cuts the circle, the cluster is vetoed.</figcaption>
</figure>

<p>
The achieved background rate, see
fig. <a href="#fig:background:background_rate_line_veto">30</a>, goes well into the
\(10⁻⁶\) range with this veto over the whole energy range, as the
influence is largest at low energies. The rate comes out to
\(\SI{7.9164(5901)e-06}{keV⁻¹.cm⁻².s⁻¹}\) over the whole range up to
\(\SI{8}{keV}\), to \(\SI{7.5645(6660)e-06}{keV⁻¹.cm⁻².s⁻¹}\) if starting from
\(\SI{2}{keV}\) and \(\SI{3.7823(5768)e-06}{keV⁻¹.cm⁻².s⁻¹}\) if starting from
\(\SI{4}{keV}\).
</p>


<figure id="fig:background:background_rate_line_veto">
<img alt="background_rate_crGold_scinti_fadc_septem_line.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/background_rate_crGold_scinti_fadc_septem_line.svg" />

<figcaption>Figure 30: <span class="figure-number">Figure 76: </span>Background rate of all CAST 2017/18 data in the center \(\SI[parse-numbers=false]{5 \times 5}{mm²}\) using all vetoes including the line veto. Improvements are seen especially at low energies. The mean background rate in the full range below \(\SI{8}{keV}\) is \(\SI{7.9164(5901)e-06}{keV⁻¹.cm⁻².s⁻¹}\).</figcaption>
</figure>
</div>
<div id="outline-container-sec:background:line_veto:eccentricity_cutoff" class="outline-5">
<h5 id="sec:background:line_veto:eccentricity_cutoff">  <a href="#sec:background:line_veto:eccentricity_cutoff">  <span class="section-number-5">12.5.4.1.</span> Possible eccentricity cutoff  <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:line_veto:eccentricity_cutoff" class="outline-text-5">
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>DISCUSS THE ECCENTRICITY CUTOFF IN LITTLE MORE DETAIL, SHOW THE
EFFICIENCIES AS DESCRIBED IN SUBSECTION BELOW SOMEWHERE!</b></li>

<li class="off"><code>[ ]</code> This should be <b>after</b> we talk about random coincidence and
efficiency etc etc</li>
</ul>

<p>
-&gt; The plot shows that the efficiency scaling is much steeper for real
data than for fake data (purple vs green). This implies that we gain
more in efficiency for real data than we lose in signal efficiency due
to random coincidence.
</p>

<p>
For the final usage in the thesis we ended up allowing all clusters to
participate in the line veto. While the 
</p>

<p>
Fig. <a href="#fig:background:fraction_passing_line_veto_ecc_cut">33</a> shows the
behavior of the line veto efficiency for real data and fake
bootstrapped data depending on what eccentricity a cluster needs to
participate as an active cluster in the line veto. The ideal choice is
somewhere in the middle.
</p>
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>DECIDE ON A FINAL VALUE WE USE AND GIVE ARGUMENTS!</b>
-&gt; Also see ratio plot! But to decide need to think through what we
actually care about most! Direct ratio may not be optimal.</li>
</ul>


<figure id="fig:background:fraction_passing_line_veto_ecc_cut">
<img alt="fraction_passing_line_veto_ecc_cut.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/estimateSeptemVetoRandomCoinc/fraction_passing_line_veto_ecc_cut.svg" /> 

<figcaption>Figure 33: <span class="figure-number">Figure 77: </span>Fraction of events in Run-3 data (green), which pass (i.e. not rejected) the line veto depending on the eccentricity cut used, which decides how eccentric a cluster needs to be in order to be used for the veto. The purple points are using fake bootstrapped data from real clusters passing the \(\ln\mathcal{L}\) cut together with real outer GridPix data from <b>other</b> events. The fraction of events being vetoed in the latter is a measure for the random coincidence rate. (See ratio plot to see that 1.4 - 1.5 is likely best?)</figcaption>
</figure>
</div>
</div>
<div id="outline-container-org82b0b25" class="outline-5">
<h5 id="org82b0b25">  <a href="#org82b0b25">  <span class="section-number-5">12.5.4.2.</span> Generate example plot of &apos;line veto&apos; event   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-4-2" class="outline-text-5">
<p>
The septemboard event for the line veto is generated as a byproduct of
the <code>likelihood</code> program, if it is being run with <code>--plotSeptem</code> as a
command line argument.
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">PLOT_SEPTEM_E_CUTOFF</span>=1.0 <span style="color: #FD971F;">PLOT_SEPTEM_EVENT_NUMBER</span>=44109 <span style="color: #E6DB74; font-weight: bold;">\</span>
    likelihood <span style="color: #E6DB74; font-weight: bold;">\</span>
    -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --run 261 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --h5out /tmp/dummy.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lnL <span style="color: #E6DB74; font-weight: bold;">\</span>
    --signalEfficiency 0.8 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lineveto <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotSeptem <span style="color: #E6DB74; font-weight: bold;">\</span>
    --plotPath ~/phd/Figs/background/exampleEvents/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX
</pre>
</div>
<p>
-&gt; Works.
-&gt; To produce thesis plot, run command in <code>/phd/Figs/&lt;subdirectory&gt;</code>
</p>
</div>
</div>
<div id="outline-container-sec:background:gen_bck_all_vetoes_compared" class="outline-5">
<h5 id="sec:background:gen_bck_all_vetoes_compared">  <a href="#sec:background:gen_bck_all_vetoes_compared">  <span class="section-number-5">12.5.4.3.</span> Generate plot for background rate with all vetoes   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:gen_bck_all_vetoes_compared" class="outline-text-5">
<p>
Let&apos;s generate the plot:
</p>

<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_septem_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_septem_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;FADC&quot;</span> --names <span style="color: #E6DB74;">&quot;FADC&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;Septem&quot;</span> --names <span style="color: #E6DB74;">&quot;Septem&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;Line&quot;</span> --names <span style="color: #E6DB74;">&quot;Line&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate from CAST data, incl. scinti, FADC, septem, line veto&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_crGold_scinti_fadc_septem_line.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/background/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
</pre>
</div>

<p>
(old files:
  ~/phd/resources/background/autoGen/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold.h5</sub> \
 ~/phd/resources/background/autoGen/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold</sub><sub>scinti.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run2</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>line</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
 ~/org/resources/lhood<sub>limits</sub><sub>automation</sub><sub>preliminary</sub>/lhood<sub>outputs</sub><sub>adaptive</sub><sub>fadc</sub>/likelihood<sub>cdl2018</sub><sub>Run3</sub><sub>crGold</sub><sub>scinti</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>line</sub><sub>vetoPercentile</sub><sub>0.99.h5</sub> \
)
</p>
</div>
</div>
</div>
<div id="outline-container-sec:background:estimate_veto_efficiency" class="outline-4">
<h4 id="sec:background:estimate_veto_efficiency">  <a href="#sec:background:estimate_veto_efficiency">  <span class="section-number-4">12.5.5.</span> Estimating the random coincidence rate of the septem &amp; line veto</a></h4>
<div id="text-sec:background:estimate_veto_efficiency" class="outline-text-4">
<p>
One potential issue with the septem and line veto is that the shutter
times we ran with at CAST are very long (\(&gt; \SI{2}{s}\)), but only the
center chip is triggered by the FADC. This means that the outer chips
can record cluster data not correlated to what the center chip
sees. When applying one of these two vetoes the chance for random
coincidence might be non negligible. This random coincidence rate
needs to be treated as a dead time for the detector, reducing the
effective solar tracking time available.
</p>

<p>
In order to estimate this we bootstrap fake events with guaranteed
random coincidence. For each data taking run, we take events with
clusters on the center chip, which pass the likelihood or MLP cuts and
combine these with data from the outer chips from different
events. The outer GridPix ring data is sampled from <i>all</i> events in
that run, including empty events as otherwise we would bias the
coincidence rate. The vetoes (either only septem veto, only line veto or
both) are then applied to this bootstrapped dataset. The fraction of
rejected events due to the veto is the random coincidence rate,
because there is no physical correlation between center cluster and
outer GridPix ring clusters.
</p>

<p>
By default \(\num{2000}\) such events are generated for each run. The
random coincidence rate is the mean of all obtained values. Further,
this can be used to study the effectiveness of the eccentricity cutoff
\(ε_{\text{cut}}\) mentioned for the line veto. It is expected that a
stricter eccentricity cut should yield less random coincidences. We
can then compare the obtained random coincidence rates to the real
fractions of events removed by the same veto setup. The ratio of real
effectiveness to random coincidence rate is a measure of the potency
of the veto setup.
</p>

<p>
Fig. <a href="#fig:background:fraction_passing_line_veto_ecc_cut">33</a> shows how the
fraction of passing events changes for the line veto as a function of
the eccentricity cluster cutoff. The cases of line veto only for fake
and real data, as well as septem veto plus line veto for the same are
shown. We can see that the fraction that passes the veto setups (y
axis) drops the further we go towards a low eccentricity cut (x
axis). For the real data (<code>Real</code> suffix in the legend) the drop is
<span class="underline">faster</span> than for fake boostrapped data (<code>Fake</code> suffix in the legend),
which means that we can use the lowest eccentricity cut possible
(effectively disabling the cut at \(ε_\text{cut} = 1.0\)). In any case,
the difference is very minor independent of the cutoff value. The
septem veto without line veto is also shown in the plot (<code>SeptemFake</code>
and <code>SeptemReal</code>) with only a single point at \(ε_{\text{cut}} = 1.0\)
for completeness.
</p>

<p>
Table <a href="#tab:background:veto_random_coinc_rate">25</a> shows the percentages of
clusters left over after the line veto (at \(ε_\text{cut} = 1.0\)) and
septem veto setups, both for real data (column &apos;Real&apos;) and
bootstrapped fake data (column &apos;Fake&apos;). For fake data this corresponds
to the random coincidence rate of the veto setup (strictly speaking
\(1 - x\) of course). For real data it is the real percentage of
clusters left, meaning the vetoes achieve reduction to \(\frac{1}{10}\)
of background, depending on setup. The random coincidence rates are
between \(\SI{78.6}{\%}\) in case both outer GridPix vetoes are used,
\(\SI{83.1}{\%}\) for the septem veto and \(\SI{85.4}{\%}\) for the line
veto only. The line veto has lower random coincidence rate, but by
itself is also less efficient. 
</p>

<p>
This effect on the signal efficiency is one of the major reasons the
choice of vetoes is not finalized outside the context of a limit
calculation.
</p>


<figure id="fig:background:fraction_passing_line_veto_ecc_cut">
<img alt="fraction_passing_line_veto_ecc_cut_only_relevant.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/estimateSeptemVetoRandomCoinc/fraction_passing_line_veto_ecc_cut_only_relevant.svg" />

<figcaption>Figure 33: <span class="figure-number">Figure 78: </span>Fraction of events in Run-3 data, which pass (i.e. not rejected) the line veto depending on the eccentricity cut used, which decides how eccentric a cluster needs to be in order to be used for the veto. &apos;Real&apos; and &apos;Fake&apos; suffixes refer to application of the vetoes to the bootstrapped fake data discussed or real data. For fake data the percentage corresponds to the random coincidence rate. Septem veto without line veto shown as single points (<code>SeptemFake/Real</code>). </figcaption>
</figure>

<table id="tab:background:veto_random_coinc_rate">
<caption class="t-above"><span class="table-number">Table 25:</span> Overview of the percentages of clusters left after the line veto and and septem veto setup combinations (y = activated, n = not activated) for bootstrapped (&apos;Fake&apos;) data and real background data. The percentages for fake data correspond to the random coincidence rate.</caption>

<colgroup>
<col class="org-left" />

<col class="org-left" />

<col class="org-left" />

<col class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">Septem veto</th>
<th scope="col" class="org-left">Line veto</th>
<th scope="col" class="org-left">Real [%]</th>
<th scope="col" class="org-left">Fake [%]</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">y</td>
<td class="org-left">n</td>
<td class="org-left">\(\num{14.12}\)</td>
<td class="org-left">\(\num{83.11}\)</td>
</tr>

<tr>
<td class="org-left">n</td>
<td class="org-left">y</td>
<td class="org-left">\(\num{25.32}\)</td>
<td class="org-left">\(\num{85.39}\)</td>
</tr>

<tr>
<td class="org-left">y</td>
<td class="org-left">y</td>
<td class="org-left">\(\num{9.17}\)</td>
<td class="org-left">\(\num{78.63}\)</td>
</tr>
</tbody>
</table>
</div>
<div id="outline-container-sec:background:calculate_random_coincidences" class="outline-5">
<h5 id="sec:background:calculate_random_coincidences">  <a href="#sec:background:calculate_random_coincidences">  <span class="section-number-5">12.5.5.1.</span> Calculate random coincidence rates and real efficiencies of veto setups   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-sec:background:calculate_random_coincidences" class="outline-text-5">
<p>
The following code snippet runs the
<code>TimepixAnalysis/Analysis/ingrid/likelihood.nim</code> program with
different arguments in order to compute the real efficiency and random
coincidence rate for different veto setups. For the line veto, the
different eccentricity cutoffs are studied.
</p>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> shell, strutils, os, sequtils, times

<span style="color: #75715E;"># </span><span style="color: #75715E;">for multiprocessing</span>
<span style="color: #F92672;">import</span> cligen / <span style="color: #AE81FF;">[</span>procpool, mslice, osUt<span style="color: #AE81FF;">]</span>

<span style="color: #F92672;">type</span>
  <span style="color: #66D9EF;">Veto</span> = <span style="color: #F92672;">enum</span>
    <span style="color: #66D9EF;">Septem</span>, <span style="color: #66D9EF;">Line</span>

  <span style="color: #66D9EF;">RealOrFake</span> = <span style="color: #F92672;">enum</span>
    <span style="color: #66D9EF;">Real</span>, <span style="color: #66D9EF;">Fake</span>

<span style="color: #E6DB74;">## Ecc = 1.0 is not in `</span><span style="color: #AE81FF;">eccs</span><span style="color: #E6DB74;">` because by default we run `</span><span style="color: #AE81FF;">ε = 1.0</span><span style="color: #E6DB74;">`.</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">eccStudy </span>= @<span style="color: #AE81FF;">[</span><span style="font-style: italic;">1</span>.<span style="font-style: italic;">1</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">2</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">3</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">4</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">5</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">6</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">7</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">8</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">9</span>, <span style="font-style: italic;">2</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">]</span>
<span style="color: #E6DB74;">## Line veto kind is not needed anymore. We select the correct kind in `</span><span style="color: #AE81FF;">likelihood</span><span style="color: #E6DB74;">`</span>
<span style="color: #E6DB74;">## -&gt; lvRegular for line veto without septem</span>
<span style="color: #E6DB74;">## -&gt; lvRegularNoHLC for line veto with septem</span>
<span style="color: #E6DB74;">## UseRealLayout is also set accordingly.</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">cmd </span>= <span style="color: #E6DB74;">&quot;&quot;&quot;</span>
<span style="color: #E6DB74;">ECC_LINE_VETO_CUT=</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;"> \</span>
<span style="color: #E6DB74;">    likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 \</span>
<span style="color: #E6DB74;">    --h5out </span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">/lhood_2018_crAll_80eff_septem_line_ecc_cutoff_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.h5 \</span>
<span style="color: #E6DB74;">    --region crAll --cdlYear 2018 --readOnly --lnL --signalEfficiency 0.8 \</span>
<span style="color: #E6DB74;">    --septemLineVetoEfficiencyFile </span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;"> \</span>
<span style="color: #E6DB74;">    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 </span><span style="color: #F92672;">$#</span>
<span style="color: #E6DB74;">&quot;&quot;&quot;</span>
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">toName</span><span style="color: #AE81FF;">(</span>rf: <span style="color: #66D9EF;">RealOrFake</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> =
  <span style="color: #F92672;">if</span> rf == <span style="color: #66D9EF;">Real</span>: <span style="color: #E6DB74;">&quot;&quot;</span> <span style="color: #F92672;">else</span>: <span style="color: #E6DB74;">&quot;fake_events&quot;</span>
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">toName</span><span style="color: #AE81FF;">(</span>vetoes: <span style="color: #66D9EF;">set</span><span style="color: #66D9EF;">[</span><span style="color: #66D9EF;">Veto</span><span style="color: #66D9EF;">]</span>, rf: <span style="color: #66D9EF;">RealOrFake</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> = <span style="color: #AE81FF;">(</span>toSeq<span style="color: #66D9EF;">(</span>vetoes<span style="color: #66D9EF;">)</span>.mapIt<span style="color: #66D9EF;">(</span>$it<span style="color: #66D9EF;">)</span>.join<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;_&quot;</span><span style="color: #66D9EF;">)</span> &amp; <span style="color: #E6DB74;">&quot;_&quot;</span> &amp; toName<span style="color: #66D9EF;">(</span>rf<span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span>.strip<span style="color: #AE81FF;">(</span>chars = <span style="color: #66D9EF;">{</span><span style="color: #E6DB74;">&apos;_&apos;</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">toCommand</span><span style="color: #AE81FF;">(</span>rf: <span style="color: #66D9EF;">RealOrFake</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> =
  <span style="color: #F92672;">if</span> rf == <span style="color: #66D9EF;">Real</span>: <span style="color: #E6DB74;">&quot;&quot;</span> <span style="color: #F92672;">else</span>: <span style="color: #E6DB74;">&quot;--estimateRandomCoinc&quot;</span>
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">toCommand</span><span style="color: #AE81FF;">(</span>veto: <span style="color: #66D9EF;">Veto</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> =
  <span style="color: #F92672;">case</span> veto
  <span style="color: #F92672; font-style: italic;">of</span> <span style="color: #66D9EF;">Septem</span>: <span style="color: #E6DB74;">&quot;--septemveto&quot;</span>
  <span style="color: #F92672; font-style: italic;">of</span> <span style="color: #66D9EF;">Line</span>: <span style="color: #E6DB74;">&quot;--lineveto&quot;</span>
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">toCommand</span><span style="color: #AE81FF;">(</span>vetoes: <span style="color: #66D9EF;">set</span><span style="color: #66D9EF;">[</span><span style="color: #66D9EF;">Veto</span><span style="color: #66D9EF;">]</span>, rf: <span style="color: #66D9EF;">RealOrFake</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> = <span style="color: #AE81FF;">(</span>toSeq<span style="color: #66D9EF;">(</span>vetoes<span style="color: #66D9EF;">)</span>.mapIt<span style="color: #66D9EF;">(</span>toCommand it<span style="color: #66D9EF;">)</span>.join<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot; &quot;</span><span style="color: #66D9EF;">)</span> &amp; <span style="color: #E6DB74;">&quot; &quot;</span> &amp; toCommand<span style="color: #66D9EF;">(</span>rf<span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span>.strip
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">toEffFile</span><span style="color: #AE81FF;">(</span>vetoes: <span style="color: #66D9EF;">set</span><span style="color: #66D9EF;">[</span><span style="color: #66D9EF;">Veto</span><span style="color: #66D9EF;">]</span>, rf: <span style="color: #66D9EF;">RealOrFake</span>, ecc: <span style="color: #66D9EF;">float</span>, outpath: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> =
  <span style="color: #FD971F;">result</span> = &amp;<span style="color: #E6DB74;">&quot;{outpath}/septem_veto_before_after_septem_line_ecc_cutoff_{ecc}_{toName(vetoes, rf)}.txt&quot;</span>

<span style="color: #E6DB74;">## `</span><span style="color: #AE81FF;">flatBuffers</span><span style="color: #E6DB74;">` to copy objects containing string to `</span><span style="color: #AE81FF;">procpool</span><span style="color: #E6DB74;">`</span>
<span style="color: #F92672;">import</span> flatBuffers
<span style="color: #F92672;">type</span>
  <span style="color: #66D9EF;">Command</span> = <span style="color: #F92672;">object</span>
    cmd: <span style="color: #66D9EF;">string</span>
    outputFile: <span style="color: #66D9EF;">string</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">runCommand</span><span style="color: #AE81FF;">(</span>r, w: <span style="color: #66D9EF;">cint</span><span style="color: #AE81FF;">)</span> =
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">o </span>= <span style="color: #F92672;">open</span><span style="color: #AE81FF;">(</span>w, fmWrite<span style="color: #AE81FF;">)</span>  
  <span style="color: #F92672;">for</span> cmdBuf <span style="color: #F92672;">in</span> getLenPfx<span style="color: #AE81FF;">[</span><span style="color: #66D9EF;">int</span><span style="color: #AE81FF;">](</span>r.<span style="color: #F92672;">open</span><span style="color: #AE81FF;">)</span>:
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">cmdObj </span>= flatTo<span style="color: #AE81FF;">[</span><span style="color: #66D9EF;">Command</span><span style="color: #AE81FF;">](</span>fromString<span style="color: #66D9EF;">(</span>cmdBuf<span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">cmdStr </span>= cmdObj.cmd
    <span style="color: #F92672;">let</span> <span style="color: #AE81FF;">(</span>res, err<span style="color: #AE81FF;">)</span> = shellVerbose:
      one:
        cd /tmp
        <span style="color: #AE81FF;">(</span>$cmdStr<span style="color: #AE81FF;">)</span>
    <span style="color: #75715E;"># </span><span style="color: #75715E;">write the program output as a logfile</span>
    <span style="color: #F92672;">writeFile</span><span style="color: #AE81FF;">(</span>cmdObj.outputFile, res<span style="color: #AE81FF;">)</span>
    o.urite <span style="color: #E6DB74;">&quot;Processing done for: &quot;</span> &amp; $cmdObj

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">main</span><span style="color: #AE81FF;">(</span>septem = <span style="color: #AE81FF;">false</span>,
          line = <span style="color: #AE81FF;">false</span>,
          septemLine = <span style="color: #AE81FF;">false</span>,
          eccs = <span style="color: #AE81FF;">false</span>,
          dryRun = <span style="color: #AE81FF;">false</span>,
          outpath = <span style="color: #E6DB74;">&quot;/tmp/&quot;</span>,
          jobs = <span style="font-style: italic;">6</span><span style="color: #AE81FF;">)</span> =
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">vetoSetups</span>: <span style="color: #66D9EF;">seq</span><span style="color: #AE81FF;">[</span><span style="color: #66D9EF;">set</span><span style="color: #66D9EF;">[</span><span style="color: #66D9EF;">Veto</span><span style="color: #66D9EF;">]</span><span style="color: #AE81FF;">]</span>
  <span style="color: #F92672;">if</span> septem:     vetoSetups.<span style="color: #F92672;">add</span> <span style="color: #AE81FF;">{</span><span style="color: #66D9EF;">Septem</span><span style="color: #AE81FF;">}</span>
  <span style="color: #F92672;">if</span> line:       vetoSetups.<span style="color: #F92672;">add</span> <span style="color: #AE81FF;">{</span><span style="color: #66D9EF;">Line</span><span style="color: #AE81FF;">}</span>
  <span style="color: #F92672;">if</span> septemLine: vetoSetups.<span style="color: #F92672;">add</span> <span style="color: #AE81FF;">{</span><span style="color: #66D9EF;">Septem</span>, <span style="color: #66D9EF;">Line</span><span style="color: #AE81FF;">}</span>                            
  <span style="color: #75715E;"># </span><span style="color: #75715E;">First run individual at `</span><span style="color: #AE81FF;">ε = 1.0</span><span style="color: #75715E;">`</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">cmds </span>= <span style="color: #F92672;">newSeq</span><span style="color: #AE81FF;">[</span><span style="color: #66D9EF;">Command</span><span style="color: #AE81FF;">]()</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">eccVals </span>= @<span style="color: #AE81FF;">[</span><span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">]</span>
  <span style="color: #F92672;">if</span> eccs:
    eccVals.<span style="color: #F92672;">add</span> eccStudy
  <span style="color: #F92672;">for</span> rf <span style="color: #F92672;">in</span> <span style="color: #66D9EF;">RealOrFake</span>:
    <span style="color: #F92672;">for</span> ecc <span style="color: #F92672;">in</span> eccVals:    
      <span style="color: #F92672;">for</span> vetoes <span style="color: #F92672;">in</span> vetoSetups:
        <span style="color: #F92672;">if</span> <span style="color: #66D9EF;">Line</span> <span style="color: #F92672;">in</span> vetoes <span style="color: #F92672;">or</span> ecc == <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span>: <span style="color: #E6DB74;">## If only septem veto do not perform eccentricity study!</span>
          <span style="color: #F92672;">let</span> <span style="color: #FD971F;">final </span>= cmd % <span style="color: #AE81FF;">[</span> $ecc, outpath, $ecc, toName<span style="color: #66D9EF;">(</span>vetoes, rf<span style="color: #66D9EF;">)</span>, toEffFile<span style="color: #66D9EF;">(</span>vetoes, rf, ecc, outpath<span style="color: #66D9EF;">)</span>, toCommand<span style="color: #66D9EF;">(</span>vetoes, rf<span style="color: #66D9EF;">)</span> <span style="color: #AE81FF;">]</span>
          <span style="color: #F92672;">let</span> <span style="color: #FD971F;">outputFile </span>= &amp;<span style="color: #E6DB74;">&quot;{outpath}/logL_output_septem_line_ecc_cutoff_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.txt&quot;</span> % <span style="color: #AE81FF;">[</span>$ecc, toName<span style="color: #66D9EF;">(</span>vetoes, rf<span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">]</span>
          cmds.<span style="color: #F92672;">add</span> <span style="color: #66D9EF;">Command</span><span style="color: #AE81FF;">(</span>cmd: final, outputFile: outputFile<span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;Commands to run:&quot;</span>
  <span style="color: #F92672;">for</span> cmd <span style="color: #F92672;">in</span> cmds:
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\tCommand:  &quot;</span>, cmd.cmd
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\t\tOutput: &quot;</span>, cmd.outputFile
  <span style="color: #75715E;"># </span><span style="color: #75715E;">now run if desired</span>
  <span style="color: #F92672;">if</span> <span style="color: #F92672;">not</span> dryRun:
    <span style="color: #75715E;"># </span><span style="color: #75715E;">1. fill the channel completely (so workers simply work until channel empty, then stop</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">t0 </span>= epochTime<span style="color: #AE81FF;">()</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">cmdBufs </span>= cmds.mapIt<span style="color: #AE81FF;">(</span>asFlat<span style="color: #66D9EF;">(</span>it<span style="color: #66D9EF;">)</span>.toString<span style="color: #66D9EF;">()</span><span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">pp </span>= initProcPool<span style="color: #AE81FF;">(</span>runCommand, framesLenPfx, jobs<span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">var</span> <span style="color: #FD971F;">readRes </span>= <span style="color: #F92672;">proc</span><span style="color: #AE81FF;">(</span>s: <span style="color: #66D9EF;">MSlice</span><span style="color: #AE81FF;">)</span> = <span style="color: #F92672;">echo</span> $s
    pp.evalLenPfx cmdBufs, readRes
    <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;Running all commands took: &quot;</span>, epochTime<span style="color: #AE81FF;">()</span> - t0, <span style="color: #E6DB74;">&quot; s&quot;</span>

<span style="color: #F92672;">when</span> <span style="color: #AE81FF;">isMainModule</span>:
  <span style="color: #F92672;">import</span> cligen
  dispatch main
</pre>
</div>

<p>
To run it: 
</p>
<div class="org-src-container">
<pre class="src src-sh">code/analyze_random_coinc_and_efficiency_vetoes  <span style="color: #E6DB74; font-weight: bold;">\</span>
    --septem --line --septemLine --eccs <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/resources/estimateRandomCoinc/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --jobs 16 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --dryRun
</pre>
</div>
<p>
(remove the <code>--dryRun</code> to actually run it! This way it just prints the
commands it would run)
Note that it uses much less RAM than regular <code>likelihood</code>. So we can
get away with 16 jobs.
</p>

<p>
<span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-11-20 Mon 11:29&gt;</span></span>: I&apos;ll rerun the code now with all
eccentricities. I remove the eccentricity 1.0 values, because they
were produced before we fixed the septem veto geometry stuff!
-&gt; Rerun finished.
</p>

<p>
Now we need to parse the data and compute the results from the output
text files.
</p>

<p>
Let&apos;s calculate the fraction in all cases:
</p>
<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> strutils

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">parseFile</span><span style="color: #AE81FF;">(</span>fname: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">float</span> =
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">lines </span>= fname.<span style="color: #F92672;">readFile</span>.strip.splitLines<span style="color: #AE81FF;">()</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">line </span>= <span style="font-style: italic;">0</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">numRuns </span>= <span style="font-style: italic;">0</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">outputs </span>= <span style="font-style: italic;">0</span>
  <span style="color: #75715E;"># </span><span style="color: #75715E;">if file has more than 68 lines, remove everything before, as that means</span>
  <span style="color: #75715E;"># </span><span style="color: #75715E;">those were from a previous run</span>
  <span style="color: #F92672;">if</span> <span style="color: #F92672;">lines</span>.<span style="color: #F92672;">len</span> &gt; <span style="font-style: italic;">68</span>:
    <span style="color: #F92672;">lines</span> = <span style="color: #F92672;">lines</span><span style="color: #AE81FF;">[</span>^<span style="font-style: italic;">68</span> .. ^<span style="font-style: italic;">1</span><span style="color: #AE81FF;">]</span>
    <span style="color: #F92672;">doAssert</span> <span style="color: #F92672;">lines</span>.<span style="color: #F92672;">len</span> == <span style="font-style: italic;">68</span>
  <span style="color: #F92672;">while</span> line &lt; <span style="color: #F92672;">lines</span>.<span style="color: #F92672;">len</span>:
    <span style="color: #F92672;">if</span> <span style="color: #F92672;">lines</span><span style="color: #AE81FF;">[</span>line<span style="color: #AE81FF;">]</span>.<span style="color: #F92672;">len</span> == <span style="font-style: italic;">0</span>: <span style="color: #F92672;">break</span>
    <span style="color: #75715E;"># </span><span style="color: #75715E;">parse input</span>
    <span style="color: #75715E;"># </span><span style="color: #75715E;">`</span><span style="color: #AE81FF;">Septem events before: 1069 (L,F) = (false, false)</span><span style="color: #75715E;">`</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">input </span>= <span style="color: #F92672;">lines</span><span style="color: #AE81FF;">[</span>line<span style="color: #AE81FF;">]</span>.split<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&apos;:&apos;</span><span style="color: #AE81FF;">)[</span><span style="font-style: italic;">1</span><span style="color: #AE81FF;">]</span>.strip.split<span style="color: #AE81FF;">()[</span><span style="font-style: italic;">0</span><span style="color: #AE81FF;">]</span>.parseInt
    <span style="color: #75715E;"># </span><span style="color: #75715E;">parse output</span>
    <span style="color: #75715E;"># </span><span style="color: #75715E;">`</span><span style="color: #AE81FF;">Septem events after fake cut: 137</span><span style="color: #75715E;">`</span>
    <span style="color: #F92672;">inc</span> line
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">output </span>= <span style="color: #F92672;">lines</span><span style="color: #AE81FF;">[</span>line<span style="color: #AE81FF;">]</span>.split<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&apos;:&apos;</span><span style="color: #AE81FF;">)[</span><span style="font-style: italic;">1</span><span style="color: #AE81FF;">]</span>.strip.parseInt
    <span style="color: #FD971F;">result</span> += output.<span style="color: #66D9EF;">float</span> / input.<span style="color: #66D9EF;">float</span>
    outputs += output
    <span style="color: #F92672;">inc</span> numRuns
    <span style="color: #F92672;">inc</span> line
  <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\tMean output = &quot;</span>, outputs.<span style="color: #66D9EF;">float</span> / numRuns.<span style="color: #66D9EF;">float</span>
  <span style="color: #FD971F;">result</span> = <span style="color: #FD971F;">result</span> / numRuns.<span style="color: #66D9EF;">float</span>

<span style="color: #75715E;"># </span><span style="color: #75715E;">now all files in our eccentricity cut run directory</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">path </span>= <span style="color: #E6DB74;">&quot;/home/basti/phd/resources/estimateRandomCoinc/&quot;</span>
<span style="color: #F92672;">import</span> std / <span style="color: #AE81FF;">[</span>os, parseutils<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> ggplotnim
<span style="color: #F92672;">import</span> strscans
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">parseEccentricityCutoff</span><span style="color: #AE81FF;">(</span>f: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">float</span> =
  <span style="color: #F92672;">let</span> <span style="color: #AE81FF;">(</span>success, _, ecc<span style="color: #AE81FF;">)</span> = scanTuple<span style="color: #AE81FF;">(</span>f, <span style="color: #E6DB74;">&quot;$+ecc_cutoff_$f_&quot;</span><span style="color: #AE81FF;">)</span>
  <span style="color: #FD971F;">result</span> = ecc

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">determineType</span><span style="color: #AE81FF;">(</span>f: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> =
  <span style="color: #E6DB74;">## I&apos;m sorry for this. :)</span>
  <span style="color: #F92672;">if</span> <span style="color: #E6DB74;">&quot;Septem_Line&quot;</span> <span style="color: #F92672;">in</span> f:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;SeptemLine&quot;</span>
  <span style="color: #F92672;">elif</span> <span style="color: #E6DB74;">&quot;Septem&quot;</span> <span style="color: #F92672;">in</span> f:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;Septem&quot;</span>
  <span style="color: #F92672;">elif</span> <span style="color: #E6DB74;">&quot;Line&quot;</span> <span style="color: #F92672;">in</span> f:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;Line&quot;</span>
    
  <span style="color: #F92672;">if</span> <span style="color: #E6DB74;">&quot;_fake_events.txt&quot;</span> <span style="color: #F92672;">in</span> f:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;Fake&quot;</span>
  <span style="color: #F92672;">else</span>:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;Real&quot;</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">hasSeptem</span><span style="color: #AE81FF;">(</span>f: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">bool</span> = <span style="color: #E6DB74;">&quot;Septem&quot;</span> <span style="color: #F92672;">in</span> f
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">hasLine</span><span style="color: #AE81FF;">(</span>f: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">bool</span> = <span style="color: #E6DB74;">&quot;Line&quot;</span> <span style="color: #F92672;">in</span> f
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">isFake</span><span style="color: #AE81FF;">(</span>f: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> =
  <span style="color: #F92672;">if</span> <span style="color: #E6DB74;">&quot;fake_events&quot;</span> <span style="color: #F92672;">in</span> f: <span style="color: #E6DB74;">&quot;Fake&quot;</span> <span style="color: #F92672;">else</span>: <span style="color: #E6DB74;">&quot;Real&quot;</span>

<span style="color: #F92672;">var</span> <span style="color: #FD971F;">df </span>= newDataFrame<span style="color: #AE81FF;">()</span>
<span style="color: #75715E;"># </span><span style="color: #75715E;">walk all files and determine the type</span>
<span style="color: #F92672;">for</span> f <span style="color: #F92672;">in</span> walkFiles<span style="color: #AE81FF;">(</span>path / <span style="color: #E6DB74;">&quot;septem_veto_before_after*.txt&quot;</span><span style="color: #AE81FF;">)</span>:
  <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;File: &quot;</span>, f
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">frac </span>= parseFile<span style="color: #AE81FF;">(</span>f<span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">eccCut </span>= parseEccentricityCutoff<span style="color: #AE81FF;">(</span>f<span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\tFraction of events left = &quot;</span>, frac
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">typ </span>= determineType<span style="color: #AE81FF;">(</span>f<span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\tFraction of events left = &quot;</span>, frac
  df.<span style="color: #F92672;">add</span> toDf<span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">{</span><span style="color: #E6DB74;">&quot;Type&quot;</span> : typ, <span style="color: #E6DB74;">&quot;Septem&quot;</span> : hasSeptem<span style="color: #A6E22E;">(</span>f<span style="color: #A6E22E;">)</span>, <span style="color: #E6DB74;">&quot;Line&quot;</span> : hasLine<span style="color: #A6E22E;">(</span>f<span style="color: #A6E22E;">)</span>, <span style="color: #E6DB74;">&quot;Fake&quot;</span> : isFake<span style="color: #A6E22E;">(</span>f<span style="color: #A6E22E;">)</span>, <span style="color: #E6DB74;">&quot;ε_cut&quot;</span> : eccCut, <span style="color: #E6DB74;">&quot;FractionPass&quot;</span> : frac<span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>

<span style="color: #75715E;"># </span><span style="color: #75715E;">Now write the table we want to use in the thesis for the efficiencies &amp; random coinc</span>
<span style="color: #75715E;"># </span><span style="color: #75715E;">rate</span>
<span style="color: #F92672;">import</span> std / strformat  
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">convert</span><span style="color: #AE81FF;">(</span>x: <span style="color: #66D9EF;">float</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> =
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">s </span>= &amp;<span style="color: #E6DB74;">&quot;{x * 100.0:.2f}&quot;</span>
  <span style="color: #FD971F;">result</span> = r<span style="color: #E6DB74;">&quot;$\num{&quot;</span> &amp; s &amp; <span style="color: #E6DB74;">&quot;}$&quot;</span>
  
<span style="color: #F92672;">echo</span> df.filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span>`ε_cut` == <span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
  .mutate<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">float</span> -&gt; <span style="color: #66D9EF;">string</span>: <span style="color: #E6DB74;">&quot;FractionPass&quot;</span> ~ convert<span style="color: #A6E22E;">(</span>idx<span style="color: #E6DB74;">(</span><span style="color: #E6DB74;">&quot;FractionPass&quot;</span><span style="color: #E6DB74;">)</span><span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
  .drop<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Type&quot;</span>, <span style="color: #E6DB74;">&quot;ε_cut&quot;</span><span style="color: #AE81FF;">)</span>
  .spread<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Fake&quot;</span>, <span style="color: #E6DB74;">&quot;FractionPass&quot;</span><span style="color: #AE81FF;">)</span>.toOrgTable<span style="color: #AE81FF;">()</span>
<span style="color: #75715E;"># </span><span style="color: #75715E;">And finally create the plots and output CSV file</span>
<span style="color: #F92672;">if</span> <span style="color: #AE81FF;">true</span>:
  df.writeCsv<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;/home/basti/phd/resources/septem_line_random_coincidences_ecc_cut.csv&quot;</span>, precision = <span style="font-style: italic;">8</span><span style="color: #AE81FF;">)</span>  

<span style="color: #F92672;">block</span> <span style="color: #66D9EF;">PlotFromCsv</span>:
  <span style="color: #F92672;">block</span> <span style="color: #66D9EF;">OldPlot</span>:
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">oldFile </span>= <span style="color: #E6DB74;">&quot;/home/basti/org/resources/septem_line_random_coincidences_ecc_cut.csv&quot;</span>
    <span style="color: #F92672;">if</span> fileExists<span style="color: #AE81FF;">(</span>oldFile<span style="color: #AE81FF;">)</span>:
      <span style="color: #F92672;">let</span> <span style="color: #FD971F;">df </span>= readCsv<span style="color: #AE81FF;">(</span>oldFile<span style="color: #AE81FF;">)</span>
        .filter<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span>`<span style="color: #66D9EF;">Type</span>` <span style="color: #F92672;">notin</span> <span style="color: #A6E22E;">[</span><span style="color: #E6DB74;">&quot;LinelvRegularNoHLCReal&quot;</span>, <span style="color: #E6DB74;">&quot;LinelvRegularNoHLCFake&quot;</span><span style="color: #A6E22E;">]</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
        .mutate<span style="color: #AE81FF;">(</span>f<span style="color: #66D9EF;">{</span><span style="color: #66D9EF;">string</span>: <span style="color: #E6DB74;">&quot;Type&quot;</span> ~ `<span style="color: #66D9EF;">Type</span>`.replace<span style="color: #A6E22E;">(</span><span style="color: #E6DB74;">&quot;lvRegular&quot;</span>, <span style="color: #E6DB74;">&quot;&quot;</span><span style="color: #A6E22E;">)</span>.replace<span style="color: #A6E22E;">(</span><span style="color: #E6DB74;">&quot;NoHLC&quot;</span>, <span style="color: #E6DB74;">&quot;&quot;</span><span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>
      ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;ε_cut&quot;</span>, <span style="color: #E6DB74;">&quot;FractionPass&quot;</span>, color = <span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
        geom_point<span style="color: #AE81FF;">()</span> +
        ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Fraction of events passing line veto based on ε cutoff&quot;</span><span style="color: #AE81FF;">)</span> +
        <span style="color: #75715E;">#</span><span style="color: #75715E;">margin(right = 9) +</span>
        themeLatex<span style="color: #AE81FF;">(</span>fWidth = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">9</span>, width = <span style="font-style: italic;">600</span>, baseTheme = singlePlot<span style="color: #AE81FF;">)</span> + 
        ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Figs/background/estimateSeptemVetoRandomCoinc/fraction_passing_line_veto_ecc_cut_only_relevant.pdf&quot;</span>,
               width = <span style="font-style: italic;">600</span>, height = <span style="font-style: italic;">420</span>, useTeX = <span style="color: #AE81FF;">true</span>, standalone = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>
        <span style="color: #75715E;">#</span><span style="color: #75715E;">ggsave(&quot;/tmp/fraction_passing_line_veto_ecc_cut.pdf&quot;, width = 800, height = 480)</span>
  <span style="color: #F92672;">block</span> <span style="color: #66D9EF;">NewPlot</span>:
    ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;ε_cut&quot;</span>, <span style="color: #E6DB74;">&quot;FractionPass&quot;</span>, color = <span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
      geom_point<span style="color: #AE81FF;">()</span> +
      ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Fraction of events passing line veto based on ε cutoff&quot;</span><span style="color: #AE81FF;">)</span> +
      <span style="color: #75715E;">#</span><span style="color: #75715E;">margin(right = 9) +</span>
      margin<span style="color: #AE81FF;">(</span>right = <span style="font-style: italic;">5</span>.<span style="font-style: italic;">5</span><span style="color: #AE81FF;">)</span> + 
      xlab<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Eccentricity cut &apos;ε_cut&apos;&quot;</span><span style="color: #AE81FF;">)</span> + ylab<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Fraction passing [%]&quot;</span><span style="color: #AE81FF;">)</span> + 
      themeLatex<span style="color: #AE81FF;">(</span>fWidth = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">9</span>, width = <span style="font-style: italic;">600</span>, baseTheme = singlePlot<span style="color: #AE81FF;">)</span> +       
      ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Figs/background/estimateSeptemVetoRandomCoinc/fraction_passing_line_veto_ecc_cut_only_relevant.pdf&quot;</span>,
             width = <span style="font-style: italic;">600</span>, height = <span style="font-style: italic;">420</span>, useTeX = <span style="color: #AE81FF;">true</span>, standalone = <span style="color: #AE81FF;">true</span><span style="color: #AE81FF;">)</span>
    
  
<span style="color: #E6DB74;">## XXX: we probably don&apos;t need the following plot for the real data, as the eccentricity</span>
<span style="color: #E6DB74;">## cut does not cause anything to get worse at lower values. Real improvement better than</span>
<span style="color: #E6DB74;">## fake coincidence rate.</span>
<span style="color: #75715E;">#</span><span style="color: #75715E;">df = df.spread(&quot;Type&quot;, &quot;FractionPass&quot;).mutate(f{float: &quot;Ratio&quot; ~ `</span><span style="color: #AE81FF;">Real</span><span style="color: #75715E;">` / `</span><span style="color: #AE81FF;">Fake</span><span style="color: #75715E;">`})</span>
<span style="color: #75715E;">#</span><span style="color: #75715E;">ggplot(df, aes(&quot;ε_cut&quot;, &quot;Ratio&quot;)) +</span>
<span style="color: #75715E;">#  </span><span style="color: #75715E;">geom_point() +</span>
<span style="color: #75715E;">#  </span><span style="color: #75715E;">ggtitle(&quot;Ratio of fraction of events passing line veto real/fake based on ε cutoff&quot;) + </span>
<span style="color: #75715E;">#  </span><span style="color: #75715E;">#ggsave(&quot;Figs/background/estimateSeptemVetoRandomCoinc/ratio_real_fake_fraction_passing_line_veto_ecc_cut.pdf&quot;)</span>
<span style="color: #75715E;">#  </span><span style="color: #75715E;">ggsave(&quot;/tmp/ratio_real_fake_fraction_passing_line_veto_ecc_cut.pdf&quot;)</span>
</pre>
</div>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">ESCAPE_LATEX</span>=true code/analyze_random_coincidence_results
</pre>
</div>

<p>
Old values used in <code>mcmc_limit_calculation</code>:
</p>
<div class="org-src-container">
<pre class="src src-nim">    septemVetoRandomCoinc = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">7841029411764704</span>, <span style="color: #75715E;"># </span><span style="color: #75715E;">only septem veto random coinc based on bootstrapped fake data</span>
    lineVetoRandomCoinc = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">8601764705882353</span>,   <span style="color: #75715E;"># </span><span style="color: #75715E;">lvRegular based on bootstrapped fake data</span>
    septemLineVetoRandomCoinc = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">732514705882353</span>, <span style="color: #75715E;"># </span><span style="color: #75715E;">lvRegularNoHLC based on bootstrapped fake data</span>
</pre>
</div>
<p>
New values:
</p>
<div class="org-src-container">
<pre class="src src-nim">    septemVetoRandomCoinc = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">8311</span>, <span style="color: #75715E;"># </span><span style="color: #75715E;">only septem veto random coinc based on bootstrapped fake data</span>
    lineVetoRandomCoinc = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">8539</span>,   <span style="color: #75715E;"># </span><span style="color: #75715E;">lvRegular based on bootstrapped fake data</span>
    septemLineVetoRandomCoinc = <span style="font-style: italic;">0</span>.<span style="font-style: italic;">7863</span>, <span style="color: #75715E;"># </span><span style="color: #75715E;">lvRegularNoHLC based on bootstrapped fake data</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-org662a46a" class="outline-5">
<h5 id="org662a46a">  <a href="#org662a46a">  <span class="section-number-5">12.5.5.2.</span> Initial thoughts and explanations   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-5-2" class="outline-text-5">
<p>
This way we bootstrap a larger number of events than otherwise
available and knowing that the geometric data cannot be
correlated. Any vetoing in these cases therefore <b>must</b> be a random
coincidence.
</p>

<p>
As the <code>likelihood</code> tool already uses effectively an index to map the
cluster indices for each chip to their respective event number, we&apos;ve
implemented this there (<code>--estimateRandomCoinc</code>) by rewriting the
index.
</p>

<p>
It is a good idea to also run it together with the <code>--plotseptem</code>
option to actually see some events and verify with your own eyes that
the events are actually &quot;correct&quot; (i.e. not the original ones). You
will note that there are many events that &quot;clearly&quot; look as if the
bootstrapping is not working correctly, because they look way too much
as if they are &quot;obviously correlated&quot;. To give yourself a better sense
that this is indeed just coincidence, you can run the tool with the
<code>--estFixedEvents</code> option, which bootstraps events using a fixed
cluster in the center for each run. Checking out the event displays of
those is a convincing affair that unfortunately random coincidences
are even convincing to our own eyes.
</p>

<div class="org-src-container">
<pre class="src src-sh">likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --h5out /tmp/lhood_2018_crAll_80eff_septem_fake.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --septemveto --estimateRandomCoinc
</pre>
</div>

<p>
which writes the file <code>/tmp/septem_fake_veto.txt</code>, which for this case
is found
<a href="./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_fake_events_septem_old.txt">./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_fake_events_septem_old.txt</a>
(note: updated numbers from latest state of code is the same file
without <code>_old</code> suffix)
</p>

<p>
Mean value of and fraction (from script in next section):
File: /home/basti/org/resources/septem<sub>veto</sub><sub>random</sub><sub>coincidences</sub>/septem<sub>veto</sub><sub>before</sub><sub>after</sub><sub>fake</sub><sub>events</sub><sub>septem.txt</sub>
	Mean output = 1674.705882352941
	Fraction of events left = 0.8373529411764704
</p>

<p>
From this file the method seems to remove typically a bit more than
300 out of 2000 bootstrapped fake events. This seems to imply a random
coincidence rate of about 17% (or effectively a reduction of further
17% in efficiency / 17% increase in dead time).
</p>

<p>
Of course this does not even include the line veto, which will drop it
further. Before we combine both of them, let&apos;s run it with the line
veto <span class="underline">alone</span>:
</p>
<div class="org-src-container">
<pre class="src src-sh">likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --h5out /tmp/lhood_2018_crAll_80eff_line_fake.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --lineveto --estimateRandomCoinc
</pre>
</div>
<p>
this results in:
<a href="./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_fake_events_line.txt">./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_fake_events_line.txt</a>
</p>

<p>
Mean value of:
File: /home/basti/org/resources/septem<sub>veto</sub><sub>random</sub><sub>coincidences</sub>/septem<sub>veto</sub><sub>before</sub><sub>after</sub><sub>fake</sub><sub>events</sub><sub>line.txt</sub>
	Mean output = 1708.382352941177
	Fraction of events left = 0.8541911764705882
</p>

<p>
And finally both together:
</p>
<div class="org-src-container">
<pre class="src src-sh">likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --h5out /tmp/lhood_2018_crAll_80eff_septem_line_fake.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --septemveto --lineveto --estimateRandomCoinc
</pre>
</div>

<p>
which generated the following output:
<a href="./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_fake_events_septem_line.txt">./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_fake_events_septem_line.txt</a>
</p>

<p>
Mean value of: 
File: /home/basti/org/resources/septem<sub>veto</sub><sub>random</sub><sub>coincidences</sub>/septem<sub>veto</sub><sub>before</sub><sub>after</sub><sub>fake</sub><sub>events</sub><sub>septem</sub><sub>line.txt</sub>
	Mean output = 1573.676470588235
	Fraction of events left = 0.7868382352941178
</p>

<p>
This comes out to a fraction of 78.68% of the events left after
running the vetoes on our bootstrapped fake events. Combining it with
a software efficiency of ε = 80% the total combined efficiency then
would be \(ε_\text{total} = 0.8 · 0.7868 = 0.629\), so about 63%.
</p>


<p>
Finally now let&apos;s prepare some event displays for the case of using
different center clusters and using the same ones. We run the
<code>likelihood</code> tool with the <code>--plotSeptem</code> option and stop the program
after we have enough plots.
In this context note the energy cut range for the <code>--plotseptem</code>
option (by default set to 5 keV), adjustable by the
<code>PLOT_SEPTEM_E_CUTOFF</code> environment variable.
</p>

<p>
First with different center clusters:
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">PLOT_SEPTEM_E_CUTOFF</span>=10.0 likelihood <span style="color: #E6DB74; font-weight: bold;">\</span>
                          -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                          --h5out /tmp/dummy.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                          --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
                          --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                          --septemveto --lineveto --estimateRandomCoinc --plotseptem
</pre>
</div>
<p>
which are wrapped up using <code>pdfunite</code> and stored in:
<img alt="fake_events_septem_line_veto_all_outer_events.png" src="./figs/home/basti/phd/Figs/background/estimateSeptemVetoRandomCoinc/fake_events_septem_line_veto_all_outer_events.png" />
</p>

<p>
and now with fixed clusters:
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">PLOT_SEPTEM_E_CUTOFF</span>=10.0 likelihood <span style="color: #E6DB74; font-weight: bold;">\</span>
                          -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                          --h5out /tmp/dummy.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                          --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
                          --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
                          --septemveto --lineveto --estimateRandomCoinc --estFixedEvent --plotseptem
</pre>
</div>
<p>
(Note that the cluster that is chosen can be set using
<code>SEPTEM_FAKE_FIXED_CLUSTER</code> to a different index, by default it just
uses <code>5</code>).
These events are here:
<img alt="fake_events_fixed_cluster_septem_line_veto_all_outer_events.png" src="./figs/home/basti/phd/Figs/background/estimateSeptemVetoRandomCoinc/fake_events_fixed_cluster_septem_line_veto_all_outer_events.png" />
</p>
</div>
</div>
<div id="outline-container-orgc298e7e" class="outline-5">
<h5 id="orgc298e7e">  <a href="#orgc298e7e">  <span class="section-number-5">12.5.5.3.</span> <span class="todo TODO">TODO</span> Rewrite the whole estimation to a proper program <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-5-3" class="outline-text-5">
<p>
<b>IMPORTANT</b>
</p>

<p>
That program should call <code>likelihood</code> alone, and <code>likelihood</code> needs to
be rewritten such that it outputs the septem random coincidence (or
real removal) into the H5 output file. Maybe just add a type that
stores the information which we serialize.
With the serialized info about the veto settings we can then
reconstruct in code what is what.
</p>

<p>
Or possibly better if the output is written to a separate file such
that we don&apos;t store all the cluster data.
</p>

<p>
Anyhow, then rewrite the code snippet in the section below that prints
the information about the random coincidence rates and creates the
plot.
</p>
</div>
</div>
<div id="outline-container-orgc184333" class="outline-5">
<h5 id="orgc184333">  <a href="#orgc184333">  <span class="section-number-5">12.5.5.4.</span> Run a whole bunch more cases   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-5-4" class="outline-text-5">
<p>
The below is running now <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-02-10 Fri 01:43&gt;</span></span>.
Still running as of <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-02-10 Fri 11:55&gt;</span></span>, damn this is slow.
</p>
<ul class="org-ul">
<li class="on"><code>[X]</code> <b>INVESTIGATE PERFORMANCE AFTER IT&apos;S DONE</b></li>
<li class="off"><code>[ ]</code> We should be able to run ~4 (depending on choice even more) in
parallel, no?</li>
</ul>

<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> shell, strutils, os

<span style="color: #75715E;">#</span><span style="color: #75715E;">let vals = @[1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2.0]</span>
<span style="color: #75715E;">#</span><span style="color: #75715E;">let vals = @[1.0, 1.1]</span>
<span style="color: #F92672;">let</span> <span style="color: #FD971F;">vals </span>= @<span style="color: #AE81FF;">[</span><span style="font-style: italic;">1</span>.<span style="font-style: italic;">0</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">1</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">2</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">3</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">4</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">5</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">6</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">7</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">8</span>, <span style="font-style: italic;">1</span>.<span style="font-style: italic;">9</span>, <span style="font-style: italic;">2</span>.<span style="font-style: italic;">0</span><span style="color: #AE81FF;">]</span>
<span style="color: #75715E;">#</span><span style="color: #75715E;">let vetoes = @[&quot;--lineveto&quot;, &quot;--lineveto --estimateRandomCoinc&quot;]</span>
<span style="color: #F92672;">let</span> <span style="color: #FD971F;">vetoes </span>= @<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;--septemveto --lineveto&quot;</span>, <span style="color: #E6DB74;">&quot;--septemveto --lineveto --estimateRandomCoinc&quot;</span><span style="color: #AE81FF;">]</span>

<span style="color: #E6DB74;">## XXX: ADD CODE DIFFERENTIATING SEPTEM + LINE &amp; LINE ONLY IN NAMES AS WELL!</span>
<span style="color: #75715E;">#</span><span style="color: #75715E;">const lineVeto = &quot;lvRegular&quot;</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">lineVeto </span>= <span style="color: #E6DB74;">&quot;lvRegularNoHLC&quot;</span>
<span style="color: #F92672;">let</span> <span style="color: #FD971F;">cmd </span>= <span style="color: #E6DB74;">&quot;&quot;&quot;</span>
<span style="color: #E6DB74;">LINE_VETO_KIND=</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;"> \</span>
<span style="color: #E6DB74;">    ECC_LINE_VETO_CUT=</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;"> \</span>
<span style="color: #E6DB74;">    USE_REAL_LAYOUT=true \</span>
<span style="color: #E6DB74;">    likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 \</span>
<span style="color: #E6DB74;">    --h5out /t/lhood_2018_crAll_80eff_septem_line_ecc_cutoff_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">_real_layout</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.h5 \</span>
<span style="color: #E6DB74;">    --region crAll --cdlYear 2018 \</span>
<span style="color: #E6DB74;">    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 </span><span style="color: #F92672;">$#</span>
<span style="color: #E6DB74;">&quot;&quot;&quot;</span>
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">toName</span><span style="color: #AE81FF;">(</span>veto: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> = <span style="color: #AE81FF;">(</span><span style="color: #F92672;">if</span> <span style="color: #E6DB74;">&quot;estimateRandomCoinc&quot;</span> <span style="color: #F92672;">in</span> veto: <span style="color: #E6DB74;">&quot;_fake_events&quot;</span> <span style="color: #F92672;">else</span>: <span style="color: #E6DB74;">&quot;&quot;</span><span style="color: #AE81FF;">)</span>
<span style="color: #F92672;">for</span> val <span style="color: #F92672;">in</span> vals:
  <span style="color: #F92672;">for</span> veto <span style="color: #F92672;">in</span> vetoes:
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">final </span>= cmd % <span style="color: #AE81FF;">[</span> lineVeto, $val, $val, lineVeto, toName<span style="color: #66D9EF;">(</span>veto<span style="color: #66D9EF;">)</span>,
                        $veto <span style="color: #AE81FF;">]</span>
    <span style="color: #F92672;">let</span> <span style="color: #AE81FF;">(</span>res, err<span style="color: #AE81FF;">)</span> = shellVerbose:
      one:
        cd /tmp
        <span style="color: #AE81FF;">(</span>$final<span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">writeFile</span><span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;/tmp/logL_output_septem_line_ecc_cutoff_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">_real_layout</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.txt&quot;</span> % <span style="color: #66D9EF;">[</span>$val, lineVeto, toName<span style="color: #A6E22E;">(</span>veto<span style="color: #A6E22E;">)</span><span style="color: #66D9EF;">]</span>, res<span style="color: #AE81FF;">)</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">outpath </span>= <span style="color: #E6DB74;">&quot;/home/basti/org/resources/septem_veto_random_coincidences/autoGen/&quot;</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">outfile </span>= <span style="color: #E6DB74;">&quot;septem_veto_before_after_septem_line_ecc_cutoff_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">_</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">_real_layout</span><span style="color: #F92672;">$#</span><span style="color: #E6DB74;">.txt&quot;</span> % <span style="color: #AE81FF;">[</span>$val, lineVeto, toName<span style="color: #66D9EF;">(</span>veto<span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">]</span>
    copyFile<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;/tmp/septem_veto_before_after.txt&quot;</span>, outpath / outfile<span style="color: #AE81FF;">)</span>
    removeFile<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;/tmp/septem_veto_before_after.txt&quot;</span><span style="color: #AE81FF;">)</span> <span style="color: #75715E;"># </span><span style="color: #75715E;">remove file to not append more and more to file</span>
</pre>
</div>

<p>
It has finally finished some time before <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-02-10 Fri 20:02&gt;</span></span>. Holy
moly how slow.
</p>

<p>
We will keep the generated <code>lhood_*</code> and <code>logL_output_*</code> files in
<a href="./../org/resources/septem_veto_random_coincidences/autoGen/">./../org/resources/septem_veto_random_coincidences/autoGen/</a> together
with the <code>septem_veto_befor_after_*</code> files.
</p>

<p>
See the code in one of the next sections for the &apos;analysis&apos; of this
dataset.
</p>
</div>
</div>
<div id="outline-container-orgdb12673" class="outline-5">
<h5 id="orgdb12673">  <a href="#orgdb12673">  <span class="section-number-5">12.5.5.5.</span> Number of events removed in real usage   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-5-5" class="outline-text-5">
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>MAYBE EXTEND CODE SNIPPET ABOVE TO ALLOW CHOOSING BETWEEN ε<sub>cut</sub>
ANALYSIS AND REAL FRACTIONS</b></li>
</ul>


<p>
As a reference let&apos;s quickly run the code also for the normal use case
where we don&apos;t do any bootstrapping:
</p>

<div class="org-src-container">
<pre class="src src-sh">likelihood <span style="color: #E6DB74; font-weight: bold;">\</span>
  -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --h5out /tmp/dummy_real.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --septemveto
</pre>
</div>

<p>
which results in
<a href="./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_only_septem.txt">./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_only_septem.txt</a>
</p>

<p>
Next the line veto alone:
</p>
<div class="org-src-container">
<pre class="src src-sh">likelihood <span style="color: #E6DB74; font-weight: bold;">\</span>
  -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --h5out /tmp/dummy_real.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --lineveto
</pre>
</div>

<p>
which results in:
<a href="./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_only_line.txt">./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_only_line.txt</a>
</p>

<p>
And finally both together:
</p>
<div class="org-src-container">
<pre class="src src-sh">likelihood <span style="color: #E6DB74; font-weight: bold;">\</span>
  -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --h5out /tmp/dummy_real_2.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --septemveto --lineveto 
</pre>
</div>

<p>
and this finally yields:
</p>

<p>
<a href="./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_septem_line.txt">./../org/resources/septem_veto_random_coincidences/septem_veto_before_after_septem_line.txt</a>
</p>


<p>
And further for reference let&apos;s compute the fake rate when only using the
septem veto (as we have no eccentricity dependence, hence a single
value):
</p>
<div class="org-src-container">
<pre class="src src-sh">likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --h5out /tmp/lhood_2018_crAll_80eff_septem_real_layout.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --region crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
  --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
  --septemveto <span style="color: #E6DB74; font-weight: bold;">\</span>
  --estimateRandomCoinc
</pre>
</div>


<p>
Run the line veto with new features:
</p>
<ul class="org-ul">
<li>real septemboard layout</li>
<li>eccentricity cut off for tracks participating (ecc &gt; 1.6)</li>
</ul>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">LINE_VETO_KIND</span>=lvRegularNoHLC <span style="color: #E6DB74; font-weight: bold;">\</span>
    <span style="color: #FD971F;">ECC_LINE_VETO_CUT</span>=1.6 <span style="color: #E6DB74; font-weight: bold;">\</span>
    <span style="color: #FD971F;">USE_REAL_LAYOUT</span>=true <span style="color: #E6DB74; font-weight: bold;">\</span>
    likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --h5out /tmp/lhood_2018_crAll_80eff_line_ecc_cutof_1.6_real_layout.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crAll <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lineveto
</pre>
</div>

<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>WE SHOULD REALLY LOOK INTO RUNNING THE LINE VETO ONLY USING
DIFFERENT ε CUTOFFS!</b>
-&gt; Then compare the real application with the fake bootstrap
application and see if there is a sweet spot in terms of S/N.</li>
</ul>

<p>
Let&apos;s calculate the fraction in all cases:
</p>
<div class="org-src-container">
<pre class="src src-nim"><span style="color: #F92672;">import</span> strutils
<span style="color: #F92672;">let</span> <span style="color: #FD971F;">files </span>= @<span style="color: #AE81FF;">[</span><span style="color: #E6DB74;">&quot;/home/basti/org/resources/septem_veto_random_coincidences/septem_veto_before_after_only_septem.txt&quot;</span>,
              <span style="color: #E6DB74;">&quot;/home/basti/org/resources/septem_veto_random_coincidences/septem_veto_before_after_only_line.txt&quot;</span>,              
              <span style="color: #E6DB74;">&quot;/home/basti/org/resources/septem_veto_random_coincidences/septem_veto_before_after_septem_line.txt&quot;</span>,
              <span style="color: #E6DB74;">&quot;/home/basti/org/resources/septem_veto_random_coincidences/septem_veto_before_after_fake_events_septem.txt&quot;</span>,
              <span style="color: #E6DB74;">&quot;/home/basti/org/resources/septem_veto_random_coincidences/septem_veto_before_after_fake_events_line.txt&quot;</span>,              
              <span style="color: #E6DB74;">&quot;/home/basti/org/resources/septem_veto_random_coincidences/septem_veto_before_after_fake_events_septem_line.txt&quot;</span><span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">parseFile</span><span style="color: #AE81FF;">(</span>fname: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">float</span> =
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">lines </span>= fname.<span style="color: #F92672;">readFile</span>.strip.splitLines<span style="color: #AE81FF;">()</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">line </span>= <span style="font-style: italic;">0</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">numRuns </span>= <span style="font-style: italic;">0</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">outputs </span>= <span style="font-style: italic;">0</span>
  <span style="color: #75715E;"># </span><span style="color: #75715E;">if file has more than 68 lines, remove everything before, as that means</span>
  <span style="color: #75715E;"># </span><span style="color: #75715E;">those were from a previous run</span>
  <span style="color: #F92672;">if</span> <span style="color: #F92672;">lines</span>.<span style="color: #F92672;">len</span> &gt; <span style="font-style: italic;">68</span>:
    <span style="color: #F92672;">lines</span> = <span style="color: #F92672;">lines</span><span style="color: #AE81FF;">[</span>^<span style="font-style: italic;">68</span> .. ^<span style="font-style: italic;">1</span><span style="color: #AE81FF;">]</span>
    <span style="color: #F92672;">doAssert</span> <span style="color: #F92672;">lines</span>.<span style="color: #F92672;">len</span> == <span style="font-style: italic;">68</span>
  <span style="color: #F92672;">while</span> line &lt; <span style="color: #F92672;">lines</span>.<span style="color: #F92672;">len</span>:
    <span style="color: #F92672;">if</span> <span style="color: #F92672;">lines</span><span style="color: #AE81FF;">[</span>line<span style="color: #AE81FF;">]</span>.<span style="color: #F92672;">len</span> == <span style="font-style: italic;">0</span>: <span style="color: #F92672;">break</span>
    <span style="color: #75715E;"># </span><span style="color: #75715E;">parse input</span>
    <span style="color: #75715E;"># </span><span style="color: #75715E;">`</span><span style="color: #AE81FF;">Septem events before: 1069 (L,F) = (false, false)</span><span style="color: #75715E;">`</span>
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">input </span>= <span style="color: #F92672;">lines</span><span style="color: #AE81FF;">[</span>line<span style="color: #AE81FF;">]</span>.split<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&apos;:&apos;</span><span style="color: #AE81FF;">)[</span><span style="font-style: italic;">1</span><span style="color: #AE81FF;">]</span>.strip.split<span style="color: #AE81FF;">()[</span><span style="font-style: italic;">0</span><span style="color: #AE81FF;">]</span>.parseInt
    <span style="color: #75715E;"># </span><span style="color: #75715E;">parse output</span>
    <span style="color: #75715E;"># </span><span style="color: #75715E;">`</span><span style="color: #AE81FF;">Septem events after fake cut: 137</span><span style="color: #75715E;">`</span>
    <span style="color: #F92672;">inc</span> line
    <span style="color: #F92672;">let</span> <span style="color: #FD971F;">output </span>= <span style="color: #F92672;">lines</span><span style="color: #AE81FF;">[</span>line<span style="color: #AE81FF;">]</span>.split<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&apos;:&apos;</span><span style="color: #AE81FF;">)[</span><span style="font-style: italic;">1</span><span style="color: #AE81FF;">]</span>.strip.parseInt
    <span style="color: #FD971F;">result</span> += output.<span style="color: #66D9EF;">float</span> / input.<span style="color: #66D9EF;">float</span>
    outputs += output
    <span style="color: #F92672;">inc</span> numRuns
    <span style="color: #F92672;">inc</span> line
  <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\tMean output = &quot;</span>, outputs.<span style="color: #66D9EF;">float</span> / numRuns.<span style="color: #66D9EF;">float</span>
  <span style="color: #FD971F;">result</span> = <span style="color: #FD971F;">result</span> / numRuns.<span style="color: #66D9EF;">float</span>

<span style="color: #75715E;"># </span><span style="color: #75715E;">first the predefined files:  </span>
<span style="color: #F92672;">for</span> f <span style="color: #F92672;">in</span> files:
  <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;File: &quot;</span>, f
  <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\tFraction of events left = &quot;</span>, parseFile<span style="color: #AE81FF;">(</span>f<span style="color: #AE81FF;">)</span>

<span style="color: #75715E;"># </span><span style="color: #75715E;">now all files in our eccentricity cut run directory</span>
<span style="color: #F92672;">const</span> <span style="color: #FD971F;">path </span>= <span style="color: #E6DB74;">&quot;/home/basti/org/resources/septem_veto_random_coincidences/autoGen/&quot;</span>
<span style="color: #F92672;">import</span> std / <span style="color: #AE81FF;">[</span>os, parseutils, strutils<span style="color: #AE81FF;">]</span>
<span style="color: #F92672;">import</span> ggplotnim
<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">parseEccentricityCutoff</span><span style="color: #AE81FF;">(</span>f: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">float</span> =
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">str </span>= <span style="color: #E6DB74;">&quot;ecc_cutoff_&quot;</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">startIdx </span>= <span style="color: #F92672;">find</span><span style="color: #AE81FF;">(</span>f, str<span style="color: #AE81FF;">)</span> + str.<span style="color: #F92672;">len</span>
  <span style="color: #F92672;">var</span> <span style="color: #FD971F;">res </span>= <span style="color: #E6DB74;">&quot;&quot;</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">stopIdx </span>= parseUntil<span style="color: #AE81FF;">(</span>f, res, until = <span style="color: #E6DB74;">&quot;_&quot;</span>, start = startIdx<span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">echo</span> res
  <span style="color: #FD971F;">result</span> = parseFloat<span style="color: #AE81FF;">(</span>res<span style="color: #AE81FF;">)</span>

<span style="color: #F92672;">proc</span> <span style="color: #A6E22E;">determineType</span><span style="color: #AE81FF;">(</span>f: <span style="color: #66D9EF;">string</span><span style="color: #AE81FF;">)</span>: <span style="color: #66D9EF;">string</span> =
  <span style="color: #E6DB74;">## I&apos;m sorry for this. :) </span>
  <span style="color: #F92672;">if</span> <span style="color: #E6DB74;">&quot;only_line_ecc&quot;</span> <span style="color: #F92672;">in</span> f:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;Line&quot;</span>
  <span style="color: #F92672;">elif</span> <span style="color: #E6DB74;">&quot;septem_line_ecc&quot;</span> <span style="color: #F92672;">in</span> f:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;SeptemLine&quot;</span>
  <span style="color: #F92672;">else</span>:
    <span style="color: #F92672;">doAssert</span> <span style="color: #AE81FF;">false</span>, <span style="color: #E6DB74;">&quot;What? &quot;</span> &amp; $f
  <span style="color: #F92672;">if</span> <span style="color: #E6DB74;">&quot;lvRegularNoHLC&quot;</span> <span style="color: #F92672;">in</span> f:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;lvRegularNoHLC&quot;</span>
  <span style="color: #F92672;">elif</span> <span style="color: #E6DB74;">&quot;lvRegular&quot;</span> <span style="color: #F92672;">in</span> f:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;lvRegular&quot;</span>
  <span style="color: #F92672;">else</span>: <span style="color: #75715E;"># </span><span style="color: #75715E;">also lvRegularNoHLC, could use else above, but clearer this way. Files </span>
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;lvRegularNoHLC&quot;</span> <span style="color: #75715E;"># </span><span style="color: #75715E;">without veto kind are older, therefore no HLC</span>
  <span style="color: #F92672;">if</span> <span style="color: #E6DB74;">&quot;_fake_events.txt&quot;</span> <span style="color: #F92672;">in</span> f:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;Fake&quot;</span>
  <span style="color: #F92672;">else</span>:
    <span style="color: #FD971F;">result</span>.<span style="color: #F92672;">add</span> <span style="color: #E6DB74;">&quot;Real&quot;</span>

<span style="color: #F92672;">var</span> <span style="color: #FD971F;">df </span>= newDataFrame<span style="color: #AE81FF;">()</span>
<span style="color: #75715E;"># </span><span style="color: #75715E;">walk all files and determine the type</span>
<span style="color: #F92672;">for</span> f <span style="color: #F92672;">in</span> walkFiles<span style="color: #AE81FF;">(</span>path / <span style="color: #E6DB74;">&quot;septem_veto_before_after*.txt&quot;</span><span style="color: #AE81FF;">)</span>:
  <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;File: &quot;</span>, f
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">frac </span>= parseFile<span style="color: #AE81FF;">(</span>f<span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">eccCut </span>= parseEccentricityCutoff<span style="color: #AE81FF;">(</span>f<span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">let</span> <span style="color: #FD971F;">typ </span>= determineType<span style="color: #AE81FF;">(</span>f<span style="color: #AE81FF;">)</span>
  <span style="color: #F92672;">echo</span> <span style="color: #E6DB74;">&quot;\tFraction of events left = &quot;</span>, frac
  df.<span style="color: #F92672;">add</span> toDf<span style="color: #AE81FF;">(</span><span style="color: #66D9EF;">{</span><span style="color: #E6DB74;">&quot;Type&quot;</span> : typ, <span style="color: #E6DB74;">&quot;ε_cut&quot;</span> : eccCut, <span style="color: #E6DB74;">&quot;FractionPass&quot;</span> : frac<span style="color: #66D9EF;">}</span><span style="color: #AE81FF;">)</span>

df.writeCsv<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;/home/basti/org/resources/septem_line_random_coincidences_ecc_cut.csv&quot;</span>, precision = <span style="font-style: italic;">8</span><span style="color: #AE81FF;">)</span>  
ggplot<span style="color: #AE81FF;">(</span>df, aes<span style="color: #66D9EF;">(</span><span style="color: #E6DB74;">&quot;ε_cut&quot;</span>, <span style="color: #E6DB74;">&quot;FractionPass&quot;</span>, color = <span style="color: #E6DB74;">&quot;Type&quot;</span><span style="color: #66D9EF;">)</span><span style="color: #AE81FF;">)</span> +
  geom_point<span style="color: #AE81FF;">()</span> +
  ggtitle<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Fraction of events passing line veto based on ε cutoff&quot;</span><span style="color: #AE81FF;">)</span> +
  margin<span style="color: #AE81FF;">(</span>right = <span style="font-style: italic;">9</span><span style="color: #AE81FF;">)</span> + 
  ggsave<span style="color: #AE81FF;">(</span><span style="color: #E6DB74;">&quot;Figs/background/estimateSeptemVetoRandomCoinc/fraction_passing_line_veto_ecc_cut.pdf&quot;</span>,
         width = <span style="font-style: italic;">800</span>, height = <span style="font-style: italic;">480</span><span style="color: #AE81FF;">)</span>
  <span style="color: #75715E;">#</span><span style="color: #75715E;">ggsave(&quot;/tmp/fraction_passing_line_veto_ecc_cut.pdf&quot;, width = 800, height = 480)</span>

<span style="color: #E6DB74;">## XXX: we probably don&apos;t need the following plot for the real data, as the eccentricity</span>
<span style="color: #E6DB74;">## cut does not cause anything to get worse at lower values. Real improvement better than</span>
<span style="color: #E6DB74;">## fake coincidence rate.</span>
<span style="color: #75715E;">#</span><span style="color: #75715E;">df = df.spread(&quot;Type&quot;, &quot;FractionPass&quot;).mutate(f{float: &quot;Ratio&quot; ~ `</span><span style="color: #AE81FF;">Real</span><span style="color: #75715E;">` / `</span><span style="color: #AE81FF;">Fake</span><span style="color: #75715E;">`})</span>
<span style="color: #75715E;">#</span><span style="color: #75715E;">ggplot(df, aes(&quot;ε_cut&quot;, &quot;Ratio&quot;)) +</span>
<span style="color: #75715E;">#  </span><span style="color: #75715E;">geom_point() +</span>
<span style="color: #75715E;">#  </span><span style="color: #75715E;">ggtitle(&quot;Ratio of fraction of events passing line veto real/fake based on ε cutoff&quot;) + </span>
<span style="color: #75715E;">#  </span><span style="color: #75715E;">#ggsave(&quot;Figs/background/estimateSeptemVetoRandomCoinc/ratio_real_fake_fraction_passing_line_veto_ecc_cut.pdf&quot;)</span>
<span style="color: #75715E;">#  </span><span style="color: #75715E;">ggsave(&quot;/tmp/ratio_real_fake_fraction_passing_line_veto_ecc_cut.pdf&quot;)</span>
</pre>
</div>

<p>
(about the first set of files)
So about 14.8% in the only septem case and 9.9% in the septem + line
veto case.
</p>


<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>MOVE BELOW TO PROPER THESIS PART!</b></li>
</ul>
<p>
(about the ε cut)
</p>

<figure id="fig:background:fraction_passing_line_veto_ecc_cut">
<img alt="fraction_passing_line_veto_ecc_cut.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/estimateSeptemVetoRandomCoinc/fraction_passing_line_veto_ecc_cut.svg" />

<figcaption>Figure 33: <span class="figure-number">Figure 79: </span>Fraction of events in Run-3 data (green), which pass (i.e. not rejected) the line veto depending on the eccentricity cut used, which decides how eccentric a cluster needs to be in order to be used for the veto. The purple points are using fake bootstrapped data from real clusters passing the \(\ln\mathcal{L}\) cut together with real outer GridPix data from <b>other</b> events. The fraction of events being vetoed in the latter is a measure for the random coincidence rate.</figcaption>
</figure>


<figure id="org71dc44f">
<img alt="ratio_real_fake_fraction_passing_line_veto_ecc_cut.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/estimateSeptemVetoRandomCoinc/ratio_real_fake_fraction_passing_line_veto_ecc_cut.svg" />

<figcaption>Figure 34: <span class="figure-number">Figure 80: </span>Ratio of the events passing in the real line veto application to the fake data application for different \(ε_{\text{cut}}\) cutoff values. The optimum seems to be in the range of 1.4 to 1.5</figcaption>
</figure>
</div>
<ol class="org-ol">
<li><a id="org955d399"> </a>Investigate significantly lower fake event fraction passing<br />
<div id="text-12-5-5-5-1" class="outline-text-6">
<p>
<b>UPDATE</b>: <span class="timestamp-wrapper">  <span class="timestamp">&lt;2023-02-13 Mon 16:50&gt;</span></span>
</p>

<p>
The numbers visible in the plot are <b>MUCH LOWER</b> than what we had
previously after implementing the line veto alone!!
</p>

<p>
Let&apos;s run with the equivalent of the old parameters:
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">LINE_VETO_KIND</span>=lvRegular <span style="color: #E6DB74; font-weight: bold;">\</span>
    <span style="color: #FD971F;">ECC_LINE_VETO_CUT</span>=1.0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    <span style="color: #FD971F;">USE_REAL_LAYOUT</span>=false <span style="color: #E6DB74; font-weight: bold;">\</span>
    likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --h5out /t/lhood_2018_crAll_80eff_line_ecc_cutof_1.0_tight_layout_lvRegular.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 --lineveto --estimateRandomCoinc
</pre>
</div>

<p>
-&gt; As it turns out this was a bug in our logic that decides which
cluster is of interest to the line veto. We accidentally always deemed
it interesting, if the original cluster was on its own… Fixed now.
</p>
</div>
</li>
</ol>
</div>
</div>
<div id="outline-container-org7db09f3" class="outline-4">
<h4 id="org7db09f3">  <a href="#org7db09f3">  <span class="section-number-4">12.5.6.</span> On the line veto without septem veto   <span class="tag">    <span class="extended">extended</span>  </span></a></h4>
<div id="text-12-5-6" class="outline-text-4">
<p>
When dealing with the line veto without the septem veto there are
multiple questions that come up of course.
</p>

<p>
First of all, what is the cluster that you&apos;re actually targeting with
our &apos;line&apos;? The original cluster (OC) that passed lnL or a
hypothetical larger cluster that was found during the septem event
reconstruction (HLC).
</p>

<p>
Assuming the former, the next question is whether we want to allow an
HLC to veto our OC? In a naive implementation this is precisely what&apos;s
happening, because in the regular use case of septem veto + line veto,
the line veto would never have any effect anymore, as an HLC would
almost certainly be vetoed by the septem veto! But without the septem
veto, this decision is fully up to the line veto and the question
becomes relevant.
(we will implement a switch, maybe based on an environment variable or
flag)
</p>

<p>
In the latter case the tricky part is mainly just identifying the
correct cluster which to test for in order to find its
center. However, this needs to be implemented to avoid the HLC in the
above mentioned case. With it done, we then have 3 different ways to
do the line veto:
</p>

<ol class="org-ol">
<li>&apos;regular&apos; line veto. <b>Every</b> cluster checks the line to the center
cluster. Without septem veto this includes HLC checking OC.</li>
<li>&apos;regular without HLC&apos; line veto: Lines check the OC, but the HLC is
explicitly <b>not</b> considered.</li>
<li>&apos;checking the HLC&apos; line veto: In this case <b>all</b> clusters check the
center of the HLC.</li>
</ol>


<p>
Thoughts on LvCheckHLC:
</p>
<ul class="org-ul">
<li>The radii around the new HLC become so large that in practice this
won&apos;t be a very good idea I think!</li>
<li>The <code>lineVetoRejected</code> part of the title seems to be &quot;true&quot; in too
many cases. What&apos;s going on here? See:
<img alt="septem_events_only_line_veto_check_hlc_not_final.svg" class="org-svg" src="./figs/home/basti/org/Figs/statusAndProgress/estimateSeptemVetoRandomCoinc/septem_events_only_line_veto_check_hlc_not_final.svg" />
for example &quot;2882 and run 297&quot; on page 31. Like huh?
My first guess is that the distance calculation is off somehow?
Similar page 33 and probably many more. Even worse is page 34: &quot;event 30 and run 297&quot;!
-&gt; Yeah, as it turns out the problem was just that our
<code>inRegionOfInterest</code> check had become outdated due to our change of</li>

<li class="off"><code>[ ]</code> Select example events for each of the &apos;line veto kinds&apos; to
demonstrate their differences.</li>
</ul>

<p>
OC: Original Cluster (passing lnL cut on center chip)
HCL: Hypothetical Large Cluster (new cluster that OC is part of after
septemboard reco)
Regular:  
<img alt="example_event_line_veto_regular.svg" class="org-svg" src="./figs/home/basti/org/Figs/statusAndProgress/estimateSeptemVetoRandomCoinc/example_event_line_veto_regular.svg" />
is an example event in which we see the &quot;regular&quot; line veto without
using the septem veto. Things to note:
</p>
<ul class="org-ul">
<li>the black circle shows the &apos;radius&apos; of the OC, <b>not</b> the HLC</li>
<li>the OC is actually part of a HLC</li>
<li>because of this and because the HLC is a nice track, the event is
<b>vetoed</b>, not by the green track, but by the HLC itself!</li>
</ul>
<p>
This wouldn&apos;t be a problem if we also used the septem veto, as this
event would already be removed due to the septem veto!
(More plots: <img alt="septem_events_only_line_veto_regular_fixed_check.png" src="./figs/home/basti/org/Figs/statusAndProgress/estimateSeptemVetoRandomCoinc/septem_events_only_line_veto_regular_fixed_check.png" />)
</p>

<p>
Regular no HLC:
<img alt="example_event_line_veto_regular_noHLC.svg" class="org-svg" src="./figs/home/basti/org/Figs/statusAndProgress/estimateSeptemVetoRandomCoinc/example_event_line_veto_regular_noHLC.svg" />
The reference cluster to check for is still the regular OC with the
same radius. And again the OC is part of an HLC. However, in contrast
to the &apos;regular&apos; case, this event is not vetoed. The green and purple
clusters simply don&apos;t point at the black circle and the HLC itself is
<b>not considered</b> here. This defines the &apos;regular no HLC&apos; veto.
<img alt="example_event_line_veto_regular_noHLC_close_hit.svg" class="org-svg" src="./figs/home/basti/org/Figs/statusAndProgress/estimateSeptemVetoRandomCoinc/example_event_line_veto_regular_noHLC_close_hit.svg" />
is just an example of an event that proves the method works &amp; a nice
example of a cluster <span class="underline">barely</span> hitting the radius of the OC.
On the other hand though this is also a good example for why we should
have an eccentricity cut on those clusters that we use to check for
lines! The green cluster in this second event is not even remotely
eccentric enough and indeed is actually part of the orange track!
(More plots: <img alt="septem_events_only_line_veto_regular_noHCL_fixed_check.png" src="./figs/home/basti/org/Figs/statusAndProgress/estimateSeptemVetoRandomCoinc/septem_events_only_line_veto_regular_noHCL_fixed_check.png" />)
</p>

<p>
Check HLC cluster:
<img alt="example_event_line_veto_check_hlc_is_problematic.svg" class="org-svg" src="./figs/home/basti/org/Figs/statusAndProgress/estimateSeptemVetoRandomCoinc/example_event_line_veto_check_hlc_is_problematic.svg" />
Is an example event where we can see how ridiculous the &quot;check HLC&quot;
veto kind can become. There is a very large cluster that the OC is
actually part of (in red). But because of that the radius is <b>SO
LARGE</b> that it even encapsulates a whole other cluster (that
technically should ideally be part of the &apos;lower&apos; of the tracks!).
For this reason I don&apos;t think this method is particularly useful. In
other events of course it looks more reasonable, but still. There
probably isn&apos;t a good way to make this work reliably. In any case
though, for events that are significant in size, they would almost
certainly never pass any lnL cuts anyhow.
(More plots: <img alt="septem_events_only_line_veto_check_hlc_fixed_check.png" src="./figs/home/basti/org/Figs/statusAndProgress/estimateSeptemVetoRandomCoinc/septem_events_only_line_veto_check_hlc_fixed_check.png" />)
</p>


<p>
The following is a broken event. THe purple cluster is not used for
line veto. Why?
/t/problem<sub>event</sub><sub>12435</sub><sub>run</sub><sub>297.pdf</sub>
</p>


<ul class="org-ul">
<li class="on"><code>[X]</code> Implement a cutoff for the eccentricity that a cluster must have
in order to partake in the line veto. Currently this can only be set
via an environment variable (<code>ECC_LINE_VETO_CUT</code>). A good value is
around the 1.4 - 1.6 range I think (anything that rules out most
X-ray like clusters!)</li>
</ul>
</div>
<div id="outline-container-org031763d" class="outline-5">
<h5 id="org031763d">  <a href="#org031763d">  <span class="section-number-5">12.5.6.1.</span> Note on real septemboard spacing being important   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-6-1" class="outline-text-5">

<figure id="org2c73ed2">
<img alt="example_line_veto_needs_chip_spacing.svg" class="org-svg" src="./figs/home/basti/org/Figs/statusAndProgress/exampleEvents/example_line_veto_needs_chip_spacing.svg" />

</figure>

<p>
is an example event that shows we need to introduce the correct chip
spacing <b>for the line veto</b>. For the septem veto it&apos;s not very
important, because the distance is way more important than the angle
of how things match up. But for the line veto it&apos;s essential, as can
be seen in that example (note that it uses <code>lvRegularNoHLC</code> and no
septem veto, i.e. that&apos;s why the veto is false, despite the purple HLC of
course &quot;hitting&quot; the original cluster)
-&gt; This has been implemented now. Activated (for now) via an
environment variable <code>USE_REAL_LAYOUT</code>.
An example event for the spacing &amp; the eccentricity cutoff is:
<a href="file:///home/basti/org/Figs/statusAndProgress/exampleEvents/example_event_with_line_spacing_and_ecc_cutoff.pdf">file:///home/basti/org/Figs/statusAndProgress/exampleEvents/example_event_with_line_spacing_and_ecc_cutoff.pdf</a>
which was generated using:
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">LINE_VETO_KIND</span>=lvRegularNoHLC <span style="color: #E6DB74; font-weight: bold;">\</span>
    <span style="color: #FD971F;">ECC_LINE_VETO_CUT</span>=1.6 <span style="color: #E6DB74; font-weight: bold;">\</span>
    <span style="color: #FD971F;">USE_REAL_LAYOUT</span>=true <span style="color: #E6DB74; font-weight: bold;">\</span>
    likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --h5out /tmp/lhood_2018_crAll_80eff_line.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --lineveto --plotseptem
</pre>
</div>

<p>
and then just extract it from the <code>/plots/septemEvents</code>
directory. Note the definition of the environment variables like this!
</p>
</div>
</div>
<div id="outline-container-org756bb2d" class="outline-5">
<h5 id="org756bb2d">  <a href="#org756bb2d">  <span class="section-number-5">12.5.6.2.</span> Outdated: Estimation using subset of outer ring events   <span class="tag">    <span class="extended">extended</span>  </span></a></h5>
<div id="text-12-5-6-2" class="outline-text-5">
<p>
The text here was written when we were still bootstrapping events only
from the subset of <b>event numbers</b> that actually have a cluster that
passes lnL on the center chip. This subset is of course biased even on
the outer chip. Assuming that center clusters often come with activity
on the outer chips, means there are less events representing those
cases where there isn&apos;t even any activity in the center. This
over represents activity on the outer chip.
</p>


<div class="org-src-container">
<pre class="src src-sh">likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --h5out /tmp/lhood_2018_crAll_80eff_septem_fake.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --septemveto --estimateRandomCoinc
</pre>
</div>

<p>
which writes the file <code>/tmp/septem_fake_veto.txt</code>, which for this case
is found
<a href="./../org/resources/septem_veto_random_coincidences/estimates_septem_veto_random_coincidences.txt">./../org/resources/septem_veto_random_coincidences/estimates_septem_veto_random_coincidences.txt</a>
</p>

<p>
Mean value of: 1522.61764706.
</p>

<p>
From this file the method seems to remove typically a bit less than
500 out of 2000 bootstrapped fake events. This seems to imply a random
coincidence rate of almost 25% (or effectively a reduction of further
25% in efficiency / 25% increase in dead time). Pretty scary stuff.
</p>

<p>
Of course this does not even include the line veto, which will drop it
further. Let&apos;s run that:
</p>
<div class="org-src-container">
<pre class="src src-sh">likelihood -f ~/CastData/data/DataRuns2018_Reco.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --h5out /tmp/lhood_2018_crAll_80eff_septem_line_fake.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --region crAll --cdlYear 2018 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --cdlFile ~/CastData/data/CDL_2019/calibration-cdl-2018.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
           --septemveto --lineveto --estimateRandomCoinc
</pre>
</div>

<p>
which generated the following output:
<a href="./../org/resources/septem_veto_random_coincidences/estimates_septem_line_veto_random_coincidences.txt">./../org/resources/septem_veto_random_coincidences/estimates_septem_line_veto_random_coincidences.txt</a>
</p>

<p>
Mean value of: 1373.70588235.
</p>

<p>
This comes out to a fraction of 68.68% of the events left after
running the vetoes on our bootstrapped fake events. Combining it with
a software efficiency of ε = 80% the total combined efficiency then
would be \(ε_\text{total} = 0.8 · 0.6868 = 0.5494\), so about 55%.
</p>
</div>
</div>
</div>
<div id="outline-container-org73a8da1" class="outline-4">
<h4 id="org73a8da1">  <a href="#org73a8da1">  <span class="section-number-4">12.5.7.</span> Veto setups of interest <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h4>
<div id="text-12-5-7" class="outline-text-4">
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>TABLE OF FEATURE, EFFICIENCY, BACKGROUND RATE</b>
-&gt; That table essentially motivates looking at different veto setups.</li>

<li class="off"><code>[ ]</code> Write a section about a) the motivation behind looking at
different setups to begin with and b) show the kind of setups we
will be looking at later in the limit part.</li>
</ul>


<p>
As a reference the random coincidence rates of:
</p>
<ul class="org-ul">
<li>pure septem veto = 0.7841029411764704</li>
<li>pure line veto   = 0.8601764705882353</li>
<li>septem + line veto = 0.732514705882353</li>
</ul>

<p>
Let&apos;s compute the total efficiencies of all the setups we look at.
</p>


<table>


<colgroup>
<col class="org-right" />

<col class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-right">limit</th>
<th scope="col" class="org-left">name</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-right">8.9258e-23</td>
<td class="org-left"> </td>
</tr>

<tr>
<td class="org-right">8.8516e-23</td>
<td class="org-left">_scinti</td>
</tr>

<tr>
<td class="org-right">8.5385e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.95</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.95</sub></td>
</tr>

<tr>
<td class="org-right">7.2889e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.95</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.95</sub><sub>line</sub><sub>vetoPercentile</sub><sub>0.95</sub></td>
</tr>

<tr>
<td class="org-right">7.538e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.95</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.95</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.95</sub></td>
</tr>

<tr>
<td class="org-right">7.2365e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.95</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.95</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.95</sub><sub>line</sub><sub>vetoPercentile</sub><sub>0.95</sub></td>
</tr>

<tr>
<td class="org-right">8.6007e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99</sub></td>
</tr>

<tr>
<td class="org-right">7.3555e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>line</sub><sub>vetoPercentile</sub><sub>0.99</sub></td>
</tr>

<tr>
<td class="org-right">7.5671e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.99</sub></td>
</tr>

<tr>
<td class="org-right">7.3249e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.99</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.99</sub><sub>line</sub><sub>vetoPercentile</sub><sub>0.99</sub></td>
</tr>

<tr>
<td class="org-right">8.4108e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.9</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.9</sub></td>
</tr>

<tr>
<td class="org-right">7.2315e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.9</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.9</sub><sub>line</sub><sub>vetoPercentile</sub><sub>0.9</sub></td>
</tr>

<tr>
<td class="org-right">7.4109e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.9</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.9</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.9</sub></td>
</tr>

<tr>
<td class="org-right">7.1508e-23</td>
<td class="org-left">_scinti<sub>vetoPercentile</sub><sub>0.9</sub><sub>fadc</sub><sub>vetoPercentile</sub><sub>0.9</sub><sub>septem</sub><sub>vetoPercentile</sub><sub>0.9</sub><sub>line</sub><sub>vetoPercentile</sub><sub>0.9</sub></td>
</tr>
</tbody>
</table>



<table>


<colgroup>
<col class="org-right" />

<col class="org-left" />

<col class="org-left" />

<col class="org-right" />

<col class="org-left" />

<col class="org-left" />

<col class="org-right" />

<col class="org-right" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-right">\(ε_{\ln\mathcal{L}, \text{eff}}\)</th>
<th scope="col" class="org-left">Scinti</th>
<th scope="col" class="org-left">FADC</th>
<th scope="col" class="org-right">\(ε_{\text{FADC, eff}}\)</th>
<th scope="col" class="org-left">Septem</th>
<th scope="col" class="org-left">Line</th>
<th scope="col" class="org-right">Efficiency</th>
<th scope="col" class="org-right">Expected limit (nmc=1000)</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-right">0.8</td>
<td class="org-left">x</td>
<td class="org-left">x</td>
<td class="org-right">-</td>
<td class="org-left">x</td>
<td class="org-left">x</td>
<td class="org-right">0.8</td>
<td class="org-right">8.9258e-23</td>
</tr>

<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">x</td>
<td class="org-right">-</td>
<td class="org-left">x</td>
<td class="org-left">x</td>
<td class="org-right">0.8</td>
<td class="org-right">8.8516e-23</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.98</td>
<td class="org-left">x</td>
<td class="org-left">x</td>
<td class="org-right">0.784</td>
<td class="org-right">8.6007e-23</td>
</tr>

<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.90</td>
<td class="org-left">x</td>
<td class="org-left">x</td>
<td class="org-right">0.72</td>
<td class="org-right">8.5385e-23</td>
</tr>

<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.80</td>
<td class="org-left">x</td>
<td class="org-left">x</td>
<td class="org-right">0.64</td>
<td class="org-right">8.4108e-23</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.98</td>
<td class="org-left">o</td>
<td class="org-left">x</td>
<td class="org-right">0.61</td>
<td class="org-right">7.5671e-23</td>
</tr>

<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.90</td>
<td class="org-left">o</td>
<td class="org-left">x</td>
<td class="org-right">0.56</td>
<td class="org-right">7.538e-23</td>
</tr>

<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.80</td>
<td class="org-left">o</td>
<td class="org-left">x</td>
<td class="org-right">0.50</td>
<td class="org-right">7.4109e-23</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.98</td>
<td class="org-left">x</td>
<td class="org-left">o</td>
<td class="org-right">0.67</td>
<td class="org-right">7.3555e-23</td>
</tr>

<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.90</td>
<td class="org-left">x</td>
<td class="org-left">o</td>
<td class="org-right">0.62</td>
<td class="org-right">7.2889e-23</td>
</tr>

<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.80</td>
<td class="org-left">x</td>
<td class="org-left">o</td>
<td class="org-right">0.55</td>
<td class="org-right">7.2315e-23</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.98</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.57</td>
<td class="org-right">7.3249e-23</td>
</tr>

<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.90</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.52</td>
<td class="org-right">7.2365e-23</td>
</tr>

<tr>
<td class="org-right">0.8</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.80</td>
<td class="org-left">o</td>
<td class="org-left">o</td>
<td class="org-right">0.47</td>
<td class="org-right">7.1508e-23</td>
</tr>
</tbody>
</table>

<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>ADD THE VERSIONS WE NOW LOOK AT</b>
<ul class="org-ul">
<li class="off"><code>[ ]</code> without FADC</li>
<li class="off"><code>[ ]</code> different lnL cut (70% &amp; 90%)</li>
</ul></li>

<li class="off"><code>[ ]</code> Maybe introduce a couple of versions with lower lnL software efficiency?</li>

<li class="off"><code>[ ]</code> Introduce all setups we care about for the limit calculation,
final decision will be done based on which yields the best
<span class="underline">expected</span> limit.</li>

<li class="off"><code>[ ]</code> Have a table with the setups, their background rates (full
range) and their efficiencies!</li>
</ul>
</div>
</div>
</div>
<div id="outline-container-sec:background:all_vetoes_combined" class="outline-3">
<h3 id="sec:background:all_vetoes_combined">  <a href="#sec:background:all_vetoes_combined">  <span class="section-number-3">12.6.</span> Background rates of combined vetoes and efficiencies</a></h3>
<div id="text-sec:background:all_vetoes_combined" class="outline-text-3">
<p>
Having discussed the two main classifiers (likelihood and MLP) as well
as all the different possible vetoes, let us now recap. We will look
at background rates obtained for different veto setups and
corresponding signal efficiencies. In addition, we will compare with
fig. <a href="#fig:background:background_no_vetoes_clusters">11</a> – the background
clusters left over the entire chip and local background suppression –
when using all vetoes.
</p>

<p>
For a comparison of the background rates achievable in the center
\goldArea region using the \lnL method and the MLP, see
fig. <a href="#fig:background:final_compare_LnL_mlp">36</a>. As already expected from
the MLP performance in sec. <a href="./background.html#sec:background:mlp:background_rate">12.4.8</a>, it
achieves comparable performance at significantly higher signal
efficiency. This will be a key feature in the limit calculation to
increase the total signal efficiency. The center region does not tell
the whole story though, as the background is not homogeneous.
</p>


<figure id="fig:background:final_compare_LnL_mlp">
<img alt="background_rate_crGold_scinti_fadc_septem_line_lnL80_mlp95.svg" class="org-svg" src="./figs/home/basti/phd/Figs/background/background_rate_crGold_scinti_fadc_septem_line_lnL80_mlp95.svg" />

<figcaption>Figure 36: <span class="figure-number">Figure 81: </span>Comparison of the \lnL at \(ε_{\text{eff}} = \SI{80}{\%}\) and the MLP at \(ε_{\text{eff}} = \SI{95}{\%}\) signal efficiency with all added vetoes. For the total combined efficiencies see tab. <a href="#tab:background:background_rate_eff_comparisons">26</a>.</figcaption>
</figure>

<p>
Utilizing all vetoes in addition to the \lnL or MLP classifier alone
significantly improves the background rejection over the entire
chip. See fig. <a href="#fig:background:cluster_center_comparison">37</a> comparing
all remaining X-ray like background cluster centers, without any
vetoes and with all vetoes, using the \lnL method at \(ε_{\text{eff}} =
\SI{80}{\%}\). While the center background improves, the largest
improvements are seen towards the edges and the corners. From a highly
non uniform background rate in
fig. <a href="#fig:background:cluster_centers_lnL80_without">37(a)</a>, the vetoes
produce an almost homogeneous background in
<a href="#fig:background:cluster_centers_lnL80_all">37(b)</a>. In total they yield a
\(\sim\num{13.5}\) fold reduction in
background. Fig. <a href="#fig:background:background_suppression_comparison">38</a>
visualizes the improvements by showing the local background
suppression, as a factor of the clusters left in each tile over the
total number of clusters in the raw data. At closer inspection it is
visible that the background improvements are slightly less near the
bottom edge of the chip. This is because the spacing to the bottom row
of the Septemboard is larger, decreasing the likelihood of detecting a
larger cluster.
</p>


<figure id="fig:background:cluster_center_comparison" class="figure-wrapper">
<figure id="fig:background:cluster_centers_lnL80_without" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/backgroundClusters/background_cluster_centers_lnL80_no_vetoes.svg" data-width="99%" />  <figcaption>Figure 37(a): No vetoes</figcaption></figure> <figure id="fig:background:cluster_centers_lnL80_all" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/backgroundClusters/background_cluster_centers_lnL80_all_vetoes.svg" data-width="99%" />  <figcaption>Figure 37(b): All vetoes</figcaption></figure>
<figcaption>Figure 37: Cluster centers of all X-ray like clusters in the 2017/18 CAST background
           data. <a href="#fig:background:cluster_centers_lnL80_without">37(a)</a> shows the remaining data for the \lnL method without any vetoes, <a href="#fig:background:cluster_centers_lnL80_all">37(b)</a> includes all vetoes. The vetoes lead to a dramatic reduction in background especially outside
           the center \goldArea.</figcaption>
</figure>



<figure id="fig:background:background_suppression_comparison" class="figure-wrapper">
<figure id="fig:background:suppression_lnL80_without" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/backgroundClusters/background_suppression_tile_map_lnL80_no_vetoes_zMaxS_1000.svg" data-width="99%" />  <figcaption>Figure 38(a): No vetoes</figcaption></figure> <figure id="fig:background:suppression_lnL80_all" class="subfigure" data-width="49%">  <img src="./figs/home/basti/phd/Figs/backgroundClusters/background_suppression_tile_map_lnL80_all_vetoes_zMaxS_1000.svg" data-width="99%" />  <figcaption>Figure 38(b): All vetoes</figcaption></figure>
<figcaption>Figure 38: Local background suppression of the \lnL method compared to the raw 2017/18 CAST data. <a href="#fig:background:suppression_lnL80_without">38(a)</a> is the suppression without vetoes, <a href="#fig:background:suppression_lnL80_all">38(b)</a> includes all vetoes.</figcaption>
</figure>


<p>
Tab. <a href="#tab:background:background_rate_eff_comparisons">26</a> shows a comparison
of many different possible combinations of classifier, signal
efficiency \(ε_{\text{eff}}\), choice of vetoes and total efficiency
\(ε_{\text{total}}\). The &apos;Rate&apos; column is the mean background rate in
the range \(\SIrange{0.2}{8}{keV}\) in the center \goldArea region. The
\(\ln\mathcal{L}\) produces the lowest background rates, at the cost of
low total efficiencies. Generally, many different setups reach into
below the \(\SI{1e-5}{keV^{-1}.cm^{-2}.s^{-1}}\) level in the center
region, while even large sacrifices in signal efficiency do not yield
significantly lower rates. This implies the detector is fundamentally
limited at this point in its design. The likely causes being
radioactive impurities in the detector material, imperfect cosmic
vetoes and lack of better time information in the events. See appendix
<a href="./appendix_background_rates.html#sec:appendix:background_rates">31</a>, in particular
tab. <a href="#tab:appendix:background_rates:mean_background_rates">33</a> for the same
table with background rates over the entire chip.
</p>

<p>
The choice for the optimal setup purely based on background rate and
total efficiency is difficult, especially because reducing the
background to a single number is flawed <sup>  <a id="fnr.36" href="#fn.36" class="footref" role="doc-backlink">36</a></sup>. For this reason
we will compute expected limits for each of these setups. The setup
yielding the best expected limit will then be used for the calculation
of solar tracking candidates and observed limit.
</p>

<table id="tab:background:background_rate_eff_comparisons">
<caption class="t-above"><span class="table-number">Table 26:</span> Overview of different classifier, veto and software efficiency \(ε_{\text{eff}}\) setups yielding different total efficiencies \(ε_{\text{total}}\). The background rate shown in the &apos;Rate&apos; column is between \(\SIrange{0.2}{8}{keV}\) for the center \goldArea region.</caption>

<colgroup>
<col class="org-left" />

<col class="org-right" />

<col class="org-left" />

<col class="org-left" />

<col class="org-left" />

<col class="org-left" />

<col class="org-right" />

<col class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">Classifier</th>
<th scope="col" class="org-right">ε<sub>eff</sub></th>
<th scope="col" class="org-left">Scinti</th>
<th scope="col" class="org-left">FADC</th>
<th scope="col" class="org-left">Septem</th>
<th scope="col" class="org-left">Line</th>
<th scope="col" class="org-right">ε<sub>total</sub></th>
<th scope="col" class="org-left">Rate [\(\si{keV^{-1}.cm^{-2}.s^{-1}}\)]</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.700</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-right">0.503</td>
<td class="org-left">\(\num{ 6.9015(5580)e-06}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.700</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">false</td>
<td class="org-left">true</td>
<td class="org-right">0.590</td>
<td class="org-left">\(\num{ 7.3074(5741)e-06}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.800</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-right">0.574</td>
<td class="org-left">\(\num{ 7.6683(5881)e-06}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.800</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">false</td>
<td class="org-left">true</td>
<td class="org-right">0.674</td>
<td class="org-left">\(\num{ 8.0743(6035)e-06}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.865</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-right">0.621</td>
<td class="org-left">\(\num{ 8.2096(6085)e-06}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.865</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">false</td>
<td class="org-left">true</td>
<td class="org-right">0.729</td>
<td class="org-left">\(\num{ 8.8411(6315)e-06}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.912</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-right">0.655</td>
<td class="org-left">\(\num{ 9.1117(6411)e-06}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.800</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">false</td>
<td class="org-right">0.615</td>
<td class="org-left">\(\num{ 9.3373(6490)e-06}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.900</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-right">0.646</td>
<td class="org-left">\(\num{ 9.3824(6506)e-06}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.912</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">false</td>
<td class="org-left">true</td>
<td class="org-right">0.769</td>
<td class="org-left">\(\num{ 9.6981(6614)e-06}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.900</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">false</td>
<td class="org-left">true</td>
<td class="org-right">0.759</td>
<td class="org-left">\(\num{ 9.9688(6706)e-06}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.957</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-right">0.687</td>
<td class="org-left">\(\num{1.00139(6721)e-05}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.957</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">false</td>
<td class="org-left">true</td>
<td class="org-right">0.807</td>
<td class="org-left">\(\num{1.05552(6900)e-05}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.983</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-right">0.706</td>
<td class="org-left">\(\num{1.10063(7046)e-05}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.983</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">false</td>
<td class="org-left">true</td>
<td class="org-right">0.829</td>
<td class="org-left">\(\num{1.16378(7245)e-05}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.800</td>
<td class="org-left">true</td>
<td class="org-left">true</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-right">0.784</td>
<td class="org-left">\(\num{1.55621(8378)e-05}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.865</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-right">0.865</td>
<td class="org-left">\(\num{1.56523(8403)e-05}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.700</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-right">0.700</td>
<td class="org-left">\(\num{1.65545(8641)e-05}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.912</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-right">0.912</td>
<td class="org-left">\(\num{1.74566(8874)e-05}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.800</td>
<td class="org-left">true</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-right">0.800</td>
<td class="org-left">\(\num{1.91256(9288)e-05}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.957</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-right">0.957</td>
<td class="org-left">\(\num{2.01631(9537)e-05}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.800</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-right">0.800</td>
<td class="org-left">\(\num{2.06142(9643)e-05}\)</td>
</tr>

<tr>
<td class="org-left">MLP</td>
<td class="org-right">0.983</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-right">0.983</td>
<td class="org-left">\(\num{ 2.3862(1037)e-05}\)</td>
</tr>

<tr>
<td class="org-left">LnL</td>
<td class="org-right">0.900</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-left">false</td>
<td class="org-right">0.900</td>
<td class="org-left">\(\num{ 2.7561(1115)e-05}\)</td>
</tr>
</tbody>
</table>
</div>
<div id="outline-container-orgac3666f" class="outline-4">
<h4 id="orgac3666f">  <a href="#orgac3666f">  <span class="section-number-4">12.6.1.</span> Generate background rate plots   <span class="tag">    <span class="extended">extended</span>  </span></a></h4>
<div id="text-12-6-1" class="outline-text-4">
<p>
What plots do we want exactly?
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">ESCAPE_LATEX</span>=true plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL@80+V&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL@80+V&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@95+V&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@95+V&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate from CAST data, LnL@80% and MLP@95%, all vetoes&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_crGold_scinti_fadc_septem_line_lnL80_mlp95.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/background/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
<span style="color: #75715E;">#    </span><span style="color: #75715E;">--applyEfficiencyNormalization \</span>
</pre>
</div>

<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_tanh300_outer_chip_training_30_10_23/lhood_c18_R2_crAll_sEff_0.85_mlp_mlp_tanh_sigmoid_MSE_SGD_300_2.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_tanh300_outer_chip_training_30_10_23/lhood_c18_R3_crAll_sEff_0.85_mlp_mlp_tanh_sigmoid_MSE_SGD_300_2.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_tanh300_outer_chip_training_30_10_23/lhood_c18_R2_crAll_sEff_0.85_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_SGD_300_2_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_tanh300_outer_chip_training_30_10_23/lhood_c18_R3_crAll_sEff_0.85_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_SGD_300_2_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_tanh300_outer_chip_training_30_10_23/lhood_c18_R2_crAll_sEff_0.95_mlp_mlp_tanh_sigmoid_MSE_SGD_300_2.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_tanh300_outer_chip_training_30_10_23/lhood_c18_R3_crAll_sEff_0.95_mlp_mlp_tanh_sigmoid_MSE_SGD_300_2.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_tanh300_outer_chip_training_30_10_23/lhood_c18_R2_crAll_sEff_0.95_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_SGD_300_2_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_tanh300_outer_chip_training_30_10_23/lhood_c18_R3_crAll_sEff_0.95_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_SGD_300_2_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run2_crAll_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run3_crAll_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run2_crAll_lnL_scinti_fadc_line_vetoPercentile_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run3_crAll_lnL_scinti_fadc_line_vetoPercentile_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@80&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@80&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP8+Sc+F+L&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP8+Sc+F+L&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@91&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@91&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP9+Sc+F+L&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP9+Sc+F+L&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL@80&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL@80&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL+Sc+F+L&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL+Sc+F+L&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate from CAST data&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_crGold_scinti_fadc_septem_line.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --hideErrors <span style="color: #E6DB74; font-weight: bold;">\</span>
    --hidePoints <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
<span style="color: #75715E;">#    </span><span style="color: #75715E;">--applyEfficiencyNormalization \</span>
</pre>
</div>



<p>
And now a plot with data &apos;normalized&apos; by the total signal
efficiency. Meaning that if we have e.g. 80% total efficiency, we
multiply the total time by that factor, reducing the &apos;live&apos; time.
</p>

<p>
This is a decent way to get an idea what veto setting is useful and
which isn&apos;t (i.e. losing too much efficiency for too little gain in efficiency).
</p>
<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run2_crAll_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run3_crAll_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run2_crAll_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run3_crAll_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run2_crAll_lnL_scinti_fadc_vetoPercentile_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run3_crAll_lnL_scinti_fadc_vetoPercentile_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run2_crAll_lnL_scinti_fadc_septem_vetoPercentile_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run3_crAll_lnL_scinti_fadc_septem_vetoPercentile_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run2_crAll_lnL_scinti_fadc_line_vetoPercentile_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run3_crAll_lnL_scinti_fadc_line_vetoPercentile_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run2_crAll_lnL_scinti_fadc_septem_line_vetoPercentile_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_limits_automation_with_nn_support/lhood/likelihood_cdl2018_Run3_crAll_lnL_scinti_fadc_septem_line_vetoPercentile_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> --names <span style="color: #E6DB74;">&quot;No vetoes&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> --names <span style="color: #E6DB74;">&quot;Scinti&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;FADC&quot;</span> --names <span style="color: #E6DB74;">&quot;FADC&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;Septem&quot;</span> --names <span style="color: #E6DB74;">&quot;Septem&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;noSeptem&quot;</span> --names <span style="color: #E6DB74;">&quot;noSeptem&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;Line&quot;</span> --names <span style="color: #E6DB74;">&quot;Line&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate from CAST data, incl. scinti, FADC, septem, line veto&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_crGold_scinti_fadc_septem_line.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTeX <span style="color: #E6DB74; font-weight: bold;">\</span>
    --applyEfficiencyNormalization <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
</pre>
</div>
<p>
-&gt; Ok, the above works with, the files all have the <code>logCtx</code> group.
</p>
</div>
</div>
<div id="outline-container-sec:background:gen_bck_cluster_plots_all_vetoes" class="outline-4">
<h4 id="sec:background:gen_bck_cluster_plots_all_vetoes">  <a href="#sec:background:gen_bck_cluster_plots_all_vetoes">  <span class="section-number-4">12.6.2.</span> Generate plot of background clusters left over with all vetoes <code>[/]</code>   <span class="tag">    <span class="extended">extended</span>  </span></a></h4>
<div id="text-sec:background:gen_bck_cluster_plots_all_vetoes" class="outline-text-4">
<ul class="org-ul">
<li class="off"><code>[ ]</code> <b>MAYBE MAKE TEXT FONT WHITE AGAIN?</b></li>
</ul>

<p>
LnL @ 80% and no vetoes, but using the same zMax as for the all veto case below.
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">ESCAPE_LATEX</span>=true plotBackgroundClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMax 5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMaxSuppression 1000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;CAST data, LnL@80%, no vetoes&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/backgroundClusters/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --filterNoisyPixels <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 --energyMax 12.0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showGoldRegion <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundSuppression <span style="color: #E6DB74; font-weight: bold;">\</span>
    --suffix <span style="color: #E6DB74;">&quot;_lnL80_no_vetoes_zMaxS_1000&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTikZ
</pre>
</div>
<p>
LnL @ 80% and all vetoes:
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">ESCAPE_LATEX</span>=true plotBackgroundClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMax 5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMaxSuppression 1000 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;CAST data, LnL@80%, all vetoes&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/backgroundClusters/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --filterNoisyPixels <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 --energyMax 12.0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showGoldRegion <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundSuppression <span style="color: #E6DB74; font-weight: bold;">\</span>
    --suffix <span style="color: #E6DB74;">&quot;_lnL80_all_vetoes_zMaxS_1000&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTikZ
</pre>
</div>

<p>
Background clusters, 95%:
</p>
<div class="org-src-container">
<pre class="src src-sh"><span style="color: #FD971F;">ESCAPE_LATEX</span>=true plotBackgroundClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --zMax 5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Cluster centers CAST data, MLP@95%, all vetoes&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath ~/phd/Figs/backgroundClusters/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --filterNoisyPixels <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showGoldRegion <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 --energyMax 12.0 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --suffix <span style="color: #E6DB74;">&quot;_mlp_95_all_vetoes_adam_tanh30_sigmoid_mse_82k&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --backgroundSuppression <span style="color: #E6DB74; font-weight: bold;">\</span>
    --useTikZ
</pre>
</div>
</div>
</div>
<div id="outline-container-sec:background:generate_background_rate_table" class="outline-4">
<h4 id="sec:background:generate_background_rate_table">  <a href="#sec:background:generate_background_rate_table">  <span class="section-number-4">12.6.3.</span> Generate background rate table   <span class="tag">    <span class="extended">extended</span>  </span></a></h4>
<div id="text-sec:background:generate_background_rate_table" class="outline-text-4">
<p>
Uhhh, excuse this massive wall. I could have written a scrip that
reads the files from two directories, but well…
</p>

<div class="org-src-container">
<pre class="src src-sh">plotBackgroundRate <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_septem_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_septem_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.8_lnL_scinti_fadc_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.8_lnL_scinti_fadc_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL80&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL80&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL80+Sc&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL80+Sc&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL80+F&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL80+F&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL80+S&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL80+S&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL80+SL&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL80+SL&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL80+L&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL80+L&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.7_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.7_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.7_lnL_scinti_fadc_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.7_lnL_scinti_fadc_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.7_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.7_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL70&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL70&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL70+L&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL70+L&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL70+S&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL70+S&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.9_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.9_lnL.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.9_lnL_scinti_fadc_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.9_lnL_scinti_fadc_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R2_crAll_sEff_0.9_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_lnL_17_11_23_septem_fixed/lhood_c18_R3_crAll_sEff_0.9_lnL_scinti_fadc_septem_line_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL90&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL90&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL90+L&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL90+L&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;LnL90+S&quot;</span> --names <span style="color: #E6DB74;">&quot;LnL90+S&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.85_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.85_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.85_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.85_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.85_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.85_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@85&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@85&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@85+L&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@85+L&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@85+S&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@85+S&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.9_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.9_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.9_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.9_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.9_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.9_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@9&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@9&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@9+L&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@9+L&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@9+S&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@9+S&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.95_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.95_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@95&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@95&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@95+L&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@95+L&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@95+S&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@95+S&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.98_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.98_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.98_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.98_scinti_fadc_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R2_crAll_sEff_0.98_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    ~/org/resources/lhood_mlp_17_11_23_adam_tanh30_sigmoid_mse_82k/lhood_c18_R3_crAll_sEff_0.98_scinti_fadc_septem_line_mlp_mlp_tanh_sigmoid_MSE_Adam_30_2checkpoint_epoch_82000_loss_0.0249_acc_0.9662_vQ_0.99.h5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@98&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@98&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@98+L&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@98+L&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --names <span style="color: #E6DB74;">&quot;MLP@98+S&quot;</span> --names <span style="color: #E6DB74;">&quot;MLP@98+S&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --centerChip 3 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --title <span style="color: #E6DB74;">&quot;Background rate from CAST data, incl. scinti, FADC, septem, line veto&quot;</span> <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showNumClusters <span style="color: #E6DB74; font-weight: bold;">\</span>
    --showTotalTime <span style="color: #E6DB74; font-weight: bold;">\</span>
    --topMargin 1.5 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyDset energyFromCharge <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outfile background_rate_crGold_scinti_fadc_septem_line.pdf <span style="color: #E6DB74; font-weight: bold;">\</span>
    --outpath /tmp/ <span style="color: #E6DB74; font-weight: bold;">\</span>
    --region crGold <span style="color: #E6DB74; font-weight: bold;">\</span>
    --energyMin 0.2 <span style="color: #E6DB74; font-weight: bold;">\</span>
    --rateTable ~/phd/resources/background_rate_comparisons.org <span style="color: #E6DB74; font-weight: bold;">\</span>
    --noPlot <span style="color: #E6DB74; font-weight: bold;">\</span>
    --quiet
</pre>
</div>
</div>
</div>
</div>
</div>

<div id="footnotes"><h2 class="footnotes">Footnotes: </h2><div class="footdef"><sup>  <a id="fn.1" href="#fnr.1" class="footnum" role="doc-backlink">1</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Of course the set of candidates
contains background itself. The terminology &apos;candidate&apos; intends
to communicate that each candidate may be a background event or
potentially a signal due to axions. But that is part of chapter
<a href="#sec:limit">13</a>.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.2" href="#fnr.2" class="footnum" role="doc-backlink">2</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
In hindsight it seems obvious that a
similarly shielded LEMO cable should have been used for the
measurements at CAST. Unfortunately, due to lack of experience with
electromagnetic interference this was not considered before.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.3" href="#fnr.3" class="footnum" role="doc-backlink">3</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
The measurement campaign at the CDL was done in
February. However, the weather was nice and sunny most of the week, if
I remember correctly. The laboratory has windows towards the
south-east. With reasonably cold outside temperatures and sunshine
warming the lab in addition to machines and people, opening and
closing windows changed the temperature on short time scales, likely
significantly contributing to the detector instability. In the
extended version of this thesis you&apos;ll find a section below this one
containing the weather data at the time of the CDL data taking.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.4" href="#fnr.4" class="footnum" role="doc-backlink">4</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Note that similar fits can be performed for
the pixel spectra as well. However, as these are not needed for
anything further they are only presented in the extended version of
the thesis.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.5" href="#fnr.5" class="footnum" role="doc-backlink">5</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
The restriction to what is &apos;visually
identifiable&apos; is to avoid the need to fit lines lost in detector
resolution and signal to noise ratio. In addition, it avoids
overfitting to large numbers of parameters (which brings its own
challenges and problems).
</p></div></div>

<div class="footdef"><sup>  <a id="fn.6" href="#fnr.6" class="footnum" role="doc-backlink">6</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
We can of course apply the regular
energy calibration based on the multiple fits to the CAST calibration
data as explained in
sec. <a href="#sec:calib:final_energy_calibration">11.3</a>. However, due to the very
different gas gains observed in the CDL, the applicability is not
ideal. Add to that the fact that we know for certain the energy of the
clusters in the main fluorescence peak there is simply no need either.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.7" href="#fnr.7" class="footnum" role="doc-backlink">7</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Keep in mind that all energies undergo roughly the same
amount of diffusion (aside from the energy dependent absorption length
to a lesser extent), so lower energy events are more sparse.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.8" href="#fnr.8" class="footnum" role="doc-backlink">8</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
The \(-\ln\mathcal{L}\) values in the
distributions give a good idea of why. Roughly speaking the values go
from 5 to 20, meaning the actual likelihood values are in the range
from \(e^{-5} \approx \num{7e-3}\) to \(e^{-20} \approx \num{2e-9}\)!
While 64-bit floating point numbers nowadays in principle provide
enough precision for these numbers, human readability is improved. But
32-bit floats would already accrue serious floating point errors due
to only about 7 decimal digits accuracy. But even with 64 bit floats,
slight changes to the likelihood definition might run into trouble
as well.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.9" href="#fnr.9" class="footnum" role="doc-backlink">9</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
It&apos;s not half the energy as the lines are
not evenly spaced in energy.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.10" href="#fnr.10" class="footnum" role="doc-backlink">10</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
The only conceptual difference in the techniques is
our inclusion of the interpolation between reference distributions in energy.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.11" href="#fnr.11" class="footnum" role="doc-backlink">11</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Due to the rich and long history of artificial neural
networks picking &quot;a&quot; or only a few citations is tricky. Amari&apos;s work (<a href="#citeproc_bib_item_7">Amari 1967</a>)
was, as far as I&apos;m aware, the first to combine a perceptron with
non-linear activation functions and using gradient descent for
training. See Schmidhuber&apos;s recent overview for a detailed history
leading up to modern deep learning (<a href="#citeproc_bib_item_194">Schmidhuber 2022</a>). 
</p></div></div>

<div class="footdef"><sup>  <a id="fn.12" href="#fnr.12" class="footnum" role="doc-backlink">12</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
For performance reasons to utilize the
parallel nature of GPUs, training and inference of NNs is done in
&apos;mini-batches&apos;. As still only a single loss value is needed for
training, the loss is computed as the mean of all losses for each
mini-batch element.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.13" href="#fnr.13" class="footnum" role="doc-backlink">13</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
The gradients in a neural network are
usually computed using &apos;automatic differentiation&apos; (or &apos;autograd&apos;,
&apos;autodiff&apos;). There are two forms of automatic differentiation: forward
mode and reverse mode. These differ by the practical evaluation order
of the chain rule. Forward mode computes the chain rule from
left-to-right (input to output), while reverse mode computes it from
right-to-left (output to input). Computationally these differ in their
complexity in terms of the required number of evaluations given N
inputs and M outputs. Forward mode computes all M output derivatives
for a single input, whereas reverse mode computes all input
derivatives for a single output. Thus, forward mode is efficient for
cases with few inputs and many outputs, while the opposite is true for
reverse mode. Neural networks are classical cases of many inputs to
few outputs (scalar loss function!). As such, reverse mode autograd is
the standard way to compute the gradients during NN training. In the
context it is effectively synonymous with &apos;backpropagation&apos; (due to
its output-to-input evaluation of the chain rule).
</p></div></div>

<div class="footdef"><sup>  <a id="fn.14" href="#fnr.14" class="footnum" role="doc-backlink">14</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
The acute reader may wonder why we care less
about this for the \(\ln\mathcal{L}\) method. The reason is simply that
the effect of gas gain on the three properties used there is
comparatively small. But if (almost) all properties are to be used,
that is less true.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.15" href="#fnr.15" class="footnum" role="doc-backlink">15</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Generating \(\num{100000}\) fake events of the \cefe
photopeak takes around \(\SI{35}{s}\) on a single thread using an AMD
Ryzen 9 5950X. The code is not exactly optimized either.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.16" href="#fnr.16" class="footnum" role="doc-backlink">16</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
\(σ_T\) would be more appropriate following
eq. \eqref{eq:gas_physics:diffusion_after_drift} in
sec. <a href="#sec:theory:gas_diffusion">6.3.4</a>.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.17" href="#fnr.17" class="footnum" role="doc-backlink">17</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Note that it is important to consider the
<span class="underline">transverse</span> standard deviation and not the longitudinal one. Even for
an X-ray there will be a short track like signature during production
of the primary electrons from the initial photoelectron. This can
define the long axis of the cluster in certain cases (depending on gas
mixture). Similarly, for background events of cosmic muons the long
axis corresponds to the direction of travel while the transverse
direction is purely due to the diffusion process.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.18" href="#fnr.18" class="footnum" role="doc-backlink">18</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
For example a number provided by Magboltz for the used
gas mixture at normal temperature and used chamber pressure. For the
CAST gas mixture and settings we start with \(D_T = \SI{660}{μm.cm^{-1/2}}\).
</p></div></div>

<div class="footdef"><sup>  <a id="fn.19" href="#fnr.19" class="footnum" role="doc-backlink">19</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
In principle the event generation is the same
algorithm as explained previously, but it does not sample from a Pólya
for the gas gain, nor compute any cluster properties. Only the
conversion point, number of target electrons based on target energy
and their positions is simulated. From these electron positions the
long and short axes are determined and \(σ_T\) returned.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.20" href="#fnr.20" class="footnum" role="doc-backlink">20</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Initially I tried to compute the
gradient using automatic differentiation using dual numbers for the
\(D_T\) input and the MC code, but the calculation of the test
statistics is effectively independent of the input numbers due to the
calculation of the ECDF. As such the derivative information is lost. I
didn&apos;t manage to come up with a way to compute Cramér-von-Mises based
on the actual input numbers directly in a timely manner.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.21" href="#fnr.21" class="footnum" role="doc-backlink">21</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
One optimization takes on the order of
\(&lt;\SI{30}{s}\) for most runs. But the diffusion parameters are used
many times.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.22" href="#fnr.22" class="footnum" role="doc-backlink">22</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
This network is considered a &quot;reference&quot;
implementation, as it is essentially the simplest ANN layout possible.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.23" href="#fnr.23" class="footnum" role="doc-backlink">23</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
<code>libtorch</code> is the C++ library PyTorch is built on.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.24" href="#fnr.24" class="footnum" role="doc-backlink">24</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Yes, only \(\num{30}\) neurons. When using SGD as an
optimizer more neurons are useful. With Adam anything more leads to
excessive overtraining. Model performance and training time is anyhow
better with Adam. Indeed, even a \(\num{10}\) neuron model performs
almost as good. The latter is fun, because it allows to write the
weight matrices on a piece of paper and almost do a forward pass by hand.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.25" href="#fnr.25" class="footnum" role="doc-backlink">25</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
For training X-rays a total of \(\num{500 000}\)
events are simulated. For background \(\num{1 000}\) events are randomly
selected from each outer chip for each background runs (so \(\num{6
000}\) clusters per background run), totaling \(\num{576 000}\)
events. Training and validation is split 50-50.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.26" href="#fnr.26" class="footnum" role="doc-backlink">26</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
One could of course interpolate on the cut values
itself, but that is less well motivated and harder to
cross-check. Predicting a cut value from two known, neighboring
energies would likely not work very well. 
</p></div></div>

<div class="footdef"><sup>  <a id="fn.27" href="#fnr.27" class="footnum" role="doc-backlink">27</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
The number quoted here for \lnL at \(\SI{80}{\%}\)
differs slightly from sec. <a href="#sec:background:likelihood_cut">12.3</a>, because the
lower energy range is \(\SI{0.2}{keV}\) here.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.28" href="#fnr.28" class="footnum" role="doc-backlink">28</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
A ball park estimate yields a coverage
of about \(\SI{35}{°}\) around the zenith. Assuming \(\cos²(θ)\) muons it
covers about \(\SI{68}{\%}\) of muons in that axis. With a rate of
\(\sim\SI{1}{cm⁻².min⁻¹}\), \(\SI{1125}{h}\) of data with scintillator and
\(\SI{4.2}{cm²}\) of active area in front of center GridPix roughly
\(\num{194000}\) muons are expected. About \(\num{70000}\) non trivial
triggers were recorded. Efficiency of \(\sim\SI{35}{\%}\).
</p></div></div>

<div class="footdef"><sup>  <a id="fn.29" href="#fnr.29" class="footnum" role="doc-backlink">29</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Estimating the number of expected
muons for the SiPM is much more difficult, because there is little
literature on the actual muon rate at angles close to \(\SI{90}{°}\). The
extended version of this thesis contains some calculations, which try
to estimate energy distribution and rate under these angles in the
detector based on numerically transporting muons from the upper
atmosphere to the surface undergoing relativistic effects, changing
atmospheric pressures and the related energy loss per distance.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.30" href="#fnr.30" class="footnum" role="doc-backlink">30</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
The properties were calculated with
<code>Magboltz</code> using <code>Nimboltz</code>, a Nim interfacer for <code>Magboltz</code>. The code can be found
here:
<a href="https://github.com/Vindaar/TimepixAnalysis/Tools/septemboardCastGasNimboltz">https://github.com/Vindaar/TimepixAnalysis/Tools/septemboardCastGasNimboltz</a>
Further the numbers used here are for a temperature of \(\SI{26}{°C}\),
slightly above room temperature due to the heating of the Septemboard
itself into the gas.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.31" href="#fnr.31" class="footnum" role="doc-backlink">31</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Of course a properly calibrated
scintillator behind the magnet should detect any muons traversing
orthogonally. However, it is likely a \(4π\) veto system would be used
in a coincidence setup between opposite scintillators. A detector with
good time resolution can also detect such events in principle. Despite
this, it might be a decent idea to include some lead shielding on the
opposite end of future helioscopes to provide some shielding against
muons coming from such a direction. In particular a future experiment
that tracks the Sun to much higher altitudes will see such muon
background significantly more (\(\cos²(θ)\)!).
</p></div></div>

<div class="footdef"><sup>  <a id="fn.32" href="#fnr.32" class="footnum" role="doc-backlink">32</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
An alternative could be to attempt to fill in
the area between the chips with simulated data based on what is seen
at the chip edges. But that is a complex problem. A fun (but possibly
questionable) idea would be to train a diffusion model (generative AI)
to predict missing data between chips.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.33" href="#fnr.33" class="footnum" role="doc-backlink">33</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
In principle this has a small impact on the
calculation of the RMS of the cluster, if there is now a large gap
between two parts of the &apos;same&apos; cluster. However, as we don&apos;t really
rely on the properties too much it is of no real concern. Technically
we still calculate the \(\ln\mathcal{L}\) of each cluster, but any
cluster passing the \(\ln\mathcal{L}\) cut before being added to a
larger cluster almost certainly won&apos;t pass it afterwards. The slight
bias in the RMS won&apos;t change that.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.34" href="#fnr.34" class="footnum" role="doc-backlink">34</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
The downside is that DBSCAN is <span class="underline">significantly</span> slower,
which is part of the reason it is not utilized for the initial reconstruction.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.35" href="#fnr.35" class="footnum" role="doc-backlink">35</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
This is roughly the radius of the cluster going by
most extended active pixels of the cluster.
</p></div></div>

<div class="footdef"><sup>  <a id="fn.36" href="#fnr.36" class="footnum" role="doc-backlink">36</a></sup> <div class="footpara" role="doc-footnote">  <p class="footpara">
Flawed, because a) the background is not homogeneous over
the entire chip and the effect of vetoes outside the center region is
not visible and b) because the most important range for axion searches
is between \(\SIrange{0.5}{5}{keV}\) (depending on the model).
</p></div></div>

</div>
<div class="hint-message">Click on any heading marked '<span class="extended">extended</span>' to open it</div>
</div>
</body>
</html>
